<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.23">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>8&nbsp; Additional information – Statistical analysis of mass spectrometry-based proteomics data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./07-mouse_diet.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac32dca00065cc45e71a60b6df2d2e12.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./99-end.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figs/msqrob2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistical analysis of mass spectrometry-based proteomics data</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preamble</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-benchmarking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-workflow_optimisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-francisella.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-heart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-mouse_diet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-end.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#citation" id="toc-citation" class="nav-link active" data-scroll-target="#citation"><span class="header-section-number">8.1</span> Citation</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  <li><a href="#data-sets" id="toc-data-sets" class="nav-link" data-scroll-target="#data-sets"><span class="header-section-number">8.2</span> Data sets</a>
  <ul class="collapse">
  <li><a href="#e.-coli-lfq-spike-in-data-set" id="toc-e.-coli-lfq-spike-in-data-set" class="nav-link" data-scroll-target="#e.-coli-lfq-spike-in-data-set"><span class="header-section-number">8.2.1</span> E. Coli LFQ spike-in data set</a></li>
  <li><a href="#tmt-spike-in-data-set" id="toc-tmt-spike-in-data-set" class="nav-link" data-scroll-target="#tmt-spike-in-data-set"><span class="header-section-number">8.2.2</span> TMT spike-in data set</a></li>
  <li><a href="#francisella-data-set" id="toc-francisella-data-set" class="nav-link" data-scroll-target="#francisella-data-set"><span class="header-section-number">8.2.3</span> Francisella data set</a></li>
  <li><a href="#heart-data-set" id="toc-heart-data-set" class="nav-link" data-scroll-target="#heart-data-set"><span class="header-section-number">8.2.4</span> Heart data set</a></li>
  <li><a href="#mouse-diet-data-set" id="toc-mouse-diet-data-set" class="nav-link" data-scroll-target="#mouse-diet-data-set"><span class="header-section-number">8.2.5</span> Mouse diet data set</a></li>
  </ul></li>
  <li><a href="#license" id="toc-license" class="nav-link" data-scroll-target="#license"><span class="header-section-number">8.3</span> License</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">8.4</span> Session Info</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-end" class="quarto-section-identifier"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="citation" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="citation"><span class="header-section-number">8.1</span> Citation</h2>
<p>Please cite this book as:</p>
<p><strong>TODO</strong>: add citation once published</p>
<p>Please cite the <code>msqrob2</code> package as:</p>
<blockquote class="blockquote">
<p>Goeminne L, Gevaert K, Clement L (2016). “Peptide-level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-dependent Quantitative Label-free Shotgun Proteomics.” <em>Molecular &amp; Cellular Proteomics</em>, <em>15</em>(2), 657-668. doi:<a href="https://doi.org/10.1074/mcp.m115.055897">10.1074/mcp.m115.055897</a>.</p>
</blockquote>
<p>If you opt for a summarisation-based workflow, you can also cite:</p>
<blockquote class="blockquote">
<p>Sticker A, Goeminne L, Martens L, Clement L (2020). “Robust Summarization and Inference in Proteome-wide Label-free Quantification.” <em>Molecular &amp; Cellular Proteomics</em>, <em>19</em>(7), 1209-1219. doi:<a href="https://doi.org/10.1074/mcp.ra119.001624">10.1074/mcp.ra119.001624</a>.</p>
</blockquote>
<p>If you use TMT-based workflows, please cite</p>
<blockquote class="blockquote">
<p>Vandenbulcke S, Vanderaa C, Crook O, Martens L, Clement L. msqrob2TMT: robust linear mixed models for inferring differential abundant proteins in labelled experiments with arbitrarily complex design. bioRxiv. Published online March 29, 2024:2024.03.29.587218. doi:10.1101/2024.03.29.587218</p>
</blockquote>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Gatto2023-kk" class="csl-entry" role="listitem">
Gatto, Laurent, Ruedi Aebersold, Juergen Cox, Vadim Demichev, Jason
Derks, Edward Emmott, Alexander M Franks, et al. 2023. <span>“Initial
Recommendations for Performing, Benchmarking and Reporting Single-Cell
Proteomics Experiments.”</span> <em>Nat. Methods</em> 20 (3): 375–86.
</div>
<div id="ref-Goeminne2016-tr" class="csl-entry" role="listitem">
Goeminne, Ludger J E, Kris Gevaert, and Lieven Clement. 2016.
<span>“Peptide-Level Robust Ridge Regression Improves Estimation,
Sensitivity, and Specificity in Data-Dependent Quantitative Label-Free
Shotgun Proteomics.”</span> <em>Mol. Cell. Proteomics</em> 15 (2):
657–68.
</div>
<div id="ref-Huang2020-lu" class="csl-entry" role="listitem">
Huang, Ting, Meena Choi, Manuel Tzouros, Sabrina Golling, Nikhil Janak
Pandya, Balazs Banfai, Tom Dunkley, and Olga Vitek. 2020.
<span>“<span>MSstatsTMT</span>: Statistical Detection of Differentially
Abundant Proteins in Experiments with Isobaric Labeling and Multiple
Mixtures.”</span> <em>Mol. Cell. Proteomics</em> 19 (10): 1706–23.
</div>
<div id="ref-O-Brien2024-lr" class="csl-entry" role="listitem">
O’Brien, Jonathon J, Anil Raj, Aleksandr Gaun, Adam Waite, Wenzhou Li,
David G Hendrickson, Niclas Olsson, and Fiona E McAllister. 2024.
<span>“A Data Analysis Framework for Combining Multiple Batches
Increases the Power of Isobaric Proteomics Experiments.”</span> <em>Nat.
Methods</em> 21 (2): 290–300.
</div>
<div id="ref-Plubell2017-th" class="csl-entry" role="listitem">
Plubell, Deanna L, Phillip A Wilmarth, Yuqi Zhao, Alexandra M Fenton,
Jessica Minnier, Ashok P Reddy, John Klimek, Xia Yang, Larry L David,
and Nathalie Pamir. 2017. <span>“Extended Multiplexing of Tandem Mass
Tags (<span>TMT</span>) Labeling Reveals Age and High Fat Diet Specific
Proteome Changes in Mouse Epididymal Adipose Tissue.”</span> <em>Mol.
Cell. Proteomics</em> 16 (5): 873–90.
</div>
<div id="ref-Ramond2015-rz" class="csl-entry" role="listitem">
Ramond, Elodie, Gael Gesbert, Ida Chiara Guerrera, Cerina Chhuon, Marion
Dupuis, Mélanie Rigard, Thomas Henry, Monique Barel, and Alain Charbit.
2015. <span>“Importance of Host Cell Arginine Uptake in Francisella
Phagosomal Escape and Ribosomal Protein Amounts.”</span> <em>Mol. Cell.
Proteomics</em> 14 (4): 870–81.
</div>
<div id="ref-Savitski2011-qi" class="csl-entry" role="listitem">
Savitski, Mikhail M, Gavain Sweetman, Manor Askenazi, Jarrod A Marto,
Manja Lang, Nico Zinn, and Marcus Bantscheff. 2011. <span>“Delayed
Fragmentation and Optimized Isolation Width Settings for Improvement of
Protein Identification and Accuracy of Isobaric Mass Tag Quantification
on Orbitrap-Type Mass Spectrometers.”</span> <em>Anal. Chem.</em> 83
(23): 8959–67.
</div>
<div id="ref-Segers2025-ce" class="csl-entry" role="listitem">
Segers, Alexandre, Cristian Castiglione, Christophe Vanderaa, Elfride De
Baere, Lennart Martens, Davide Risso, and Lieven Clement. 2025.
<span>“<span class="nocase">omicsGMF</span>: A Multi-Tool for
Dimensionality Reduction, Batch Correction and Imputation Applied to
Bulk- and Single Cell Proteomics Data.”</span> <em>bioRxiv</em>, March,
2025.03.24.644996.
</div>
<div id="ref-Shen2018-gw" class="csl-entry" role="listitem">
Shen, Xiaomeng, Shichen Shen, Jun Li, Qiang Hu, Lei Nie, Chengjian Tu,
Xue Wang, et al. 2018. <span>“<span>IonStar</span> Enables
High-Precision, Low-Missing-Data Proteomics Quantification in Large
Biological Cohorts.”</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 115
(21): E4767–76.
</div>
<div id="ref-Sticker2020-rl" class="csl-entry" role="listitem">
Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement.
2020. <span>“Robust Summarization and Inference in Proteome-Wide
Label-Free Quantification.”</span> <em>Mol. Cell. Proteomics</em> 19
(7): 1209–19.
</div>
<div id="ref-Vandenbulcke2025-sj" class="csl-entry" role="listitem">
Vandenbulcke, Stijn, Christophe Vanderaa, Oliver Crook, Lennart Martens,
and Lieven Clement. 2025. <span>“<span>Msqrob2TMT</span>: Robust Linear
Mixed Models for Inferring Differential Abundant Proteins in Labeled
Experiments with Arbitrarily Complex Design.”</span> <em>Mol. Cell.
Proteomics</em> 24 (7): 101002.
</div>
</div>
</section>
<section id="data-sets" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="data-sets"><span class="header-section-number">8.2</span> Data sets</h2>
<p>We refer here the data sources used in the book:</p>
<section id="e.-coli-lfq-spike-in-data-set" class="level3" data-number="8.2.1">
<h3 data-number="8.2.1" class="anchored" data-anchor-id="e.-coli-lfq-spike-in-data-set"><span class="header-section-number">8.2.1</span> E. Coli LFQ spike-in data set</h3>
<p>Original study: Shen X, Shen S, Li J, Hu Q, Nie L, Tu C, et al.&nbsp;(2018) Ionstar enables high-precision, low-missing-data proteomics quantification in large bio- logical cohorts. Proc. Natl. Acad. Sci. U.S.A. 115, E4767–E4776</p>
<p>Reanalysis study: Sticker A, Goeminne L, Martens L, Clement L. Robust Summarization and Inference in Proteome-wide Label-free Quantification. Mol Cell Proteomics. 2020;19(7):1209-1219.</p>
<p>Link to data: https://github.com/statOmics/MSqRobSumPaper/raw/refs/heads/master/spikein/data/maxquant/peptides.zip</p>
<p>Link to data in archive: <strong>TODO</strong> add link to Zenodo</p>
<p>Used in <a href="01-basics.html" class="quarto-xref"><span>Chapter 1</span></a> and <a href="03-benchmarking.html" class="quarto-xref"><span>Chapter 3</span></a>.</p>
</section>
<section id="tmt-spike-in-data-set" class="level3" data-number="8.2.2">
<h3 data-number="8.2.2" class="anchored" data-anchor-id="tmt-spike-in-data-set"><span class="header-section-number">8.2.2</span> TMT spike-in data set</h3>
<p>Original study: Huang T, Choi M, Tzouros M, Golling S, Pandya NJ, Banfai B, et al.&nbsp;MSstatsTMT: Statistical Detection of Differentially Abundant Proteins in Experiments with Isobaric Labeling and Multiple Mixtures. Mol Cell Proteomics. 2020;19(10):1706-1723.</p>
<p>Reanalysis study: Vandenbulcke S, Vanderaa C, Crook O, Martens L, Clement L. Msqrob2TMT: Robust linear mixed models for inferring differential abundant proteins in labeled experiments with arbitrarily complex design. Mol Cell Proteomics. 2025;24(7):101002.</p>
<p>Data source: MassIVE repository (RMSV000000265)</p>
<p>Link to data from archive: https://zenodo.org/records/14767905</p>
<p>Used in <a href="02-advanced.html" class="quarto-xref"><span>Chapter 2</span></a>.</p>
</section>
<section id="francisella-data-set" class="level3" data-number="8.2.3">
<h3 data-number="8.2.3" class="anchored" data-anchor-id="francisella-data-set"><span class="header-section-number">8.2.3</span> Francisella data set</h3>
<p><strong>TODO</strong></p>
</section>
<section id="heart-data-set" class="level3" data-number="8.2.4">
<h3 data-number="8.2.4" class="anchored" data-anchor-id="heart-data-set"><span class="header-section-number">8.2.4</span> Heart data set</h3>
<p><strong>TODO</strong></p>
</section>
<section id="mouse-diet-data-set" class="level3" data-number="8.2.5">
<h3 data-number="8.2.5" class="anchored" data-anchor-id="mouse-diet-data-set"><span class="header-section-number">8.2.5</span> Mouse diet data set</h3>
<p><strong>TODO</strong></p>
</section>
</section>
<section id="license" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="license"><span class="header-section-number">8.3</span> License</h2>
<p><a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons Licence" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"></a><br>This material is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/">Creative Commons Attribution-ShareAlike 4.0 International License</a>. You are free to <strong>share</strong> (copy and redistribute the material in any medium or format) and <strong>adapt</strong> (remix, transform, and build upon the material) for any purpose, even commercially, as long as you give appropriate credit and distribute your contributions under the same license as the original.</p>
</section>
<section id="session-info" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="session-info"><span class="header-section-number">8.4</span> Session Info</h2>
<p>The following packages have been used to generate this document.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.5.2 (2025-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.12.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.12.0  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Brussels
tzcode source: system (glibc)

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] bookdown_0.45               tidyr_1.3.1                
 [3] scater_1.38.0               scuttle_1.20.0             
 [5] SingleCellExperiment_1.32.0 patchwork_1.3.2            
 [7] MsDataHub_1.10.0            impute_1.84.0              
 [9] ggrepel_0.9.6               ggplot2_4.0.1              
[11] ExploreModelMatrix_1.22.0   dplyr_1.1.4                
[13] ComplexHeatmap_2.26.0       BiocFileCache_3.0.0        
[15] dbplyr_2.5.1                BiocParallel_1.44.0        
[17] msqrob2book_0.0.99          msqrob2_1.18.0             
[19] QFeatures_1.20.0            MultiAssayExperiment_1.36.1
[21] SummarizedExperiment_1.40.0 Biobase_2.70.0             
[23] GenomicRanges_1.62.0        Seqinfo_1.0.0              
[25] IRanges_2.44.0              S4Vectors_0.48.0           
[27] BiocGenerics_0.56.0         generics_0.1.4             
[29] MatrixGenerics_1.22.0       matrixStats_1.5.0          

loaded via a namespace (and not attached):
  [1] splines_4.5.2           later_1.4.4             filelock_1.0.3         
  [4] tibble_3.3.0            lifecycle_1.0.4         httr2_1.2.1            
  [7] Rdpack_2.6.4            doParallel_1.0.17       rprojroot_2.1.1        
 [10] lattice_0.22-7          MASS_7.3-65             magrittr_2.0.4         
 [13] limma_3.66.0            rmarkdown_2.30          yaml_2.3.11            
 [16] remotes_2.5.0           httpuv_1.6.16           otel_0.2.0             
 [19] sessioninfo_1.2.3       pkgbuild_1.4.8          cowplot_1.2.0          
 [22] MsCoreUtils_1.22.1      DBI_1.2.3               minqa_1.2.8            
 [25] RColorBrewer_1.1-3      abind_1.4-8             pkgload_1.4.1          
 [28] purrr_1.2.0             AnnotationFilter_1.34.0 rappdirs_0.3.3         
 [31] circlize_0.4.16         irlba_2.3.5.1           codetools_0.2-20       
 [34] DelayedArray_0.36.0     DT_0.34.0               tidyselect_1.2.1       
 [37] shape_1.4.6.1           farver_2.1.2            lme4_1.1-37            
 [40] ScaledMatrix_1.18.0     viridis_0.6.5           jsonlite_2.0.0         
 [43] GetoptLong_1.1.0        BiocNeighbors_2.4.0     ellipsis_0.3.2         
 [46] iterators_1.0.14        foreach_1.5.2           tools_4.5.2            
 [49] Rcpp_1.1.0              glue_1.8.0              gridExtra_2.3          
 [52] SparseArray_1.10.3      BiocBaseUtils_1.12.0    xfun_0.54              
 [55] usethis_3.2.1           shinydashboard_0.7.3    withr_3.0.2            
 [58] BiocManager_1.30.27     fastmap_1.2.0           boot_1.3-31            
 [61] shinyjs_2.1.0           digest_0.6.39           rsvd_1.0.5             
 [64] R6_2.6.1                mime_0.13               colorspace_2.1-2       
 [67] RSQLite_2.4.5           httr_1.4.7              htmlwidgets_1.6.4      
 [70] S4Arrays_1.10.0         pkgconfig_2.0.3         gtable_0.3.6           
 [73] blob_1.2.4              S7_0.2.1                XVector_0.50.0         
 [76] htmltools_0.5.8.1       ProtGenerics_1.42.0     rintrojs_0.3.4         
 [79] clue_0.3-66             scales_1.4.0            png_0.1-8              
 [82] reformulas_0.4.2        knitr_1.50              rstudioapi_0.17.1      
 [85] reshape2_1.4.5          rjson_0.2.23            nlme_3.1-168           
 [88] curl_7.0.0              nloptr_2.2.1            cachem_1.1.0           
 [91] GlobalOptions_0.1.3     stringr_1.6.0           BiocVersion_3.22.0     
 [94] parallel_4.5.2          vipor_0.4.7             AnnotationDbi_1.72.0   
 [97] desc_1.4.3              pillar_1.11.1           vctrs_0.6.5            
[100] promises_1.5.0          BiocSingular_1.26.1     beachmat_2.26.0        
[103] xtable_1.8-4            cluster_2.1.8.1         beeswarm_0.4.0         
[106] evaluate_1.0.5          cli_3.6.5               compiler_4.5.2         
[109] rlang_1.1.6             crayon_1.5.3            plyr_1.8.9             
[112] fs_1.6.6                ggbeeswarm_0.7.3        stringi_1.8.7          
[115] viridisLite_0.4.2       Biostrings_2.78.0       lazyeval_0.2.2         
[118] devtools_2.4.6          Matrix_1.7-4            ExperimentHub_3.0.0    
[121] bit64_4.6.0-1           KEGGREST_1.50.0         statmod_1.5.1          
[124] shiny_1.11.1            AnnotationHub_4.0.0     rbibutils_2.4          
[127] igraph_2.2.1            memoise_2.0.1           bit_4.6.0              </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./07-mouse_diet.html" class="pagination-link" aria-label="The mouse diet use case: a Skyline TMT DDA dataset">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav>
</div> <!-- /content -->




</body></html>