<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.23">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2&nbsp; Advanced statistical analysis with msqrob2 – Statistical analysis of mass spectrometry-based proteomics data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./03-benchmarking.html" rel="next">
<link href="./01-basics.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac32dca00065cc45e71a60b6df2d2e12.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./02-advanced.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figs/msqrob2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistical analysis of mass spectrometry-based proteomics data</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preamble</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-advanced.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-benchmarking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-workflow_optimisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-francisella.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-heart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-mouse_diet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-end.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background"><span class="header-section-number">2.1</span> Background</a>
  <ul class="collapse">
  <li><a href="#sec-tmt_workflow" id="toc-sec-tmt_workflow" class="nav-link" data-scroll-target="#sec-tmt_workflow"><span class="header-section-number">2.1.1</span> TMT workflow</a></li>
  <li><a href="#challenges" id="toc-challenges" class="nav-link" data-scroll-target="#challenges"><span class="header-section-number">2.1.2</span> Challenges</a></li>
  <li><a href="#experimental-context" id="toc-experimental-context" class="nav-link" data-scroll-target="#experimental-context"><span class="header-section-number">2.1.3</span> Experimental context</a></li>
  </ul></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages"><span class="header-section-number">2.2</span> Load packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2.3</span> Data</a>
  <ul class="collapse">
  <li><a href="#sec-caching" id="toc-sec-caching" class="nav-link" data-scroll-target="#sec-caching"><span class="header-section-number">2.3.1</span> File caching</a></li>
  <li><a href="#sec-psm_table" id="toc-sec-psm_table" class="nav-link" data-scroll-target="#sec-psm_table"><span class="header-section-number">2.3.2</span> PSM table</a></li>
  <li><a href="#sample-annotation-table" id="toc-sample-annotation-table" class="nav-link" data-scroll-target="#sample-annotation-table"><span class="header-section-number">2.3.3</span> Sample annotation table</a></li>
  <li><a href="#sec_advanced_readqf" id="toc-sec_advanced_readqf" class="nav-link" data-scroll-target="#sec_advanced_readqf"><span class="header-section-number">2.3.4</span> Convert to QFeatures</a></li>
  </ul></li>
  <li><a href="#data-preprocessing" id="toc-data-preprocessing" class="nav-link" data-scroll-target="#data-preprocessing"><span class="header-section-number">2.4</span> Data preprocessing</a>
  <ul class="collapse">
  <li><a href="#encoding-missing-values" id="toc-encoding-missing-values" class="nav-link" data-scroll-target="#encoding-missing-values"><span class="header-section-number">2.4.1</span> Encoding missing values</a></li>
  <li><a href="#log2-transformation" id="toc-log2-transformation" class="nav-link" data-scroll-target="#log2-transformation"><span class="header-section-number">2.4.2</span> Log2 transformation</a></li>
  <li><a href="#sample-filtering" id="toc-sample-filtering" class="nav-link" data-scroll-target="#sample-filtering"><span class="header-section-number">2.4.3</span> Sample filtering</a></li>
  <li><a href="#psm-filtering" id="toc-psm-filtering" class="nav-link" data-scroll-target="#psm-filtering"><span class="header-section-number">2.4.4</span> PSM filtering</a></li>
  <li><a href="#normalisation" id="toc-normalisation" class="nav-link" data-scroll-target="#normalisation"><span class="header-section-number">2.4.5</span> Normalisation</a></li>
  <li><a href="#summarisation" id="toc-summarisation" class="nav-link" data-scroll-target="#summarisation"><span class="header-section-number">2.4.6</span> Summarisation</a></li>
  </ul></li>
  <li><a href="#data-exploration" id="toc-data-exploration" class="nav-link" data-scroll-target="#data-exploration"><span class="header-section-number">2.5</span> Data exploration</a></li>
  <li><a href="#sec-run_model" id="toc-sec-run_model" class="nav-link" data-scroll-target="#sec-run_model"><span class="header-section-number">2.6</span> Data modelling</a>
  <ul class="collapse">
  <li><a href="#sec-fixed_effect" id="toc-sec-fixed_effect" class="nav-link" data-scroll-target="#sec-fixed_effect"><span class="header-section-number">2.6.1</span> Effect of treatment of interest</a></li>
  <li><a href="#sec-random_effect" id="toc-sec-random_effect" class="nav-link" data-scroll-target="#sec-random_effect"><span class="header-section-number">2.6.2</span> Effect of run</a></li>
  <li><a href="#effect-of-tmt-label" id="toc-effect-of-tmt-label" class="nav-link" data-scroll-target="#effect-of-tmt-label"><span class="header-section-number">2.6.3</span> Effect of TMT label</a></li>
  <li><a href="#sec-tmt_protein_model" id="toc-sec-tmt_protein_model" class="nav-link" data-scroll-target="#sec-tmt_protein_model"><span class="header-section-number">2.6.4</span> Effect of replication</a></li>
  <li><a href="#sec-tmt_ion_model" id="toc-sec-tmt_ion_model" class="nav-link" data-scroll-target="#sec-tmt_ion_model"><span class="header-section-number">2.6.5</span> Ion-level modelling</a></li>
  <li><a href="#sec-robust" id="toc-sec-robust" class="nav-link" data-scroll-target="#sec-robust"><span class="header-section-number">2.6.6</span> Why robust regression?</a></li>
  <li><a href="#sec-ridge" id="toc-sec-ridge" class="nav-link" data-scroll-target="#sec-ridge"><span class="header-section-number">2.6.7</span> Why ridge regression?</a></li>
  </ul></li>
  <li><a href="#statistical-inference" id="toc-statistical-inference" class="nav-link" data-scroll-target="#statistical-inference"><span class="header-section-number">2.7</span> Statistical inference</a>
  <ul class="collapse">
  <li><a href="#hypothesis-testing" id="toc-hypothesis-testing" class="nav-link" data-scroll-target="#hypothesis-testing"><span class="header-section-number">2.7.1</span> Hypothesis testing</a></li>
  <li><a href="#volcano-plots" id="toc-volcano-plots" class="nav-link" data-scroll-target="#volcano-plots"><span class="header-section-number">2.7.2</span> Volcano plots</a></li>
  <li><a href="#fold-change-distributions" id="toc-fold-change-distributions" class="nav-link" data-scroll-target="#fold-change-distributions"><span class="header-section-number">2.7.3</span> Fold change distributions</a></li>
  <li><a href="#detail-plots" id="toc-detail-plots" class="nav-link" data-scroll-target="#detail-plots"><span class="header-section-number">2.7.4</span> Detail plots</a></li>
  </ul></li>
  <li><a href="#protein-level-inference" id="toc-protein-level-inference" class="nav-link" data-scroll-target="#protein-level-inference"><span class="header-section-number">2.8</span> Protein-level inference</a></li>
  <li><a href="#dealing-with-fiterrors" id="toc-dealing-with-fiterrors" class="nav-link" data-scroll-target="#dealing-with-fiterrors"><span class="header-section-number">2.9</span> Dealing with <code id="sec-fiterror">fitErrors</code></a>
  <ul class="collapse">
  <li><a href="#sec-msqrob_refit" id="toc-sec-msqrob_refit" class="nav-link" data-scroll-target="#sec-msqrob_refit"><span class="header-section-number">2.9.1</span> Removing the random effect of sample</a></li>
  <li><a href="#manual-inspection" id="toc-manual-inspection" class="nav-link" data-scroll-target="#manual-inspection"><span class="header-section-number">2.9.2</span> Manual inspection</a></li>
  <li><a href="#imputation" id="toc-imputation" class="nav-link" data-scroll-target="#imputation"><span class="header-section-number">2.9.3</span> Imputation</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">2.10</span> Conclusion</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-advanced" class="quarto-section-identifier"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This chapter builds upon the chapter with the <a href="01-basics.html">basics</a> of differential proteomics data analysis and provides more advanced concepts using <code>msqrob2</code>. To illustrate these advanced concepts, we will use the spike-in study published by <span class="citation" data-cites="Huang2020-lu">Huang et al. (<a href="99-end.html#ref-Huang2020-lu" role="doc-biblioref">2020</a>)</span>. We chose this data set because:</p>
<ol type="1">
<li>Spike-in data contain ground truth information about which proteins are differentially abundant, enabling us to show the impact of different analysis strategies.</li>
<li>It has been acquired with a TMT-labelling strategy that require a complex experimental design. This provides an excellent example to explain different sources of variability in an MS experiment and demonstrate the flexibility of <code>msqrob2</code> to model these sources of variability.</li>
</ol>
<section id="background" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="background"><span class="header-section-number">2.1</span> Background</h2>
<p>Labelling strategies in mass spectrometry (MS)-based proteomics enhance sample throughput by enabling the acquisition of multiple samples within a single run. The labelling strategy that allows the highest multiplexing is the tandem mass tag (TMT) labelling and will be the focus of the current tutorial.</p>
<section id="sec-tmt_workflow" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" class="anchored" data-anchor-id="sec-tmt_workflow"><span class="header-section-number">2.1.1</span> TMT workflow</h3>
<p>TMT-based workflow highly overlap with <a href="01-basics.html#sec-lfq_workflow">label-free workflows</a>. However, TMT-based workflows have an additional sample preparation step, where the digested peptides from each sample are labelled with a TMT reagent and samples with different TMT reagents are pooled in a single TMT mixture<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. The signal processing is also slightly affected since the quantification no longer occurs in the MS1 space but at the MS2 level. It is important to understand that TMT reagent are isobaric, meaning that the same peptide with different TMT labels will have the same mass for the intact ion, as recorded during MS1. However, the TMT fragments that are released upon fragmentation during MS2, also called TMT reporter ions, have label-specific masses. The TMT fragments have an expected mass and are distributed in a low-mass range of the MS2 space. The intensity of each TMT fragment is directly proportional to the peptide quantity in the original sample before pooling. The TMT fragment intensities measured during MS2 are used as quantitative data. The higher mass range contains the peptide fragments that compose the peptide fingerprint, similarly to LFQ. This data range is therefore used for peptide identification. Interestingly, the peptide fingerprint originates from the same peptide across multiple samples. This leads to a signal boost for low abundant peptides and hence should improve data sensitivity and consistency.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figs/tmt_workflow.png" class="img-fluid figure-img" style="width:60.0%"></p>
<figcaption>Overview of an TMT-based proteomics workflow.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="challenges" class="level3" data-number="2.1.2">
<h3 data-number="2.1.2" class="anchored" data-anchor-id="challenges"><span class="header-section-number">2.1.2</span> Challenges</h3>
<p>The analysis of TMT-based proteomics data shares the same challenges as the <a href="01-basics.html#sec-lfq_challenges">data analysis challenges for LFQ</a>. However, TMT workflows impose additional challenges:</p>
<ul>
<li>Contemporary experiments often involve increasingly complex designs, where the number of samples exceeds the capacity of a single TMT mixture, resulting in a complex correlation structure that must be addressed for accurate statistical inference. We will describe in the modelling section the different sources of variation.</li>
<li>We also recommend modelling TMT data at the lowest level, that is at the peptide ion level, for optimal performance <span class="citation" data-cites="Vandenbulcke2025-sj">(<a href="99-end.html#ref-Vandenbulcke2025-sj" role="doc-biblioref">Vandenbulcke et al. 2025</a>)</span>. These ion-level models are more complex and include additional sources of variation instead of relying on the summarised protein values. We have shown for LFQ data that a two-step approach where data are first summarised (using a <a href="01-basics.html#sec-summarisation">model-based method</a>) and then modelled with <code>msqrob2</code> leads to similar results, and hence provides more accessible models for non-specialised data analysts <span class="citation" data-cites="Sticker2020-rl">(<a href="99-end.html#ref-Sticker2020-rl" role="doc-biblioref">Sticker et al. 2020</a>)</span>.</li>
</ul>
</section>
<section id="experimental-context" class="level3" data-number="2.1.3">
<h3 data-number="2.1.3" class="anchored" data-anchor-id="experimental-context"><span class="header-section-number">2.1.3</span> Experimental context</h3>
<p>The data set used in this chapter is a spike-in experiment (PXD0015258) published by <span class="citation" data-cites="Huang2020-lu">Huang et al. (<a href="99-end.html#ref-Huang2020-lu" role="doc-biblioref">2020</a>)</span>. It consists of controlled mixtures with known ground truth. UPS1 peptides at concentrations of 500, 333, 250, and 62.5 fmol were spiked into 50 g of SILAC HeLa peptides, each in duplicate. These concentrations form a dilution series of 1, 0.667, 0.5, and 0.125 relative to the highest UPS1 peptide amount (500 fmol). A reference sample was created by combining the diluted UPS1 peptide samples with 50g of SILAC HeLa peptides. All dilutions and the reference sample were prepared in duplicate, resulting in a total of ten samples. These samples were then treated with TMT10-plex reagents and combined before LC-MS/MS analysis. This protocol was repeated five times, each with three technical replicates, totaling 15 MS runs.</p>
<p>We will start from the PSM data generated by Skyline and infer protein-level differences between samples. To achieve this goal, we will apply an msqrob2TMT workflow, a data processing and modelling workflow dedicated to the analysis of TMT-based proteomics datasets. We will demonstrate how the workflow can highlight the spiked-in proteins. Before delving into the analysis, let us prepare our computational environment.</p>
</section>
</section>
<section id="load-packages" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="load-packages"><span class="header-section-number">2.2</span> Load packages</h2>
<p>We load the <code>msqrob2</code> package, along with additional packages for data manipulation and visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"patchwork"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also configure the <a href="01-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="data"><span class="header-section-number">2.3</span> Data</h2>
<section id="sec-caching" class="level3" data-number="2.3.1">
<h3 data-number="2.3.1" class="anchored" data-anchor-id="sec-caching"><span class="header-section-number">2.3.1</span> File caching</h3>
<p>The data have been deposited by the authors in the <code>MSV000084264</code> MASSiVE repository, but we will retrieve the time stamped data from our <a href="https://zenodo.org/records/14767905">Zenodo repository</a>. We need 2 files: the Skyline identification and quantification table generated by the authors and the sample annotation files.</p>
<p>To facilitate management of the files, we download the required files using the <code>BiocFileCache</code> package. The package will set up a local database in a cache directory<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. <code>BiocFileCache()</code> creates a connection to that database and <code>bfcrpath()</code> will query the database for the required URL. If that URL is not present in the database, the function will automatically download the URL target file and store it in the cache directory. If the URL is already present in the database, the function will retrieve the associated file from the local cache directory. This procedure ensures that the files are downloaded only once while providing a direct link to its source link. When these links point to permanent archives (like Zenodo) or large public databases (like PRIDE or MASSiVE), this approach promotes reproducible analyses.</p>
<p>The chunk below will take some time to complete the first time you run it as it needs to download the (large) file locally, but will fetch the local copy the following times.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocFileCache"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> <span class="fu">BiocFileCache</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>psmFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://zenodo.org/records/14767905/files/spikein1_psms.txt?download=1"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>annotFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://zenodo.org/records/14767905/files/spikein1_annotations.csv?download=1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now the files are downloaded, we can load the two tables.</p>
</section>
<section id="sec-psm_table" class="level3" data-number="2.3.2">
<h3 data-number="2.3.2" class="anchored" data-anchor-id="sec-psm_table"><span class="header-section-number">2.3.2</span> PSM table</h3>
<p>An MS experiment generates spectra. Each MS2 spectrum is used to infer the peptide identity using a search engine. When an observed spectrum is matched to a theoretical peptide spectrum, we have a peptide-to-spectrum match (PSM). The identification software compiles all the PSMs inside a table. Hence, the PSM data is the lowest possible level to perform data modelling.</p>
<p>Each row in the PSM data table contains information for one PSM (the table below shows the first 6 rows). The columns contains various information about the PSM, such as the peptide sequence and charge, the quantified value, the inferred protein group, the measured and predicted retention time and precursor mass, the score of the match, … In the case of Skyline TMT data, the quantification values are provides in multiple columns (start with <code>"Abundance."</code>), one for each TMT label. Regardless of TMT or LFQ experiments, the PSM table stacks the quantitative values from samples in different runs below each other. We must therefore split the table by run to ensure that every quantitative column contains data from a single sample. This is performed during the conversion to a <code>QFeatures</code> object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>psms <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(psmFile)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>qcols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Abundance"</span>, <span class="fu">colnames</span>(psms), <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 8%">
<col style="width: 1%">
<col style="width: 8%">
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 4%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Checked</th>
<th style="text-align: left;">Confidence</th>
<th style="text-align: left;">Identifying.Node</th>
<th style="text-align: left;">PSM.Ambiguity</th>
<th style="text-align: left;">Annotated.Sequence</th>
<th style="text-align: left;">Modifications</th>
<th style="text-align: left;">Marked.as</th>
<th style="text-align: right;">X..Protein.Groups</th>
<th style="text-align: right;">X..Proteins</th>
<th style="text-align: left;">Master.Protein.Accessions</th>
<th style="text-align: left;">Master.Protein.Descriptions</th>
<th style="text-align: left;">Protein.Accessions</th>
<th style="text-align: left;">Protein.Descriptions</th>
<th style="text-align: right;">X..Missed.Cleavages</th>
<th style="text-align: right;">Charge</th>
<th style="text-align: right;">DeltaScore</th>
<th style="text-align: right;">DeltaCn</th>
<th style="text-align: right;">Rank</th>
<th style="text-align: right;">Search.Engine.Rank</th>
<th style="text-align: right;">m.z..Da.</th>
<th style="text-align: right;">MH…Da.</th>
<th style="text-align: right;">Theo..MH…Da.</th>
<th style="text-align: right;">DeltaM..ppm.</th>
<th style="text-align: right;">Deltam.z..Da.</th>
<th style="text-align: left;">Activation.Type</th>
<th style="text-align: left;">MS.Order</th>
<th style="text-align: right;">Isolation.Interference….</th>
<th style="text-align: right;">Average.Reporter.S.N</th>
<th style="text-align: right;">Ion.Inject.Time..ms.</th>
<th style="text-align: right;">RT..min.</th>
<th style="text-align: right;">First.Scan</th>
<th style="text-align: left;">Spectrum.File</th>
<th style="text-align: left;">File.ID</th>
<th style="text-align: right;">Abundance..126</th>
<th style="text-align: right;">Abundance..127N</th>
<th style="text-align: right;">Abundance..127C</th>
<th style="text-align: right;">Abundance..128N</th>
<th style="text-align: right;">Abundance..128C</th>
<th style="text-align: right;">Abundance..129N</th>
<th style="text-align: right;">Abundance..129C</th>
<th style="text-align: right;">Abundance..130N</th>
<th style="text-align: right;">Abundance..130C</th>
<th style="text-align: right;">Abundance..131</th>
<th style="text-align: left;">Quan.Info</th>
<th style="text-align: right;">Ions.Score</th>
<th style="text-align: right;">Identity.Strict</th>
<th style="text-align: right;">Identity.Relaxed</th>
<th style="text-align: right;">Expectation.Value</th>
<th style="text-align: right;">Percolator.q.Value</th>
<th style="text-align: right;">Percolator.PEP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (O4)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[K].gFQQILAGEYDHLPEQAFYMVGPIEEAVAk.[A]</td>
<td style="text-align: left;">N-Term(TMT6plex); K30(TMT6plex)</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">P06576</td>
<td style="text-align: left;">ATP synthase subunit beta, mitochondrial OS=Homo sapiens GN=ATP5B PE=1 SV=3</td>
<td style="text-align: left;">P06576</td>
<td style="text-align: left;">ATP synthase subunit beta, mitochondrial OS=Homo sapiens GN=ATP5B PE=1 SV=3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1270.3249</td>
<td style="text-align: right;">3808.960</td>
<td style="text-align: right;">3808.966</td>
<td style="text-align: right;">-1.51</td>
<td style="text-align: right;">-0.00192</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">47.955590</td>
<td style="text-align: right;">8.7</td>
<td style="text-align: right;">50.000</td>
<td style="text-align: right;">212.2487</td>
<td style="text-align: right;">112815</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_03.raw</td>
<td style="text-align: left;">F1</td>
<td style="text-align: right;">2548.326</td>
<td style="text-align: right;">3231.929</td>
<td style="text-align: right;">2760.839</td>
<td style="text-align: right;">4111.639</td>
<td style="text-align: right;">3127.254</td>
<td style="text-align: right;">1874.163</td>
<td style="text-align: right;">2831.423</td>
<td style="text-align: right;">2298.401</td>
<td style="text-align: right;">3798.876</td>
<td style="text-align: right;">3739.067</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">90</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000140</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (K2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[R].qYPWGVAEVENGEHcDFTILr.[N]</td>
<td style="text-align: left;">N-Term(TMT6plex); C15(Carbamidomethyl); R21(Label:13C(6)15N(4))</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Q16181</td>
<td style="text-align: left;">Septin-7 OS=Homo sapiens GN=SEPT7 PE=1 SV=2</td>
<td style="text-align: left;">Q16181</td>
<td style="text-align: left;">Septin-7 OS=Homo sapiens GN=SEPT7 PE=1 SV=2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">920.4493</td>
<td style="text-align: right;">2759.333</td>
<td style="text-align: right;">2759.332</td>
<td style="text-align: right;">0.31</td>
<td style="text-align: right;">0.00028</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">9.377507</td>
<td style="text-align: right;">8.1</td>
<td style="text-align: right;">3.242</td>
<td style="text-align: right;">164.7507</td>
<td style="text-align: right;">87392</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture3_03.raw</td>
<td style="text-align: left;">F5</td>
<td style="text-align: right;">22861.765</td>
<td style="text-align: right;">25817.946</td>
<td style="text-align: right;">23349.498</td>
<td style="text-align: right;">29449.609</td>
<td style="text-align: right;">25995.929</td>
<td style="text-align: right;">22955.769</td>
<td style="text-align: right;">30578.971</td>
<td style="text-align: right;">30660.488</td>
<td style="text-align: right;">38728.853</td>
<td style="text-align: right;">25047.280</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">76</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">0.0000001</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000003</td>
</tr>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (K2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[R].dkPSVEPVEEYDYEDLk.[E]</td>
<td style="text-align: left;">N-Term(TMT6plex); K2(Label); K17(Label)</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Q9Y450</td>
<td style="text-align: left;">HBS1-like protein OS=Homo sapiens GN=HBS1L PE=1 SV=1</td>
<td style="text-align: left;">Q9Y450</td>
<td style="text-align: left;">HBS1-like protein OS=Homo sapiens GN=HBS1L PE=1 SV=1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.9730</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">920.1605</td>
<td style="text-align: right;">2758.467</td>
<td style="text-align: right;">2758.461</td>
<td style="text-align: right;">2.08</td>
<td style="text-align: right;">0.00192</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">38.317050</td>
<td style="text-align: right;">17.8</td>
<td style="text-align: right;">13.596</td>
<td style="text-align: right;">143.4534</td>
<td style="text-align: right;">74786</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture3_03.raw</td>
<td style="text-align: left;">F5</td>
<td style="text-align: right;">25504.083</td>
<td style="text-align: right;">27740.450</td>
<td style="text-align: right;">25144.974</td>
<td style="text-align: right;">25754.579</td>
<td style="text-align: right;">29923.176</td>
<td style="text-align: right;">34097.637</td>
<td style="text-align: right;">31650.255</td>
<td style="text-align: right;">27632.692</td>
<td style="text-align: right;">23886.881</td>
<td style="text-align: right;">35331.092</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0.0000004</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000010</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (F2)</td>
<td style="text-align: left;">Selected</td>
<td style="text-align: left;">[R].hEHQVMLmr.[Q]</td>
<td style="text-align: left;">N-Term(TMT6plex); M8(Oxidation); R9(Label:13C(6)15N(4))</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Q15233</td>
<td style="text-align: left;">Non-POU domain-containing octamer-binding protein OS=Homo sapiens GN=NONO PE=1 SV=4</td>
<td style="text-align: left;">Q15233</td>
<td style="text-align: left;">Non-POU domain-containing octamer-binding protein OS=Homo sapiens GN=NONO PE=1 SV=4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.5250</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">359.6898</td>
<td style="text-align: right;">1435.737</td>
<td style="text-align: right;">1435.738</td>
<td style="text-align: right;">-0.04</td>
<td style="text-align: right;">-0.00002</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">21.390040</td>
<td style="text-align: right;">36.5</td>
<td style="text-align: right;">50.000</td>
<td style="text-align: right;">21.6426</td>
<td style="text-align: right;">6458</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture4_02.raw</td>
<td style="text-align: left;">F10</td>
<td style="text-align: right;">13493.228</td>
<td style="text-align: right;">14674.490</td>
<td style="text-align: right;">11187.900</td>
<td style="text-align: right;">12831.495</td>
<td style="text-align: right;">13839.426</td>
<td style="text-align: right;">12441.353</td>
<td style="text-align: right;">13450.885</td>
<td style="text-align: right;">14777.844</td>
<td style="text-align: right;">13039.995</td>
<td style="text-align: right;">12057.121</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">0.0003351</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0001175</td>
</tr>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (K2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[R].dNLTLWTADNAGEEGGEAPQEPQS.[-]</td>
<td style="text-align: left;">N-Term(TMT6plex)</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">P31947</td>
<td style="text-align: left;">14-3-3 protein sigma OS=Homo sapiens GN=SFN PE=1 SV=1</td>
<td style="text-align: left;">P31947</td>
<td style="text-align: left;">14-3-3 protein sigma OS=Homo sapiens GN=SFN PE=1 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1.0000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">920.0943</td>
<td style="text-align: right;">2758.268</td>
<td style="text-align: right;">2758.264</td>
<td style="text-align: right;">1.53</td>
<td style="text-align: right;">0.00141</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: right;">16.7</td>
<td style="text-align: right;">6.723</td>
<td style="text-align: right;">174.1863</td>
<td style="text-align: right;">92950</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture3_03.raw</td>
<td style="text-align: left;">F5</td>
<td style="text-align: right;">64582.786</td>
<td style="text-align: right;">50576.417</td>
<td style="text-align: right;">47126.037</td>
<td style="text-align: right;">56285.129</td>
<td style="text-align: right;">46257.310</td>
<td style="text-align: right;">52634.885</td>
<td style="text-align: right;">49716.850</td>
<td style="text-align: right;">60660.574</td>
<td style="text-align: right;">55830.488</td>
<td style="text-align: right;">40280.577</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">38</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">0.0002153</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000138</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (K2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[R].aLVAIGTHDLDTLSGPFTYTAk.[R]</td>
<td style="text-align: left;">N-Term(TMT6plex); K22(Label)</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">Q9NSD9</td>
<td style="text-align: left;">Phenylalanine–tRNA ligase beta subunit OS=Homo sapiens GN=FARSB PE=1 SV=3</td>
<td style="text-align: left;">Q9NSD9</td>
<td style="text-align: left;">Phenylalanine–tRNA ligase beta subunit OS=Homo sapiens GN=FARSB PE=1 SV=3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.9783</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">919.8502</td>
<td style="text-align: right;">2757.536</td>
<td style="text-align: right;">2757.532</td>
<td style="text-align: right;">1.48</td>
<td style="text-align: right;">0.00136</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">30.619960</td>
<td style="text-align: right;">26.7</td>
<td style="text-align: right;">8.958</td>
<td style="text-align: right;">176.4863</td>
<td style="text-align: right;">94294</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture3_03.raw</td>
<td style="text-align: left;">F5</td>
<td style="text-align: right;">35404.709</td>
<td style="text-align: right;">31905.852</td>
<td style="text-align: right;">30993.941</td>
<td style="text-align: right;">36854.351</td>
<td style="text-align: right;">37506.001</td>
<td style="text-align: right;">25703.444</td>
<td style="text-align: right;">38626.598</td>
<td style="text-align: right;">35447.942</td>
<td style="text-align: right;">33788.409</td>
<td style="text-align: right;">32031.516</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">46</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">0.0002060</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000720</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>There is a peculiarity with the dataset: the spectra have been identified with 2 nodes. In one node, the authors searched the SwissProt database for proteins with static modifications related to the metabolic labelling, in the other node they searched the Sigma_UPS protein database without these static modifications. However, some spectra were identified by both nodes leading to duplicate PSMs. We here remove these duplicated PSMs that are identification artefacts.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>duplicatesQuants <span class="ot">&lt;-</span> <span class="fu">duplicated</span>(psms[, qcols]) <span class="sc">|</span> <span class="fu">duplicated</span>(psms[, qcols], <span class="at">fromLast =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>psms <span class="ot">&lt;-</span> psms[<span class="sc">!</span>duplicatesQuants, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will also subset the data set to reduce computational costs. If you want to run the analysis on the full data set, you can skip this code chunk. The subsetting will keep all UPS proteins, known to be differentially abundant by experimental design, and we will keep 500 background proteins known to be unchanged across condition.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>allProteins <span class="ot">&lt;-</span> <span class="fu">unique</span>(psms<span class="sc">$</span>Protein.Accessions)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>upsProteins <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"ups"</span>, allProteins, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>helaProteins <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"ups"</span>, allProteins, <span class="at">value =</span> <span class="cn">TRUE</span>, <span class="at">invert =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>keepProteins <span class="ot">&lt;-</span> <span class="fu">c</span>(upsProteins, <span class="fu">sample</span>(helaProteins, <span class="dv">500</span>))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>psms <span class="ot">&lt;-</span> psms[psms<span class="sc">$</span>Protein.Accessions <span class="sc">%in%</span> keepProteins, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sample-annotation-table" class="level3" data-number="2.3.3">
<h3 data-number="2.3.3" class="anchored" data-anchor-id="sample-annotation-table"><span class="header-section-number">2.3.3</span> Sample annotation table</h3>
<p>The purpose and structure of the sample annotation table is identical across proteomics experiments (see introduction to the <a href="01-basics.html#sec-annotation_table">sample annotation table</a>). The annotation table used in this tutorial has been generated by the authors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(annotFile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We perform a little cleanup:</p>
<ol type="1">
<li>We keep only the sample annotations that are meaningful to the experiment and that are not redundant with other annotations.</li>
<li>We extract the run identifier from the MS file name (which we store as the <code>File.Name</code> annotation).</li>
<li>The TMT used for labelling each sample is stored in the <code>Channel</code> column. We however prefer to use the less esoteric term <code>Label</code> for more clarity with the main text when we’ll discuss labelling effects.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 1.</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> coldata[, <span class="fu">c</span>(<span class="st">"Run"</span>, <span class="st">"Channel"</span>, <span class="st">"Condition"</span>, <span class="st">"Mixture"</span>, <span class="st">"TechRepMixture"</span>)]</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 2.</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>File.Name <span class="ot">&lt;-</span> coldata<span class="sc">$</span>Run</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>Run <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^.*(Mix.*).raw"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, coldata<span class="sc">$</span>Run)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="do">## 3.</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(coldata)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"Label"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 12%">
<col style="width: 6%">
<col style="width: 10%">
<col style="width: 9%">
<col style="width: 15%">
<col style="width: 46%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Run</th>
<th style="text-align: left;">Label</th>
<th style="text-align: left;">Condition</th>
<th style="text-align: left;">Mixture</th>
<th style="text-align: right;">TechRepMixture</th>
<th style="text-align: left;">File.Name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">126</td>
<td style="text-align: left;">Norm</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">127N</td>
<td style="text-align: left;">0.667</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">127C</td>
<td style="text-align: left;">0.125</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">128N</td>
<td style="text-align: left;">0.5</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">128C</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mixture1_01</td>
<td style="text-align: left;">129N</td>
<td style="text-align: left;">0.125</td>
<td style="text-align: left;">Mixture1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="sec_advanced_readqf" class="level3" data-number="2.3.4">
<h3 data-number="2.3.4" class="anchored" data-anchor-id="sec_advanced_readqf"><span class="header-section-number">2.3.4</span> Convert to QFeatures</h3>
<p>We use <code>readQFeatures()</code> to create a <a href="01-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>. Since we start from the PSM-level data, the approach is somewhat more elaborate<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>. First, recall that every quantitative column in the PSM table contains information for multiple runs. Therefore, the function split the table based on the run identifier, given by the <code>runCol</code> argument (for Skyline, that identifier is contained in <code>Spectrum.File</code>). So, the <code>QFeatures</code> object after import will contain as many sets as there are runs. Next, the function links the annotation table with the PSM data. To achieve this, the annotation table must contain a <code>runCol</code> column that provides the run identifier in which each sample has been acquired, and this information will be used to match the identifiers in the <code>Spectrum.File</code> column of the PSM table. The annotation table must also contain a <code>quantCols</code> column that tells the function which column in the PSM table contains the quantitative information for a given sample. In this case, the <code>quantCols</code> depend on</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>runCol <span class="ot">&lt;-</span> coldata<span class="sc">$</span>File.Name</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>quantCols <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Abundance.."</span>, coldata<span class="sc">$</span>Label)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  psms, <span class="at">colData =</span> coldata,</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">runCol =</span> <span class="st">"Spectrum.File"</span>,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">quantCols =</span> qcols</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 15 sets:

 [1] 161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw: SummarizedExperiment with 1905 rows and 10 columns 
 [2] 161117_SILAC_HeLa_UPS1_TMT10_Mixture1_02.raw: SummarizedExperiment with 1902 rows and 10 columns 
 [3] 161117_SILAC_HeLa_UPS1_TMT10_Mixture1_03.raw: SummarizedExperiment with 1952 rows and 10 columns 
 ...
 [13] 161117_SILAC_HeLa_UPS1_TMT10_Mixture5_01.raw: SummarizedExperiment with 1919 rows and 10 columns 
 [14] 161117_SILAC_HeLa_UPS1_TMT10_Mixture5_02.raw: SummarizedExperiment with 1909 rows and 10 columns 
 [15] 161117_SILAC_HeLa_UPS1_TMT10_Mixture5_03.raw: SummarizedExperiment with 1844 rows and 10 columns </code></pre>
</div>
</div>
<p>We now have a <code>QFeatures</code> object with 15 sets, each containing data associated with an MS run. The name of each set is defined by the name of the corresponding file name of the run, which is unnecessarily long. We simplify the set names, although this step is optional and only meant to improve the clarity of the output.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This is optional</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(spikein) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^.*(Mix.*).raw"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">names</span>(spikein))</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>(inputNames <span class="ot">&lt;-</span> <span class="fu">names</span>(spikein))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Mixture1_01" "Mixture1_02" "Mixture1_03" "Mixture2_01" "Mixture2_02"
 [6] "Mixture2_03" "Mixture3_01" "Mixture3_02" "Mixture3_03" "Mixture4_01"
[11] "Mixture4_02" "Mixture4_03" "Mixture5_01" "Mixture5_02" "Mixture5_03"</code></pre>
</div>
</div>
</section>
</section>
<section id="data-preprocessing" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="data-preprocessing"><span class="header-section-number">2.4</span> Data preprocessing</h2>
<p>Similar to the <a href="01-basics.html#sec-basic_preprocess">basic concepts chapter</a>, we will use the <code>QFeatures</code>’ data preprocessing functionality. The data preprocessing workflow for TMT data is similar to the workflow for LFQ data, but there are subtle differences associated with the fact that we start from PSM-level data, namely the PSM filtering is mode complex and the data preprocessing will be applied for each run separately to remove part of the run effect.</p>
<section id="encoding-missing-values" class="level3" data-number="2.4.1">
<h3 data-number="2.4.1" class="anchored" data-anchor-id="encoding-missing-values"><span class="header-section-number">2.4.1</span> Encoding missing values</h3>
<p>Any zero value needs to be <a href="01-basics.html#sec-encode_missing">encoded by a missing value</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(spikein, inputNames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="log2-transformation" class="level3" data-number="2.4.2">
<h3 data-number="2.4.2" class="anchored" data-anchor-id="log2-transformation"><span class="header-section-number">2.4.2</span> Log2 transformation</h3>
<p>Similar to any MS-based proteomics data, TMT data are heteroskedastic, with a strong mean-variance relationship. This is illustrated by the intensities and log2-intensities for one of the peptide ions, the triply charged DLLHVLAFSK, in function of the UPS spike-in dilution factor.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_log_plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Log2-transformation solves the heteroskedasticity issue, but also provides a scale that directly relates to biological interpretation (see the <a href="01-basics.html#sec-log2">basic concepts chapter</a>). We perform log2-transformation with <code>logTransform()</code> from the <code>QFeatures</code> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>logNames  <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputNames, <span class="st">"_log"</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    spikein, inputNames, <span class="at">name =</span> logNames, <span class="at">base =</span> <span class="dv">2</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sample-filtering" class="level3" data-number="2.4.3">
<h3 data-number="2.4.3" class="anchored" data-anchor-id="sample-filtering"><span class="header-section-number">2.4.3</span> Sample filtering</h3>
<p>We first remove the reference samples. These samples were used by the MSstatsTMT authors to obtain normalisation factors <span class="citation" data-cites="Huang2020-lu">(<a href="99-end.html#ref-Huang2020-lu" role="doc-biblioref">Huang et al. 2020</a>)</span>. However, this approach ignores the uncertainty associated with the measurement with these reference samples, potentially inflating the noise in the samples of interest. Hence, <code>msqrob2</code> workflows do not use reference normalisation. In practice, we found no impact on model performance <span class="citation" data-cites="Vandenbulcke2025-sj">(<a href="99-end.html#ref-Vandenbulcke2025-sj" role="doc-biblioref">Vandenbulcke et al. 2025</a>)</span>, hence we favor a more parsimonious approach. The information is available from the <code>colData</code>, under the <code>Condition</code> column. We remove any sample that is marked as <code>Norm</code> using <code>subsetByColData()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">subsetByColData</span>(spikein, spikein<span class="sc">$</span>Condition <span class="sc">!=</span> <span class="st">"Norm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="psm-filtering" class="level3" data-number="2.4.4">
<h3 data-number="2.4.4" class="anchored" data-anchor-id="psm-filtering"><span class="header-section-number">2.4.4</span> PSM filtering</h3>
<p>Filtering removes low-quality and unreliable PSMs that would otherwise introduce noise and artefacts in the data. Conceptually, PSM filtering is identical to <a href="01-basics.html#sec-filter">peptide filtering</a>, but we will here apply filtering criteria for which some are not readily available in the data. Therefore, we will add custom filtering variable to the <code>rowData</code> that will then be used with <code>filterFeatures()</code>. This provides an ideal use case to demonstrate the customisation of a filtering workflow.</p>
<section id="remove-ambiguous-identifications" class="level4">
<h4 class="anchored" data-anchor-id="remove-ambiguous-identifications">Remove ambiguous identifications</h4>
<p>The background proteins originate from HeLa cells, which also contain UPS proteins. The background UPS proteins and the spiked-in UPS proteins differ in metabolic labelling, so we should be able to distinguish them. We used the PSM-level data searched with mascot, as provided by the MSstatsTMT authors who used two mascot identification nodes. In one node they searched the SwissProt database for proteins with static modifications related to the metabolic labelling, in the other node they searched the Sigma_UPS protein database without these static modifications. Ideally, this should separate the spiked-in UPS proteins and the UPS proteins from the HeLa cells, however, this is not always the case. The SwissProt search is expected to return peptide-spectrum matches (PSMs) for all proteins, including non-UPS HeLa, UPS HeLa, and spike-in UPS proteins. Conversely, the Sigma_UPS search is expected to return PSMs exclusively for spike-in UPS proteins. However, a PSM that matches a UPS protein in the SwissProt search but is not identified as such in the Sigma_UPS search could either correctly originate from a HeLa protein or represent a spiked-in UPS protein that was not recognised as such in the Sigma_UPS search. Additionally, there are ambiguous PSMs that are not matched to a UPS protein in the HeLa search but are matched to a UPS protein in the SwissProt search. To address this, we exclude these ambiguous proteins from the analysis.</p>
<p>To define the amibiguous PSMs, we retrieve the PSM annotations from the <code>rowData</code> and create a new colum indicating whether a PSM belongs to a UPS protein or not, based on the protein SwissProt identifiers. For this, we apply a custom filtering worklow:</p>
<ol type="1">
<li><strong>Collect data</strong>: combine all the <code>rowData</code> information in a single table. We will apply the filter on the</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>rowdata <span class="ot">&lt;-</span> <span class="fu">rbindRowData</span>(spikein, logNames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li><strong>Compute new variable</strong>: (2a) define whether the PSM’s protein group is a UPS protein and then (2b) define an ambiguous PSM as a PSM that is marked as UPS by the SwissProt identifier but not by the Sigma_UPS node (<code>Marked.as</code> column), and inversely.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 2a.</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>rowdata<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="st">"no"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>isUpsProtein <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, rowdata<span class="sc">$</span>Protein.Accessions)</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>rowdata<span class="sc">$</span>isUps[isUpsProtein] <span class="ot">&lt;-</span> <span class="st">"yes"</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="do">## 2b.</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>rowdata<span class="sc">$</span>isUps[<span class="sc">!</span>isUpsProtein <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"UPS"</span>, rowdata<span class="sc">$</span>Marked.as)] <span class="ot">&lt;-</span> <span class="st">"amb"</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>rowdata<span class="sc">$</span>isUps[isUpsProtein <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"UPS"</span>, rowdata<span class="sc">$</span>Marked.as)] <span class="ot">&lt;-</span> <span class="st">"amb"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="3" type="1">
<li><strong>Reinsert in the rowData</strong>: insert the modified table with new information back in the <code>rowData</code> of the different sets. This means that the single table with <code>rowData</code> information needs to be split by each set. <code>split()</code> will produce a named list of tables and each table will be iteratively inserted as <code>rowData</code> of the set.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein) <span class="ot">&lt;-</span> <span class="fu">split</span>(rowdata, rowdata<span class="sc">$</span>assay)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="4" type="1">
<li><strong>Apply the filter</strong>: the filtering is performed by <code>filterFeatures()</code> using the new information from the <code>rowData</code>. We specify <code>keep = TRUE</code> because the input sets (before log-transformation) do not contain the filtering variable, so we tell the function to keep all PSMs for the sets that don’t have the variable <code>isUps</code>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(spikein, <span class="sc">~</span> isUps <span class="sc">!=</span> <span class="st">"amb"</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-failed-protein-inference" class="level4">
<h4 class="anchored" data-anchor-id="remove-failed-protein-inference">Remove failed protein inference</h4>
<p>Next, we remove PSMs that could not be mapped to a protein or that map to multiple proteins, i.e.&nbsp;a protein group. For the latter, the protein identifier contains multiple identifiers separated by a <code>;</code>). This information is readily available in the <code>rowData</code>, so there is no need for a custom filtering.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    spikein, <span class="sc">~</span> Protein.Accessions <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Protein.Accessions)) <span class="do">## Remove protein groups</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-inconsistent-protein-inference" class="level4">
<h4 class="anchored" data-anchor-id="remove-inconsistent-protein-inference">Remove inconsistent protein inference</h4>
<p>We also remove peptide ions that map to a different protein depending on the run. Again, this requires a custom filtering and we apply the same filtering workflow as above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 1. Collect data</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>rowdata <span class="ot">&lt;-</span> <span class="fu">rbindRowData</span>(spikein, logNames)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 2. Compute new variable</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Annotated.Sequence, Charge) <span class="sc">|&gt;</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">nProtsMapped =</span> <span class="fu">length</span>(<span class="fu">unique</span>(Protein.Accessions)))</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="do">## 3. Reinsert in the rowData</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein) <span class="ot">&lt;-</span> <span class="fu">split</span>(rowdata, rowdata<span class="sc">$</span>assay)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 4. Apply the filter</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(spikein, <span class="sc">~</span> nProtsMapped <span class="sc">==</span> <span class="dv">1</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-one-run-wonders" class="level4">
<h4 class="anchored" data-anchor-id="remove-one-run-wonders">Remove one-run wonders</h4>
<p>We also remove proteins that can only be found in one run as such proteins may not be trustworthy. In this case,</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 1. Collect data</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>rowdata <span class="ot">&lt;-</span> <span class="fu">rbindRowData</span>(spikein, logNames)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 2. Compute new variable</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Protein.Accessions) <span class="sc">|&gt;</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">nRuns =</span> <span class="fu">length</span>(<span class="fu">unique</span>(assay)))</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="do">## 3. Reinsert in the rowData</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein) <span class="ot">&lt;-</span> <span class="fu">split</span>(rowdata, rowdata<span class="sc">$</span>assay)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="do">## 4. Apply the filter</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(spikein, <span class="sc">~</span> nRuns <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-duplicated_psms" class="level4">
<h4 class="anchored" data-anchor-id="sec-duplicated_psms">Remove duplicated PSMs</h4>
<p>Finally, peptide ions that were identified with multiple PSMs in a run are collapsed to the PSM with the highest summed intensity over the TMT labels, a strategy that is also used by MSstats.</p>
<p>This filtering requires a more complex workflow because it mixes information from the <code>rowData</code> (to obtain ion identities) with quantitative data (to obtain PSM intensity ranks). We therefore compute the filtering variable for every set iteratively:</p>
<ol type="1">
<li>Get the <code>rowData</code> for the current set.</li>
<li>Make a new variable <code>ionID</code>.</li>
<li>We calculate the <code>rowSums</code> for each ion.</li>
<li>Make a new variable <code>psmRank</code> that ranks the PSMs for each ion identifier based on the summed intensity.</li>
<li>We store the new information back in the <code>rowData</code>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> logNames) { <span class="do">## for each set of interest</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    rowdata <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[i]]) <span class="do">## 1.</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    rowdata<span class="sc">$</span>ionID <span class="ot">&lt;-</span> <span class="fu">paste0</span>(rowdata<span class="sc">$</span>Annotated.Sequence, rowdata<span class="sc">$</span>Charge) <span class="do">## 2.</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    rowdata<span class="sc">$</span>rowSums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(spikein[[i]]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="do">## 3.</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>    rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(ionID) <span class="sc">|&gt;</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">psmRank =</span> <span class="fu">rank</span>(<span class="sc">-</span>rowSums)) <span class="do">## 4.</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowData</span>(spikein[[i]]) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(rowdata) <span class="do">## 5.</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For each ion that maps to multiple PSMs, we keep the PSM with the highest summed intensity, that is that ranks first.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(spikein, <span class="sc">~</span> psmRank <span class="sc">==</span> <span class="dv">1</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-highly-missing-psms" class="level4">
<h4 class="anchored" data-anchor-id="remove-highly-missing-psms">Remove highly missing PSMs</h4>
<p>We then remove PSMs with five or more missing values out of the ten TMT labels (&gt;= 50%). This is an arbitrary value that may need to be adjusted depending on the experiment and the data set.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(spikein, logNames, <span class="at">pNA =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filtering-wrap-up" class="level4">
<h4 class="anchored" data-anchor-id="filtering-wrap-up">Filtering wrap-up</h4>
<p>We have demonstrated the different procedures to perform feature filtering. Here is a summary (from simple to complex):</p>
<ol type="1">
<li>If you want to filter on missing values, use <code>filterNA()</code>.</li>
<li>If you want to filter based on a <code>rowData</code> column, use <code>filterFeatures()</code>.</li>
<li>If you want to filter based on information that needs to be built from <code>rowData</code> information, use the following workflow: i. collect the <code>rowData</code> in a table; ii. compute the new variable; iii. reinsert the updated table in the <code>rowData</code>; iv. apply the filter with <code>filterFeatures()</code>.</li>
<li>If you want to filter based on <code>rowData</code> and quantitative information, iterate the following workflow over each set:
<ol type="i">
<li>Get the <code>rowData</code> for the current set; ii. Compute the filtering variable based on <code>rowData</code> and/or quantitative information; iii. store the new information back in the <code>rowData</code>. Then, the filtering can be performed by <code>filterFeatures()</code>.</li>
</ol></li>
</ol>
<p>When using <code>filterFeatures()</code>, specify <code>keep = TRUE</code> to select all features for which a custom variable is not available or has not been computed. By default, the function will remove all the feature of a set for which the information is not available.</p>
<p>These standard and custom filtering procedures have been demonstrated on PSM-level data, but the same procedures can be performed at any data level, e.g.&nbsp;also at peptide or protein level.</p>
</section>
</section>
<section id="normalisation" class="level3" data-number="2.4.5">
<h3 data-number="2.4.5" class="anchored" data-anchor-id="normalisation"><span class="header-section-number">2.4.5</span> Normalisation</h3>
<p>Before performing normalisation, we explore the systematic shifts across the samples (using the pipeline described in the <a href="01-basics.html#sec-norm">previous chapter</a>). To facilitate interpretation, we facet the data by TMT mixture.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>spikein[, , logNames] <span class="sc">|&gt;</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">c</span>(<span class="st">"Mixture"</span>, <span class="st">"TechRepMixture"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">colour =</span> <span class="fu">as.factor</span>(TechRepMixture), <span class="at">group =</span> colname) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Intensity distribution for each observational unit"</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">subtitle =</span> <span class="st">"Before normalisation"</span>,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">colour =</span> <span class="st">"Technical replicate"</span>) <span class="sc">+</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(Mixture <span class="sc">~</span> .) <span class="sc">+</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_before_norm-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p>Similarly to the previous chapter, we again observe misalignments of the intensity distributions across samples. We see the intensity distributions cluster by technical replicate. Since each replicate contains all experimental conditions, we know that these difference stem from technical variability and not biological variability. We normalise the data by median centering.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>normNames  <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputNames, <span class="st">"_norm"</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">normalize</span>(</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    spikein, logNames, <span class="at">name =</span> normNames,</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"center.median"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>TODO</strong> think about using the Median of Ratio normalisation for tmt data.</p>
<p>And we confirm that the normalisation resulted in a better alignment of the intensity distribution across samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>spikein[, , normNames] <span class="sc">|&gt;</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">c</span>(<span class="st">"Mixture"</span>, <span class="st">"TechRepMixture"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">colour =</span> <span class="fu">as.factor</span>(TechRepMixture), <span class="at">group =</span> colname) <span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Intensity distribution for each observational unit"</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">subtitle =</span> <span class="st">"Before normalisation"</span>,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">colour =</span> <span class="st">"Technical replicate"</span>) <span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(Mixture <span class="sc">~</span> .) <span class="sc">+</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_after_norm-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p>Up to now, the data from different runs were kept in separate assays. We can now join the normalised sets into an <code>ions</code> set using <code>joinAssays()</code>. Sets are joined by stacking the columns (samples) in a matrix and rows (features) are matched according to a row identifier, here the <code>ionID</code> from the <code>rowData</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>(</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>    spikein, normNames, <span class="at">fcol =</span> <span class="st">"ionID"</span>, <span class="at">name =</span> <span class="st">"ions"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 46 sets:

 [1] Mixture1_01: SummarizedExperiment with 1719 rows and 8 columns 
 [2] Mixture1_02: SummarizedExperiment with 1722 rows and 8 columns 
 [3] Mixture1_03: SummarizedExperiment with 1776 rows and 8 columns 
 ...
 [44] Mixture5_02_norm: SummarizedExperiment with 1646 rows and 8 columns 
 [45] Mixture5_03_norm: SummarizedExperiment with 1578 rows and 8 columns 
 [46] ions: SummarizedExperiment with 4066 rows and 120 columns </code></pre>
</div>
</div>
<p>We have a new set contain 15 runs <span class="math inline">\(\times\)</span> 8 labelled sample = 120 data columns. Note that the 15 sets have 4066 ions in common, leading to a joined set with <code>r nrows(spikein)[["ions"]]</code> rows.</p>
<p>If we want to use protein-level data for modelling, we will need a summarisation step. Note that this last step is optional.</p>
</section>
<section id="summarisation" class="level3" data-number="2.4.6">
<h3 data-number="2.4.6" class="anchored" data-anchor-id="summarisation"><span class="header-section-number">2.4.6</span> Summarisation</h3>
<p>While this chapter will focus on ion-level data modelling, modelling of protein-level data is possible upon summarisation. Below, we illustrate the challenges of <a href="01-basics.html#sec-summarisation">summarising</a> TMT data using one of the UPS proteins in Mixture 1 (separating the data for each technical replicate). We also focus on the 0.125x and the 1x spike-in conditions. We illustrate the different peptide ions on the x axis and plot the log2 normalised intensities across samples on y axis. All the points belonging to the same sample are linked through a grey line.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_summarise-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>We see that the same challenges observed for <a href="01-basics.html#sec-summarisation">LFQ data</a> also apply to TMT data. Briefly:</p>
<ol type="1">
<li>Data for a protein can consist of many peptide ions.</li>
<li>Peptide ions have different intensity baselines.</li>
<li>There is strong missingness across runs (compare points between replicates), but the missingness is mitigated within runs (compare points within replicates<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>).</li>
<li>Subtle intensity shifts for the same peptide across different replicates, called spectrum effects, are caused by small run-to-run fluctuations.</li>
<li>Presence of outliers. For instance, the first peptide ion doesn’t show the same change in intensity between conditions compared to majority of the peptides.</li>
</ol>
<p><strong>TODO</strong>: use robust summary instead of median polish for consistency with previous vignette? I known median polish is faster, but still?</p>
<p>Here, we summarise the ion-level data into protein intensities through the median polish approach, which alternately removes the peptide-ions and the sample medians from the data until the summaries stabilise. Removing the peptide-ion medians will solve issue 2. as it removes the ion-specific effects. Using the median instead of the mean will solve issue 5. Note that we perform summarisation for each run separately, hence the ion effect will be different for each run, effectively allowing for a spectrum effect and solving issue 4.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>summNames <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputNames, <span class="st">"_proteins"</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    spikein, <span class="at">i =</span> normNames,  <span class="at">name =</span> summNames,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>, <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>medianPolish,</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 61 sets:

 [1] Mixture1_01: SummarizedExperiment with 1719 rows and 8 columns 
 [2] Mixture1_02: SummarizedExperiment with 1722 rows and 8 columns 
 [3] Mixture1_03: SummarizedExperiment with 1776 rows and 8 columns 
 ...
 [59] Mixture5_01_proteins: SummarizedExperiment with 307 rows and 8 columns 
 [60] Mixture5_02_proteins: SummarizedExperiment with 296 rows and 8 columns 
 [61] Mixture5_03_proteins: SummarizedExperiment with 299 rows and 8 columns </code></pre>
</div>
</div>
<p>We can now join the different protein sets into a single set. We omit the <code>fcol</code> argument, meaning that the set rows will be matched based on the row names (generated by <code>aggregateFeatures()</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>( </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>    spikein, summNames, <span class="st">"proteins"</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-exploration" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="data-exploration"><span class="header-section-number">2.5</span> Data exploration</h2>
<p>We perform data exploration using MDS, using the same pipeline as in <a href="01-basics.html#sec_data_exploration">the previous chapter</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(spikein, <span class="st">"ions"</span>)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(<span class="fu">as</span>(se, <span class="st">"SingleCellExperiment"</span>), <span class="at">exprs_values =</span> <span class="dv">1</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Condition"</span>) <span class="sc">+</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Run"</span>) <span class="sc">+</span> </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Mixture"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_mds-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>There is a strong run-to-run effect, which is partly explained by a mixture effect as the runs from the same mixture tend to be closer than runs from different mixtures. The condition effect is much more subtle to find, probably because we know only a few UPS proteins were spiked in while the majority of the background proteins are unchanged.</p>
<p>As discussed above, the median polish summarisation should remove part of the run to run effect. We repeat the data exploration, but using the protein-level data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(spikein, <span class="st">"proteins"</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(<span class="fu">as</span>(se, <span class="st">"SingleCellExperiment"</span>), <span class="at">exprs_values =</span> <span class="dv">1</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Condition"</span>) <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Run"</span>) <span class="sc">+</span> </span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Mixture"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_mds_proteins-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<p>While the run effects are smaller (i.e.&nbsp;points within a run are more scattered than on the previous plots) on the protein-level MDS compared to the ion-level MDS (the samples are less clustered per run), we can see that normalisation and summarisation alone are not sufficient to correct for these unwanted effects. We will take care of these effects during the data modelling.</p>
</section>
<section id="sec-run_model" class="level2" data-number="2.6">
<h2 data-number="2.6" class="anchored" data-anchor-id="sec-run_model"><span class="header-section-number">2.6</span> Data modelling</h2>
<p>Proteomics data contain several sources of variation that need to be accounted for by the model. Before delving into these sources of variation, we here show how to run the model that accounts for all relevant sources of variation in the spike-in experiment, which we have shown performs best <span class="citation" data-cites="Vandenbulcke2025-sj">(<a href="99-end.html#ref-Vandenbulcke2025-sj" role="doc-biblioref">Vandenbulcke et al. 2025</a>)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>    spikein,  <span class="at">i =</span> <span class="st">"ions"</span>,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## random effect for run</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>Label) <span class="sc">+</span> <span class="do">## random effect for PSMs for the same protein in a label of a run</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for ions in the same spectrum of an MS run</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumnName =</span> <span class="st">"msqrob_psms_rrilm"</span>,</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will now build the model by progressively adding the different sources of variation.</p>
<section id="sec-fixed_effect" class="level3" data-number="2.6.1">
<h3 data-number="2.6.1" class="anchored" data-anchor-id="sec-fixed_effect"><span class="header-section-number">2.6.1</span> Effect of treatment of interest</h3>
<p>We model the source of variation induced by the experimental treatment of interest as a <strong>fixed effect</strong>, which we consider non-random, i.e.&nbsp;the treatment effect is assumed to be the same in repeated experiments, but it is unknown and has to be estimated. When modelling a typical label-free experiment at the protein level, the model boils down to a linear model, again we suppress the index for protein:</p>
<p><span class="math display">\[
y_i = \mathbf{x}^T_i \boldsymbol{\beta} + \epsilon_i,
\]</span></p>
<p>with <span class="math inline">\(y_i\)</span> the <span class="math inline">\(\log_2\)</span>-normalised protein intensities in sample <span class="math inline">\(i\)</span> out of <span class="math inline">\(N\)</span> samples; <span class="math inline">\(\mathbf{x}_i\)</span> a vector with the covariate pattern for the sample in run <span class="math inline">\(r\)</span> encoding the intercept, treatment, potential batch effects and confounders; <span class="math inline">\(\boldsymbol{\beta}\)</span> the vector of parameters that model the association between the covariates and the outcome; and <span class="math inline">\(\epsilon_i\)</span> the residuals reflecting variation that is not captured by the fixed effects. Note that <span class="math inline">\(\mathbf{x}_i\)</span> allows for a flexible parameterisation of the treatment beyond a single covariate, i.e. including a 1 for the intercept, continuous and categorical variables as well as their interactions. For all models considered in this work, we assume the residuals to be independent and identically distributed (i.i.d) according to a normal distribution with zero mean and constant variance, i.e.&nbsp;<span class="math inline">\(\epsilon_{i} \sim N(0,\sigma_\epsilon^2)\)</span>, that can differ from protein to protein.</p>
<p>We could estimate this model from the data using <code>msqrob()</code> (described in the <a href="01-basics.html#sec-msqrob">previous chapter</a>), i.e.&nbsp;the model translates into the following code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># spikein &lt;- msqrob(</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     spikein,  i = "proteins",</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     formula = ~ Condition, ## fixed effect for experimental condition</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     robust = TRUE, ridge = TRUE</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This model, however, does not model all sources of variation in the experiment and relying on its results would lead to incorrect conclusions. We therefore did not run the modelling command and will expand the model.</p>
</section>
<section id="sec-random_effect" class="level3" data-number="2.6.2">
<h3 data-number="2.6.2" class="anchored" data-anchor-id="sec-random_effect"><span class="header-section-number">2.6.2</span> Effect of run</h3>
<p>As label-free experiments contain only a single sample per run, run-specific effects will be absorbed in the residuals. However, the data analysis of multiplexed experiments involving multiple MS runs has to account for run-specific effects, explicitly. If all treatments are present in each run, then the model parameters can be estimated using fixed run effects. Indeed, for these designs run acts as a blocking variable as all treatment effects can be estimated within each run.</p>
<p>However, for more complex designs this is no longer possible and the uncertainty in the estimation of the mean model parameters can involve both within and between run variability. For these designs we can resort to mixed models where the run effect is modelled using <strong>random effects</strong>, i.e.&nbsp;they are considered as a random sample from the population of all possible runs, which are assumed to be i.i.d normally distributed with mean 0 and constant variance, <span class="math inline">\(u_{run} \sim N(0,\sigma^2_\text{run})\)</span>. The use of random effects thus models the correlation in the data, explicitly. Indeed, protein intensities that are measured within the same run will be more similar than protein intensities between runs.</p>
<p>Hence, the model is extended to:</p>
<p><span class="math display">\[
y_{i} =
\mathbf{x}^T_{i} \boldsymbol{\beta} + u_r^\text{run} + \epsilon_{i}
\]</span> with <span class="math inline">\(y_{i}\)</span> the normalised <span class="math inline">\(\log_2\)</span> protein intensities measured in sample <span class="math inline">\(i\)</span> that has been acquired in run <span class="math inline">\(r\)</span> out of <span class="math inline">\(R\)</span> runs, and <span class="math inline">\(u_r^\text{run}\)</span> the effect introduced by run <span class="math inline">\(r\)</span>.</p>
<p>We can also write the model in matrix form:</p>
<p><span class="math display">\[
\mathbf{Y} = \mathbf{X}\boldsymbol{\beta} + \mathbf{Zu}^\text{run} + \boldsymbol{\epsilon}
\]</span> with <span class="math display">\[
\mathbf{Y}=\left[
  \begin{array}{c} y_{1} \\\vdots\\ y_{i} \\\vdots\\ y_{N}\end{array}
\right],
\mathbf{X}=\left[
  \begin{array}{cccc}
    1 &amp; x_{1,1} &amp; \ldots &amp; x_{1,P} \\
    \vdots &amp; \vdots &amp; &amp; \vdots \\
    1 &amp; x_{i,1} &amp; \ldots &amp; x_{i,P} \\
    \vdots &amp; \vdots &amp; &amp; \vdots \\
    1 &amp; x_{N,1} &amp; \ldots &amp; x_{N,P}
  \end{array}
\right],
\boldsymbol{\beta}=\left[
  \begin{array}{c} \beta_0 \\ \beta_1 \\ \vdots \\ \beta_P \end{array}
\right],
\mathbf{Z}=\left[
  \begin{array}{ccc}z_{1,1}&amp;\ldots&amp;z_{1, R}\\\vdots&amp;&amp;\vdots\\z_{N,1}&amp;\ldots&amp;z_{N,R}\end{array}
\right],
\mathbf{u}^\text{run}=\left[
  \begin{array}{c} u^\text{run}_1 \\\vdots\\ u^\text{run}_{r} \\\vdots\\ u^\text{run}_{R}\end{array}
\right]
\]</span></p>
<p>Hence, with the mixed model, the variance covariance matrix of the intensities becomes</p>
<p><span class="math display">\[
\begin{array}{rcl}
\boldsymbol{\Sigma}_\mathbf{Y} &amp;=&amp; \text{var}\left(\mathbf{Y}\right) \\
&amp;=&amp; \text{var}\left(\mathbf{X}\boldsymbol{\beta} + \mathbf{Zu} + \boldsymbol{\epsilon}\right) \\
&amp;=&amp; \mathbf{Z}\text{var}\left(\mathbf{u}\right)\mathbf{Z}^T + \mathbf{I}\sigma_\epsilon^2\\
&amp;=&amp; \mathbf{Z}\mathbf{Z}^T\sigma^2_\text{run} + \mathbf{I}\sigma_\epsilon^2
\end{array}
\]</span></p>
<p>So, we see that the correlation of the data from the same run are correctly addressed and that the data from distinct runs are assumed to be independent. Hence, the variance-covariance matrix of <span class="math inline">\(\mathbf{Y}\)</span> has a block diagonal structure, with as variance <span class="math inline">\(\sigma^2_\text{run} + \sigma_\epsilon^2\)</span> and the covariance between intensities from the same run equals <span class="math inline">\(\sigma^2_\text{run}\)</span>. Suppose every run contains three samples, then</p>
<p><span class="math display">\[
\boldsymbol{\Sigma}_\mathbf{Y} = \left[
\begin{array}{cccccccccc}\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
\sigma^2_\text{run}&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;0&amp;0&amp;0&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}&amp;\ldots&amp;0&amp;0&amp;0\\
0&amp;0&amp;0&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\ldots&amp;0&amp;0&amp;0\\
\ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots &amp; \ldots \\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}&amp;\sigma^2_u\\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon&amp;\sigma^2_\text{run}\\
0&amp;0&amp;0&amp;0&amp;0&amp;0&amp;\dots&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}&amp;\sigma^2_\text{run}+\sigma^2_\epsilon\\
\end{array}\right]
\]</span></p>
<p>This translates in the following code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># spikein &lt;- msqrob(</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     spikein,  i = "proteins",</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     formula = ~ Condition + ## fixed effect for experimental condition</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co">#         (1 | Run), ## random effect for MS run</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co">#     robust = TRUE, ridge = TRUE</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This model is still incomplete and is not executed as we still need to account that multiple samples are acquired within the same run thanks to TMT labelling.</p>
</section>
<section id="effect-of-tmt-label" class="level3" data-number="2.6.3">
<h3 data-number="2.6.3" class="anchored" data-anchor-id="effect-of-tmt-label"><span class="header-section-number">2.6.3</span> Effect of TMT label</h3>
<p>Acquiring multiple samples in a single run is possible because the peptides from each samples are labelled with chemical tags<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>. Peptide labelling can introduce label-specific effects that also need to be modelled. In principle, the effect of adding a chemical label to a peptide should be reproducible from experiment to experiment, and hence could be modelled using a fixed effect (given that TMT label swaps are performed so as to avoid confounding between label and treatment). However, TMT aliquotes contain impurities during production and these impurities may lead to unreproducible effects. Hence, we also include a random effect to account for labelling effects, i.e. <span class="math inline">\(u^\text{label}_l \sim N(0, \sigma^{2,\text{label}})\)</span>. The model thus extends to:</p>
<p><span class="math display">\[
y_{rl} =
\mathbf{x}^T_{rl} \boldsymbol{\beta} +
u_r^\text{run} + u_l^\text{label} + \epsilon_{rlm}
\]</span> with <span class="math inline">\(y_{rl}\)</span> the normalised <span class="math inline">\(\log_2\)</span> protein intensities in run <span class="math inline">\(r\)</span>, labelled with TMT <span class="math inline">\(l\)</span>, <span class="math inline">\(u_l^\text{label}\)</span> the effect introduced by label <span class="math inline">\(l\)</span>. Note that the <span class="math inline">\(rl\)</span> indexing is similar to the previous <span class="math inline">\(i\)</span> indexing, but we now explicitly define the observational unit as the sample that has been measured in run <span class="math inline">\(r\)</span> and labelled with label <span class="math inline">\(l\)</span>.</p>
<p>Specifying label as a random effect translates in the following code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># spikein &lt;- msqrob(</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     spikein,  i = "proteins",</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     formula = ~ Condition + ## fixed effect for experimental condition</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co">#         (1 | Run) + ## random effect for MS run</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co">#         (1 | Label), ## random effect for label</span></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     robust = TRUE, ridge = TRUE</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This model is still incomplete and is not executed as we still need to account that each mixture has been replicated three times.</p>
</section>
<section id="sec-tmt_protein_model" class="level3" data-number="2.6.4">
<h3 data-number="2.6.4" class="anchored" data-anchor-id="sec-tmt_protein_model"><span class="header-section-number">2.6.4</span> Effect of replication</h3>
<p>Some experiments also include technical replication where a TMT mixture can be acquired multiple times. This again will induce correlation. Indeed, protein intensities from the same mixture will be more alike than those of different mixtures. Hence, we also include a random effect to account for this pseudo-replication, i.e. <span class="math inline">\(u^\text{mix}_m \sim N(0, \sigma^{2,\text{mix}})\)</span>. The model thus extends to:</p>
<p><span class="math display">\[
y_{rlm} =
\mathbf{x}^T_{rlm} \boldsymbol{\beta} +
u_r^\text{run} + u_l^\text{label} + u_m^\text{mix} + \epsilon_{rlm}
\]</span></p>
<p>with <span class="math inline">\(y_{rm}\)</span> the normalised <span class="math inline">\(\log_2\)</span> protein intensities in run <span class="math inline">\(r\)</span> with label <span class="math inline">\(l\)</span> in mixture <span class="math inline">\(m\)</span>, <span class="math inline">\(u_m^\text{mix}\)</span> the effect introduced by mixture <span class="math inline">\(m\)</span>.</p>
<p>The model translates to the following code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    spikein,  <span class="at">i =</span> <span class="st">"proteins"</span>,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>      (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## random effect for MS run</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>      (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>      (<span class="dv">1</span> <span class="sc">|</span> Mixture), <span class="do">## random effect for mixture</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span>,</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">overwrite =</span> <span class="cn">TRUE</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This model provides a sensible representation of the sources of variation in the data if we were to model the data at the protein level. This time, we executed the code and will use the result in a later section. However, we found that modelling protein-level effects from ion-level data leads to improved performance <span class="citation" data-cites="Vandenbulcke2025-sj">(<a href="99-end.html#ref-Vandenbulcke2025-sj" role="doc-biblioref">Vandenbulcke et al. 2025</a>)</span>. This will require an additional model expansion.</p>
</section>
<section id="sec-tmt_ion_model" class="level3" data-number="2.6.5">
<h3 data-number="2.6.5" class="anchored" data-anchor-id="sec-tmt_ion_model"><span class="header-section-number">2.6.5</span> Ion-level modelling</h3>
<p>Estimating the treatment effect from ion-level data will again induce additional levels of correlation. Indeed, the intensities for the different reporter ions in a TMT run within the same spectrum (PSM) will be more similar than the intensities between PSMs. We therefore need to add a random effect term to account for the within PSM correlation structure, i.e.&nbsp;<span class="math inline">\(u^\text{PSM}_{rp} \sim
N(0,\sigma^{2,\text{PSM}})\)</span>. Moreover, in each label of a run multiple PSM intensities are picked up for each protein. Hence, intensities from different PSMs for a protein in the same label of a run will be more alike than intensities of different PSMs for the same protein between labels of runs, and we will address this correlation with a label-specific random effect nested in run, i.e.&nbsp;<span class="math inline">\(u_{rl}^{label} \sim
N(0,\sigma^{2,\text{label}})\)</span>. The model then becomes:</p>
<p><span class="math display">\[
y_{rlmp} =
\mathbf{x}^T_{rlmp} \beta + u_r^\text{run} + u_{l}^\text{label} +
u_m^\text{mix}  +
u_{rl}^\text{label} + u_{rp}^\text{PSM} + \epsilon_{rlmp}
\]</span> with <span class="math inline">\(y_{rlmp}\)</span> the <span class="math inline">\(\log_2\)</span>-normalised PSM intensities for run <span class="math inline">\(r\)</span> with label <span class="math inline">\(l\)</span> in mixture <span class="math inline">\(m\)</span> and peptide ion <span class="math inline">\(p\)</span>. Note, that the peptide ion random effect is also nested within each run since each spectrum is described by run-specific characteristics.</p>
<p><code>msqrobAggregate()</code>enables the fitting of an ion-level model to obtain protein-level estimates. The function behaves similarly to <code>msqrob()</code> and shares most of the arguments. The notable difference is the <code>fcol</code> argument that tells the function how to group the ion-level data into protein-level data. Here, we will group ions by the <code>Protein.Accessions</code>. The results will be stored in a new protein-level set, which we call <code>proteins_msqrob</code>. <code>msqrobAggregate()</code> will fetch annotations from the <code>colData</code> (i.e. <code>"Condition"</code>, <code>"Label"</code>, <code>"Run"</code>, <code>"Mixture"</code>), but contrarily to <code>msqrob()</code>, it can also fetch anntations from the <code>rowData</code> (i.e. <code>"ionID"</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>    spikein,  <span class="at">i =</span> <span class="st">"ions"</span>,</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## random effect for Run</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>Label) <span class="sc">+</span> <span class="do">## random effect for label nested in run</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for ion nested in run</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"proteins_msqrob"</span>,</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>So, we built the model shown at the beginning of this section, effectively accounting for all sources of variation in TMT-based proteomics data.</p>
</section>
<section id="sec-robust" class="level3" data-number="2.6.6">
<h3 data-number="2.6.6" class="anchored" data-anchor-id="sec-robust"><span class="header-section-number">2.6.6</span> Why robust regression?</h3>
<p>Throughout this book, we estimate the model parameters using robust regression through M-estimation (<code>robust = TRUE</code>). This class of estimation does not need the assumption that the residuals <span class="math inline">\(\epsilon\)</span> are normally distributed. However, if the residuals are normal, the M-estimators have a high efficiency.</p>
<p>In ordinary least squares (OLS), the loss function that is minimised is:</p>
<p><span class="math display">\[
\sum\limits_{i = 1}^n \left(y_i - \mathbf{x}_i^T \boldsymbol{\beta}\right)^2
\]</span></p>
<p>While the M-estimation minimises the following loss function:</p>
<p><span class="math display">\[
\sum\limits_{i = 1}^n \rho \left(y_i - \mathbf{x}_i^T \boldsymbol{\beta}\right)
\]</span></p>
<p>where <span class="math inline">\(\rho(z)\)</span> is a symmetric function with a minimum at <span class="math inline">\(\rho(0) =
0\)</span> and increases as <span class="math inline">\(|z|\)</span> increases. So the robust regression through M-estimation minimises the maximal bias of the estimators as the derivative of <span class="math inline">\(\rho\)</span> is bounded. A popular function for robust regression (and which is used by <code>msqrob2</code>) is the Huber function:</p>
<p><span class="math display">\[
\rho(e) = \left\{
\begin{array}{cl}
e^2 / 2, &amp; \text{if}\ |e| \leq k  \\
k (|e| - k/2), &amp; \text{if}\ |e| \gt k  \\
\end{array}
\right.
\]</span> where <span class="math inline">\(k\)</span> is a tuning constant (defaults to <span class="math inline">\(k = 1.345\)</span>)</p>
<p>The estimation is performed using iteratively reweighted least square, i.e.&nbsp;</p>
<p><span class="math display">\[
\sum\limits_{i = 1}^n w_i\left(y_i - \mathbf{x}_i^T \boldsymbol{\beta}\right)^2
\]</span> where the weigths <span class="math inline">\(w_i\)</span> for each iteration are defined for each data point using the function :</p>
<p><span class="math display">\[
w(e) = \left\{
\begin{array}{cl}
1, &amp; \text{if}\ |e| \leq k  \\
\frac{k}{|e|}, &amp; \text{if}\ |e| \gt k  \\
\end{array}
\right.
\]</span></p>
<p>Intuitively, observations for which the fitting error (absolute residual) is smaller than the constant <span class="math inline">\(k\)</span> will contribute contribute to the fitting as for the OLS, while the contribution of the remaining observations will decrease linearly as the error exceeds <span class="math inline">\(k\)</span>. This means that a strong <strong>outlier</strong>, i.e.&nbsp;an observation for which the fitting error is large, will barely contribute to the fit. Since the last iteration upon covergence is still a weighted least square estimation, we can use the same statistical testing framework as for OLS, although the tests and confidence intervals are based on asymptotic theory.</p>
<p>Note, that robust regression also extends to linear mixed models, by adopting weighted maximum likelihood e.g.&nbsp;<strong>TODO</strong> ref https://www.tandfonline.com/doi/abs/10.1080/03610920802677216.</p>
</section>
<section id="sec-ridge" class="level3" data-number="2.6.7">
<h3 data-number="2.6.7" class="anchored" data-anchor-id="sec-ridge"><span class="header-section-number">2.6.7</span> Why ridge regression?</h3>
<p><code>msqrob2</code> also allows for ridge regression (when <code>ridge = TRUE</code>). In ridge regression, model parameters are estimated by minimising a penalised version of the OLS loss:</p>
<p><span class="math display">\[
\sum\limits_{i = 1}^n \left(y_i - \mathbf{x}_i^T \boldsymbol{\beta}\right)^2 + \lambda \boldsymbol{\beta}^T\mathbf{D}\boldsymbol{\beta},
\]</span> with <span class="math inline">\(\mathbf{D}\)</span> a diagonal matrix that include a 1 on the diagonal elements for the model parameters that are penalised and a 0 otherwise.</p>
<p>Note, that in msqrob2 we leave the intercept <span class="math inline">\(\beta_0\)</span> unpenalised so the first diagonal element of <span class="math inline">\(\mathbf{D}\)</span> is set at zero and we only penalise the remaining <span class="math inline">\(m-1\)</span> slope parameters <span class="math inline">\(\boldsymbol{\beta}_s=[\beta_1 \ldots \beta_{m-1}]^T\)</span>. Hence, <span class="math inline">\(\boldsymbol{\beta}^T\mathbf{D}\boldsymbol{\beta}\)</span> reduces to the squared L2 norm of <span class="math inline">\(\boldsymbol{\beta}_s\)</span>, i.e.&nbsp;<span class="math inline">\(||\boldsymbol{\beta}_s||_2^2 =
\sum\limits_{j = 1}^{m-1} \beta_j^2\)</span>.</p>
<p>Due to the penality term in the loss function, the estimates of penalised slope terms <span class="math inline">\(\boldsymbol{\beta}_s\)</span> will be shrunken towards zero, especially for irrelevant covariates in <span class="math inline">\(\mathbf{X}\)</span>. In other words, the parameters are stabilised by reducing the variance of the estimation, which prevents overfitting. Note, that the penalty term in the loss function thus implies that the estimates of <span class="math inline">\(\boldsymbol{\beta}\)</span> will be biased towards zero. This is known as the variance-bias trade-off. In our experience, msqrob’s ridge regression mainly shrinks the log2-fold change estimates (<span class="math inline">\(\beta\)</span>) for features with low evidence for differential abundance to zero while leaving those for features with high evidence for differential abundance largely unaltered.</p>
<p>Interestingly, there exists a link between Bayesian regression, ridge regression and mixed models, i.e.&nbsp;the ridge regression can be adopted by placing a normal prior on the slope terms that have to be penalised, i.e.&nbsp;<span class="math inline">\(\beta_j \sim N(0,\sigma_\beta^2)\)</span> for <span class="math inline">\(j= 1 \ldots m-1\)</span>. It can than be shown that the best linear unbiased predictor from the linear mixed model with the slope terms defined as random effects provides the ridge regression estimates with a ridge penalty <span class="math inline">\(\lambda = \hat{\sigma}^2_\epsilon/\hat{\sigma}^2_\beta\)</span>). Hence, mixed model software can be used to tune the ridge penalty in a data driven way. Note, that this comes at price of increased computational complexity. msqrob2 estimation with ridge regression will thus be slower than without ridge regression.</p>
<p>Also note, that ridge regression cannot be performed when <span class="math inline">\(\mathbf{X}\)</span> contains an intercept and a single covariate. Indeed, based on a single slope term the random effect variance <span class="math inline">\(\sigma^2_\beta\)</span> cannot be estimated. We demonstrate this by intentionally triggering an error after subsetting only spike-in condition <code>1</code> and <code>0.5</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>spikeinSubset <span class="ot">&lt;-</span> <span class="fu">subsetByColData</span>(spikein, spikein<span class="sc">$</span>Condition <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"0.5"</span>))</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">try</span>(<span class="fu">msqrobAggregate</span>(</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>  spikeinSubset,  <span class="at">i =</span> <span class="st">"ions"</span>,</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span>  Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## random effect for Run</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>Label) <span class="sc">+</span> <span class="do">## random effect for label nested in run</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for ion nested in run</span></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"proteins_msqrob"</span>,</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Error : BiocParallel errors
  1 remote errors, element index: 1
  406 unevaluated and other errors
  first remote error:
Error in (function (y, rowdata = NULL, formula, coldata, doQR, robust, : The mean model must have more than two parameters for ridge regression.
              if you really want to adopt ridge regression when your factor has only two levels
              rerun the function with a formula where you drop the intercept. e.g. ~-1+condition
            </code></pre>
</div>
</div>
<p>As the error message suggests, either the ridge argument has to be set at FALSE to obtain an unpenalised fit, or the intercept can be suppressed in order to obtain a fit with two slope terms (one for each group) so as to enable the estimation of the ridge penalty.</p>
</section>
</section>
<section id="statistical-inference" class="level2" data-number="2.7">
<h2 data-number="2.7" class="anchored" data-anchor-id="statistical-inference"><span class="header-section-number">2.7</span> Statistical inference</h2>
<p>We can now convert the biological question “does the spike-in condition affect the protein intensities?” into a statistical hypothesis. In other words, we need to translate this question in a null and alternative hypothesis on a single model parameter or a linear combination of model parameters, which is also referred to with a <a href="01-basics.html#sec-inference">contrast</a>. We plot an overview of the model parameters.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sampleData =</span>  <span class="fu">colData</span>(spikein),</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> Condition,</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>vd<span class="sc">$</span>plotlist</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_VisualizeDesign-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Note that with ExploreModelMatrix we can only visualise fixed effects part of the model. This is fine as the mean protein abundances can only systematically differ from each other according to the <code>Condition</code> (fixed effect).</p>
<section id="hypothesis-testing" class="level3" data-number="2.7.1">
<h3 data-number="2.7.1" class="anchored" data-anchor-id="hypothesis-testing"><span class="header-section-number">2.7.1</span> Hypothesis testing</h3>
<p>The average difference in the log2-intensity between the 1x and the 0.5x conditions is provided by the contrast <code>ridgeCondition1 - ridgeCondition0.5</code>. This is, however, not the only difference we could assess. As described in the <a href="01-basics.html#sec-multiple_contrasts">previous chapter</a>, we will generate a contrast matrix that assess all possible pairwise comparisons between spike-in conditions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>allHypotheses <span class="ot">&lt;-</span> <span class="fu">createPairwiseContrasts</span>(</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> Condition, <span class="fu">colData</span>(spikein), <span class="st">"Condition"</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>  allHypotheses,</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">parameterNames =</span> <span class="fu">paste0</span>(<span class="st">"ridgeCondition"</span>, <span class="fu">c</span>(<span class="st">"0.5"</span>, <span class="st">"0.667"</span>, <span class="st">"1"</span>))</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We test our null hypotheses using <code>hypothesisTest()</code> and the estimated ion-level model stored in <code>proteins_msqrob</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(spikein, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, <span class="at">contrast =</span> L)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins_msqrob"</span>]])[, <span class="fu">colnames</span>(L)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can retrieve the results for the comparison between the 1x and the 0.5x conditions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(inferences<span class="sc">$</span><span class="st">"ridgeCondition1 - ridgeCondition0.5"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               logFC           se         df             t        pval
O00151 -2.549859e-02 2.153969e-02  471.97751 -1.183796e+00 0.237089769
O00220  1.440133e-02 5.774331e-02   94.45995  2.494026e-01 0.803590901
O00244 -3.706545e-10 5.416412e-06   84.16092 -6.843175e-05 0.999945561
O00299 -3.154427e-02 1.172633e-02 1380.19537 -2.690038e+00 0.007230547
O00330 -2.881986e-02 3.682363e-02  193.93129 -7.826456e-01 0.434789869
O00399            NA           NA         NA            NA          NA
          adjPval
O00151 0.58896109
O00220 1.00000000
O00244 1.00000000
O00299 0.03771935
O00330 0.87236685
O00399         NA</code></pre>
</div>
</div>
<p>The last row is filled with missing values because data modelling resulted in a <code>fitError</code> (we will explore in a <a href="#sec-fiterror">later section</a> how we can deal with proteins that could not be fit).</p>
</section>
<section id="volcano-plots" class="level3" data-number="2.7.2">
<h3 data-number="2.7.2" class="anchored" data-anchor-id="volcano-plots"><span class="header-section-number">2.7.2</span> Volcano plots</h3>
<p>We generate volcano plots for all pairwise comparison between conditions. First, we add new columns to the tables and join them in a single table.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(inferences), <span class="cf">function</span>(i) {</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  inference <span class="ot">&lt;-</span> inferences[[i]]</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>Comparison <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, i)</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>Comparison <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^([0-9.]*)$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1 - 0.125"</span>, inference<span class="sc">$</span>Comparison)</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  inference</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, inferences) <span class="do">## combine in a single table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then, we plot the volcano plots with each comparison in a separate facet.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> isUps,</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_volcao-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fold-change-distributions" class="level3" data-number="2.7.3">
<h3 data-number="2.7.3" class="anchored" data-anchor-id="fold-change-distributions"><span class="header-section-number">2.7.3</span> Fold change distributions</h3>
<p>As this is a spike-in study with known ground truth, we can also plot the log2 fold change distributions against the expected values, in this case 0 for the HeLa proteins and the difference of the log concentration for the spiked-in UPS standards. We first create a small table with the real values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>realLogFC <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> <span class="fu">unique</span>(inferences<span class="sc">$</span>Comparison))</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>realLogFC<span class="sc">$</span>logFC <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"log2("</span>, <span class="fu">gsub</span>(<span class="st">"-"</span>, <span class="st">"/"</span>, realLogFC<span class="sc">$</span>Comparison), <span class="st">")"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sapply</span>(<span class="cf">function</span>(x) <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now create the boxplots with the estimated log2-fold changes, adding horizontal lines with the corresponding target values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> logFC,</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> isUps,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> isUps) <span class="sc">+</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">data =</span> realLogFC, <span class="fu">aes</span>(<span class="at">yintercept =</span> logFC), </span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison) <span class="sc">+</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Distribution of the log2 fold changes"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_boxplot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Estimated log2 fold change for HeLa proteins are closely distributed around 0, as expected. log2 fold changes for UPS standard proteins are biased towards zero probably due to ratio compression effects, as reported previously for labeled strategies <span class="citation" data-cites="Savitski2011-qi">(<a href="99-end.html#ref-Savitski2011-qi" role="doc-biblioref">Savitski et al. 2011</a>)</span>.</p>
</section>
<section id="detail-plots" class="level3" data-number="2.7.4">
<h3 data-number="2.7.4" class="anchored" data-anchor-id="detail-plots"><span class="header-section-number">2.7.4</span> Detail plots</h3>
<p>We can explore the PSM intensities for a protein to validate the statistical inference results. For example, let’s explore the intensities for the protein with the most significant difference.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>(targetProtein <span class="ot">&lt;-</span> inferences<span class="sc">$</span>Protein[<span class="fu">which.min</span>(inferences<span class="sc">$</span>adjPval)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "P02788ups"</code></pre>
</div>
</div>
<p>To obtain the required data, we perform a <a href="01-basics.html#sec-detail_plot">data manipulation pipeline</a>. We plot the log2 normalised intensities for each sample. Since the protein is modelled at the peptide ion level, multiple ion intensities are recorded in each sample. Each ion is linked across samples using a grey line. Samples are colored according to UPS spike-in condition. Finally, we split the plot in facets, one for each mixture, to visualise the heterogeneity induced by sample preparation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>spikein[targetProtein, , <span class="st">"ions"</span>] <span class="sc">|&gt;</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein)),</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">rowvars =</span> <span class="fu">c</span>(<span class="st">"Protein.Accessions"</span>, <span class="st">"ionID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    <span class="do">## We reorder the sample identifiers to improve visualisation</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">colname =</span> <span class="fu">factor</span>(colname, <span class="at">levels =</span> <span class="fu">unique</span>(colname[<span class="fu">order</span>(Condition)]))) <span class="sc">|&gt;</span> </span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> ionID), <span class="at">linewidth =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Condition)) <span class="sc">+</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span> Mixture, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_detail_plot-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="protein-level-inference" class="level2" data-number="2.8">
<h2 data-number="2.8" class="anchored" data-anchor-id="protein-level-inference"><span class="header-section-number">2.8</span> Protein-level inference</h2>
<p>We here show how to perform a two-step approach, where the data are first summarised then modelled. We perform the same statistical workflow as above, but starting from the protein level model estimated in <a href="#sec-tmt_protein_model">the section above</a><a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(spikein, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins"</span>]])[, <span class="fu">colnames</span>(L)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We build the volcano using the same code as above:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(inferences), <span class="cf">function</span>(i) {</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  inference <span class="ot">&lt;-</span> inferences[[i]]</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  inference<span class="sc">$</span>Comparison <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, i)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  inference</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, inferences) <span class="do">## combine in a single table</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> isUps) <span class="sc">+</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>,</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="st">"Protein-level modelling"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_volcano_proteins-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We plot the fold change distributions:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> logFC,</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> isUps,</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> isUps) <span class="sc">+</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>( <span class="do">## Adding the expected Log2 fold changes </span></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">data =</span> <span class="fu">group_by</span>(inferences, Comparison, isUps) <span class="sc">|&gt;</span> </span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarise</span>(<span class="at">logFC =</span> <span class="fu">ifelse</span>(isUps, <span class="fu">log2</span>(<span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">sub</span>(<span class="st">"-"</span>, <span class="st">"/"</span>, Comparison)))), <span class="dv">0</span>)),</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> <span class="dv">10</span>, <span class="at">size =</span> <span class="dv">4</span></span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison) <span class="sc">+</span></span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Distribution of the log2 fold changes"</span>,</span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="st">"Protein-level modelling"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_boxplot_proteins-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Exploring the intensities at the protein level is simplified compared to PSM-level exploration since every sample now contains a single observation, the protein intensity.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>spikein[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein))) <span class="sc">|&gt;</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>    <span class="do">## We reorder the sample identifiers to improve visualisation</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">colname =</span> <span class="fu">factor</span>(colname, <span class="at">levels =</span> <span class="fu">unique</span>(colname[<span class="fu">order</span>(Condition)]))) <span class="sc">|&gt;</span> </span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Condition)) <span class="sc">+</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span> Mixture, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein,</span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="st">"Summarised protein data"</span>) <span class="sc">+</span></span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_detail_plot_proteins-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>Notice how the summarisation-based approach hides the variation associated with the measurement of different peptide ions within the same protein, as well as discrepancies between peptide identification rates across mixtures.</p>
</section>
<section id="dealing-with-fiterrors" class="level2" data-number="2.9">
<h2 data-number="2.9" class="anchored" data-anchor-id="dealing-with-fiterrors"><span class="header-section-number">2.9</span> Dealing with <code id="sec-fiterror">fitErrors</code></h2>
<p>Missing value patterns in the data may lead to non-estimable parameters. This is recognised by <code>msqrob2</code> and will lead to <code>fitError</code>s which is a type of model output where the model could not be fit. This information is available from the <code>StatModel</code> objects.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein[[<span class="st">"proteins_msqrob"</span>]])[[<span class="st">"msqrobModels"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(<span class="cf">function</span>(x) x<span class="sc">@</span>type) <span class="sc">|&gt;</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">table</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
fitError     lmer 
      94      313 </code></pre>
</div>
</div>
<p>There are 3 possible strategies for dealing with these <code>fitErrors</code>.</p>
<section id="sec-msqrob_refit" class="level3" data-number="2.9.1">
<h3 data-number="2.9.1" class="anchored" data-anchor-id="sec-msqrob_refit"><span class="header-section-number">2.9.1</span> Removing the random effect of sample</h3>
<p>This strategy only applies for PSM-level models. Some proteins are difficult to detect and may be quantified by a single peptide ion species. In these cases, every sample contains a single observation for the protein and hence no random effect of <code>Run:Label</code> can be estimated. While the results for such one-hit wonders are questionable, we provide <code>msqrobRefit()</code> to refit a new model for a subset of proteins of interest.</p>
<p>In this case, we want to refit a model without a sample effect for one-hit-wonder proteins. This information can be retrieved from the aggregation results, using <code>aggcounts()</code>. This getter function provides the number of features used when performing summarisation for each protein in each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">aggcounts</span>(spikein[[<span class="st">"proteins_msqrob"</span>]])</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw_Abundance..127N
O00151                                                            4
O00220                                                            1
O00244                                                            0
O00299                                                           15
O00330                                                            2
       161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw_Abundance..127C
O00151                                                            4
O00220                                                            2
O00244                                                            0
O00299                                                           15
O00330                                                            2
       161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw_Abundance..128N
O00151                                                            4
O00220                                                            2
O00244                                                            0
O00299                                                           15
O00330                                                            2
       161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw_Abundance..128C
O00151                                                            4
O00220                                                            1
O00244                                                            0
O00299                                                           15
O00330                                                            2
       161117_SILAC_HeLa_UPS1_TMT10_Mixture1_01.raw_Abundance..129N
O00151                                                            4
O00220                                                            2
O00244                                                            0
O00299                                                           15
O00330                                                            2</code></pre>
</div>
</div>
<p>One-hit wonder proteins are proteins for which the number of feature used for summarisation does not exceed 1 peptide ion across samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>oneHitProteins <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts)[<span class="fu">rowMax</span>(counts) <span class="sc">==</span> <span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Using <code>msqrobRefit()</code> is very similar to <code>msqrobAggregate()</code>, see here however that we adapted the formula to remove the random effect for label nested within run. We also mention which proteins must be refit using the <code>subset</code> argument.</p>
<p><strong>TODO</strong>: include msqrobRefit in <code>msqrob2</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrobRefit</span>(</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>    spikein, <span class="at">i =</span> <span class="st">"ions"</span>,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">subset =</span> oneHitProteins,</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## fixed effect for label</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run ) <span class="sc">+</span> <span class="do">## random effect for run</span></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for PSM nested in MS run</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>        <span class="do">## random effect for label nested in run has been removed</span></span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"proteins_msqrob"</span>,</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s see how removing the random effect of label within run for one-hit-wonder proteins reduced the number of <code>fitError</code>s.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>fitTypes <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins_msqrob"</span>]])[[<span class="st">"msqrobModels"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(<span class="cf">function</span>(x) x<span class="sc">@</span>type)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(fitTypes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>fitTypes
fitError     lmer 
       1      406 </code></pre>
</div>
</div>
</section>
<section id="manual-inspection" class="level3" data-number="2.9.2">
<h3 data-number="2.9.2" class="anchored" data-anchor-id="manual-inspection"><span class="header-section-number">2.9.2</span> Manual inspection</h3>
<p>One protein is still non-estimable upon refitting and requires additional data exploration to understand why the model cannot be estimated. Let us take the protein that cannot be fitted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>proteinError <span class="ot">&lt;-</span> <span class="fu">names</span>(fitTypes[fitTypes <span class="sc">==</span> <span class="st">"fitError"</span>])[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To understand the problem, we plot the data for that protein using the same <code>QFeatures</code> pipeline described above. We here plot the data in function of the <code>Label</code> (x-axis), <code>Condition</code> (colour) and <code>Mixture</code> (shape).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>spikein[proteinError, , <span class="st">"ions"</span>] <span class="sc">|&gt;</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein)),</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">rowvars =</span> <span class="fu">c</span>(<span class="st">"Protein.Accessions"</span>, <span class="st">"ionID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">colname =</span> <span class="fu">factor</span>(colname, <span class="at">levels =</span> <span class="fu">unique</span>(colname[<span class="fu">order</span>(Condition)]))) <span class="sc">|&gt;</span> </span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Condition)) <span class="sc">+</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span> Mixture, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02-advanced_files/figure-html/advanced_fiterror_manual-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can immediately spot that PSM intensities are only present in mixture 3. Hence, the mixed model cannot be fitted with a random effect for mixture. However, we don’t want to rely on a result that has been measured in a single replicate and <code>msqrob2</code> flags these problematic proteins instead of defining ad-hoc heuristics, avoiding potentially misleading conclusions.</p>
</section>
<section id="imputation" class="level3" data-number="2.9.3">
<h3 data-number="2.9.3" class="anchored" data-anchor-id="imputation"><span class="header-section-number">2.9.3</span> Imputation</h3>
<p>The last popular strategy to deal with fit errors is to impute missing values so that all models can be estimated. Note, that our general advice is to avoid imputation as this typically comes at the expense of additional and often unrealistic assumptions<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a>.</p>
<p><code>QFeatures</code> provides a large panel of imputation strategies through <code>impute()</code>. Identifying which imputation strategy is most suited for this data set is outside the scope of this book. For illustration purposes, we here arbitrarily use KNN imputation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">impute</span>(</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>    spikein, <span class="at">i =</span> <span class="st">"ions"</span>, <span class="at">name =</span> <span class="st">"ions_imputed"</span>,</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"knn"</span>, <span class="at">colmax =</span> <span class="dv">1</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 64 sets:

 [1] Mixture1_01: SummarizedExperiment with 1719 rows and 8 columns 
 [2] Mixture1_02: SummarizedExperiment with 1722 rows and 8 columns 
 [3] Mixture1_03: SummarizedExperiment with 1776 rows and 8 columns 
 ...
 [62] proteins: SummarizedExperiment with 407 rows and 120 columns 
 [63] proteins_msqrob: SummarizedExperiment with 407 rows and 120 columns 
 [64] ions_imputed: SummarizedExperiment with 4066 rows and 120 columns </code></pre>
</div>
</div>
<p>The function added a new set <code>ions_imputed</code> which we can use to fit the ion-level model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>    spikein,  <span class="at">i =</span> <span class="st">"ions_imputed"</span>,</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## random effect for run</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>Label) <span class="sc">+</span> <span class="do">## random effect for PSMs from the same protein in a label of a run</span></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for ions in the same spectrum of an MS run</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumnName =</span> <span class="st">"msqrob_psm_rrilmm"</span>,</span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"proteins_msqrob_imputed"</span>,</span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We here assess how many models have been estimated for all proteins upon imputation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein[[<span class="st">"proteins_msqrob_imputed"</span>]])[[<span class="st">"msqrob_psm_rrilmm"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(<span class="cf">function</span>(x) x<span class="sc">@</span>type) <span class="sc">|&gt;</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">table</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
fitError     lmer 
      59      348 </code></pre>
</div>
</div>
<p>Again <code>fitError</code>s were generated for one-hit-wonder proteins.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">aggcounts</span>(spikein[[<span class="st">"proteins_msqrob_imputed"</span>]])</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>oneHitProteins <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts)[<span class="fu">rowMax</span>(counts) <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrobRefit</span>(</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>    spikein, <span class="at">i =</span> <span class="st">"ions_imputed"</span>,</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">subset =</span> oneHitProteins,</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> <span class="sc">~</span> Condition <span class="sc">+</span> <span class="do">## fixed effect for experimental condition</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## random effect for label</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## random effect for mixture</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run ) <span class="sc">+</span> <span class="do">## random effect for run</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID), <span class="do">## random effect for PSM nested in MS run</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a>        <span class="do">## random effect for label nested in run has been removed</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumnName =</span> <span class="st">"msqrob_psm_rrilmm"</span>,</span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"proteins_msqrob_imputed"</span>,</span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(spikein[[<span class="st">"proteins_msqrob_imputed"</span>]])[[<span class="st">"msqrob_psm_rrilmm"</span>]] <span class="sc">|&gt;</span></span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sapply</span>(<span class="cf">function</span>(x) x<span class="sc">@</span>type) <span class="sc">|&gt;</span></span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">table</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
lmer 
 407 </code></pre>
</div>
</div>
<p>Upon refit, no <code>fitError</code>s were generated for any proteins, as expected. Be mindful that the results upon imputation will be highly depended on the suitability of the imputation approach.</p>
</section>
</section>
<section id="conclusion" class="level2" data-number="2.10">
<h2 data-number="2.10" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">2.10</span> Conclusion</h2>
<p>In this chapter, we expanded upon the basic concepts with a more complex analysis. First, we started our analysis from PSM-level data, showing that our tools are amenable to read different levels of input format such as the <a href="01-basics.html#sec-peptide_table">peptides table</a> or the <a href="#sec-psm_table">PSM table</a>. Starting from the PSM level data also enabled better control of the PSM filtering compared to starting with peptide-level data. We demonstrated how to conduct a comprehensive feature filtering workflow, with different level of customisation to enable filtering based on any criterion.</p>
<p>The complexity of the analysis is the reflection of the complexity of the experimental designs, as TMT-labelling includes the modelling of several additional sources of variation compared to LFQ experiments: effect of treatment of interest, effect of TMT labelling, effect of the MS acquisition run, and the effect of replication. We built protein-level models, but we have also shown that we can build PSM-level models if we also include a spectrum effect and an effect for TMT label nested within run.</p>
<p>Finally, we have demonstrated how to deal with proteins that cannot be modelled due to missing values. For PSM-level models, we can remove the random effects for TMT label within run that cannot be estimated for one-hit-wonder proteins. We can also manually inspect how missing values can influence the model design, and refit a simplified model upon expert’s intervention. Finally, we can impute missing values, which unlocks model fitting, but imposes strong assumption on the validity of the imputation approach and the reliability of the predicted values.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Huang2020-lu" class="csl-entry" role="listitem">
Huang, Ting, Meena Choi, Manuel Tzouros, Sabrina Golling, Nikhil Janak Pandya, Balazs Banfai, Tom Dunkley, and Olga Vitek. 2020. <span>“<span>MSstatsTMT</span>: Statistical Detection of Differentially Abundant Proteins in Experiments with Isobaric Labeling and Multiple Mixtures.”</span> <em>Mol. Cell. Proteomics</em> 19 (10): 1706–23.
</div>
<div id="ref-Savitski2011-qi" class="csl-entry" role="listitem">
Savitski, Mikhail M, Gavain Sweetman, Manor Askenazi, Jarrod A Marto, Manja Lang, Nico Zinn, and Marcus Bantscheff. 2011. <span>“Delayed Fragmentation and Optimized Isolation Width Settings for Improvement of Protein Identification and Accuracy of Isobaric Mass Tag Quantification on Orbitrap-Type Mass Spectrometers.”</span> <em>Anal. Chem.</em> 83 (23): 8959–67.
</div>
<div id="ref-Sticker2020-rl" class="csl-entry" role="listitem">
Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement. 2020. <span>“Robust Summarization and Inference in Proteome-Wide Label-Free Quantification.”</span> <em>Mol. Cell. Proteomics</em> 19 (7): 1209–19.
</div>
<div id="ref-Vandenbulcke2025-sj" class="csl-entry" role="listitem">
Vandenbulcke, Stijn, Christophe Vanderaa, Oliver Crook, Lennart Martens, and Lieven Clement. 2025. <span>“<span>Msqrob2TMT</span>: Robust Linear Mixed Models for Inferring Differential Abundant Proteins in Labeled Experiments with Arbitrarily Complex Design.”</span> <em>Mol. Cell. Proteomics</em> 24 (7): 101002.
</div>
</div>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>Depending on the reagents used, 6, 10, 11 up to 18 samples can be pooled in one mixture<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>You will be prompted to create a new folder the first time you use the package.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>You can find an illustrated step-by-step guide in the <a href="https://rformassspectrometry.github.io/QFeatures/articles/read_QFeatures.html">QFeatures vignette</a><a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>Note that the data points from one peptide ion in one replicate has been extracted from a single MS2 spectrum.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>Recall the <a href="#sec-tmt_workflow">section above</a>).<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p>Note that the contrasts remain the same, as the fixed effect part of the model (spike-in treatment effect)is the same for the models estimated with <code>msqrob()</code> or <code>msqrobAggregate()</code>, so we do not need to build a new contrast matrix.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7"><p>Indeed, we have shown that state-of-the-art imputation methods profoundly change the distribution of the intensities in bulk proteomics, which can have a detrimental impact on the downstream analysis. Our take is to start from the ion or peptide intensity data as is and to model the data from the same protein together while correcting for ion/peptide species. These ion/peptide-level models can either be used to directly infer differential abundance at the protein level or to obtain protein-level abundance values in the summarisation step. Note, that this approach assumes random missingness upon correction for ion/peptide species.<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./01-basics.html" class="pagination-link" aria-label="Statistical analysis with msqrob2">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./03-benchmarking.html" class="pagination-link" aria-label="Benchmarking workflows">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>