<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.23">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Benchmarking workflows â€“ Statistical analysis of mass spectrometry-based proteomics data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./04-workflow_optimisation.html" rel="next">
<link href="./02-advanced.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac32dca00065cc45e71a60b6df2d2e12.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./03-benchmarking.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figs/msqrob2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistical analysis of mass spectrometry-based proteomics data</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preamble</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-benchmarking.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-workflow_optimisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-francisella.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-heart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-mouse_diet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-end.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages"><span class="header-section-number">3.1</span> Load packages</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">3.2</span> Data</a>
  <ul class="collapse">
  <li><a href="#data-files" id="toc-data-files" class="nav-link" data-scroll-target="#data-files"><span class="header-section-number">3.2.1</span> Data files</a></li>
  <li><a href="#convert-to-qfeatures" id="toc-convert-to-qfeatures" class="nav-link" data-scroll-target="#convert-to-qfeatures"><span class="header-section-number">3.2.2</span> Convert to QFeatures</a></li>
  </ul></li>
  <li><a href="#data-preprocessing" id="toc-data-preprocessing" class="nav-link" data-scroll-target="#data-preprocessing"><span class="header-section-number">3.3</span> Data preprocessing</a>
  <ul class="collapse">
  <li><a href="#preprocessing-workflow" id="toc-preprocessing-workflow" class="nav-link" data-scroll-target="#preprocessing-workflow"><span class="header-section-number">3.3.1</span> Preprocessing workflow</a></li>
  <li><a href="#run-the-workflow" id="toc-run-the-workflow" class="nav-link" data-scroll-target="#run-the-workflow"><span class="header-section-number">3.3.2</span> Run the workflow</a></li>
  </ul></li>
  <li><a href="#data-modelling" id="toc-data-modelling" class="nav-link" data-scroll-target="#data-modelling"><span class="header-section-number">3.4</span> Data modelling</a>
  <ul class="collapse">
  <li><a href="#model-definition" id="toc-model-definition" class="nav-link" data-scroll-target="#model-definition"><span class="header-section-number">3.4.1</span> Model definition</a></li>
  <li><a href="#modelling-workflow" id="toc-modelling-workflow" class="nav-link" data-scroll-target="#modelling-workflow"><span class="header-section-number">3.4.2</span> Modelling workflow</a></li>
  <li><a href="#run-the-workflow-1" id="toc-run-the-workflow-1" class="nav-link" data-scroll-target="#run-the-workflow-1"><span class="header-section-number">3.4.3</span> Run the workflow</a></li>
  </ul></li>
  <li><a href="#sec-performance" id="toc-sec-performance" class="nav-link" data-scroll-target="#sec-performance"><span class="header-section-number">3.5</span> Performance benchmark</a>
  <ul class="collapse">
  <li><a href="#number-of-fits" id="toc-number-of-fits" class="nav-link" data-scroll-target="#number-of-fits"><span class="header-section-number">3.5.1</span> Number of fits</a></li>
  <li><a href="#tp-and-fp-at-5-fdr" id="toc-tp-and-fp-at-5-fdr" class="nav-link" data-scroll-target="#tp-and-fp-at-5-fdr"><span class="header-section-number">3.5.2</span> TP and FP at 5% FDR</a></li>
  <li><a href="#sec-tpr_fdp" id="toc-sec-tpr_fdp" class="nav-link" data-scroll-target="#sec-tpr_fdp"><span class="header-section-number">3.5.3</span> TPR-FDP curves</a></li>
  <li><a href="#sec-benchmark_with_boxplots" id="toc-sec-benchmark_with_boxplots" class="nav-link" data-scroll-target="#sec-benchmark_with_boxplots"><span class="header-section-number">3.5.4</span> Fold change boxplots</a></li>
  </ul></li>
  <li><a href="#sec-take_home" id="toc-sec-take_home" class="nav-link" data-scroll-target="#sec-take_home"><span class="header-section-number">3.6</span> Take home messages</a>
  <ul class="collapse">
  <li><a href="#sec-preprocess_evidence" id="toc-sec-preprocess_evidence" class="nav-link" data-scroll-target="#sec-preprocess_evidence"><span class="header-section-number">3.6.1</span> Preprocessing the evidence file</a></li>
  <li><a href="#modelling-the-preprocessed-data" id="toc-modelling-the-preprocessed-data" class="nav-link" data-scroll-target="#modelling-the-preprocessed-data"><span class="header-section-number">3.6.2</span> Modelling the preprocessed data</a></li>
  </ul></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">3.7</span> Appendix</a>
  <ul class="collapse">
  <li><a href="#sec-appendix3_1" id="toc-sec-appendix3_1" class="nav-link" data-scroll-target="#sec-appendix3_1"><span class="header-section-number">3.7.1</span> TPR-FDP curves using all proteins in the data</a></li>
  <li><a href="#sec-appendix3_2" id="toc-sec-appendix3_2" class="nav-link" data-scroll-target="#sec-appendix3_2"><span class="header-section-number">3.7.2</span> TPR-FDP curves using only the proteins fit by all approaches</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-benchmarking" class="quarto-section-identifier"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>In the previous chapters we have seen three different approaches to model the data:</p>
<ul>
<li><a href="02-advanced.html#sec-tmt_protein_model">Approach 1</a>: start from a PSM-level table, compute the protein summaries and model the data at the protein level.</li>
<li><a href="02-advanced.html#sec-tmt_ion_model">Approach 2</a>: start from a PSM-level table and model the ion-level data to obtain protein-level results.</li>
<li><a href="01-basics.html">Approach 3</a>: start from a peptide-level table, compute the protein summaries and model the data at the protein level.</li>
</ul>
<p>There are further possible approaches to model the data:</p>
<ul>
<li>Approach 4: start from a peptide-level table and model the peptide-level data to obtain protein-level results.</li>
<li>Approach 5: start from a protein-level table (generated by MaxQuant using maxLFQ) and model the protein-level data to obtain protein-level results.</li>
</ul>
<p>In this chapter, we will attempt to understand whether these different approaches lead to difference in modelling performance. To explore these differences, we will conduct a benchmarking experiment using the <a href="01-basics.html#sec-ecoli_experiment">E. coli spike-in experiment</a>, containing ground truth information that will be used for an objective comparison of the workflows.</p>
<p><strong>Important</strong>: the first sections of the chapter are meant for advanced users that are familiar with R scripting since benchmarking requires some degree of automation. However, for novice users interested in the key messages of the benchmarking and that want to implement the best practices, we refer to the <a href="#sec-take_home">take home message section</a> for more accessible guidelines.</p>
<section id="load-packages" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="load-packages"><span class="header-section-number">3.1</span> Load packages</h2>
<p>We load the <code>msqrob2</code> package, along with additional packages for data manipulation and visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"data.table"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"patchwork"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>TODO</strong>: move some functions from utils.R to msqrob2, eg msqrobCollect, pairwise contrasts</p>
<p>We also configure the <a href="01-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="data"><span class="header-section-number">3.2</span> Data</h2>
<p>We will reuse the data by <span class="citation" data-cites="Shen2018-gw">(<a href="99-end.html#ref-Shen2018-gw" role="doc-biblioref">Shen et al. 2018</a>)</span> as in <a href="01-basics.html" class="quarto-xref"><span>Chapter 1</span></a>. The <a href="01-basics.html#sec-ecoli_data">data</a> were reanalysed using MaxQuant, which generates results at three levels: the evidence file containing the <a href="02-advanced.html#sec-psm_table">PSM table</a>, the <a href="01-basics.html#sec-peptide_table">peptides file</a>, and the protein group file.</p>
<section id="data-files" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="data-files"><span class="header-section-number">3.2.1</span> Data files</h3>
<p>We here retrieve those three data files.</p>
<p><strong>TODO</strong>: put data on Zenodo or MsDataHub, and update code below</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"MsDataHub"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># myurl &lt;- "https://github.com/statOmics/MSqRobSumPaper/raw/refs/heads/master/spikein/data/maxquant/peptides.zip"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># download.file(myurl,"data/sticker2020/peptides.zip", method = "curl", extra = "-L")</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># unzip("data/sticker2020/peptides.zip", exdir = "data/sticker2020/")</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># myurl &lt;- "https://github.com/statOmics/MSqRobSumPaper/raw/refs/heads/master/spikein/data/maxquant/evidence.zip"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># download.file(myurl,"data/sticker2020/evidence.zip", method = "curl", extra = "-L")</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># unzip("data/sticker2020/evidence.zip", exdir = "data/sticker2020/")</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># myurl &lt;- "https://github.com/statOmics/MSqRobSumPaper/raw/refs/heads/master/spikein/data/maxquant/proteinGroups.zip"</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># download.file(myurl,"data/sticker2020/proteinGroups.zip", method = "curl", extra = "-L")</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># unzip("data/sticker2020/proteinGroups.zip", exdir = "data/sticker2020/")</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>evidenceFile <span class="ot">&lt;-</span> <span class="st">"data/sticker2020/evidence.txt"</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>peptidesFile <span class="ot">&lt;-</span> <span class="st">"data/sticker2020/peptides.txt"</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>proteinGroupsFile <span class="ot">&lt;-</span> <span class="st">"data/sticker2020/proteinGroups.txt"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also load the <a href="01-basics.html#sec-annotation_table">annotation table</a>) that has been generated by the authors. Since the evidence, peptides and protein-groups tables all contain the same samples, the annotation table will be shared across the MaxQuant data tables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>annotFile <span class="ot">&lt;-</span> <span class="st">"data/sticker2020/sticker2020_annotation.csv"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(annotFile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We retrieve all the E. coli protein identifiers to later identify which proteins are known to be differentially abundant (E. coli proteins) or constant (human) across condition.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocFileCache"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> <span class="fu">BiocFileCache</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>ecoli <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/refs/heads/master/spikein/data/fasta/ecoli_up000000625_7_06_2018.fasta"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>ecoli <span class="ot">&lt;-</span> <span class="fu">readLines</span>(ecoli)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>ecoli <span class="ot">&lt;-</span> ecoli[<span class="fu">grepl</span>(<span class="st">"^&gt;"</span>, ecoli)]</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>ecoli <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"&gt;sp</span><span class="sc">\\</span><span class="st">|(.*)</span><span class="sc">\\</span><span class="st">|.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, ecoli)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="convert-to-qfeatures" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="convert-to-qfeatures"><span class="header-section-number">3.2.2</span> Convert to QFeatures</h3>
<p>We combine each MaxQuant file with the annotation table into a <a href="01-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>.</p>
<ul>
<li>Load and convert the evidence table.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">fread</span>(evidenceFile, <span class="at">check.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>runCol <span class="ot">&lt;-</span> coldata<span class="sc">$</span>Raw.file</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    evidence, <span class="at">colData =</span> coldata, <span class="at">runCol =</span> <span class="st">"Raw.file"</span>, </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">quantCols =</span> <span class="st">"Intensity"</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Load and convert the peptide table.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>peptides <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(peptidesFile)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>quantCols <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Intensity."</span>, coldata<span class="sc">$</span>Sample)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>peptides <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    peptides, <span class="at">colData =</span> coldata, <span class="at">name =</span> <span class="st">"peptides"</span>, </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">fnames =</span> <span class="st">"Sequence"</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Load and convert the protein-groups table</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>proteinGroups <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(proteinGroupsFile)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>proteinGroups <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    proteinGroups, <span class="at">colData =</span> coldata, <span class="at">name =</span> <span class="st">"proteinGroups"</span>,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">fnames =</span> <span class="st">"Protein.IDs"</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-preprocessing" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="data-preprocessing"><span class="header-section-number">3.3</span> Data preprocessing</h2>
<p>To exclude differences in data processing between the different data files, we will create a common data processing workflow in a dedicated function<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<section id="preprocessing-workflow" class="level3" data-number="3.3.1">
<h3 data-number="3.3.1" class="anchored" data-anchor-id="preprocessing-workflow"><span class="header-section-number">3.3.1</span> Preprocessing workflow</h3>
<p>We will use the same <code>QFeatures</code> data preprocessing workflow as presented in the <a href="01-basics.html#sec-basic_preprocess">basic concepts chapter</a> starting from the peptides table. However, we will account for the increased complexity in filtering when starting from PSM-level tables, where we need to exclude PSM mapping to multiple ions (as explained in the <a href="02-advanced.html#sec-duplicated_psms">advanced concepts chapter</a>). Conversely, the protein group table (containing LFQ normalised data) already underwent several data processing steps that will be ignored. We also need to provide the correct protein identifiers across data levels. We will use the <code>Proteins</code> column for the PSM-level and the peptide-level data, while we will use the <code>Protein.IDs</code> for the protein-group-level data.</p>
<p>Here is an overview of the data processing workflow:</p>
<ol type="1">
<li><a href="01-basics.html#sec-encode_missing">Encode</a> missing values with <code>NA</code></li>
<li>Perform <a href="01-basics.html#sec-log2">log2-transformation</a></li>
<li>(Only for PSM-level data) Remove <a href="02-advanced.html#sec-duplicated_psms">duplicated PSMs</a> for each ion and join the data from different runs, effectively leading to ion-level data.</li>
<li>(Only for protein-level data) Rename <code>Protein.IDs</code> to <code>Protein</code> for consistent protein identifier column name.</li>
<li><a href="01-basics.html#sec-filter">Filter features</a> by removing failed protein inference and removing decoys and contaminants.</li>
<li>Filter missing values, keeping features that are observed in at least 4 samples. This is the last step for the workflow starting from protein group data, because maxLFQ values are already normalised and summarised to protein values.</li>
<li>Perform <a href="01-basics.html#sec-norm">normalisation</a>.</li>
<li>Perform <a href="01-basics.html#sec-summarisation">summarisation</a>. The summarised values will only be used for protein-level modelling while the normalised data prior to normalisation will be used for ion-level and peptide-level modelling.</li>
</ol>
<p>The workflow is implemented in a functions with one argument, <code>object</code>, which is the <code>QFeatures</code> we generated, either from the evidence file, the peptides file or the protein-groups file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>preprocessing_workflow <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 1. Encode missing values</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">&lt;-</span> <span class="fu">names</span>(object) <span class="do">## store the input set names</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(object, i)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 2. Log transformation</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    logName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(i, <span class="st">"_log"</span>) <span class="do">## generate the log set names</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(object, i, <span class="at">name =</span> logName, <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 3. PSM filtering (only for psm-level data)</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(i) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> (ii <span class="cf">in</span> logName) {</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>            rowdata <span class="ot">&lt;-</span> <span class="fu">rowData</span>(object[[ii]]) </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>            rowdata<span class="sc">$</span>ionID <span class="ot">&lt;-</span> <span class="fu">paste0</span>(rowdata<span class="sc">$</span>Sequence, rowdata<span class="sc">$</span>Charge) </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>            rowdata<span class="sc">$</span>rowSums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(object[[ii]]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>            rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>                <span class="fu">group_by</span>(ionID) <span class="sc">|&gt;</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">psmRank =</span> <span class="fu">rank</span>(<span class="sc">-</span>rowSums))</span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rowData</span>(object[[ii]]) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(rowdata)    </span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a>        object <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(object, <span class="sc">~</span> psmRank <span class="sc">==</span> <span class="dv">1</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>        i <span class="ot">&lt;-</span> <span class="st">"evidence"</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a>        joinName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(i, <span class="st">"_log"</span>)</span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>        object <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>(object, logName, joinName, <span class="st">"ionID"</span>)</span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>        logName <span class="ot">&lt;-</span> joinName </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 4. Match protein identifiers across data levels</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (i <span class="sc">==</span> <span class="st">"proteinGroups"</span>) { </span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a>        <span class="do">## We need to match the protein IDs in the evidence/peptide</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a>        <span class="do">## table with the protiein IDs in the protein group table</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>        rowdata <span class="ot">&lt;-</span> <span class="fu">rbindRowData</span>(object, <span class="fu">names</span>(object))</span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>        rowdata<span class="sc">$</span>Proteins <span class="ot">&lt;-</span> rowdata<span class="sc">$</span>Protein.IDs</span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rowData</span>(object) <span class="ot">&lt;-</span> <span class="fu">split</span>(rowdata, rowdata<span class="sc">$</span>assay)</span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 5. Feature filtering</span></span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>        object, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>          <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Proteins) <span class="sc">&amp;</span> <span class="do">## Remove protein groups</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>          Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> <span class="do">## Remove decoys</span></span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a>          (Potential.contaminant <span class="sc">!=</span> <span class="st">"+"</span>) <span class="do">## Remove contaminants</span></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 6. Missing value filtering</span></span>
<span id="cb9-41"><a href="#cb9-41" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(object[[logName]])</span>
<span id="cb9-42"><a href="#cb9-42" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(object, <span class="at">i =</span> logName, <span class="at">pNA =</span> (n <span class="sc">-</span> <span class="dv">4</span>) <span class="sc">/</span> n)</span>
<span id="cb9-43"><a href="#cb9-43" aria-hidden="true" tabindex="-1"></a>    <span class="do">## (The steps below are not for protein-group data)</span></span>
<span id="cb9-44"><a href="#cb9-44" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (i <span class="sc">==</span> <span class="st">"proteinGroups"</span>) <span class="fu">return</span>(object)</span>
<span id="cb9-45"><a href="#cb9-45" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 7. Normalisation</span></span>
<span id="cb9-46"><a href="#cb9-46" aria-hidden="true" tabindex="-1"></a>    normName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(i, <span class="st">"_norm"</span>)</span>
<span id="cb9-47"><a href="#cb9-47" aria-hidden="true" tabindex="-1"></a>    pseudoRef <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(object[[logName]]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-48"><a href="#cb9-48" aria-hidden="true" tabindex="-1"></a>    nfLog <span class="ot">&lt;-</span> <span class="fu">sweep</span>(<span class="fu">assay</span>(object[[logName]]), <span class="at">MARGIN =</span> <span class="dv">1</span>, pseudoRef) <span class="sc">|&gt;</span> </span>
<span id="cb9-49"><a href="#cb9-49" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colMedians</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-50"><a href="#cb9-50" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">sweep</span>(</span>
<span id="cb9-51"><a href="#cb9-51" aria-hidden="true" tabindex="-1"></a>      object, <span class="at">MARGIN =</span> <span class="dv">2</span>, <span class="at">STATS =</span> nfLog, <span class="at">i =</span> logName, <span class="at">name =</span> normName</span>
<span id="cb9-52"><a href="#cb9-52" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-53"><a href="#cb9-53" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 8. Summarisation</span></span>
<span id="cb9-54"><a href="#cb9-54" aria-hidden="true" tabindex="-1"></a>    summName <span class="ot">&lt;-</span> <span class="fu">paste0</span>(i, <span class="st">"_proteins"</span>)</span>
<span id="cb9-55"><a href="#cb9-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aggregateFeatures</span>(</span>
<span id="cb9-56"><a href="#cb9-56" aria-hidden="true" tabindex="-1"></a>        object, <span class="at">i =</span> normName, <span class="at">name =</span> summName, <span class="at">fcol =</span> <span class="st">"Proteins"</span>,</span>
<span id="cb9-57"><a href="#cb9-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary</span>
<span id="cb9-58"><a href="#cb9-58" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb9-59"><a href="#cb9-59" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-the-workflow" class="level3" data-number="3.3.2">
<h3 data-number="3.3.2" class="anchored" data-anchor-id="run-the-workflow"><span class="header-section-number">3.3.2</span> Run the workflow</h3>
<p>Now that we have defined a common data processing function, we apply to each input data level: the ion data, the peptide data and the protein-group data. We then combine all the data in a single <code>QFeatures</code> object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">preprocessing_workflow</span>(evidence)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>peptides <span class="ot">&lt;-</span> <span class="fu">preprocessing_workflow</span>(peptides)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>proteinGroups <span class="ot">&lt;-</span> <span class="fu">preprocessing_workflow</span>(proteinGroups)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">c</span>(evidence, peptides, proteinGroups)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-modelling" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="data-modelling"><span class="header-section-number">3.4</span> Data modelling</h2>
<p>Similarly to the data processing, we will use the same modelling workflow for each data level. For ion-level data (starting from the evidence file) and peptide-level data, we will both model the data before and after summarisation. In other words, we will model the data at the ion/peptide level (as described in the <a href="02-advanced.html#sec-tmt_ion_model">advanced chapter</a>) and at the protein level (as described in the <a href="01-basics.html#sec-modelling">basics chapter</a>).</p>
<section id="model-definition" class="level3" data-number="3.4.1">
<h3 data-number="3.4.1" class="anchored" data-anchor-id="model-definition"><span class="header-section-number">3.4.1</span> Model definition</h3>
<p>Depending on the input level (ion, peptide or protein), different models are required. The simplest model is the protein model where we account for the effect of <code>Condition</code> (spike-in amount group) using a <a href="02-advanced.html#sec-fixed_effect">fixed effect</a>. For the peptide and the ion models, the data contains multiple peptides per proteins, and we expect that intensities from the same peptide are more similar than intensities from different peptides. Similarly, modelling multiple peptides per protein implies that we have multiple intensities per sample and hence that intensities from the same sample are more similar that intensities from different samples. The ion and peptide models therefore include a random effect for ion (<code>ionID</code>) and peptide (<code>Sequence</code>), and a random effect for sample (<code>Sample</code>). All three model definitions are stored in a list to streamline later access.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>models <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">protein =</span> <span class="sc">~</span> Condition,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">peptide =</span> <span class="sc">~</span> Condition <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sample) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sequence),</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">ion =</span> <span class="sc">~</span> Condition <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sample) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> ionID)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will benchmark the performance of the modelling approaches by comparing all possible combination of 2 spike-in conditions<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. All models assess the same comparisons of conditions, hence the same contrasts. We therefore build a <a href="01-basics.html#sec-hypothesis_testing">contrast matrix</a> that is shared across all models.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>allContrasts <span class="ot">&lt;-</span> <span class="fu">createPairwiseContrasts</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    models<span class="sc">$</span>protein, <span class="fu">colData</span>(spikein), <span class="at">var =</span> <span class="st">"Condition"</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    allContrasts, </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"ridgeConditionB"</span>, <span class="st">"ridgeConditionC"</span>, <span class="st">"ridgeConditionD"</span>, <span class="st">"ridgeConditionE"</span>)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="modelling-workflow" class="level3" data-number="3.4.2">
<h3 data-number="3.4.2" class="anchored" data-anchor-id="modelling-workflow"><span class="header-section-number">3.4.2</span> Modelling workflow</h3>
<p>We will use the same <code>msqrob2</code> data modelling and statistical inference workflow as presented in the <a href="01-basics.html#sec-msqrob">basic concepts chapter</a>. However, for ion-level and peptide-level models, we will rely on mixed models that have been introduced in the <a href="02-advanced.html#sec-run_model">advanced chapter</a>, including the <a href="02-advanced.html#sec-msqrob_refit">refitting</a> of proteins for which there is only 1 feature (ion or peptide). We implement the workflow in a function that we will call for all modelling approaches. It has 5 arguments:</p>
<ul>
<li><code>object</code> is the preprocessed <code>QFeatures</code> object containing the data to model.</li>
<li><code>i</code> is the name of the set to start (see <code>names(spikein)</code>).</li>
<li><code>model</code> is a formula defining the model to estimate.</li>
<li><code>L</code> is the contrast matrix to perform hypothesis testing.</li>
<li><code>modelLevel</code> indicates whether the model should be fit at the <code>"ion"</code>, <code>"peptide"</code> or <code>"protein"</code> level.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>modelling_workflow <span class="ot">&lt;-</span> <span class="cf">function</span>(object, i, model, L, modelLevel) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 1. Estimate the model</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (modelLevel <span class="sc">==</span> <span class="st">"protein"</span>) {</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>        object <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>            object,  <span class="at">i =</span> i, <span class="at">formula =</span> model,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>        <span class="do">## 1a. Estimate</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>        object <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>            object,  <span class="at">i =</span> i, <span class="at">formula =</span> model, </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">fcol =</span> <span class="st">"Proteins"</span>, <span class="at">name =</span> <span class="st">"msqrob"</span>,</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>        i <span class="ot">&lt;-</span> <span class="st">"msqrob"</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>        <span class="do">## 1b. Refit one-hit-wonders</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        counts <span class="ot">&lt;-</span> <span class="fu">aggcounts</span>(object[[<span class="st">"msqrob"</span>]])</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>        oneHitProteins <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts)[<span class="fu">rowMax</span>(counts) <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>        object <span class="ot">&lt;-</span> <span class="fu">msqrobRefit</span>(</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>          object, <span class="at">i =</span> i, <span class="at">subset =</span> oneHitProteins,</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">formula =</span> <span class="sc">~</span> Condition,</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">fcol =</span> <span class="st">"Proteins"</span>, <span class="at">name =</span> <span class="st">"msqrob"</span>,</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 2. Hypothesis testing</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>    object <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(object, i, <span class="at">contrast =</span> L)</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>    <span class="do">## 3. Collect the results</span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>    out <span class="ot">&lt;-</span> <span class="fu">msqrobCollect</span>(object[[i]], L, <span class="at">combine =</span> <span class="cn">TRUE</span>)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    out</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-the-workflow-1" class="level3" data-number="3.4.3">
<h3 data-number="3.4.3" class="anchored" data-anchor-id="run-the-workflow-1"><span class="header-section-number">3.4.3</span> Run the workflow</h3>
<p>We will now run the above function for the different approaches. To do so, we create a table containing the main modelling settings<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a>:</p>
<ul>
<li>Approach 1 (called <code>evidence_proteins</code>): model the data from the evidence file using the protein level data.</li>
<li>Approach 2 (called <code>evidence_norm</code>): model the data from the evidence file using the normalised ion data</li>
<li>Approach 3 (called <code>peptides_proteins</code>): model the data from the peptides file using the protein level data.</li>
<li>Approach 4 (called <code>peptides_norm</code>): model the data from the peptides file using the normalised peptide data.</li>
<li>Approach 5 (called <code>proteinGroups_log</code>): model the data from the protein groups file using the protein level data.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>(approaches <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">inputLevel =</span> <span class="fu">c</span>(<span class="st">"evidence_proteins"</span>, <span class="st">"evidence_norm"</span>, <span class="st">"peptides_proteins"</span>, <span class="st">"peptides_norm"</span>, <span class="st">"proteinGroups_log"</span>),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelLevel =</span> <span class="fu">c</span>(<span class="st">"protein"</span>, <span class="st">"ion"</span>, <span class="st">"protein"</span>, <span class="st">"peptide"</span>, <span class="st">"protein"</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         inputLevel modelLevel
1 evidence_proteins    protein
2     evidence_norm        ion
3 peptides_proteins    protein
4     peptides_norm    peptide
5 proteinGroups_log    protein</code></pre>
</div>
</div>
<p>For each row of this table, we retrieve the different arguments and run our data modelling workflow. Notice how every approach is run using the same code. Every <code>apply</code> iteration returns a table of statistical inference results for all pairwise comparisons. We combine all the tables in a single table for result exploration.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">apply</span>(approaches, <span class="dv">1</span>, <span class="cf">function</span>(x) {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    out <span class="ot">&lt;-</span> <span class="fu">modelling_workflow</span>(</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>        spikein, <span class="at">i =</span> x[[<span class="st">"inputLevel"</span>]], <span class="at">model =</span> models[[x[[<span class="st">"modelLevel"</span>]]]],</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">L =</span> L, <span class="at">modelLevel =</span> x[[<span class="st">"modelLevel"</span>]]</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    out<span class="sc">$</span>inputLevel <span class="ot">&lt;-</span> x[[<span class="st">"inputLevel"</span>]]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    out<span class="sc">$</span>modelLevel <span class="ot">&lt;-</span> x[[<span class="st">"modelLevel"</span>]]</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    out</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also add whether each modelled protein is a E. Coli protein (known to be differentially abundant) or not. We also simplify the naming of the contrasts for visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>isEcoli <span class="ot">&lt;-</span> results<span class="sc">$</span>feature <span class="sc">%in%</span> ecoli</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>contrast <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, results<span class="sc">$</span>contrast)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>results<span class="sc">$</span>contrast <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^([B-E])$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1 - A"</span>, results<span class="sc">$</span>contrast)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="sec-performance" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="sec-performance"><span class="header-section-number">3.5</span> Performance benchmark</h2>
<p>We now can compare the performance of the different modelling approaches. We will compare the approaches based on 4 objectives criteria:</p>
<ol type="1">
<li>The number of fitted proteins across conditions.</li>
<li>The number of true positives and false positives at a 5% FDR threshold.</li>
<li>The sensitivity against the rate of missidentification.</li>
<li>The accuracy and pricision of the log2-fold change estimatione</li>
</ol>
<section id="number-of-fits" class="level3" data-number="3.5.1">
<h3 data-number="3.5.1" class="anchored" data-anchor-id="number-of-fits"><span class="header-section-number">3.5.1</span> Number of fits</h3>
<p>We compare the number of proteins that could be estimated by each approach, where a model that fits more proteins is preferred. Note that the number of fits exceeds the number of measured proteins because we summed the fits over all comparisons.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">group_by</span>(results, inputLevel) <span class="sc">|&gt;</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(adjPval))) <span class="sc">|&gt;</span> </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> inputLevel, <span class="at">y =</span> n, <span class="at">fill =</span> inputLevel) <span class="sc">+</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_nfits-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can see that the approaches that fits least proteins is when starting from the protein-group file. However, all other four approaches can fit the same number of proteins.</p>
</section>
<section id="tp-and-fp-at-5-fdr" class="level3" data-number="3.5.2">
<h3 data-number="3.5.2" class="anchored" data-anchor-id="tp-and-fp-at-5-fdr"><span class="header-section-number">3.5.2</span> TP and FP at 5% FDR</h3>
<p>An approach that fits more proteins is preferred, provided that the additional fits lead to meaningful results. Because this data set contains ground truth information, we can assess whether the modelling approaches correctly prioritised the proteins given the known differential abundant proteins. We therefore benchmark the approaches by examining the number of reported proteins that are true positive (TP) (i.e., E. Coli proteins), and false positive (FP) (i.e., human proteins) considering a 5% false discovery rate (FDR) threshold, which is typically used. We will therefore first construct the table with TPs and FPs obtained from each data modelling approach for each comparison.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>tpFpTable <span class="ot">&lt;-</span> <span class="fu">group_by</span>(results, inputLevel, contrast) <span class="sc">|&gt;</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="st">"True Positives"</span> <span class="ot">=</span> <span class="fu">sum</span>(isEcoli),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>              <span class="st">"False Positives"</span> <span class="ot">=</span> <span class="fu">sum</span>(<span class="sc">!</span>isEcoli)) <span class="sc">|&gt;</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"True Positives"</span>, <span class="st">"False Positives"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We then plot the table as a bar plot, facetting for every comparison.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(tpFpTable) <span class="sc">+</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> inputLevel,</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">fill =</span> inputLevel) <span class="sc">+</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(contrast <span class="sc">~</span> name, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">nrow =</span> <span class="dv">3</span>)) <span class="do">## to avoid legend getting cropped</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_tp_fp-1.png" class="img-fluid figure-img" width="384"></p>
</figure>
</div>
</div>
</div>
<p>The plot clearly shows that starting from MaxQuantâ€™s proteinGroups file leads to a severe decrease in performance as the number of TP is systematically lower compared to other approaches, while there is no dramatic difference with the number of FPs.</p>
<p>The four other approaches lead to comparable results. We could argue that for some specific comparisons (C-B, D-C, E-D) approaches starting from the evidence file recover slightly more TP without impact on the number of FP. Similarly modelling at the peptide/ion level leads to slight increase in performance compared to modelling at the protein level, but these difference are subtle.</p>
</section>
<section id="sec-tpr_fdp" class="level3" data-number="3.5.3">
<h3 data-number="3.5.3" class="anchored" data-anchor-id="sec-tpr_fdp"><span class="header-section-number">3.5.3</span> TPR-FDP curves</h3>
<p>Additionally, we construct true positive rate (TPR)-false discovery proportion (FDP) plots. TPR represents the fraction of truly DA proteins reported by the method, calculated as <span class="math inline">\(TPR =
\frac{TP}{TP+FN}\)</span>, with FN false negatives (i.e., E. Coli proteins that were not flagged as differential abundant). FDP denotes the proportion of false positives among all proteins flagged as differential abundant, calculated as <span class="math inline">\(FDP = \frac{FP}{TP + FP}\)</span>.</p>
<p>So we first compute the FDP and TPR using the custom functions below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>computeFDP <span class="ot">&lt;-</span> <span class="cf">function</span>(pval, tp) {</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>    ord <span class="ot">&lt;-</span> <span class="fu">order</span>(pval)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>    fdp <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="sc">!</span>tp[ord]) <span class="sc">/</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(tp)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>    fdp[<span class="fu">order</span>(ord)]</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>computeTPR <span class="ot">&lt;-</span> <span class="cf">function</span>(pval, tp, <span class="at">nTP =</span> <span class="cn">NULL</span>) {</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(nTP)) nTP <span class="ot">&lt;-</span> <span class="fu">sum</span>(tp)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>    ord <span class="ot">&lt;-</span> <span class="fu">order</span>(pval)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>    tpr <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(tp[ord]) <span class="sc">/</span> nTP</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>    tpr[<span class="fu">order</span>(ord)]</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Before computing these metrics, we first remove any failed inference<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a>. This means that we are comparing the approach based on a set of proteins that is specific to each approach. However, proteins that are fit by some approaches but not by others may be harder to estimate and hence will decrease the overall performance. So, for a fair comparison, we also created a plot that compares the approaches when considering all the proteins measured in the data set (see the <a href="#sec-appendix3_1">appendix section</a>). Any failed inference for a DA protein will results in a FP. Similarly, we could also compare the approaches based on a common set of proteins that has been fit by all approaches (see the following <a href="#sec-appendix3_2">appendix section</a>).</p>
<p>We compute the TPR and FDP for each approach (given by <code>inputLevel</code>) and each pairwise spike-in comparison (given by <code>contrast</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>performance <span class="ot">&lt;-</span> <span class="fu">group_by</span>(results, inputLevel, contrast) <span class="sc">|&gt;</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">na.exclude</span>() <span class="sc">|&gt;</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">tpr =</span> <span class="fu">computeTPR</span>(pval, isEcoli),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">fdp =</span> <span class="fu">computeFDP</span>(pval, isEcoli)) <span class="sc">|&gt;</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(fdp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also highlight the observed FDP at a 5% FDR threshold. Since the FDR represent the expected FDP, i.e.&nbsp;the average of the FDPs obtained when the spike-in experiment were to be repeated an infinite number of times, an observed FDP that is very far away from 5% is indicative for a workflow that provides poor FDR control.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>workPoints <span class="ot">&lt;-</span> <span class="fu">group_by</span>(performance, inputLevel, contrast) <span class="sc">|&gt;</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(adjPval) <span class="sc">|&gt;</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(inputLevel))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now generate the TPR-FDP curves. The best performing approach is characterised by the largest area under the curve. These curves provide the performance over a range of FDP values, but we limit the plot to the <span class="math inline">\([0, 0.2]\)</span> range because researchers are rarely interest in the performance when the FDP exceeds 20%.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance) <span class="sc">+</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> fdp,</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> tpr,</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> inputLevel) <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> workPoints, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> contrast) <span class="sc">+</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_tpr_fdp_curve1-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>The results are in line with the previous bechmark criteria, that is starting from the proteinGroups file lead to a severe backlash on performance. The other approaches show comparable performance, although for challenging comparisons where the performance is generally low (C-B, D-C, E-D) we find a subtle increase in performance for ion/peptide level models.</p>
<p>Overal, we find that all approaches lead to a good FDR control. A notable exception is for challenging comparisons when starting from protein group data, where the performance is so low that FDR cannot be controlled. We also find that the other approaches tend to be conservative for the D-C and E-D comparisons.</p>
<p>The same results apply when considering <a href="{#sec-appendix3_1">all proteins in the data</a> or when considering <a href="{#sec-appendix3_2">proteins estimated by all approaches</a>, indicating that the conclusions do not depend on the set of proteins considered.</p>
</section>
<section id="sec-benchmark_with_boxplots" class="level3" data-number="3.5.4">
<h3 data-number="3.5.4" class="anchored" data-anchor-id="sec-benchmark_with_boxplots"><span class="header-section-number">3.5.4</span> Fold change boxplots</h3>
<p>Next to correctly prioritising the differentially abundant proteins, another object is to correctly estimate the log2-fold change between conditions. Since every condition contains E. Coli proteins that have been spiked in experimentally controlled amounts, we know the real log2-fold change between any two conditions. We will explore the model accuracy, i.e.&nbsp;how close the estimations are from the true value on average, and the model precision, i.e.&nbsp;how narrow the estimations are spread around the average estimation. In this data set, there are two target values. For E. Coli proteins, the expected value is the experimentally induced log2-fold change. For human proteins, the expected value is a log2-fold change of 0, as these proteins are experimentally known to be constant.</p>
<p>We explore the results using boxplots of the estimated log2-fold changes, but we first create a small table with the expected values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>realLogFC <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">logFC =</span> <span class="fu">t</span>(L) <span class="sc">%*%</span> <span class="fu">lm</span>(<span class="fu">log2</span>(Concentration) <span class="sc">~</span> Condition, <span class="fu">colData</span>(spikein))<span class="sc">$</span>coef[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>realLogFC<span class="sc">$</span>contrast <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>,<span class="st">""</span>,<span class="fu">colnames</span>(L))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>realLogFC<span class="sc">$</span>contrast <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^([B-E])$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1 - A"</span>, realLogFC<span class="sc">$</span>contrast)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now create the boxplots with the estimated log2-fold changes, adding horizontal lines with the corresponding target values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(results) <span class="sc">+</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">y =</span> logFC,</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> isEcoli,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">colour =</span> isEcoli) <span class="sc">+</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()  <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Human proteins"</span>, <span class="st">"E. coli proteins"</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(contrast <span class="sc">~</span> inputLevel, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">data =</span> realLogFC, <span class="fu">aes</span>(<span class="at">yintercept =</span> logFC), </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">"firebrick"</span>) <span class="sc">+</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_boxplot-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p>For all approaches, the boxplots are roughly centred on the expected value, indicating good model accuracy. Starting from MaxQuantâ€™s protein-group data leads to wider boxplots, hence less stable estimation and hence decreased precision. The approach starting from the evidence file seems to provide somewhat more precise estimations as it shows less outliers compared to starting from the peptides file.</p>
</section>
</section>
<section id="sec-take_home" class="level2" data-number="3.6">
<h2 data-number="3.6" class="anchored" data-anchor-id="sec-take_home"><span class="header-section-number">3.6</span> Take home messages</h2>
<p>We found a striking drop in performance when starting from the protein-group data compared to the other approaches, suggesting that this approach leads to suboptimal results. We couldnâ€™t find strong differences in performance between the remaining apporaches. For some comparisons, we found a slight increase in performance when starting from the evidence file compared to starting from the peptides file. Similarly, for some comparisons, we found a slight increase in performance when modelling data at the ion/peptide level compared to the protein level.</p>
<p>Processing the data from the evidence file only leads to one additional step compared to processing the data from the peptides file: we need to find a low-level feature definition that is shared across rus, which we here defined as the ion level (i.e.&nbsp;the combination of the peptide sequence and its charge). This additional step has little impact on the complexity of the workflow, hence we would advice to start from the evidence file. However, modelling the data at the ion/peptide level is more advanced than modelling at the protein level, as it requires the inclusion of random effects to account for the correlation structure within and between samples and within and between ions/peptides from the same protein. Moreover, the data modelling is performed at the ion/peptide level, but the statistical inference results are reported at the protein level, meaning that no direct protein summaries are available to explore and support the statistical outcome<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a>. Therefore, we leave it to the user to decide whether the slight improvement in performance is worth the cost of more complex statistical analysis <span class="citation" data-cites="Sticker2020-rl">(<a href="99-end.html#ref-Sticker2020-rl" role="doc-biblioref">Sticker et al. 2020</a>)</span>.</p>
<p>Hence, this chapter showed how to perform benchmarking on different types of data input. Note that the same framework could be used to compare different search and quantification engines. Similarly, this framework can also be applied to compare different instruments or analytical protocols and setups. In the <a href="sec-workflow_optimisation">next chapter</a> we demonstrate how to compare the impact of analysis steps for the same data.</p>
<p>In the remainder of this section, we provide a recap on how to perform a proteomics analysis from the evidence file<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>, either at the ion-level or at the protein level.</p>
<section id="sec-preprocess_evidence" class="level3" data-number="3.6.1">
<h3 data-number="3.6.1" class="anchored" data-anchor-id="sec-preprocess_evidence"><span class="header-section-number">3.6.1</span> Preprocessing the evidence file</h3>
<p>The first step is to read the data. Remember that we need two pieces of data, the <a href="01-basics.html#sec-annotation_table">sample annotation table</a> and, in this case, the <a href="02-advanced.html#sec-psm_table">PSM table</a> obtained after reading MaxQuantâ€™s evidence file.</p>
<p>Here are the first 6 lines (first 6 samples) of the sample annotations, note we added <code>runCol</code> and <code>quantCol</code> that are required for the <a href="02-advanced.html#sec_advanced_readqf">conversion to a <code>QFeatures</code> object</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(annotFile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 67%">
<col style="width: 10%">
<col style="width: 7%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Raw.file</th>
<th style="text-align: left;">Condition</th>
<th style="text-align: left;">Sample</th>
<th style="text-align: right;">Concentration</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">B03_03_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">c1</td>
<td style="text-align: right;">6.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">B03_08_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">c2</td>
<td style="text-align: right;">6.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B03_18_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">C</td>
<td style="text-align: left;">c3</td>
<td style="text-align: right;">6.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">B03_19_150304_human_ecoli_B_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">B</td>
<td style="text-align: left;">b1</td>
<td style="text-align: right;">4.5</td>
</tr>
<tr class="odd">
<td style="text-align: left;">B03_07_150304_human_ecoli_D_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">d1</td>
<td style="text-align: right;">7.5</td>
</tr>
<tr class="even">
<td style="text-align: left;">B03_14_150304_human_ecoli_D_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">D</td>
<td style="text-align: left;">d2</td>
<td style="text-align: right;">7.5</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Here are the first 6 lines (first 6 PSMs) of the PSM table. The table contains many columns, most containing information about the identified peptide and the quality of the spectrum matching.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">fread</span>(evidenceFile, <span class="at">check.names =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 2%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Sequence</th>
<th style="text-align: right;">Length</th>
<th style="text-align: left;">Modifications</th>
<th style="text-align: left;">Modified.sequence</th>
<th style="text-align: left;">Oxidation..M..Probabilities</th>
<th style="text-align: left;">Oxidation..M..Score.Diffs</th>
<th style="text-align: right;">Acetyl..Protein.N.term.</th>
<th style="text-align: right;">Oxidation..M.</th>
<th style="text-align: right;">Missed.cleavages</th>
<th style="text-align: left;">Proteins</th>
<th style="text-align: left;">Leading.proteins</th>
<th style="text-align: left;">Leading.razor.protein</th>
<th style="text-align: left;">Gene.names</th>
<th style="text-align: left;">Protein.names</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Raw.file</th>
<th style="text-align: left;">Experiment</th>
<th style="text-align: right;">MS.MS.m.z</th>
<th style="text-align: right;">Charge</th>
<th style="text-align: right;">m.z</th>
<th style="text-align: right;">Mass</th>
<th style="text-align: right;">Resolution</th>
<th style="text-align: right;">Uncalibratedâ€¦Calibrated.m.z..ppm.</th>
<th style="text-align: right;">Uncalibratedâ€¦Calibrated.m.z..Da.</th>
<th style="text-align: right;">Mass.error..ppm.</th>
<th style="text-align: right;">Mass.error..Da.</th>
<th style="text-align: right;">Uncalibrated.mass.error..ppm.</th>
<th style="text-align: right;">Uncalibrated.mass.error..Da.</th>
<th style="text-align: right;">Max.intensity.m.z.0</th>
<th style="text-align: right;">Retention.time</th>
<th style="text-align: right;">Retention.length</th>
<th style="text-align: right;">Calibrated.retention.time</th>
<th style="text-align: right;">Calibrated.retention.time.start</th>
<th style="text-align: right;">Calibrated.retention.time.finish</th>
<th style="text-align: right;">Retention.time.calibration</th>
<th style="text-align: right;">Match.time.difference</th>
<th style="text-align: right;">Match.m.z.difference</th>
<th style="text-align: right;">Match.q.value</th>
<th style="text-align: right;">Match.score</th>
<th style="text-align: right;">Number.of.data.points</th>
<th style="text-align: right;">Number.of.scans</th>
<th style="text-align: right;">Number.of.isotopic.peaks</th>
<th style="text-align: right;">PIF</th>
<th style="text-align: right;">Fraction.of.total.spectrum</th>
<th style="text-align: right;">Base.peak.fraction</th>
<th style="text-align: right;">PEP</th>
<th style="text-align: right;">MS.MS.count</th>
<th style="text-align: right;">MS.MS.scan.number</th>
<th style="text-align: right;">Score</th>
<th style="text-align: right;">Delta.score</th>
<th style="text-align: right;">Combinatorics</th>
<th style="text-align: right;">Intensity</th>
<th style="text-align: left;">Reverse</th>
<th style="text-align: left;">Potential.contaminant</th>
<th style="text-align: right;">id</th>
<th style="text-align: left;">Protein.group.IDs</th>
<th style="text-align: right;">Peptide.ID</th>
<th style="text-align: right;">Mod..peptide.ID</th>
<th style="text-align: left;">MS.MS.IDs</th>
<th style="text-align: right;">Best.MS.MS</th>
<th style="text-align: left;">AIF.MS.MS.IDs</th>
<th style="text-align: left;">Oxidation..M..site.IDs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MULTI-SECPEP</td>
<td style="text-align: left;">B03_03_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">c1</td>
<td style="text-align: right;">532.9854</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">532.9533</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">94394.69</td>
<td style="text-align: right;">-0.5225600</td>
<td style="text-align: right;">-0.0002785</td>
<td style="text-align: right;">0.84218</td>
<td style="text-align: right;">0.0004488</td>
<td style="text-align: right;">0.31962</td>
<td style="text-align: right;">0.0001703</td>
<td style="text-align: right;">532.9553</td>
<td style="text-align: right;">76.475</td>
<td style="text-align: right;">0.19265</td>
<td style="text-align: right;">76.426</td>
<td style="text-align: right;">76.307</td>
<td style="text-align: right;">76.500</td>
<td style="text-align: right;">-0.049773</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0013949</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">31302</td>
<td style="text-align: right;">46.133</td>
<td style="text-align: right;">33.507</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4268500</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MULTI-SECPEP</td>
<td style="text-align: left;">B03_08_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">c2</td>
<td style="text-align: right;">532.9863</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">532.9533</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">89387.45</td>
<td style="text-align: right;">0.0021322</td>
<td style="text-align: right;">0.0000011</td>
<td style="text-align: right;">-0.25348</td>
<td style="text-align: right;">-0.0001351</td>
<td style="text-align: right;">-0.25135</td>
<td style="text-align: right;">-0.0001340</td>
<td style="text-align: right;">533.2871</td>
<td style="text-align: right;">76.225</td>
<td style="text-align: right;">0.16901</td>
<td style="text-align: right;">76.426</td>
<td style="text-align: right;">76.314</td>
<td style="text-align: right;">76.483</td>
<td style="text-align: right;">0.200350</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0069352</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">31103</td>
<td style="text-align: right;">38.377</td>
<td style="text-align: right;">32.662</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7099400</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MSMS</td>
<td style="text-align: left;">B03_18_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">c4</td>
<td style="text-align: right;">798.9761</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">798.9263</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">76.225</td>
<td style="text-align: right;">1.00000</td>
<td style="text-align: right;">76.639</td>
<td style="text-align: right;">76.139</td>
<td style="text-align: right;">77.139</td>
<td style="text-align: right;">0.413920</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">31830</td>
<td style="text-align: right;">98.407</td>
<td style="text-align: right;">82.183</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MSMS</td>
<td style="text-align: left;">B03_19_150304_human_ecoli_B_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">b4</td>
<td style="text-align: right;">798.9767</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">798.9263</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">75.981</td>
<td style="text-align: right;">1.00000</td>
<td style="text-align: right;">76.733</td>
<td style="text-align: right;">76.233</td>
<td style="text-align: right;">77.233</td>
<td style="text-align: right;">0.752060</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">31810</td>
<td style="text-align: right;">71.241</td>
<td style="text-align: right;">53.805</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MULTI-MATCH</td>
<td style="text-align: left;">B03_07_150304_human_ecoli_D_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">d2</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">532.9533</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">83252.61</td>
<td style="text-align: right;">-0.4043600</td>
<td style="text-align: right;">-0.0002155</td>
<td style="text-align: right;">-0.31288</td>
<td style="text-align: right;">-0.0001667</td>
<td style="text-align: right;">-0.71725</td>
<td style="text-align: right;">-0.0003823</td>
<td style="text-align: right;">532.9532</td>
<td style="text-align: right;">76.332</td>
<td style="text-align: right;">0.11982</td>
<td style="text-align: right;">76.533</td>
<td style="text-align: right;">76.433</td>
<td style="text-align: right;">76.553</td>
<td style="text-align: right;">0.200930</td>
<td style="text-align: right;">-0.016489</td>
<td style="text-align: right;">-0.0001398</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">37.285</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8563700</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAAAAAAAAAAAAAAGAGAGAK</td>
<td style="text-align: right;">22</td>
<td style="text-align: left;">Unmodified</td>
<td style="text-align: left;"><em>AAAAAAAAAAAAAAAGAGAGAK</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">P55011</td>
<td style="text-align: left;">SLC12A2</td>
<td style="text-align: left;">Solute carrier family 12 member 2</td>
<td style="text-align: left;">MULTI-MATCH</td>
<td style="text-align: left;">B03_14_150304_human_ecoli_D_3ul_3um_column_95_HCD_OT_2hrs_30B_9B</td>
<td style="text-align: left;">d3</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">532.9533</td>
<td style="text-align: right;">1595.838</td>
<td style="text-align: right;">98783.58</td>
<td style="text-align: right;">-0.3048300</td>
<td style="text-align: right;">-0.0001625</td>
<td style="text-align: right;">-0.31827</td>
<td style="text-align: right;">-0.0001696</td>
<td style="text-align: right;">-0.62310</td>
<td style="text-align: right;">-0.0003321</td>
<td style="text-align: right;">532.9531</td>
<td style="text-align: right;">76.046</td>
<td style="text-align: right;">0.10846</td>
<td style="text-align: right;">76.498</td>
<td style="text-align: right;">76.403</td>
<td style="text-align: right;">76.511</td>
<td style="text-align: right;">0.451640</td>
<td style="text-align: right;">-0.051605</td>
<td style="text-align: right;">-0.0001484</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">37.285</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">NaN</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6597000</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">2115</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Note that <code>Intensity</code> column contains the quantitative values and the <code>Raw.file</code> column indicates in which run the sample was acquired. We use the latter to link the sample annotation with the PSM table. We therefore need to add <code>runCol</code> to the sample annotation that will serve as the linker. We can then convert the tables to a <code>QFeatures</code> object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>runCol <span class="ot">&lt;-</span> coldata<span class="sc">$</span>Raw.file</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>(evidence <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>    evidence, <span class="at">colData =</span> coldata, <span class="at">runCol =</span> <span class="st">"Raw.file"</span>, </span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">quantCols =</span> <span class="st">"Intensity"</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 20 sets:

 [1] B03_02_150304_human_ecoli_B_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 40057 rows and 1 columns 
 [2] B03_03_150304_human_ecoli_C_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 41266 rows and 1 columns 
 [3] B03_04_150304_human_ecoli_D_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 41396 rows and 1 columns 
 ...
 [18] B03_19_150304_human_ecoli_B_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 39388 rows and 1 columns 
 [19] B03_20_150304_human_ecoli_A_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 39000 rows and 1 columns 
 [20] B03_21_150304_human_ecoli_A_3ul_3um_column_95_HCD_OT_2hrs_30B_9B: SummarizedExperiment with 38783 rows and 1 columns </code></pre>
</div>
</div>
<p>Note that the data from every run is contained in a separate set. We cannot yet join the sets together since we donâ€™t have a specific feature identifier, yet<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a>.</p>
<ol type="1">
<li>Encoding missing values as zeros.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(evidence, <span class="fu">names</span>(evidence))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li>Log2 transforming</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>inputNames <span class="ot">&lt;-</span> <span class="fu">names</span>(evidence)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>logNames <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputNames, <span class="st">"_log"</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(evidence, inputNames, <span class="at">name =</span> logNames, <span class="at">base =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="3" type="1">
<li>Keeping only the most intense PSM per ion (see <a href="02-advanced.html#sec-duplicated_psms">here</a> for a step-by-step explanation of the code). Upon this filtering every feature is unique to a ion identifier (peptide sequence + charge), and we hence can join sets using that identifier.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> logNames) {</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  rowdata <span class="ot">&lt;-</span> <span class="fu">rowData</span>(evidence[[i]]) </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  rowdata<span class="sc">$</span>ionID <span class="ot">&lt;-</span> <span class="fu">paste0</span>(rowdata<span class="sc">$</span>Sequence, rowdata<span class="sc">$</span>Charge) </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  rowdata<span class="sc">$</span>value <span class="ot">&lt;-</span> <span class="fu">assay</span>(evidence[[i]])[, <span class="dv">1</span>]</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(ionID) <span class="sc">|&gt;</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">psmRank =</span> <span class="fu">rank</span>(<span class="sc">-</span>value))</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowData</span>(evidence[[i]])<span class="sc">$</span>psmRank <span class="ot">&lt;-</span> rowdata<span class="sc">$</span>psmRank</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowData</span>(evidence[[i]])<span class="sc">$</span>ionID <span class="ot">&lt;-</span> rowdata<span class="sc">$</span>ionID</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(evidence, <span class="sc">~</span> psmRank <span class="sc">==</span> <span class="dv">1</span>, <span class="at">keep =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>(evidence, logNames, <span class="st">"ions_log"</span>, <span class="st">"ionID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="4" type="1">
<li>Feature filtering</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>        evidence, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>          <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Proteins) <span class="sc">&amp;</span> <span class="do">## Remove protein groups</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>          Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> <span class="do">## Remove decoys</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>          (Potential.contaminant <span class="sc">!=</span> <span class="st">"+"</span>) <span class="do">## Remove contaminants</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="5" type="1">
<li>Missing value filtering</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(evidence[[<span class="st">"ions_log"</span>]])</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(evidence, <span class="at">i =</span> <span class="st">"ions_log"</span>, <span class="at">pNA =</span> (n <span class="sc">-</span> <span class="dv">4</span>) <span class="sc">/</span> n)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="6" type="1">
<li>Normalisation</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>pseudoRef <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">assay</span>(evidence[[<span class="st">"ions_log"</span>]]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>nfLog <span class="ot">&lt;-</span> <span class="fu">sweep</span>(<span class="fu">assay</span>(evidence[[<span class="st">"ions_log"</span>]]), <span class="at">MARGIN =</span> <span class="dv">1</span>, pseudoRef) <span class="sc">|&gt;</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colMedians</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">sweep</span>(</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  evidence, <span class="at">MARGIN =</span> <span class="dv">2</span>, <span class="at">STATS =</span> nfLog, </span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">i =</span> <span class="st">"ions_log"</span>, <span class="at">name =</span> <span class="st">"ions_norm"</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="7" type="1">
<li>Summarisation</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  evidence, <span class="at">i =</span> <span class="st">"ions_norm"</span>, <span class="at">name =</span> <span class="st">"proteins"</span>, <span class="at">fcol =</span> <span class="st">"Proteins"</span>,</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="modelling-the-preprocessed-data" class="level3" data-number="3.6.2">
<h3 data-number="3.6.2" class="anchored" data-anchor-id="modelling-the-preprocessed-data"><span class="header-section-number">3.6.2</span> Modelling the preprocessed data</h3>
<p>We can model the data either at the ion level or at the protein level. Regardless of the modelling approach, we can readily generate the <a href="01-basics.html#sec-multiple_contrasts">contrast matrix</a> to assess all pairwise comparisons between the experimental spike-in conditions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>allContrasts <span class="ot">&lt;-</span> <span class="fu">createPairwiseContrasts</span>(</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    <span class="sc">~</span> Condition, <span class="fu">colData</span>(evidence), <span class="at">var =</span> <span class="st">"Condition"</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    allContrasts, </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"ridgeConditionB"</span>, <span class="st">"ridgeConditionC"</span>, <span class="st">"ridgeConditionD"</span>, <span class="st">"ridgeConditionE"</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="modelling-at-the-protein-level" class="level4">
<h4 class="anchored" data-anchor-id="modelling-at-the-protein-level">Modelling at the protein-level</h4>
<p>We first need to define the model we want to estimate, which descirbes the sources of variation in the data. For the protein-level data, the only potential source of variation identified from the experiment is the spike-in condition of interest. We model it as a fixed effect.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>modelProtein <span class="ot">&lt;-</span> <span class="er">~</span> Condition</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then, we estimate the model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  evidence,  <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">formula =</span> modelProtein,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we perform statistical inference on the estimated model parameters.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(evidence, <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>resultsProtein <span class="ot">&lt;-</span> <span class="fu">msqrobCollect</span>(evidence[[<span class="st">"proteins"</span>]], L, <span class="at">combine =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>resultsProtein<span class="sc">$</span>isEcoli <span class="ot">&lt;-</span> resultsProtein<span class="sc">$</span>feature <span class="sc">%in%</span> ecoli</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note that this workflows closely follows the workflow described in the <a href="01-basics.html#sec-multiple_contrasts">basics chapter</a>. We can report the result using a volcano plot, for instance.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(resultsProtein) <span class="sc">+</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>,</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> isEcoli) <span class="sc">+</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> contrast, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_volcano_proteins-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="sec-ion_level_modelling" class="level4">
<h4 class="anchored" data-anchor-id="sec-ion_level_modelling">Modelling at the ion-level</h4>
<p>The model definition for the ion-level data is more ellaborate than for the protein-level data. We need to account for the fact that intensities within the same sample are more correlated than intensities across samples. Similarly, we need to account for the fact that intensities for the same ion will be more similar than intensities between ions of the same proteins. On top of the condition effect that we model as a fixed effect, we will account for these sample and ion effects using a random effect.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>modelIon <span class="ot">&lt;-</span> <span class="er">~</span> Condition <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Sample) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> ionID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then, we estimate the model using <code>msqrobAggregate()</code> instead of <code>msqrob()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  evidence,  <span class="at">i =</span> <span class="st">"ions_norm"</span>, <span class="at">formula =</span> modelIon, </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">"Proteins"</span>, <span class="at">name =</span> <span class="st">"msqrob"</span>,</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Because some proteins are only measured by a single ion, its corresponding sample and ion effects cannot be estimated and hence the model for those proteins will not be estimated. We therefore refit a simplified model for those proteins using the refitting workflow described in the <a href="02-advanced.html#sec-msqrob_refit">advanced chapter</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">aggcounts</span>(evidence[[<span class="st">"msqrob"</span>]])</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>oneHitProteins <span class="ot">&lt;-</span> <span class="fu">rownames</span>(counts)[<span class="fu">rowMax</span>(counts) <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">msqrobRefit</span>(</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  evidence, <span class="at">i =</span> <span class="st">"ions_norm"</span>, <span class="at">subset =</span> oneHitProteins,</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> Condition,</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">"Proteins"</span>, <span class="at">name =</span> <span class="st">"msqrob"</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">robust =</span> <span class="cn">TRUE</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we perform statistical inference on the estimated model parameters (same as above).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>evidence <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(evidence, <span class="st">"msqrob"</span>, <span class="at">contrast =</span> L)</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>resultsIon <span class="ot">&lt;-</span> <span class="fu">msqrobCollect</span>(evidence[[<span class="st">"msqrob"</span>]], L, <span class="at">combine =</span> <span class="cn">TRUE</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>resultsIon<span class="sc">$</span>isEcoli <span class="ot">&lt;-</span> resultsIon<span class="sc">$</span>feature <span class="sc">%in%</span> ecoli</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can report the result using a volcano plot, for instance.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(resultsIon) <span class="sc">+</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">shape =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>,</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">color =</span> isEcoli) <span class="sc">+</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> contrast, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_volcano_ion-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="appendix" class="level2" data-number="3.7">
<h2 data-number="3.7" class="anchored" data-anchor-id="appendix"><span class="header-section-number">3.7</span> Appendix</h2>
<p>This section provides a complement to the <a href="#sec-tpr_fdp">TPR-FDR curves</a></p>
<section id="sec-appendix3_1" class="level3" data-number="3.7.1">
<h3 data-number="3.7.1" class="anchored" data-anchor-id="sec-appendix3_1"><span class="header-section-number">3.7.1</span> TPR-FDP curves using all proteins in the data</h3>
<p>This code is almost the same as for the main plot, except we removed the <code>na.exclude()</code> statement. This means that all proteins, even if not estimated, are included. The missing p-values will be ranked at the end (as if they were estimated at 1), meaning they will be accounted for when computing the TPR and FDP. This will inevitably lead to a decrease in the maximum TPR, especially for approaches that estimated less proteins.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>performance_all <span class="ot">&lt;-</span> <span class="fu">group_by</span>(results, inputLevel, contrast) <span class="sc">|&gt;</span> </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">tpr =</span> <span class="fu">computeTPR</span>(pval, isEcoli),</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">fdp =</span> <span class="fu">computeFDP</span>(pval, isEcoli)) <span class="sc">|&gt;</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(fdp)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>workPoints <span class="ot">&lt;-</span> <span class="fu">group_by</span>(performance_all, inputLevel, contrast) <span class="sc">|&gt;</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(adjPval) <span class="sc">|&gt;</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(inputLevel))</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_all) <span class="sc">+</span></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> fdp,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> tpr,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> inputLevel) <span class="sc">+</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> workPoints, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> contrast) <span class="sc">+</span></span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"TPR-FDP curves using all proteins in the data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_tpr_fdp_2-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="sec-appendix3_2" class="level3" data-number="3.7.2">
<h3 data-number="3.7.2" class="anchored" data-anchor-id="sec-appendix3_2"><span class="header-section-number">3.7.2</span> TPR-FDP curves using only the proteins fit by all approaches</h3>
<p>The code is almost the same as for the main plot, except we add a filtering step where we require, for each comparison, that p-values are not missing for all 5 approaches, effectively focusing on the set of proteins that have been estimated by all approaches.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>performance_common <span class="ot">&lt;-</span> <span class="fu">group_by</span>(results, feature, contrast) <span class="sc">|&gt;</span> </span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(adjPval),</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">n</span>() <span class="sc">==</span> <span class="dv">5</span>) <span class="sc">|&gt;</span> </span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(inputLevel, contrast) <span class="sc">|&gt;</span> </span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tpr =</span> <span class="fu">computeTPR</span>(pval, isEcoli),</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">fdp =</span> <span class="fu">computeFDP</span>(pval, isEcoli)) <span class="sc">|&gt;</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(fdp)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>workPoints <span class="ot">&lt;-</span> <span class="fu">group_by</span>(performance_common, inputLevel, contrast) <span class="sc">|&gt;</span></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(adjPval) <span class="sc">|&gt;</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(inputLevel))</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_common) <span class="sc">+</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">y =</span> fdp,</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> tpr,</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">colour =</span> inputLevel) <span class="sc">+</span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">data =</span> workPoints, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.05</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> contrast) <span class="sc">+</span></span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"TPR-FDP curves using only the proteins fit by all approaches"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="03-benchmarking_files/figure-html/benchmarking_tpr_fdp_3-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Shen2018-gw" class="csl-entry" role="listitem">
Shen, Xiaomeng, Shichen Shen, Jun Li, Qiang Hu, Lei Nie, Chengjian Tu, Xue Wang, et al. 2018. <span>â€œ<span>IonStar</span> Enables High-Precision, Low-Missing-Data Proteomics Quantification in Large Biological Cohorts.â€</span> <em>Proc. Natl. Acad. Sci. U. S. A.</em> 115 (21): E4767â€“76.
</div>
<div id="ref-Sticker2020-rl" class="csl-entry" role="listitem">
Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement. 2020. <span>â€œRobust Summarization and Inference in Proteome-Wide Label-Free Quantification.â€</span> <em>Mol. Cell. Proteomics</em> 19 (7): 1209â€“19.
</div>
</div>
</section>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>Another option would be to copy-paste the workflow code for every approach, but we refrain from doing so as this can lead to incoherent code when the workflow needs to be changed. This is a common malpractice that reduces code maintainability.<a href="#fnref1" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn2"><p>There are 5 conditions, so 10 unique pairwise combinations.<a href="#fnref2" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn3"><p>Note the approach names match some of the sets in the <code>QFeatures</code> object.<a href="#fnref3" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn4"><p>When statistical inference fails for a protein, <code>msqrob2</code> will fill the (adjusted) p-value and log2-fold change with missing values. So excluding NAs will ignore any failed inference.<a href="#fnref4" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn5"><p>Although these protein data are indirectly generated using a summarisation approach (c.f. <code>msqrobAggregate()</code>).<a href="#fnref5" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn6"><p>We build on the concepts introduced in the <a href="01-basics.html">basic concepts chapter</a> and the <a href="02-advanced.html">advanced concepts chapter</a><a href="#fnref6" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
<li id="fn7"><p>A PSM is generated from a spectrum, which is specific to each run and there is not unambiguous way to link a spectrum across runs<a href="#fnref7" class="footnote-back" role="doc-backlink">â†©ï¸Ž</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "î§‹";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./02-advanced.html" class="pagination-link" aria-label="Advanced statistical analysis with msqrob2">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./04-workflow_optimisation.html" class="pagination-link" aria-label="Optimisation of a data analysis workflow">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>