<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.23">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>5&nbsp; The francisella use case: a MaxQuant LFQ DDA dataset with technical replication – Statistical analysis of mass spectrometry-based proteomics data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./06-heart.html" rel="next">
<link href="./04-workflow_optimisation.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac32dca00065cc45e71a60b6df2d2e12.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./05-francisella.html"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figs/msqrob2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistical analysis of mass spectrometry-based proteomics data</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preamble</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-benchmarking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-workflow_optimisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-francisella.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-heart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-mouse_diet.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-end.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">5.1</span> Introduction</a></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages"><span class="header-section-number">5.2</span> Load packages</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data"><span class="header-section-number">5.3</span> Load data</a>
  <ul class="collapse">
  <li><a href="#experimental-context" id="toc-experimental-context" class="nav-link" data-scroll-target="#experimental-context"><span class="header-section-number">5.3.1</span> Experimental context</a></li>
  <li><a href="#getting-the-data" id="toc-getting-the-data" class="nav-link" data-scroll-target="#getting-the-data"><span class="header-section-number">5.3.2</span> Getting the data</a></li>
  <li><a href="#the-qfeatures-data-class" id="toc-the-qfeatures-data-class" class="nav-link" data-scroll-target="#the-qfeatures-data-class"><span class="header-section-number">5.3.3</span> The <code>QFeatures</code> data class</a></li>
  </ul></li>
  <li><a href="#data-preprocessing" id="toc-data-preprocessing" class="nav-link" data-scroll-target="#data-preprocessing"><span class="header-section-number">5.4</span> Data preprocessing</a>
  <ul class="collapse">
  <li><a href="#encoding-missing-values" id="toc-encoding-missing-values" class="nav-link" data-scroll-target="#encoding-missing-values"><span class="header-section-number">5.4.1</span> Encoding missing values</a></li>
  <li><a href="#peptide-filtering" id="toc-peptide-filtering" class="nav-link" data-scroll-target="#peptide-filtering"><span class="header-section-number">5.4.2</span> Peptide filtering</a></li>
  <li><a href="#standard-preprocessing-workflow" id="toc-standard-preprocessing-workflow" class="nav-link" data-scroll-target="#standard-preprocessing-workflow"><span class="header-section-number">5.4.3</span> Standard preprocessing workflow</a></li>
  </ul></li>
  <li><a href="#data-exploration" id="toc-data-exploration" class="nav-link" data-scroll-target="#data-exploration"><span class="header-section-number">5.5</span> Data exploration</a></li>
  <li><a href="#data-modelling" id="toc-data-modelling" class="nav-link" data-scroll-target="#data-modelling"><span class="header-section-number">5.6</span> Data modelling</a>
  <ul class="collapse">
  <li><a href="#sources-of-variation" id="toc-sources-of-variation" class="nav-link" data-scroll-target="#sources-of-variation"><span class="header-section-number">5.6.1</span> Sources of variation</a></li>
  <li><a href="#estimate-the-model" id="toc-estimate-the-model" class="nav-link" data-scroll-target="#estimate-the-model"><span class="header-section-number">5.6.2</span> Estimate the model</a></li>
  </ul></li>
  <li><a href="#statistical-inference" id="toc-statistical-inference" class="nav-link" data-scroll-target="#statistical-inference"><span class="header-section-number">5.7</span> Statistical inference</a>
  <ul class="collapse">
  <li><a href="#hypothesis-testing" id="toc-hypothesis-testing" class="nav-link" data-scroll-target="#hypothesis-testing"><span class="header-section-number">5.7.1</span> Hypothesis testing</a></li>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot"><span class="header-section-number">5.7.2</span> Volcano-plot</a></li>
  <li><a href="#heatmap" id="toc-heatmap" class="nav-link" data-scroll-target="#heatmap"><span class="header-section-number">5.7.3</span> Heatmap</a></li>
  <li><a href="#detail-plots" id="toc-detail-plots" class="nav-link" data-scroll-target="#detail-plots"><span class="header-section-number">5.7.4</span> Detail plots</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">5.8</span> Conclusion</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-francisella" class="quarto-section-identifier"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="introduction" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">5.1</span> Introduction</h2>
<p>In this chapter we show how to analyse LFQ data with technical replication. In the data analysis we have to acknowledge pseudo-replication, i.e.&nbsp;data for multiple technical replicates from the same biological repeat are correlated and we cannot treat these as independent repeats.</p>
<p>We will build upon the mixed model framework in msqrob2 to address this correlation by introducing a random effect for each biological repeat.</p>
<p>If you never used <code>msqrob2</code>, we suggest to familiarise yourself with the <a href="01-basics.html">concepts chapter</a> first.</p>
</section>
<section id="load-packages" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="load-packages"><span class="header-section-number">5.2</span> Load packages</h2>
<p>First, we load the <code>msqrob2</code> package and additional packages for data manipulation and visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also configure the <a href="01-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="load-data"><span class="header-section-number">5.3</span> Load data</h2>
<section id="experimental-context" class="level3" data-number="5.3.1">
<h3 data-number="5.3.1" class="anchored" data-anchor-id="experimental-context"><span class="header-section-number">5.3.1</span> Experimental context</h3>
<p>A study on the facultative pathogen Francisella tularensis was conceived by <span class="citation" data-cites="Ramond2015-rz">(<a href="99-end.html#ref-Ramond2015-rz" role="doc-biblioref">Ramond et al. 2015</a>)</span>. F. tularensis enters the cells of its host by phagocytosis. The authors showed that F. tularensis is arginine deficient and imports arginine from the host cell via an arginine transporter, ArgP, in order to efficiently escape from the phagosome and reach the cytosolic compartment, where it can actively multiply. In their study, they compared the proteome of wild type F. tularensis (WT) to ArgP-gene deleted F. tularensis (knock-out, D8). For this exercise, we use a subset of the F. tularensis dataset where bacterial cultures were grown in biological triplicate and each biorep was run in technical triplicate on a nanoRSLC-Q Exactive PLUS instrument.</p>
</section>
<section id="getting-the-data" class="level3" data-number="5.3.2">
<h3 data-number="5.3.2" class="anchored" data-anchor-id="getting-the-data"><span class="header-section-number">5.3.2</span> Getting the data</h3>
<p>The data were searched with MaxQuant version 1.4.1.2. and are available on the PRIDE repository: <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD001584">PXD001584</a>. We download the peptides file<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<p><strong>TODO</strong> put on Zenodo and use BiocFileCache</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocFileCache"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> <span class="fu">BiocFileCache</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>pepFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/master/Francisella/data/maxquant/peptides.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After downloading the files, we can load the <a href="{#sec-peptide_table}">peptide table</a>. Contrarily a <a href="02-advanced.html#sec-psm_table">PSM table</a>, the peptide table is provided in a “wide format”, meaning that each row represents a single peptide and that each quantification column (that starts with <code>"Intensity"</code>) represents a single sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>peps <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(pepFile)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>quantcols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Intensity</span><span class="sc">\\</span><span class="st">."</span>, <span class="fu">names</span>(peps), <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 3%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 8%">
<col style="width: 17%">
<col style="width: 0%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Sequence</th>
<th style="text-align: left;">Amino.acid.before</th>
<th style="text-align: left;">First.amino.acid</th>
<th style="text-align: left;">Second.amino.acid</th>
<th style="text-align: left;">Second.last.amino.acid</th>
<th style="text-align: left;">Last.amino.acid</th>
<th style="text-align: left;">Amino.acid.after</th>
<th style="text-align: right;">A.Count</th>
<th style="text-align: right;">R.Count</th>
<th style="text-align: right;">N.Count</th>
<th style="text-align: right;">D.Count</th>
<th style="text-align: right;">C.Count</th>
<th style="text-align: right;">Q.Count</th>
<th style="text-align: right;">E.Count</th>
<th style="text-align: right;">G.Count</th>
<th style="text-align: right;">H.Count</th>
<th style="text-align: right;">I.Count</th>
<th style="text-align: right;">L.Count</th>
<th style="text-align: right;">K.Count</th>
<th style="text-align: right;">M.Count</th>
<th style="text-align: right;">F.Count</th>
<th style="text-align: right;">P.Count</th>
<th style="text-align: right;">S.Count</th>
<th style="text-align: right;">T.Count</th>
<th style="text-align: right;">W.Count</th>
<th style="text-align: right;">Y.Count</th>
<th style="text-align: right;">V.Count</th>
<th style="text-align: right;">U.Count</th>
<th style="text-align: right;">Length</th>
<th style="text-align: right;">Missed.cleavages</th>
<th style="text-align: right;">Mass</th>
<th style="text-align: left;">Proteins</th>
<th style="text-align: left;">GI.number</th>
<th style="text-align: left;">Leading.razor.protein</th>
<th style="text-align: left;">Protein.names</th>
<th style="text-align: left;">Unique..Groups.</th>
<th style="text-align: left;">Unique..Proteins.</th>
<th style="text-align: left;">Charges</th>
<th style="text-align: right;">PEP</th>
<th style="text-align: left;">Score</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n3_1</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n3_2</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n3_3</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n4_1</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n4_2</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n4_3</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n5_1</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n5_2</th>
<th style="text-align: right;">Experiment.1WT_20_2h_n5_3</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n3_1</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n3_2</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n3_3</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n4_1</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n4_2</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n4_3</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n5_1</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n5_2</th>
<th style="text-align: right;">Experiment.3D8_20_2h_n5_3</th>
<th style="text-align: right;">Intensity</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n3_1</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n3_2</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n3_3</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n4_1</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n4_2</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n4_3</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n5_1</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n5_2</th>
<th style="text-align: right;">Intensity.1WT_20_2h_n5_3</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n3_1</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n3_2</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n3_3</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n4_1</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n4_2</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n4_3</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n5_1</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n5_2</th>
<th style="text-align: right;">Intensity.3D8_20_2h_n5_3</th>
<th style="text-align: left;">Reverse</th>
<th style="text-align: left;">Contaminant</th>
<th style="text-align: right;">id</th>
<th style="text-align: left;">Protein.group.IDs</th>
<th style="text-align: left;">Mod..peptide.IDs</th>
<th style="text-align: left;">Evidence.IDs</th>
<th style="text-align: left;">MS.MS.IDs</th>
<th style="text-align: right;">Best.MS.MS</th>
<th style="text-align: left;">Oxidation..M..site.IDs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">AAAEELDTR</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">T</td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">K</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">974.4669</td>
<td style="text-align: left;">WP_003038655</td>
<td style="text-align: left;">gi|118497196</td>
<td style="text-align: left;">gi|118497196</td>
<td style="text-align: left;">hypothetical protein [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">1,2</td>
<td style="text-align: right;">0.0001253</td>
<td style="text-align: left;">59.35</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1.3295e+09</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9290400</td>
<td style="text-align: right;">92996000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">98059000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">80803000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">95433000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">0</td>
<td style="text-align: left;">419</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td style="text-align: left;">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td style="text-align: right;">15</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAAGFVITASHNK</td>
<td style="text-align: left;">R</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">N</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">F</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1285.6779</td>
<td style="text-align: left;">WP_003041237</td>
<td style="text-align: left;">gi|118498194</td>
<td style="text-align: left;">gi|118498194</td>
<td style="text-align: left;">phosphoglucosamine mutase [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">0.0000530</td>
<td style="text-align: left;">45.825</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1.6719e+08</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">20679000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">17358000</td>
<td style="text-align: right;">13841000</td>
<td style="text-align: right;">12278000</td>
<td style="text-align: right;">10712000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9500700</td>
<td style="text-align: right;">11111000</td>
<td style="text-align: right;">8723500</td>
<td style="text-align: right;">8125200</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">1150</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td style="text-align: left;">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td style="text-align: right;">26</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAANEYELALAYSIEEVAPDLHK</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">H</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">Y</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2516.2435</td>
<td style="text-align: left;">WP_003038915</td>
<td style="text-align: left;">gi|118497331</td>
<td style="text-align: left;">gi|118497331</td>
<td style="text-align: left;">glycine–tRNA ligase subunit beta [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">3</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">84.327</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5.5675e+08</td>
<td style="text-align: right;">28853000</td>
<td style="text-align: right;">28091000</td>
<td style="text-align: right;">30436000</td>
<td style="text-align: right;">19476000</td>
<td style="text-align: right;">22050000</td>
<td style="text-align: right;">20687000</td>
<td style="text-align: right;">36588000</td>
<td style="text-align: right;">28654000</td>
<td style="text-align: right;">7009100</td>
<td style="text-align: right;">14174000</td>
<td style="text-align: right;">8548000</td>
<td style="text-align: right;">7859200</td>
<td style="text-align: right;">14871000</td>
<td style="text-align: right;">11690000</td>
<td style="text-align: right;">4470500</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4048100</td>
<td style="text-align: right;">3422300</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">2</td>
<td style="text-align: left;">513</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76</td>
<td style="text-align: left;">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79</td>
<td style="text-align: right;">39</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAANNPQLEAFK</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">F</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">K</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1272.6463</td>
<td style="text-align: left;">WP_003038264</td>
<td style="text-align: left;">gi|118496879</td>
<td style="text-align: left;">gi|118496879</td>
<td style="text-align: left;">molecular chaperone HtpG [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: left;">108.56</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.4830e+10</td>
<td style="text-align: right;">339830000</td>
<td style="text-align: right;">420390000</td>
<td style="text-align: right;">393930000</td>
<td style="text-align: right;">235060000</td>
<td style="text-align: right;">381090000</td>
<td style="text-align: right;">360270000</td>
<td style="text-align: right;">255710000</td>
<td style="text-align: right;">311370000</td>
<td style="text-align: right;">271440000</td>
<td style="text-align: right;">225140000</td>
<td style="text-align: right;">289880000</td>
<td style="text-align: right;">282500000</td>
<td style="text-align: right;">209490000</td>
<td style="text-align: right;">223610000</td>
<td style="text-align: right;">219700000</td>
<td style="text-align: right;">169120000</td>
<td style="text-align: right;">178150000</td>
<td style="text-align: right;">170400000</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">199</td>
<td style="text-align: left;">3</td>
<td style="text-align: left;">77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124</td>
<td style="text-align: left;">80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172</td>
<td style="text-align: right;">159</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">AAASAGLVDEK</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">E</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1030.5295</td>
<td style="text-align: left;">WP_003035781</td>
<td style="text-align: left;">gi|118497152</td>
<td style="text-align: left;">gi|118497152</td>
<td style="text-align: left;">delta-aminolevulinic acid dehydratase [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">0.0000951</td>
<td style="text-align: left;">67.385</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4.8209e+09</td>
<td style="text-align: right;">240280000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">241890000</td>
<td style="text-align: right;">222250000</td>
<td style="text-align: right;">189930000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">160440000</td>
<td style="text-align: right;">147230000</td>
<td style="text-align: right;">143340000</td>
<td style="text-align: right;">135910000</td>
<td style="text-align: right;">185740000</td>
<td style="text-align: right;">168620000</td>
<td style="text-align: right;">160240000</td>
<td style="text-align: right;">158000000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">92542000</td>
<td style="text-align: right;">102350000</td>
<td style="text-align: right;">100090000</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">4</td>
<td style="text-align: left;">388</td>
<td style="text-align: left;">4</td>
<td style="text-align: left;">125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155</td>
<td style="text-align: left;">173;174;175;176;177;178;179;180;181;182;183;184;185;186;187;188;189;190;191;192;193;194;195;196;197;198;199;200;201;202;203</td>
<td style="text-align: right;">191</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">AAASLDLYSYPK</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">A</td>
<td style="text-align: left;">P</td>
<td style="text-align: left;">K</td>
<td style="text-align: left;">V</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1297.6554</td>
<td style="text-align: left;">WP_003039212</td>
<td style="text-align: left;">gi|118497492</td>
<td style="text-align: left;">gi|118497492</td>
<td style="text-align: left;">phosphorylase [Francisella tularensis]</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">yes</td>
<td style="text-align: left;">2</td>
<td style="text-align: right;">0.0125630</td>
<td style="text-align: left;">31.675</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">1.3715e+08</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">31069000</td>
<td style="text-align: right;">31448000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">27721000</td>
<td style="text-align: right;">28657000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">18253000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: right;">5</td>
<td style="text-align: left;">624</td>
<td style="text-align: left;">5</td>
<td style="text-align: left;">156;157;158;159;160</td>
<td style="text-align: left;">204;205;206;207;208</td>
<td style="text-align: right;">205</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We now extract the <a href="01-basics.html#sec-annotation_table">sample annotations</a>. We will build a table where each row in the annotation table contains information for one sample (the table below shows the first 6 rows). This information is extracted from the sample names.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">quantCols =</span> quantcols) <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"_20_"</span>, quantCols) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"_n</span><span class="sc">\\</span><span class="st">d"</span>, quantCols)) <span class="sc">|&gt;</span> </span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype  =</span> <span class="fu">substr</span>(quantCols, <span class="dv">12</span>, <span class="dv">13</span>), </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">biorep  =</span> <span class="fu">paste0</span>(genotype, <span class="st">"_"</span>, <span class="fu">substr</span>(quantCols, <span class="dv">21</span>, <span class="dv">22</span>)),</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">run =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(quantcols))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">quantCols</th>
<th style="text-align: left;">genotype</th>
<th style="text-align: left;">biorep</th>
<th style="text-align: right;">run</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Intensity.1WT_20_2h_n3_1</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n3</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">Intensity.1WT_20_2h_n3_2</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n3</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Intensity.1WT_20_2h_n3_3</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n3</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Intensity.1WT_20_2h_n4_1</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n4</td>
<td style="text-align: right;">4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Intensity.1WT_20_2h_n4_2</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n4</td>
<td style="text-align: right;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">Intensity.1WT_20_2h_n4_3</td>
<td style="text-align: left;">WT</td>
<td style="text-align: left;">WT_n4</td>
<td style="text-align: right;">6</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="the-qfeatures-data-class" class="level3" data-number="5.3.3">
<h3 data-number="5.3.3" class="anchored" data-anchor-id="the-qfeatures-data-class"><span class="header-section-number">5.3.3</span> The <code>QFeatures</code> data class</h3>
<p>We combine the two tables into a <a href="01-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  peps, <span class="at">colData =</span> coldata, <span class="at">fnames =</span> <span class="st">"Sequence"</span>, <span class="at">name =</span> <span class="st">"peptides"</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 1 set:

 [1] peptides: SummarizedExperiment with 10693 rows and 18 columns </code></pre>
</div>
</div>
<p>We now have a <code>QFeatures</code> object with 1 set, containing <code>r nrows(pe)[[1]]</code> rows (peptides) and 18 columns (samples).</p>
</section>
</section>
<section id="data-preprocessing" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="data-preprocessing"><span class="header-section-number">5.4</span> Data preprocessing</h2>
<p><code>msqrob2</code> relies on the <code>QFeatures</code> data structure, meaning that we can directly make use of <code>QFeatures</code>’ data preprocessing functionality (see also the <code>QFeatures</code> <a href="https://rformassspectrometry.github.io/QFeatures/articles/Processing.html">documentation</a>).</p>
<section id="encoding-missing-values" class="level3" data-number="5.4.1">
<h3 data-number="5.4.1" class="anchored" data-anchor-id="encoding-missing-values"><span class="header-section-number">5.4.1</span> Encoding missing values</h3>
<p>Peptides with zero intensities should be <a href="01-basics.html#sec-encode_missing">encoded</a> using <code>NA</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">"peptides"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We calculate how many non zero intensities we have per peptide and this is often useful for filtering.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>naResults <span class="ot">&lt;-</span> <span class="fu">nNA</span>(pe, <span class="st">"peptides"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(naResults<span class="sc">$</span>nNArows) <span class="sc">|&gt;</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> nNA) <span class="sc">+</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_nNA_hist-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="peptide-filtering" class="level3" data-number="5.4.2">
<h3 data-number="5.4.2" class="anchored" data-anchor-id="peptide-filtering"><span class="header-section-number">5.4.2</span> Peptide filtering</h3>
<p>We filter peptides based on 3 criteria (see [PSM filtering]).</p>
<ol type="1">
<li>Remove failed protein inference</li>
</ol>
<p>We remove peptides that could not be uniquely mapped to a protein.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li>Remove reverse sequences and contaminants</li>
</ol>
<p>We now remove the contaminants and peptides that map to decoy sequences. These features bear no information of interest and will reduce the statistical power upon multiple test adjustment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> Contaminant <span class="sc">!=</span> <span class="st">"+"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="3" type="1">
<li>Drop peptides that were only identified in a single biorepeat</li>
</ol>
<p>Note, that in experiments without technical repeats we filter on the number of samples in which a peptide is picked up (this is typically performed using <code>filterNA()</code>). Here, we will require that a peptide is picked up in at least two biorepeats. We compute the number of biorepeats that were observed for each peptide (that is the number of biorepeats that contain at least one observed value).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">"peptides"</span>]])<span class="sc">$</span>nNonZeroBiorep <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">"peptides"</span>]]), <span class="dv">1</span>, <span class="cf">function</span>(intensity)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    pe<span class="sc">$</span>biorep[<span class="sc">!</span><span class="fu">is.na</span>(intensity)] <span class="sc">|&gt;</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">|&gt;</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">length</span>()</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We keep peptides that are observed in at least two biorepeats.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> nNonZeroBiorep <span class="sc">&gt;=</span> <span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 1 set:

 [1] peptides: SummarizedExperiment with 7542 rows and 18 columns </code></pre>
</div>
</div>
<p>We keep 7542 peptides upon filtering.</p>
</section>
<section id="standard-preprocessing-workflow" class="level3" data-number="5.4.3">
<h3 data-number="5.4.3" class="anchored" data-anchor-id="standard-preprocessing-workflow"><span class="header-section-number">5.4.3</span> Standard preprocessing workflow</h3>
<p>We can now prepare the data for modelling. The workflow ensures the data complies to <code>msqrob2</code>’s requirements:</p>
<ol type="1">
<li>Intensities are <a href="01-basics.html#sec-log2">log-transformed</a>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">"peptides"</span>, <span class="at">name =</span> <span class="st">"peptides_log"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="2" type="1">
<li>Normalisation with the Median of Ratios method.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>pseudoRef <span class="ot">&lt;-</span> <span class="fu">assay</span>(pe[[<span class="st">"peptides_log"</span>]]) <span class="sc">|&gt;</span> </span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowMeans</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#1. Calculate the row means </span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>nfLog <span class="ot">&lt;-</span> <span class="fu">sweep</span>(</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">"peptides_log"</span>]]), </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">MARGIN =</span> <span class="dv">1</span>, </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  pseudoRef) <span class="sc">|&gt;</span> <span class="co">#2. Subtract the row means row-by-row (MARGIN = 1)</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colMedians</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)  <span class="co">#3. Calculate the column median </span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sweep</span>(pe, </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">MARGIN =</span> <span class="dv">2</span>, </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">STATS =</span> nfLog , </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">i =</span> <span class="st">"peptides_log"</span>, </span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"peptides_norm"</span>) <span class="co">#4. Subtract log2 norm factor column-by-column (MARGIN = 2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<!--
2. Samples are normalised by substracting the sample median (see [Normalisation])
-->
<p>Upon the normalisation the density curves should be nicely centred. To confirm this, we will plot the intensity distributions for each biorepeat (francisella culture). <code>longForm()</code> seamlessly combines the quantification and annotation data into a table suitable for <code>ggplot2</code> visualisation. We also subset the object with the data before and after normalisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">longForm</span>(pe[, , <span class="fu">c</span>(<span class="st">"peptides_log"</span>, <span class="st">"peptides_norm"</span>)], <span class="at">colvar =</span> <span class="st">"biorep"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">group =</span> colname, <span class="at">color =</span> biorep) <span class="sc">+</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> assay, <span class="at">scale =</span> <span class="st">"free"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_after_norm-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<ol start="3" type="1">
<li><a href="01-basics.html#sec-summarisation">Summarisation</a> to protein level.</li>
</ol>
<p>We use the robust summary approach to infer protein-level data from peptide-level data, accounting for the fact that different peptides have ionisation efficiencies hence leading to different intensity baselines.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  pe, <span class="at">i =</span> <span class="st">"peptides_norm"</span>, <span class="at">fcol =</span> <span class="st">"Proteins"</span>, </span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">name =</span> <span class="st">"proteins"</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-exploration" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="data-exploration"><span class="header-section-number">5.5</span> Data exploration</h2>
<p>We will explore the main sources of variation in the data using <a href="01-basics.html#sec_data_exploration">MDS</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getWithColData</span>(pe, <span class="st">"proteins"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">"SingleCellExperiment"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">runMDS</span>(<span class="at">exprs_values =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(<span class="at">colour_by =</span> <span class="st">"genotype"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_mds-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Note that the samples upon robust summarisation show a clear separation according to the genotype in the first dimension of the MDS plot.</p>
</section>
<section id="data-modelling" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="data-modelling"><span class="header-section-number">5.6</span> Data modelling</h2>
<p>The preprocessed data can now be modelled to answer biologically relevant questions. As described above, samples (bacterial cultures) originate from either a wildtype (WT) or a ArgP knockout (D8). Each genotype was cultured in biological triplicate. Each biological triplicate was acquired in technical triplicate, leading to <span class="math inline">\(2 \times
3 \times 3 = 18\)</span> samples. In this context, we are interested in the effects of genotype on the protein abundances.</p>
<p>The table below confirms we have a balanced design for each condition and biological triplicate.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="at">genotype =</span> pe<span class="sc">$</span>genotype, <span class="at">biorep =</span> pe<span class="sc">$</span>biorep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        biorep
genotype D8_n3 D8_n4 D8_n5 WT_n3 WT_n4 WT_n5
      D8     3     3     3     0     0     0
      WT     0     0     0     3     3     3</code></pre>
</div>
</div>
<section id="sources-of-variation" class="level3" data-number="5.6.1">
<h3 data-number="5.6.1" class="anchored" data-anchor-id="sources-of-variation"><span class="header-section-number">5.6.1</span> Sources of variation</h3>
<p>We will model two <a href="01-basics.html#sec-modelling">sources of variation</a>:</p>
<ol type="1">
<li><p><strong>Genotype</strong>: we model the source of variation induced by the experimental group of interest as a <strong>fixed effect</strong>. Fixed effects are effect that are considered non-random, i.e.&nbsp;the treatment effect is assumed to be the same and reproducible across repeated experiments, but it is unknown and has to be estimated. We will include <code>genotype</code> as a fixed effect that models the fact that a change in genotype can induce changes in protein abundance.</p></li>
<li><p><strong>Biological replicate effect</strong>: the experiment involves biological replication as the bacterial cultures are repeated. Replicate-specific effects occurs due to uncontrollable factors, such as variation in the number of bacterium seeded, position in the incubator, transient contamination,… Two bacterial cultures will never provide exactly the same sample material. These effects are typically modelled as random effects which are considered as a random sample from the population of all possible mice and are assumed to be i.i.d normally distributed with mean 0 and constant variance, <span class="math inline">\(u_{biorep} \sim
N(0,\sigma^{2,\text{b}})\)</span>. The use of random effects thus models the correlation in the data, explicitly. We expect that intensities from the same bacterial culture are more alike than intensities between cultures.</p></li>
</ol>
<p>Hence, the variance-covariance matrix of the 18 protein abundance values <span class="math inline">\(\mathbf{Y}=(y_{wt\_n3,1}, y_{wt\_n3,2}, y_{wt\_n3,3},
y_{wt\_n4,1} \ldots y_{d8\_n5,1}, y_{d8\_n5,2}, y_{d8\_n5,3})^T\)</span> is assumed to have a block diagonal structure, with as variance <span class="math inline">\(\sigma_b^2 + \sigma_\epsilon^2\)</span> and the covariance between protein abundance values of technical replicates from the same biorepeat equals <span class="math inline">\(\sigma_b^2\)</span> (with _^2 the variance of the residuals). More details on mixed models can be found in the <a href="02-advanced.html#sec-random_effect">advanced chapter</a>.</p>
<p>We model the protein level expression values using <code>msqrob2</code>. <code>msqrob2</code> workflows rely on linear mixed models, which are models that can estimate and predict fixed and random effects, respectively. The fixed effect are estimated using robust regression to avoid that outliers distort the statistical outcome.</p>
<p>Note, that we cannot use ridge regression here to further stabilize the parameter estimation of the fixed effects. Ridge regression typically performs better the more slope parameters there are in the mean model (more complex designs). Here, the fixed effect consists of the genotype: knockout (D8) vs wild type (WT). By default the first group (D8) will become the reference group and its effect will be absorbed in the intercept of the model. Hence, only a single slope term is needed to model the average difference between D8 and WT (See Section <a href="#statistical-inference">Statistical inference</a> below).</p>
<p>Now we have identified the sources of variation in the experiment, we can define a model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="er">~</span> genotype <span class="sc">+</span> <span class="do">## (1) fixed effect for genotype</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> biorep)  <span class="do">## (2) random effect for biological replicate (culture)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="estimate-the-model" class="level3" data-number="5.6.2">
<h3 data-number="5.6.2" class="anchored" data-anchor-id="estimate-the-model"><span class="header-section-number">5.6.2</span> Estimate the model</h3>
<p>We estimate the model with <code>msqrob()</code> (see <a href="02-advanced.html#sec-run_model">the modelling section</a>). Recall that variables defined in <code>model</code> are automatically retrieved from the <code>colData</code> (i.e.&nbsp;<code>"genotype"</code>, and <code>"biorep"</code>). Note, that <code>msqrob2</code> also features ridge regression for stabilising the parameter estimation, but it is irrelevant in this context as the genotype factor only has 2 levels (WT and D8), so ridge regression . We will therefore leave the ridge regression disabled (default).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">formula =</span> model, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="statistical-inference" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="statistical-inference"><span class="header-section-number">5.7</span> Statistical inference</h2>
<p>Once the models are estimated, we can start answering biological questions by performing <a href="#statistical-inference">Statistical inference</a>. We must convert the biological question “does the bacterial genotype affect the protein intensities?” into a statistical hypothesis. In other words, we must convert this question in a combination of the model parameters, also referred to as a contrast. To aid defining contrasts, we will visualise the experimental design using the <code>ExploreModelMatrix</code> package. Note that with ExploreModelMatrix we can only visualise fixed effects part of the model. This is fine as the mean protein abundances can only systematically differ from each other according to the <code>genotype</code> (fixed effect).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sampleData =</span>  <span class="fu">colData</span>(pe),</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> genotype,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>vd<span class="sc">$</span>plotlist</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_VisualizeDesign-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This plot shows that the average <span class="math inline">\(\log_2\)</span> protein intensity for the D8 group is modelled by <code>(Intercept)</code>, and the the average <span class="math inline">\(\log_2\)</span> protein intensity for the WT group is modelled by <code>(Intercept) + genotypeWT</code>.</p>
<section id="hypothesis-testing" class="level3" data-number="5.7.1">
<h3 data-number="5.7.1" class="anchored" data-anchor-id="hypothesis-testing"><span class="header-section-number">5.7.1</span> Hypothesis testing</h3>
<p>Hence we can translate the research hypothesis that there is an effect of the genotype on the protein abundance to the average log2 fold change (<span class="math inline">\(\log_2 FC\)</span>) between the WT and D8 groups, which boils down to <code>(Intercept) + genotypeWT - (Intercept)</code>, and equals the <code>genotypeWT</code> parameter. The null hypothesis of the hypothesis test for this contrast is that the average <span class="math inline">\(\log2 FC\)</span> between D8 knock-out and WT is zero, or in other words that the <code>genotypeWT</code> parameter is zero.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>hypothesis <span class="ot">&lt;-</span> <span class="st">"genotypeWT = 0"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We next use <code>makeContrast()</code> to build the contrast matrix.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(hypothesis, <span class="at">parameterNames =</span> <span class="st">"genotypeWT"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           genotypeWT
genotypeWT          1</code></pre>
</div>
</div>
<p>In this case, the contrast matrix is trivial, but it becomes a matrix but for <a href="01-basics.html#sec-multiple_contrasts">more complex designs</a>. We can now test our null hypothesis:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let us retrieve the result table from the <code>rowData</code>. Note that the hypothesis testing results are stored in <code>rowData</code> columns named after the column names (here <code>genotypeWT</code>) of the contrast matrix <code>L</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">"proteins"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(inference)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   logFC         se        df          t         pval
WP_003013731  0.08607313 0.13622311 14.515249  0.6318541 0.5373074487
WP_003013860 -0.24361570 0.96333877  4.863216 -0.2528868 0.8106957489
WP_003013909 -0.29604787 0.06878765 15.007538 -4.3037941 0.0006263236
WP_003014068  0.11520522 0.15938527 14.142659  0.7228097 0.4815800326
WP_003014122  0.20050168 0.11484365 17.854759  1.7458665 0.0980174216
WP_003014123  0.11735634 0.14937755 14.599255  0.7856357 0.4446451519
                 adjPval      Protein
WP_003013731 0.740335646 WP_003013731
WP_003013860 0.909709125 WP_003013860
WP_003013909 0.006914364 WP_003013909
WP_003014068 0.695546291 WP_003014068
WP_003014122 0.253571752 WP_003014122
WP_003014123 0.666917832 WP_003014123</code></pre>
</div>
</div>
<p>Notice that some rows contain missing values. This is because data modelling resulted in a <code>fitError</code> for some proteins, probably because not enough data was available for model fitting due to missing values in the quantitative data (see <a href="#sec-fiterror">how to deal with <code>fitError</code>s</a>).</p>
</section>
<section id="volcano-plot" class="level3" data-number="5.7.2">
<h3 data-number="5.7.2" class="anchored" data-anchor-id="volcano-plot"><span class="header-section-number">5.7.2</span> Volcano-plot</h3>
<p>Volcano plots are straightforward to generate from the inference table above. We also use <code>ggrepel</code> to annotate the 20 most significant proteins.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">slice_min</span>(inference, adjPval, <span class="at">n =</span> <span class="dv">20</span>),</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on differences between WT and D8"</span>,</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>          <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">colnames</span>(L), <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_volcano-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Note, that 195 proteins are found to be differentially abundant.</p>
</section>
<section id="heatmap" class="level3" data-number="5.7.3">
<h3 data-number="5.7.3" class="anchored" data-anchor-id="heatmap"><span class="header-section-number">5.7.3</span> Heatmap</h3>
<p>We can also build a <a href="01-basics.html#sec-heatmaps">heatmap</a> for the significant proteins which are obtained by filtering the inference table. We first retrieve the data with proteins that are differentially abundant between the WT and the D8 genotype.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> inference<span class="sc">$</span>Protein[<span class="sc">!</span><span class="fu">is.na</span>(inference<span class="sc">$</span>adjPval) <span class="sc">&amp;</span> inference<span class="sc">$</span>adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>]</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(pe, <span class="st">"proteins"</span>)[sigNames, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We then plot the protein-wise standardised data as an annotated heatmap.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>quants <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(<span class="fu">assay</span>(se))))</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ComplexHeatmap"</span>)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">columnAnnotation</span>(</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> se<span class="sc">$</span>genotype</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>) <span class="do">## annotation colours are randomly generated by default</span></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a> quants, <span class="at">name =</span> <span class="st">"log2 intensity"</span>,</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a> <span class="at">top_annotation =</span> annotations</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_heatmap-1.png" class="img-fluid figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="detail-plots" class="level3" data-number="5.7.4">
<h3 data-number="5.7.4" class="anchored" data-anchor-id="detail-plots"><span class="header-section-number">5.7.4</span> Detail plots</h3>
<p>Let us visualise the most significant protein. We perform this with a little data manipulation pipeline:</p>
<ol type="1">
<li>Identify the target protein with largest logFC.</li>
<li>We use the <code>QFeatures</code> subsetting functionality to retrieve all data related to the target protein, focusing on the <code>proteins</code> set that contains the preprocessed data used for modelling.</li>
<li>We use <code>longForm()</code> to convert the object into a table suitable for plotting.</li>
<li>We remove missing values for plotting.</li>
<li>Plot the data with <code>ggplot2</code>.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)] <span class="co">#1</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>pe[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span> <span class="co">#2</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="st">"genotype"</span>) <span class="sc">|&gt;</span>  <span class="co">#3</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">|&gt;</span> <span class="co">#4</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="co">#5</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> genotype)) <span class="sc">+</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2 intensity"</span>) <span class="sc">+</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="05-francisella_files/figure-html/francisella_detail_plot-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="conclusion" class="level2" data-number="5.8">
<h2 data-number="5.8" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">5.8</span> Conclusion</h2>
<p>In this chapter, we illustrated the analysis of a label-free proteomics data set with technical replication. We followed the workflow described in the previous chapters with minimal changes, with a few exceptions.</p>
<p>First, we removed peptides that were missing across biological replicates instead of across all samples. This filtering strategy uses the experimental design to better define interesting features. Indeed, peptides that are only found in a single francisella culture batch bears no interesting biological information. The <code>QFeatures</code> framework is amenable for this custom filtering.</p>
<p>Second, we could not perform ridge regression because the fixed effects contained only two levels. However, it is possible to disable ridge regression within <code>mssqrob()</code> thanks to the argument <code>ridge = FALSE</code>.</p>
<p>Third, this data set has a more complex experimental design since there each biological replicate has been acquired in technical triplicate. Hence, we had to model</p>
<ul>
<li>the variation according to the treatment, effect of interest and modelled using a fixed effect</li>
<li>variation between biological repeats, random effect as the francisella cultures will change each time we repeat the experiment, and</li>
<li>technical variability, run to run variability and other sources of technical variability which are lumped in the residual variance.</li>
</ul>
<p>Note, that we could have performed the differential abundance analysis at the protein-level using <a href="03-benchmarking.html#sec-ion_level_modelling">ion- or peptide-level models</a>, i.e.&nbsp;by using the <code>msqrobAggregate</code> on the peptide_log assay with a formula <code>peptideModel &lt;- ~ genotype + (1|biorep) + (1|run) + (1|Sequence)</code>.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Ramond2015-rz" class="csl-entry" role="listitem">
Ramond, Elodie, Gael Gesbert, Ida Chiara Guerrera, Cerina Chhuon, Marion Dupuis, Mélanie Rigard, Thomas Henry, Monique Barel, and Alain Charbit. 2015. <span>“Importance of Host Cell Arginine Uptake in Francisella Phagosomal Escape and Ribosomal Protein Amounts.”</span> <em>Mol. Cell. Proteomics</em> 14 (4): 870–81.
</div>
</div>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>The file is <a href="02-advanced.html#sec-caching">locally cached</a><a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./04-workflow_optimisation.html" class="pagination-link" aria-label="Optimisation of a data analysis workflow">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./06-heart.html" class="pagination-link" aria-label="Heart use case: a MaxQuant LFQ DDA dataset with a more complex design">
        <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>