<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.23">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>7&nbsp; The mouse diet use case: a Skyline TMT DDA dataset – Statistical analysis of mass spectrometry-based proteomics data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./99-end.html" rel="next">
<link href="./06-heart.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-226bd0f977fa82dfae4534cac220d79a.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-ac32dca00065cc45e71a60b6df2d2e12.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./07-mouse_diet.html"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figs/msqrob2.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Statistical analysis of mass spectrometry-based proteomics data</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preamble</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-basics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-advanced.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Advanced statistical analysis with msqrob2</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-benchmarking.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Benchmarking workflows</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-workflow_optimisation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Optimisation of a data analysis workflow</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-francisella.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-heart.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-mouse_diet.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-end.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span></a>
  </div>
</li>
    </ul>
    </div>
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">7.1</span> Introduction</a></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages"><span class="header-section-number">7.2</span> Load packages</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data"><span class="header-section-number">7.3</span> Load data</a>
  <ul class="collapse">
  <li><a href="#sec-exp_design_mouse" id="toc-sec-exp_design_mouse" class="nav-link" data-scroll-target="#sec-exp_design_mouse"><span class="header-section-number">7.3.1</span> Experimental context</a></li>
  <li><a href="#getting-the-data" id="toc-getting-the-data" class="nav-link" data-scroll-target="#getting-the-data"><span class="header-section-number">7.3.2</span> Getting the data</a></li>
  <li><a href="#the-qfeatures-data-class" id="toc-the-qfeatures-data-class" class="nav-link" data-scroll-target="#the-qfeatures-data-class"><span class="header-section-number">7.3.3</span> The <code>QFeatures</code> data class</a></li>
  </ul></li>
  <li><a href="#data-preprocessing" id="toc-data-preprocessing" class="nav-link" data-scroll-target="#data-preprocessing"><span class="header-section-number">7.4</span> Data preprocessing</a>
  <ul class="collapse">
  <li><a href="#encoding-missing-values" id="toc-encoding-missing-values" class="nav-link" data-scroll-target="#encoding-missing-values"><span class="header-section-number">7.4.1</span> Encoding missing values</a></li>
  <li><a href="#sample-filtering" id="toc-sample-filtering" class="nav-link" data-scroll-target="#sample-filtering"><span class="header-section-number">7.4.2</span> Sample filtering</a></li>
  <li><a href="#psm-filtering" id="toc-psm-filtering" class="nav-link" data-scroll-target="#psm-filtering"><span class="header-section-number">7.4.3</span> PSM filtering</a></li>
  <li><a href="#standard-preprocessing-workflow" id="toc-standard-preprocessing-workflow" class="nav-link" data-scroll-target="#standard-preprocessing-workflow"><span class="header-section-number">7.4.4</span> Standard preprocessing workflow</a></li>
  </ul></li>
  <li><a href="#data-exploration" id="toc-data-exploration" class="nav-link" data-scroll-target="#data-exploration"><span class="header-section-number">7.5</span> Data exploration</a></li>
  <li><a href="#data-modelling" id="toc-data-modelling" class="nav-link" data-scroll-target="#data-modelling"><span class="header-section-number">7.6</span> Data modelling</a>
  <ul class="collapse">
  <li><a href="#sources-of-variation" id="toc-sources-of-variation" class="nav-link" data-scroll-target="#sources-of-variation"><span class="header-section-number">7.6.1</span> Sources of variation</a></li>
  <li><a href="#model-estimation" id="toc-model-estimation" class="nav-link" data-scroll-target="#model-estimation"><span class="header-section-number">7.6.2</span> Model estimation</a></li>
  <li><a href="#hypothesis-testing" id="toc-hypothesis-testing" class="nav-link" data-scroll-target="#hypothesis-testing"><span class="header-section-number">7.6.3</span> Hypothesis testing</a></li>
  <li><a href="#protein-level-model" id="toc-protein-level-model" class="nav-link" data-scroll-target="#protein-level-model"><span class="header-section-number">7.6.4</span> Protein-level model</a></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">7.7</span> Conclusion</a></li>
  </ul>
</nav>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-mouse-diet" class="quarto-section-identifier"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">The mouse diet use case: a Skyline TMT DDA dataset</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="introduction" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">7.1</span> Introduction</h2>
<p>If you never used <code>msqrob2</code>, we suggest to familiarise yourself with the <a href="01-basics.html">basic concepts chapter</a> first. Note that TMT experiments imposes complex designs, hence we also suggest reading the <a href="02-advanced.html">advanced concepts chapter</a>.</p>
<p>We will demonstrate the msqrob2TMT workflow, a data processing and modelling workflow dedicated to the analysis of TMT-based proteomics datasets. We will demonstrated the workflow using the study published by <span class="citation" data-cites="Plubell2017-th">Plubell et al. (<a href="99-end.html#ref-Plubell2017-th" role="doc-biblioref">2017</a>)</span>, illustrating the statistical concepts using a real-life use case.</p>
<p>Since this chapter uses a real-life study, we cannot objectively assess the results. The <a href="#sec-advances">advanced concepts chapter</a> demonstrates an msqrob2TMT analysis with ground truth information.</p>
<p>Before delving further into the use case, let us prepare our computational environment.</p>
</section>
<section id="load-packages" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="load-packages"><span class="header-section-number">7.2</span> Load packages</h2>
<p>First, we load the <code>msqrob2</code> package and additional packages for data manipulation and visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"patchwork"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also configure the <a href="01-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="load-data"><span class="header-section-number">7.3</span> Load data</h2>
<section id="sec-exp_design_mouse" class="level3" data-number="7.3.1">
<h3 data-number="7.3.1" class="anchored" data-anchor-id="sec-exp_design_mouse"><span class="header-section-number">7.3.1</span> Experimental context</h3>
<p>The data used in this vignette has been published by <span class="citation" data-cites="Plubell2017-th">Plubell et al. (<a href="99-end.html#ref-Plubell2017-th" role="doc-biblioref">2017</a>)</span> (PXD005953). The objective of the experiment was to explore the impact of low-fat and high-fat diets on the proteomic content of adipose tissue in mice. It also assesses whether the duration of the diet may impact the results. The authors assigned twenty mice into four groups (5 mice per group) based on their diet, either low-fat (LF) or high-fat (HF), and the duration of the diet, which was classified as short (8 weeks) or long (18 weeks). Samples from the epididymal adipose tissue were extracted from each mice. The samples were then randomly distributed across three TMT 10-plex mixtures for analysis. In each mixture, two reference labels were used, each containing pooled samples that included a range of peptides from all the samples. Not all labels were used, leading to an unbalanced design. Each TMT 10-plex mixture was fractionated into nine parts, resulting in a total of 27 MS runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="figs/mouse_experiment.png" class="img-fluid figure-img" width="353"></p>
<figcaption>Overview of the experimental design. Taken from Plubell et al.&nbsp;2017.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="getting-the-data" class="level3" data-number="7.3.2">
<h3 data-number="7.3.2" class="anchored" data-anchor-id="getting-the-data"><span class="header-section-number">7.3.2</span> Getting the data</h3>
<p>The data were reanalyzed by <span class="citation" data-cites="Huang2020-lu">Huang et al. (<a href="99-end.html#ref-Huang2020-lu" role="doc-biblioref">2020</a>)</span> and have been deposited in the <code>MSV000084264</code> MASSiVE repository, but we will retrieve the timestamped data from our <a href="https://zenodo.org/records/14767905">Zenodo repository</a>. We need 2 files: the Skyline identification and quantification table generated by the authors and the sample annotation files<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocFileCache"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> <span class="fu">BiocFileCache</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>psmFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://zenodo.org/records/14767905/files/mouse_psms.txt?download=1"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>annotFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://zenodo.org/records/14767905/files/mouse_annotations.csv?download=1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This analysis starts with the <a href="02-advanced.html#sec-psm_table">PSM table</a>. Note that columns that start with <code>"Abundance."</code> contain the quantitative values for each TMT label.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>psms <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(psmFile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 22%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 6%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 0%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Checked</th>
<th style="text-align: left;">Confidence</th>
<th style="text-align: left;">Identifying.Node</th>
<th style="text-align: left;">PSM.Ambiguity</th>
<th style="text-align: left;">Annotated.Sequence</th>
<th style="text-align: left;">Modifications</th>
<th style="text-align: right;">X..Protein.Groups</th>
<th style="text-align: right;">X..Proteins</th>
<th style="text-align: left;">Master.Protein.Accessions</th>
<th style="text-align: left;">Master.Protein.Descriptions</th>
<th style="text-align: left;">Protein.Accessions</th>
<th style="text-align: left;">Protein.Descriptions</th>
<th style="text-align: right;">X..Missed.Cleavages</th>
<th style="text-align: right;">Charge</th>
<th style="text-align: right;">DeltaScore</th>
<th style="text-align: right;">DeltaCn</th>
<th style="text-align: right;">Rank</th>
<th style="text-align: right;">Search.Engine.Rank</th>
<th style="text-align: right;">m.z..Da.</th>
<th style="text-align: right;">MH…Da.</th>
<th style="text-align: right;">Theo..MH…Da.</th>
<th style="text-align: right;">DeltaM..ppm.</th>
<th style="text-align: right;">Deltam.z..Da.</th>
<th style="text-align: left;">Activation.Type</th>
<th style="text-align: left;">MS.Order</th>
<th style="text-align: right;">Isolation.Interference….</th>
<th style="text-align: right;">Average.Reporter.S.N</th>
<th style="text-align: right;">Ion.Inject.Time..ms.</th>
<th style="text-align: right;">RT..min.</th>
<th style="text-align: right;">First.Scan</th>
<th style="text-align: left;">Spectrum.File</th>
<th style="text-align: left;">File.ID</th>
<th style="text-align: right;">Abundance..126</th>
<th style="text-align: right;">Abundance..127N</th>
<th style="text-align: right;">Abundance..127C</th>
<th style="text-align: right;">Abundance..128N</th>
<th style="text-align: right;">Abundance..128C</th>
<th style="text-align: right;">Abundance..129N</th>
<th style="text-align: right;">Abundance..129C</th>
<th style="text-align: right;">Abundance..130N</th>
<th style="text-align: right;">Abundance..130C</th>
<th style="text-align: right;">Abundance..131</th>
<th style="text-align: left;">Quan.Info</th>
<th style="text-align: right;">Ions.Score</th>
<th style="text-align: right;">Identity.Strict</th>
<th style="text-align: right;">Identity.Relaxed</th>
<th style="text-align: right;">Expectation.Value</th>
<th style="text-align: right;">Percolator.q.Value</th>
<th style="text-align: right;">Percolator.PEP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (A2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[K].aIDILDR.[SM]</td>
<td style="text-align: left;">N-Term(TMT6plex)</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q61084</td>
<td style="text-align: left;">Mitogen-activated protein kinase kinase kinase 3 OS=Mus musculus GN=Map3k3 PE=1 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">522.8161</td>
<td style="text-align: right;">1044.625</td>
<td style="text-align: right;">1044.625</td>
<td style="text-align: right;">-0.14</td>
<td style="text-align: right;">-0.00008</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">2.279464</td>
<td style="text-align: right;">10.5</td>
<td style="text-align: right;">42.390</td>
<td style="text-align: right;">63.4111</td>
<td style="text-align: right;">16103</td>
<td style="text-align: left;">PAMI-194_Mouse_U-Dd_TMT_40ug_28pctACN_25cm_120min_20160426_OT.raw</td>
<td style="text-align: left;">F3.6</td>
<td style="text-align: right;">2382.454</td>
<td style="text-align: right;">2429.068</td>
<td style="text-align: right;">2707.243</td>
<td style="text-align: right;">3317.650</td>
<td style="text-align: right;">5035.093</td>
<td style="text-align: right;">3031.7993</td>
<td style="text-align: right;">4808.5522</td>
<td style="text-align: right;">2583.115</td>
<td style="text-align: right;">3103.613</td>
<td style="text-align: right;">2795.283</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">40</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">0.0003802</td>
<td style="text-align: right;">0.0008956</td>
<td style="text-align: right;">0.0111900</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (C2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[K].aLEENNNFSk.[M]</td>
<td style="text-align: left;">N-Term(TMT6plex); K10(TMT6plex)</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">P55821</td>
<td style="text-align: left;">Stathmin-2 OS=Mus musculus GN=Stmn2 PE=1 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">812.4410</td>
<td style="text-align: right;">1623.875</td>
<td style="text-align: right;">1623.874</td>
<td style="text-align: right;">0.26</td>
<td style="text-align: right;">0.00021</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">20.489330</td>
<td style="text-align: right;">0.8</td>
<td style="text-align: right;">116.000</td>
<td style="text-align: right;">38.9028</td>
<td style="text-align: right;">7622</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_22pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">F1.3</td>
<td style="text-align: right;">1112.530</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">389.5883</td>
<td style="text-align: right;">963.0698</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">22</td>
<td style="text-align: right;">0.0000932</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000510</td>
</tr>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (C2)</td>
<td style="text-align: left;">Rejected</td>
<td style="text-align: left;">[K].eMISDIk.[F]</td>
<td style="text-align: left;">N-Term(TMT6plex); K7(TMT6plex)</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q5SQM0</td>
<td style="text-align: left;">Echinoderm microtubule-associated protein-like 6 OS=Mus musculus GN=Eml6 PE=2 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">647.3786</td>
<td style="text-align: right;">1293.750</td>
<td style="text-align: right;">1293.749</td>
<td style="text-align: right;">0.84</td>
<td style="text-align: right;">0.00054</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">62.114720</td>
<td style="text-align: right;">25.4</td>
<td style="text-align: right;">50.658</td>
<td style="text-align: right;">45.5494</td>
<td style="text-align: right;">10094</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_24pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">F1.4</td>
<td style="text-align: right;">8606.559</td>
<td style="text-align: right;">10042.301</td>
<td style="text-align: right;">7976.240</td>
<td style="text-align: right;">5196.599</td>
<td style="text-align: right;">9381.061</td>
<td style="text-align: right;">12227.7070</td>
<td style="text-align: right;">6418.2720</td>
<td style="text-align: right;">5791.740</td>
<td style="text-align: right;">9665.493</td>
<td style="text-align: right;">5747.680</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.5635688</td>
<td style="text-align: right;">0.0016120</td>
<td style="text-align: right;">0.0284500</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (C2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[KR].iIDFGLAR.[HQRT]</td>
<td style="text-align: left;">N-Term(TMT6plex)</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q3UIZ8; Q5SUV5; Q8VCR8</td>
<td style="text-align: left;">Myosin light chain kinase 3 OS=Mus musculus GN=Mylk3 PE=1 SV=1; Myosin light chain kinase family member 4 OS=Mus musculus GN=Mylk4 PE=1 SV=2; Myosin light chain kinase 2, skeletal/cardiac muscle OS=Mus musculus GN=Mylk2 PE=1 SV=2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">567.3476</td>
<td style="text-align: right;">1133.688</td>
<td style="text-align: right;">1133.688</td>
<td style="text-align: right;">-0.09</td>
<td style="text-align: right;">-0.00005</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">30.288610</td>
<td style="text-align: right;">13.5</td>
<td style="text-align: right;">41.830</td>
<td style="text-align: right;">70.4334</td>
<td style="text-align: right;">19108</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_30pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">F1.7</td>
<td style="text-align: right;">4964.429</td>
<td style="text-align: right;">5950.643</td>
<td style="text-align: right;">3133.549</td>
<td style="text-align: right;">3321.042</td>
<td style="text-align: right;">2914.397</td>
<td style="text-align: right;">3274.3197</td>
<td style="text-align: right;">3284.3334</td>
<td style="text-align: right;">4714.300</td>
<td style="text-align: right;">3846.105</td>
<td style="text-align: right;">5461.438</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">41</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">0.0002663</td>
<td style="text-align: right;">0.0000983</td>
<td style="text-align: right;">0.0004159</td>
</tr>
<tr class="odd">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (B2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[KR].iQLWDTAGQER.[FY]</td>
<td style="text-align: left;">N-Term(TMT6plex)</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">O35963</td>
<td style="text-align: left;">Ras-related protein Rab-33B OS=Mus musculus GN=Rab33b PE=1 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">515.9459</td>
<td style="text-align: right;">1545.823</td>
<td style="text-align: right;">1545.822</td>
<td style="text-align: right;">0.49</td>
<td style="text-align: right;">0.00025</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">14.174430</td>
<td style="text-align: right;">5.2</td>
<td style="text-align: right;">116.000</td>
<td style="text-align: right;">63.1792</td>
<td style="text-align: right;">15687</td>
<td style="text-align: left;">PAMI-176_Mouse_K-T_TMT_40ug_26pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">F2.5</td>
<td style="text-align: right;">2202.812</td>
<td style="text-align: right;">1313.995</td>
<td style="text-align: right;">1942.837</td>
<td style="text-align: right;">2381.060</td>
<td style="text-align: right;">1443.969</td>
<td style="text-align: right;">1701.8316</td>
<td style="text-align: right;">968.0946</td>
<td style="text-align: right;">1867.904</td>
<td style="text-align: right;">910.823</td>
<td style="text-align: right;">1371.243</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">28</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">0.0092083</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">0.0000261</td>
</tr>
<tr class="even">
<td style="text-align: left;">False</td>
<td style="text-align: left;">High</td>
<td style="text-align: left;">Mascot (A2)</td>
<td style="text-align: left;">Unambiguous</td>
<td style="text-align: left;">[K].aIDILDR.[SM]</td>
<td style="text-align: left;">N-Term(TMT6plex)</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Q61084</td>
<td style="text-align: left;">Mitogen-activated protein kinase kinase kinase 3 OS=Mus musculus GN=Map3k3 PE=1 SV=1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">522.8161</td>
<td style="text-align: right;">1044.625</td>
<td style="text-align: right;">1044.625</td>
<td style="text-align: right;">-0.14</td>
<td style="text-align: right;">-0.00008</td>
<td style="text-align: left;">CID</td>
<td style="text-align: left;">MS2</td>
<td style="text-align: right;">21.614820</td>
<td style="text-align: right;">19.0</td>
<td style="text-align: right;">116.000</td>
<td style="text-align: right;">62.4027</td>
<td style="text-align: right;">15935</td>
<td style="text-align: left;">PAMI-194_Mouse_U-Dd_TMT_40ug_26pctACN_25cm_120min_20160426_OT.raw</td>
<td style="text-align: left;">F3.5</td>
<td style="text-align: right;">6064.501</td>
<td style="text-align: right;">3790.208</td>
<td style="text-align: right;">4328.930</td>
<td style="text-align: right;">6528.889</td>
<td style="text-align: right;">6080.098</td>
<td style="text-align: right;">7705.8087</td>
<td style="text-align: right;">7836.0572</td>
<td style="text-align: right;">4606.420</td>
<td style="text-align: right;">5347.845</td>
<td style="text-align: right;">5869.469</td>
<td style="text-align: left;">NA</td>
<td style="text-align: right;">42</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">0.0002350</td>
<td style="text-align: right;">0.0005618</td>
<td style="text-align: right;">0.0067230</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We also load the <a href="01-basics.html#sec-annotation_table">annotation table</a>. Each row in the annotation table contains information for one sample (the table below shows the first 6 rows).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(annotFile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 36%">
<col style="width: 13%">
<col style="width: 10%">
<col style="width: 19%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Channel</th>
<th style="text-align: left;">Condition</th>
<th style="text-align: left;">Run</th>
<th style="text-align: left;">BioReplicate</th>
<th style="text-align: left;">Mixture</th>
<th style="text-align: left;">Fraction</th>
<th style="text-align: right;">TechRepMixture</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_30pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_30pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_28pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_28pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_90pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_90pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_40pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_40pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_22pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_22pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">126</td>
<td style="text-align: left;">Long_LF</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J_TMT_40ug_24pctACN_25cm_120min_20160223_OT.raw</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J.X126</td>
<td style="text-align: left;">PAMI-176_Mouse_A-J</td>
<td style="text-align: left;">40ug_24pctACN_25cm_120min_20160223</td>
<td style="text-align: right;">1</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We perform a little cleanup of the sample annotations to generate the information needed for later data modelling, namely</p>
<ol type="1">
<li>We extract the diet type from the condition variable.</li>
<li>We extract the diet duration from the condition variable.</li>
<li>We rename the <code>Channel</code> column to <code>Label</code> for more clarity with the main text.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>Duration <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, coldata<span class="sc">$</span>Condition) <span class="do">## 1.</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>Diet <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*_"</span>, <span class="st">""</span>, coldata<span class="sc">$</span>Condition) <span class="do">## 2.</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(coldata)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"Label"</span> <span class="do">## 3.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will also subset the data set to reduce computational costs. If you want to run the vignette on the full data set, you can skip this chunk. We here randomly sample 500 proteins from the experiment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>proteinIds <span class="ot">&lt;-</span> <span class="fu">unique</span>(psms<span class="sc">$</span>Protein.Accessions)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>psms <span class="ot">&lt;-</span> psms[psms<span class="sc">$</span>Protein.Accessions <span class="sc">%in%</span> <span class="fu">sample</span>(proteinIds, <span class="dv">500</span>), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="the-qfeatures-data-class" class="level3" data-number="7.3.3">
<h3 data-number="7.3.3" class="anchored" data-anchor-id="the-qfeatures-data-class"><span class="header-section-number">7.3.3</span> The <code>QFeatures</code> data class</h3>
<p>We combine the two tables into a <a href="01-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>. We need to point to the column containing the run information. For the annotation table, this is simply the <code>Run</code> column. For the Skyline table, this is the <code>Spectrum.File</code> column. We also add a <code>quantCols</code> column in the annotation table. We also simply the run names upon conversion for conciseness.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>runCol <span class="ot">&lt;-</span> coldata<span class="sc">$</span>Run</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>coldata<span class="sc">$</span>quantCols <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Abundance.."</span>, coldata<span class="sc">$</span>Label)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(psms, <span class="at">colData =</span> coldata,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">quantCols =</span> <span class="fu">unique</span>(coldata<span class="sc">$</span>quantCols),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">runCol =</span> <span class="st">"Spectrum.File"</span>, <span class="at">name =</span> <span class="st">"psms"</span>)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mouse) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^.*(Mouse.*ACN).*raw"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">names</span>(mouse))</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>mouse</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An instance of class QFeatures (type: bulk) with 27 sets:

 [1] Mouse_A-J_TMT_40ug_14pctACN: SummarizedExperiment with 198 rows and 10 columns 
 [2] Mouse_A-J_TMT_40ug_20pctACN: SummarizedExperiment with 481 rows and 10 columns 
 [3] Mouse_A-J_TMT_40ug_22pctACN: SummarizedExperiment with 514 rows and 10 columns 
 ...
 [25] Mouse_U-Dd_TMT_40ug_30pctACN: SummarizedExperiment with 726 rows and 10 columns 
 [26] Mouse_U-Dd_TMT_40ug_40pctACN: SummarizedExperiment with 566 rows and 10 columns 
 [27] Mouse_U-Dd_TMT_40ug_90pctACN: SummarizedExperiment with 251 rows and 10 columns </code></pre>
</div>
</div>
<p>We now have a <code>QFeatures</code> object with 27 sets, each containing data associated with an MS run.</p>
</section>
</section>
<section id="data-preprocessing" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="data-preprocessing"><span class="header-section-number">7.4</span> Data preprocessing</h2>
<p><code>msqrob2</code> relies on the <code>QFeatures</code> data structure, meaning that we can directly make use of <code>QFeatures</code>’ data preprocessing functionality (see also the <code>QFeatures</code> <a href="https://rformassspectrometry.github.io/QFeatures/articles/Processing.html">documentation</a>).</p>
<section id="encoding-missing-values" class="level3" data-number="7.4.1">
<h3 data-number="7.4.1" class="anchored" data-anchor-id="encoding-missing-values"><span class="header-section-number">7.4.1</span> Encoding missing values</h3>
<p>Peptides with zero intensities are missing peptides and should be represent with a <code>NA</code> value rather than <code>0</code> (see <a href="#encoding-missing-values">Encoding missing values</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(mouse, <span class="fu">names</span>(mouse))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sample-filtering" class="level3" data-number="7.4.2">
<h3 data-number="7.4.2" class="anchored" data-anchor-id="sample-filtering"><span class="header-section-number">7.4.2</span> Sample filtering</h3>
<p>We remove the reference samples that were used by the MSstatsTMT authors to obtain normalisation factors since msqrob2TMT workflows do not require normalisation from reference label. The information about which samples are normalisation samples is available from the <code>colData</code>, in the <code>Condition</code> column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(mouse<span class="sc">$</span>Condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 Long_HF  Long_LF   Long_M     Norm Short_HF Short_LF 
      45       45       36       54       45       45 </code></pre>
</div>
</div>
<p>We remove any sample that is marked as <code>Norm</code>. We also remove sample that are annotated as <code>Long_M</code> since we could not find documentation for this group.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">subsetByColData</span>(</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>    mouse, mouse<span class="sc">$</span>Condition <span class="sc">!=</span> <span class="st">"Norm"</span> <span class="sc">&amp;</span> mouse<span class="sc">$</span>Condition <span class="sc">!=</span> <span class="st">"Long_M"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="psm-filtering" class="level3" data-number="7.4.3">
<h3 data-number="7.4.3" class="anchored" data-anchor-id="psm-filtering"><span class="header-section-number">7.4.3</span> PSM filtering</h3>
<p>We <a href="01-basics.html#sec-filter">filter features</a> for which more than 70% of the intensities are missing in a run. We keep the spectrum as soon as the reporter ions are observed in at least 3 out of 10 TMT labels of the run (same cut-off as applied in <span class="citation" data-cites="Huang2020-lu">Huang et al. (<a href="99-end.html#ref-Huang2020-lu" role="doc-biblioref">2020</a>)</span>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(mouse, <span class="fu">names</span>(mouse), <span class="at">pNA =</span> <span class="fl">0.7</span>) <span class="do">## 2.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We next remove PSMs that could not be mapped to a protein or that map to multiple proteins (the protein identifier contains multiple identifiers separated by a <code>;</code>). We use <code>filterFeatures()</code> that will keep the row that fulfill the condition below. Note that <code>Protein.Accessions</code> is a column generated by Skyline that is available in the <code>rowData</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="sc">~</span> Protein.Accessions <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>        <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Protein.Accessions)) <span class="do">## Remove protein groups</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Peptide ions that were identified with multiple PSMs in a run are collapsed to the PSM with the highest summed intensity over the labels, a strategy that is also used by MSstats. We will again use <code>filterFeatures()</code>, but the highest summed intensity for each PSM is not available in the <code>rowData</code>, so we need to create it manually.</p>
<p>We therefore loop over each set:</p>
<ol type="1">
<li>Make a new variable for ionID in the <code>rowData</code>, which is defined as the peptide sequence and its charge.</li>
<li>We calculate the <code>rowSums</code> for each ion.</li>
<li>Make a new variable <code>psmRank</code> that ranks the PSMs for each ion based on the summed intensity.</li>
<li>We store the new information back in the <code>rowData</code>.</li>
<li>We keep the PSM with the highest summed intensity, that is that ranks first (note that PSM unique to an ion will always rank first).</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(mouse)) {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>    rowdata <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[i]])</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    rowdata<span class="sc">$</span>ionID <span class="ot">&lt;-</span> <span class="fu">paste0</span>(rowdata<span class="sc">$</span>Annotated.Sequence, rowdata<span class="sc">$</span>Charge) <span class="do">## 1.</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    rowdata<span class="sc">$</span>rowSums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">assay</span>(mouse[[i]]), <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="do">## 2.</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    rowdata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(rowdata) <span class="sc">|&gt;</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">group_by</span>(ionID) <span class="sc">|&gt;</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mutate</span>(<span class="at">psmRank =</span> <span class="fu">rank</span>(<span class="sc">-</span>rowSums)) <span class="do">## 3.</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rowData</span>(mouse[[i]]) <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(rowdata) <span class="do">## 4.</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(mouse, <span class="sc">~</span> psmRank <span class="sc">==</span> <span class="dv">1</span>) <span class="do">## 5.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>So, we implicitly collapsed the PSM-level data into <strong>peptide-ion</strong> data, where each row represents a PSM, but also a unique ion within a run. So we will refer to the data as “ion-level”.</p>
</section>
<section id="standard-preprocessing-workflow" class="level3" data-number="7.4.4">
<h3 data-number="7.4.4" class="anchored" data-anchor-id="standard-preprocessing-workflow"><span class="header-section-number">7.4.4</span> Standard preprocessing workflow</h3>
<p>We can now prepare the data for modelling. The workflow ensures the data complies to <code>msqrob2</code>’s requirements:</p>
<ol type="1">
<li>Intensities are log-transformed.</li>
<li>Samples are normalised.</li>
<li>(optionally) PSMs intensities are summarised into protein abundance values for protein-level workflows.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>sNames <span class="ot">&lt;-</span> <span class="fu">names</span>(mouse)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">logTransform</span>( <span class="do">## 1.</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>    mouse, sNames, <span class="at">name =</span> <span class="fu">paste0</span>(sNames, <span class="st">"_log"</span>), <span class="at">base =</span> <span class="dv">2</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">normalize</span>( <span class="do">## 2.</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="fu">paste0</span>(sNames, <span class="st">"_log"</span>), <span class="at">name =</span> <span class="fu">paste0</span>(sNames, <span class="st">"_norm"</span>),</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">"center.median"</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>( <span class="do">## 3.</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="fu">paste0</span>(sNames, <span class="st">"_norm"</span>), <span class="at">name =</span> <span class="fu">paste0</span>(sNames, <span class="st">"_proteins"</span>),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>, <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>medianPolish,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.rm=</span><span class="cn">TRUE</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We conclude the preprocessing by joining the assays of the different runs in a single ion set for ion-level models. In order to correctly match peptide ions across rus, we use <code>ionID</code> as a row identifier.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>(mouse, <span class="fu">paste0</span>(sNames, <span class="st">"_norm"</span>), <span class="at">fcol =</span> <span class="st">"ionID"</span>, <span class="st">"ions_norm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also join the protein sets for protein-level models. We omit <code>fcol</code> and will merge rows based on their row name (protein identifier).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">joinAssays</span>(mouse, <span class="fu">paste0</span>(sNames, <span class="st">"_proteins"</span>), <span class="st">"proteins"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-exploration" class="level2" data-number="7.5">
<h2 data-number="7.5" class="anchored" data-anchor-id="data-exploration"><span class="header-section-number">7.5</span> Data exploration</h2>
<p>As <a href="#sec-exp_design_mouse">described above</a>, the samples originate from mice that were either subject to a low-fat (<code>LF</code>) or high-fat (<code>HF</code>) diet. Moreover, each diet was maintained for a short duration (<code>Short</code>) or a long duration (<code>Long</code>). Note that each group contains 5 mice but the peptides from each sample have been fractionated in 9 fractions, leading to 45 units per group.</p>
<p>In this case, we are interested in the effects of diet type and the effect of diet duration. The table below confirms we have a balanced design for each condition.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="at">Diet =</span> mouse<span class="sc">$</span>Diet, <span class="at">Duration =</span> mouse<span class="sc">$</span>Duration)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    Duration
Diet Long Short
  HF   45    45
  LF   45    45</code></pre>
</div>
</div>
<p>Furthermore, there are potential unwanted sources of variation: the experimental unit (i.e.&nbsp;the mouse, <code>BioReplicate</code>), the fraction (<code>Fraction</code>), the run (<code>Run</code>), the TMT mixture (<code>Mixture</code>).</p>
<p>We will explore the main sources of variation in the data (see <a href="#data-exploration">Data exploration</a>). Unfortunately, the peptide ion data contains too many missing values and cannot be explored using standard dimension reduction approaches. <code>omicsGMF</code> <span class="citation" data-cites="Segers2025-ce">(<a href="99-end.html#ref-Segers2025-ce" role="doc-biblioref">Segers et al. 2025</a>)</span> provides an interesting alternative, but is still in its early stage and will not be included here. We will therefore run a MDS analysis on the protein-level data instead, which contains less missing values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(mouse, <span class="st">"proteins"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">"SingleCellExperiment"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">runMDS</span>(<span class="at">exprs_values =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now plot the MDS and colour each sample based on different potential sources of variation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Run"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by Run"</span>) <span class="sc">+</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">rainbow</span>(<span class="dv">27</span>)) <span class="sc">+</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Fraction"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by Fraction"</span>) <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"BioReplicate"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by BioReplicate"</span>) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Mixture"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by Mixture"</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Diet"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by Diet"</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"Duration"</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Coloured by Duration"</span>) <span class="sc">&amp;</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The data exploration leads to several observations:</p>
<ul>
<li>The strongest source of variation is associated with the MS acquisition run.</li>
<li>Part of this run effect is influenced by which fraction it contains since samples from the same fraction tend to be closer than samples from different fractions. (<strong>TODO DISCUSS</strong>)</li>
<li>It is difficult to identify an effect of mouse (biological replicate) because every run contains distinct mice. However, this does not exclude an effect of mice which has been identified as a potential source of variation and hence should still be modelled. (<strong>TODO DISCUSS</strong>)</li>
<li>There is potentially also an effect of TMT mixture since samples from the same mixtures tend to cluster together (in the center of the plot). However, this effect are more subtle to detect and difficult to disentangle from the run and fraction effects.</li>
<li>Although again very subtle, we can see within each run that samples from the mice with the same diet tends to group together. However, these effects are overwhelmed by the run effects. An effect of duration is to subtle to pinpoint from the current data exploration.</li>
</ul>
<p>Data modelling disentangles the different sources of variation, given their are properly defined in the model, hence the next section.</p>
</section>
<section id="data-modelling" class="level2" data-number="7.6">
<h2 data-number="7.6" class="anchored" data-anchor-id="data-modelling"><span class="header-section-number">7.6</span> Data modelling</h2>
<p>The preprocessed data can now be modelled to answer biologically relevant questions.</p>
<section id="sources-of-variation" class="level3" data-number="7.6.1">
<h3 data-number="7.6.1" class="anchored" data-anchor-id="sources-of-variation"><span class="header-section-number">7.6.1</span> Sources of variation</h3>
<p>Proteomics data contain several sources of variation that need to be accounted for by the model:</p>
<ol type="1">
<li><p><strong>Treatment of interest</strong>: we model the source of variation induced by the experimental treatment of interest as a <strong>fixed effect</strong>. Fixed effects are effect that are considered non-random, i.e.&nbsp;the treatment effect is assumed to be the same and reproducible across repeated experiments, but it is unknown and has to be estimated. We will include <code>Diet</code> as a fixed effect that models the fact that a change in diet type can induce changes in protein abundance. Similarly, we also include <code>Duration</code> as a fixed effect to model the change in protein abundance induces by the diet duration. Finally, we will also include an interaction between the two variables allowing that the changes in protein abundance induced by diet type can be different whether the mice were fed for a short or long duration.</p></li>
<li><p><strong>Pseudo-replication</strong>: the experiment involves biological replication as the adipose tissue extracts were sampled from 20 mice (5 mice per Diet x Duration combination). The tissue from each mouse was prepared in a single TMT mixture, but each mixture was acquired in 9 fractions, so we have 9 measures for each mice. While the treatment is applied at the mice level (experimental unit), we actually measure 9 fractions as an outcome (observational unit). We refer to <strong>pseudo-replication</strong> when the observational unit is different from the experimental unit. We therefore need to account for the potential correlation among pseudo-replicates from the same mouse compared to between different mice. These effects are typically modelled as <a href="02-advanced.html#sec-random_effect">random effects</a>, and are assumed to be i.i.d normally distributed with mean 0 and constant variance, <span class="math inline">\(u_{mouse} \sim
N(0,\sigma^{2,\text{mouse}})\)</span>. This random effect models the correlation of the log2-intensities of pseudo replicates, explicitly.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mouse<span class="sc">$</span>BioReplicate))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 20</code></pre>
</div>
</div>
<ol start="3" type="1">
<li><strong>Labelling effects</strong>: the 20 mouse adipose tissue samples have been labelled using 18-plex TMT. We can expect that samples measured within the same TMT label may be more similar than samples measured within different TMT labels. Since these effects may not be reproducible from one experiment to another, for instance because each TMT kit may potentially contain different impurity ratios, we can account for this source of variation using a random effect for TMT label.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mouse<span class="sc">$</span>Label))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 10</code></pre>
</div>
</div>
<ol start="4" type="1">
<li><strong>Mixture effects</strong>: the 20 mouse samples were assigned to one out of 3 mixtures. Again, we expect that protein intensities from the same mixture will be more alike than those of different mixtures. Hence, we will add a random effect for mixture.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(mouse<span class="sc">$</span>Mixture)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 PAMI-176_Mouse_A-J  PAMI-176_Mouse_K-T PAMI-194_Mouse_U-Dd 
                 54                  63                  63 </code></pre>
</div>
</div>
<ol start="5" type="1">
<li><strong>Run effects</strong>: log2-intensities that are measured within the same run will be more similar than log2-intensities between runs. We will use a random effect for run to explicitly model this correlation in the data. Note that each sample has been acquired in 9 fractions, each fraction being measured in a separate run. Accounting for the effects of run will also absorb the effects of fraction.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(mouse<span class="sc">$</span>Run))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 27</code></pre>
</div>
</div>
<ol start="6" type="1">
<li><p><strong>Spectrum effects</strong>: we will directly estimate the treatment effect at the protein-level from ion-level data. This will again induce additional levels of correlation. The intensities for the different TMT labels in the same spectrum (PSM) within a run will be more similar than the intensities between spectra. We therefore need to add a random effect term to account for the within spectrum correlation structure. Note that a spectrum here contains the data from one peptide ion within a run. Hence, modelling a random effect for spectrum boils down to modelling a random effect for peptide ion nested within run.</p></li>
<li><p><strong>Labelling effects nested in run</strong>: modelling the data at the ion-level also implies that a label in a run contains multiple ion intensities for each protein. Hence, intensities from different peptide ions for a protein with the same label within a run will be more alike than intensities of different PSMs for the same protein with different labels and/or runs, and we will address this correlation with a random effect for label nested in run.</p></li>
</ol>
<p><code>msqrob2</code> workflows rely on linear mixed models, which are models that can estimate and predict fixed and random effects, respectively.</p>
<p>Now we have identified the sources of variation, we can define a model. We will model the main effects for <code>Diet</code> and <code>Duration</code>, and a <code>Diet:Duration</code> interaction, to account for proteins for which the <code>Diet</code> effect changes according to <code>Duration</code>, and vice versa, which can be written as <code>Diet + Duration + Diet:Duration</code>, shortened into <code>Diet * Duration</code>. Adding the technical sources of variation, the model becomes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="er">~</span> Diet <span class="sc">*</span> Duration <span class="sc">+</span> <span class="do">## (1) fixed effect for Diet and Duration with interaction</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> BioReplicate) <span class="sc">+</span>  <span class="do">## (2) random effect for biological replicate (mouse)      </span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## (3) random effect for label</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## (4) random effect for mixture</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## (5) random effect for MS run</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>ionID) <span class="sc">+</span> <span class="do">## (6) random effect for spectrum, i.e. ionID nested in run</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> Run<span class="sc">:</span>Label)  <span class="do">## (7) random effect for label nested in MS run</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can run the <code>msqrob2</code> statistical workflow.</p>
</section>
<section id="model-estimation" class="level3" data-number="7.6.2">
<h3 data-number="7.6.2" class="anchored" data-anchor-id="model-estimation"><span class="header-section-number">7.6.2</span> Model estimation</h3>
<p>We estimate the peptide-ion-level model with <code>msqrobAggregate()</code> (see <a href="02-advanced.html#sec-run_model">the modelling section</a>). Recall that variables defined in <code>model</code> are automatically retrieved from the <code>colData</code> (<code>"Diet"</code>, <code>"Duration"</code>, <code>"Label"</code>, <code>"Mixture"</code>) and from the <code>rowData</code> (<code>"ionID"</code>). We also enable M-estimation (<code>robust = TRUE</code>) for improved robustness against outliers and ridge penalisation (<code>ridge = TRUE</code>) to stabilise the parameter estimation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">msqrobAggregate</span>(</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"ions_norm"</span>,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> model,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">fcol =</span> <span class="st">"Protein.Accessions"</span>,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumnName =</span> <span class="st">"msqrob_ion"</span>,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"proteins_msqrob"</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once the model is estimated, we can start answering biological questions.</p>
</section>
<section id="hypothesis-testing" class="level3" data-number="7.6.3">
<h3 data-number="7.6.3" class="anchored" data-anchor-id="hypothesis-testing"><span class="header-section-number">7.6.3</span> Hypothesis testing</h3>
<p>In this section, you will learn how to convert a biological question into a statistical hypothesis.</p>
<section id="difference-between-low-fat-and-high-fat-diet-after-short-duration" class="level4">
<h4 class="anchored" data-anchor-id="difference-between-low-fat-and-high-fat-diet-after-short-duration">Difference between low fat and high fat diet after short duration</h4>
<p>A first question one can ask is: how are protein abundance affected by diet when only considering a short diet duration? We need to convert this question in a combination of the model parameters, also referred to as a contrast. To aid defining contrasts, we will visualise the experimental design using the <code>ExploreModelMatrix</code> package. Note that with ExploreModelMatrix we can only visualise fixed effects part of the model. This is fine as the mean protein abundances can only systematically differ from each other according to the main effects for <code>Diet</code> and <code>Duration</code> and the <code>Diet:Duration</code> interaction.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sampleData =</span>  <span class="fu">colData</span>(mouse),</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> Diet <span class="sc">*</span> Duration,</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>vd<span class="sc">$</span>plotlist[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_VisualizeDesign-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Assessing the difference between low-fat and high-fat diets for short duration boils down to assessing the difference between the <code>Short_LF</code> and <code>Short_HF</code>. The mean for the short low-fat diet group is defined by <code>(Intercept) + DietLF + DurationShort + DietLF:DurationShort</code>. The mean for the short high-fat diet group is defined by <code>(Intercept) + DurationShort</code>. The difference between the two results in the contrast below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="st">"ridgeDietLF + ridgeDietLF:DurationShort"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note that because we used ridge regression for modelling, we need to prefix the parameter names with <code>ridge</code>. We can further specify the null hypothesis, that is we are interest whether the differences between the two groups is different from zero.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>(hypothesis1 <span class="ot">&lt;-</span> <span class="fu">paste</span>(contrast, <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ridgeDietLF + ridgeDietLF:DurationShort = 0"</code></pre>
</div>
</div>
<p>We next use <code>makeContrast()</code> to build a contrast matrix.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>    hypothesis1,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">"ridgeDietLF"</span>,<span class="st">"ridgeDurationShort"</span>,<span class="st">"ridgeDietLF:DurationShort"</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                          ridgeDietLF + ridgeDietLF:DurationShort
ridgeDietLF                                                     1
ridgeDurationShort                                              0
ridgeDietLF:DurationShort                                       1</code></pre>
</div>
</div>
<p>We can now test our null hypothesis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, L, <span class="at">modelColumn =</span> <span class="st">"msqrob_ion"</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let us retrieve the result table from the <code>rowData</code>. Note that the model column is named after the column names of the contrast matrix <code>L</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins_msqrob"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(inference, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               logFC           se       df             t      pval adjPval
A2AJB7  3.491872e-09 9.284189e-05 67.35093  3.761096e-05 0.9999701       1
A2AJK6            NA           NA       NA            NA        NA      NA
A2AQP0            NA           NA       NA            NA        NA      NA
A2AWP8            NA           NA       NA            NA        NA      NA
A6H8H2            NA           NA       NA            NA        NA      NA
B1AVY7            NA           NA       NA            NA        NA      NA
B2RSH2  2.296647e-09 8.045105e-05 52.17788  2.854713e-05 0.9999773       1
C0HKD8            NA           NA       NA            NA        NA      NA
D3Z5L6            NA           NA       NA            NA        NA      NA
E9Q5C9 -1.798058e-09 6.822108e-05 29.11140 -2.635634e-05 0.9999792       1
       Protein
A2AJB7  A2AJB7
A2AJK6  A2AJK6
A2AQP0  A2AQP0
A2AWP8  A2AWP8
A6H8H2  A6H8H2
B1AVY7  B1AVY7
B2RSH2  B2RSH2
C0HKD8  C0HKD8
D3Z5L6  D3Z5L6
E9Q5C9  E9Q5C9</code></pre>
</div>
</div>
<p>The table contains the hypothesis testing results for every protein. Notice that several rows contain missing values. This is because data modelling resulted in a <code>fitError</code>. The model cannot be estimated for some proteins either because of patterns in missing values, or because a protein was measured from a single peptide ion alleviating the estimation of spectrum effects (see <a href="#sec-fiterror">how to deal with <code>fitError</code>s</a>).</p>
<p>We can use the table above directly to build a volcano plot using <code>ggplot2</code> functionality.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on differences between LF and HF (short duration)"</span>,</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(L)), <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_short-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>In this example (remember this is a subset of the complete data set), only a few proteins pass the significance threshold of 5%. Let us visualise the protein with the largest fold change.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>(targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.max</span>(inference<span class="sc">$</span>logFC)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Q5SWU9"</code></pre>
</div>
</div>
<p>To obtain the required data, we perform a little data manipulation pipeline:</p>
<ol type="1">
<li>We use the <code>QFeatures</code> subsetting functionality to retrieve all data related to Q5SWU9 and focusing on the <code>ions_norm</code> set that contains the preprocessed peptide ion data used for modelling.</li>
<li>We use <code>longForm()</code> to convert the object into a table suitable for plotting.</li>
<li>We remove missing values for plotting and focus only on the data with short diet duration.</li>
<li>We reorder the sample identifiers to improve visualisation.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ionData <span class="ot">&lt;-</span> mouse[targetProtein, , <span class="st">"ions_norm"</span>] <span class="sc">|&gt;</span> <span class="co">#1</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(mouse)), <span class="co">#2</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">rowvars =</span> <span class="fu">c</span>(<span class="st">"Protein.Accessions"</span>, <span class="st">"ionID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value) <span class="sc">&amp;</span> Duration <span class="sc">==</span> <span class="st">"Short"</span>) <span class="sc">|&gt;</span> <span class="co">#3</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">colname =</span> <span class="fu">factor</span>(colname, <span class="at">levels =</span> <span class="fu">unique</span>(colname[<span class="fu">order</span>(Condition)]))) <span class="co">#4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can now plot the log normalised intensities. Since the protein is modelled at the peptide ion level, multiple ion intensities are recorded in each sample. Each ion is linked across samples using a grey line. Samples are coloured according to the diet type. Finally, we split the plot in facets, one for each mixture, to visualise the heterogeneity induced by different pools of mice.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ionData) <span class="sc">+</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> ionID), <span class="at">linewidth =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Condition)) <span class="sc">+</span></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="sc">~</span> Mixture, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2 intensity"</span>) <span class="sc">+</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_detail_short-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The statistical analysis revealed a significant increase (positive log fold change) of the abundance for Q5SWU9 in the group fed with a low-fat diet compared to the high-fat diet fed group (upon early diet duration). This finding can be visually validated as there is a systematic increase in peptide ion intensities between the low-fat diet group (blue) compared to the high-fat diet group (red).</p>
</section>
<section id="difference-between-low-fat-and-high-fat-diet-after-long-duration" class="level4">
<h4 class="anchored" data-anchor-id="difference-between-low-fat-and-high-fat-diet-after-long-duration">Difference between low fat and high fat diet after long duration</h4>
<p>The second question one can ask is what proteins are affected by diet when only considering, this time, a long diet duration. Following the same approach as above, the contrast becomes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>hypothesis2 <span class="ot">&lt;-</span> <span class="st">"ridgeDietLF = 0"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We run the same statistical analysis pipeline as above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>    hypothesis2,</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">"ridgeDietLF"</span>,<span class="st">"ridgeDurationShort"</span>,<span class="st">"ridgeDietLF:DurationShort"</span>)</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, L, <span class="at">modelColumn =</span> <span class="st">"msqrob_ion"</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins_msqrob"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we plot the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on differences between LF and HF (long duration)"</span>,</span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(L)), <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_long-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Again, only a few proteins come out differentially abundant between the two diets, but after a long diet duration. Surprisingly, there is only a small overlap between differential protein after short duration and after long duration. One hypothesis is that there is not sufficient data to detect a reliable difference. A solution would be to combine both short and long diet duration to retrieve an averaged systematic effect between diets that combine all available data. Another hypothesis is that diet duration may influence the effect of diet on the protein abundances. We will explore the two hypothesis in the following two sub-sections.</p>
</section>
<section id="average-difference-between-low-fat-and-high-fat-diet" class="level4">
<h4 class="anchored" data-anchor-id="average-difference-between-low-fat-and-high-fat-diet">Average difference between low fat and high fat diet</h4>
<p>One may want to identify the set of proteins that are systematically differentially abundant between diets, irrespective of the duration. To answer this question, we want to infer on the average difference between group <code>LF</code> and group <code>HF</code>. The average low-fat diet is defined by <code>((Intercept) + DietLF + DurationShort + DietLF:DurationShort + (Intercept) + DietLF)/2</code>. The average high-fat diet group is defined by <code>((Intercept) + DurationShort + (Intercept))/2</code>. The difference between the two results in the hypothesis below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>hypothesis3 <span class="ot">&lt;-</span> <span class="st">"ridgeDietLF + (ridgeDietLF:DurationShort)/2 = 0"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We next run again the same statistical analysis pipeline as above.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>    hypothesis3,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">"ridgeDietLF"</span>,<span class="st">"ridgeDurationShort"</span>,<span class="st">"ridgeDietLF:DurationShort"</span>)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, L, <span class="at">modelColumn =</span> <span class="st">"msqrob_ion"</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins_msqrob"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we plot the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on average difference between LF and HF"</span>,</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(L)), <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_average-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We find much more significant proteins when combining all available data to infer the differences between low-fat and high-fat diets, irrespective of duration. We also retrieve a good overlap between this set of significant proteins and the two previous sets, indicating that more data helped improving the statistical power.</p>
</section>
<section id="interaction-does-the-diet-effect-change-according-to-duration" class="level4">
<h4 class="anchored" data-anchor-id="interaction-does-the-diet-effect-change-according-to-duration">Interaction: does the diet effect change according to duration?</h4>
<p>We will now explore whether the effect of diet on protein abundance may be affected by duration, i.e.&nbsp;we want to infer on the difference of differences. The difference between hypothesis 1 and 2 is <code>(DietLF + DietLF:DurationShort) - (DietLF)</code> and results in the hypothesis below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>hypothesis4 <span class="ot">&lt;-</span> <span class="st">"ridgeDietLF:DurationShort = 0"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can proceed with the same statistical pipeline.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>    hypothesis4,</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">"ridgeDietLF"</span>,<span class="st">"ridgeDurationShort"</span>,<span class="st">"ridgeDietLF:DurationShort"</span>)</span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, L, <span class="at">modelColumn =</span> <span class="st">"msqrob_ion"</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins_msqrob"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we plot the results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on the effect of duration on the differences between diets"</span>,</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">gsub</span>(<span class="st">"ridgeCondition"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(L)), <span class="st">"= 0"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_interaction-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>There are only 3 proteins for which the effect of diet changes according to the duration. Let us visually explore this changes for the most significant protein.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>(targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Q924P3"</code></pre>
</div>
</div>
<p>We use again <code>Qfeatures</code>’s data manipulation pipeline.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>ionData <span class="ot">&lt;-</span> mouse[targetProtein, , <span class="st">"ions_norm"</span>] <span class="sc">|&gt;</span> <span class="co">#1</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(mouse)), <span class="co">#2</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">rowvars =</span> <span class="fu">c</span>(<span class="st">"Protein.Accessions"</span>, <span class="st">"ionID"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">|&gt;</span> <span class="co">#3</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">colname =</span> <span class="fu">factor</span>(colname, <span class="at">levels =</span> <span class="fu">unique</span>(colname[<span class="fu">order</span>(Condition)]))) <span class="co">#4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And we explore the peptide ion data by visualising the differences between LF and HF separately for each mixture and diet duration in order to highlight changes in direction between these differences according to duration. We link data points belonging to the same peptide ion using a grey line.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ionData) <span class="sc">+</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> ionID), <span class="at">linewidth =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> Diet)) <span class="sc">+</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(Mixture <span class="sc">~</span> Duration, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2 intensity"</span>) <span class="sc">+</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_detail_interaction-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The graph hints towards a slight increase in protein abundance in the low-fat diet group compared to the high-fat diet group during a short diet duration, but this increase disappears after a long diet duration. However, the visual inspection of the results also shows that the result rely on sparse and highly unbalanced data. The results may hence require further experimental validation.</p>
<p>Note that we performed the statistical analysis for each hypothesis separately. However, <code>msqrob2</code> can assess multiple hypotheses at once.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(hypothesis1, hypothesis2, hypothesis3, hypothesis4),</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">parameterNames =</span> <span class="fu">c</span>(<span class="st">"ridgeDietLF"</span>,<span class="st">"ridgeDurationShort"</span>,<span class="st">"ridgeDietLF:DurationShort"</span>)</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins_msqrob"</span>, L,</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumn =</span> <span class="st">"msqrob_ion"</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note that since we already generated results for the contrast, we overwrite the results with the argument <code>overwrite = TRUE</code>.</p>
<p>We retrieve the inference tables from the <code>rowData</code> to generate the volcano plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>inferenceTables <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins_msqrob"</span>]])[, <span class="fu">colnames</span>(L)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We here use a <code>lapply()</code> loop to generate the plots. The code chunk is elaborate, but follows the same structure as in the previous section. This generates a list of volcano plots, one for each hypothesis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>volcanoPlots <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(inferenceTables), <span class="cf">function</span>(i) {</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>    inference <span class="ot">&lt;-</span> inferenceTables[[i]]</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">"Hypothesis test:"</span>,</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">paste</span>(<span class="fu">gsub</span>(<span class="st">"ridge"</span>, <span class="st">""</span>, i), <span class="st">"= 0"</span>))</span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We combine all the plots in a single figure using the <code>patchwork</code> packages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(volcanoPlots)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_all-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="protein-level-model" class="level3" data-number="7.6.4">
<h3 data-number="7.6.4" class="anchored" data-anchor-id="protein-level-model"><span class="header-section-number">7.6.4</span> Protein-level model</h3>
<p>This section illustrates data modelling starting from protein data instead of ion data.</p>
<p>Here, the workflow will use the summarised peptide ion intensities, as performed during the preprocessing. Note, that we no longer have multiple quantitative values for a protein in the same label of a run. Hence, we can omit the nested effects for label and ionID in run.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>modelSum <span class="ot">&lt;-</span> <span class="er">~</span> Diet <span class="sc">*</span> Duration <span class="sc">+</span> <span class="do">## fixed effect for Diet and Duration with interaction</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Label) <span class="sc">+</span> <span class="do">## (1) random effect for label</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Mixture) <span class="sc">+</span> <span class="do">## (2) random effect for mixture</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> Run) <span class="sc">+</span> <span class="do">## (3) random effect for MS run</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>        (<span class="dv">1</span> <span class="sc">|</span> BioReplicate)  <span class="do">## (6) random effect for biorepeat (mouse)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For protein-level modelling, we use <code>msqrob()</code> instead of <code>msqrobAggregate()</code>, but their function arguments closely overlap.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins"</span>,</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">formula =</span> modelSum,</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">modelColumnName =</span> <span class="st">"msqrob_rrilmm"</span>,</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We perform hypothesis tests for the early, late, average and interaction effects. Note that the contrasts remain unchanged.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>mouse <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>    mouse, <span class="at">i =</span> <span class="st">"proteins"</span>, L, <span class="at">modelColumn =</span> <span class="st">"msqrob_rrilmm"</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The inference tables were all stored in the <code>rowData</code> as separate columns, like previously.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>inferenceTablesSum <span class="ot">&lt;-</span> <span class="fu">rowData</span>(mouse[[<span class="st">"proteins"</span>]])[, <span class="fu">colnames</span>(L)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We here use again the <code>lapply()</code> loop that generates the list of volcano plots, one for each hypothesis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>volcanoPlotsSum <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(inferenceTablesSum), <span class="cf">function</span>(i) {</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>    inference <span class="ot">&lt;-</span> inferenceTablesSum[[i]]</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>    inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>        <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">filter</span>(inference, adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>),</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">"Hypothesis test:"</span>,</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">paste</span>(<span class="fu">gsub</span>(<span class="st">"tests_|ridge"</span>, <span class="st">""</span>, i), <span class="st">"= 0"</span>))</span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wrap_plots</span>(volcanoPlotsSum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="07-mouse_diet_files/figure-html/mouse_diet_volcano_all_proteins-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="conclusion" class="level2" data-number="7.7">
<h2 data-number="7.7" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">7.7</span> Conclusion</h2>
<p>In this chapter, we have demonstrated the application of msqrob2TMT workflows on a real-life case study.</p>
<p>The preprocessing workflow relies on the the <code>QFeatures</code> package. The package provides functionality to carry out many steps like data filtering, missing values management, normalisation, log-transformation, imputation, summarisation, etc. The functions also provide different methods for each step, meaning that the preprocessing pipeline can be easily adapted to the researcher’s needs based on their experiment and data set.</p>
<p>Once preprocessed, we used the <code>msqrob2</code> package to model all sources of variability as identified from the experimental design (and in part validated by data exploration): effect of diet and duration, effect of the MS acquisition run, effect of TMT mixture, effect of spectrum, and effect of sample. Modelling these different sources of variability allows to correctly infer changes in protein abundances between groups of interest while using ion-level data, although we also illustrate how to model the data at the protein level.</p>
<p>The experiment aims to understand the proteomic changes in mouse adipose tissue that occur upon feeding the mice with low-fat or high-fat diets, during a short or a long duration. We showed how to model the effect of these two factors using main effects and an interaction, which allows that the effect of diet can change according to the duration, and vice versa. We also showed how to translate biological questions into statistical hypothesis and corresponding contrast matrices using our msqrob2TMT workflow.</p>
<p>A unique feature of <code>msqrob2</code> is that its flexible approach can include more than 2 variables (with multiple interaction terms) as well as including numerical variables, which may be essential in other experimental contexts.</p>
<p>Modelling TMT-based data with biological replication leads to one of the most complex designs. These have to be further complexified with upcoming single-cell proteomics design where a new source of variability arises as cells belonging to the same experimental unit (subject or cell culture) are more similar than cells belonging to different experimental units. However, the model simplifies for other use cases. For instance, we saw how a protein-level model upon aggregation simplifies the model. However, spectrum effects can no longer be accounted for. This means that an appropriate summarisation approach is needed to correctly account for these spectrum effects when computing the protein-level summaries. Median polish (exemplified here) or robust summary do account for this, at least partially depending on the experimental design.</p>
<p>Label-free experiments, which do not perform chemical labelling of the samples, do not contain labelling effects which therefore are omitted. Moreover, every sample is acquired as part of a single run hence no run effect can be modelled. This simplified model is easier to understand but bear in mind that these sources of variation (eg. spectrum effects or run effect) end up in the residual variance, which might reduce statistical power.</p>
<p>Hence, we here demonstrated the power and flexibility of <code>msqrob2</code> and the msqrob2TMT workflows to help researchers answer biologically-relevant questions from their MS proteomics data.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-Huang2020-lu" class="csl-entry" role="listitem">
Huang, Ting, Meena Choi, Manuel Tzouros, Sabrina Golling, Nikhil Janak Pandya, Balazs Banfai, Tom Dunkley, and Olga Vitek. 2020. <span>“<span>MSstatsTMT</span>: Statistical Detection of Differentially Abundant Proteins in Experiments with Isobaric Labeling and Multiple Mixtures.”</span> <em>Mol. Cell. Proteomics</em> 19 (10): 1706–23.
</div>
<div id="ref-Plubell2017-th" class="csl-entry" role="listitem">
Plubell, Deanna L, Phillip A Wilmarth, Yuqi Zhao, Alexandra M Fenton, Jessica Minnier, Ashok P Reddy, John Klimek, Xia Yang, Larry L David, and Nathalie Pamir. 2017. <span>“Extended Multiplexing of Tandem Mass Tags (<span>TMT</span>) Labeling Reveals Age and High Fat Diet Specific Proteome Changes in Mouse Epididymal Adipose Tissue.”</span> <em>Mol. Cell. Proteomics</em> 16 (5): 873–90.
</div>
<div id="ref-Segers2025-ce" class="csl-entry" role="listitem">
Segers, Alexandre, Cristian Castiglione, Christophe Vanderaa, Elfride De Baere, Lennart Martens, Davide Risso, and Lieven Clement. 2025. <span>“<span class="nocase">omicsGMF</span>: A Multi-Tool for Dimensionality Reduction, Batch Correction and Imputation Applied to Bulk- and Single Cell Proteomics Data.”</span> <em>bioRxiv</em>, March, 2025.03.24.644996.
</div>
</div>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>note that these files are <a href="02-advanced.html#sec-caching">locally cached</a><a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./06-heart.html" class="pagination-link" aria-label="Heart use case: a MaxQuant LFQ DDA dataset with a more complex design">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./99-end.html" class="pagination-link" aria-label="Additional information">
        <span class="nav-page-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Additional information</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>