<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 5 The francisella use case: a MaxQuant LFQ DDA dataset | Statistical analysis for proteomics data" />
<meta property="og:type" content="book" />




<meta name="author" content="Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement" />

<meta name="date" content="2025-10-07" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="Chapter 5 The francisella use case: a MaxQuant LFQ DDA dataset | Statistical analysis for proteomics data">

<title>Chapter 5 The francisella use case: a MaxQuant LFQ DDA dataset | Statistical analysis for proteomics data</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }

code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Statistical analysis for proteomics data<p><p class="author">Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html" id="toc-preamble"><span class="toc-section-number">1</span> Preamble</a>
<a href="sec-basics.html" id="toc-sec-basics"><span class="toc-section-number">2</span> Statistical analysis with msqrob2</a>
<a href="sec-advanced.html" id="toc-sec-advanced"><span class="toc-section-number">3</span> Advanced statistical analysis with msqrob2</a>
<a href="sec-data_levels.html#sec-data_levels" id="toc-sec-data_levels"><span class="toc-section-number">4</span> PSM, peptide or protein-level modelling?</a>
<a id="active-page" href="sec-francisella.html" id="toc-sec-francisella"><span class="toc-section-number">5</span> The francisella use case: a MaxQuant LFQ DDA dataset</a><ul class="toc-sections">
<li class="toc"><a href="#introduction"> Introduction</a></li>
<li class="toc"><a href="#load-packages-3"> Load packages</a></li>
<li class="toc"><a href="#load-data"> Load data</a></li>
<li class="toc"><a href="#data-preprocessing-2"> Data preprocessing</a></li>
<li class="toc"><a href="#data-exploration-1"> Data exploration</a></li>
<li class="toc"><a href="#data-modelling-2"> Data modelling</a></li>
<li class="toc"><a href="#statistical-inference-1"> Statistical inference</a></li>
<li class="toc"><a href="#conclusion-2"> Conclusion</a></li>
</ul>
<a href="sec-mouse_diet.html#sec-mouse_diet" id="toc-sec-mouse_diet"><span class="toc-section-number">6</span> The mouse diet use case: a Skyline TMT DDA dataset</a>
<a href="sec-end.html" id="toc-sec-end"><span class="toc-section-number">7</span> Usefull links and session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec-francisella" class="section level1" number="5">
<h1>
<span class="header-section-number">Chapter 5</span> The francisella use case: a MaxQuant LFQ DDA dataset</h1>
<div id="introduction" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Introduction<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('introduction')" onmouseout="reset_tooltip('introduction-tooltip')"><span class="tooltiptext" id="introduction-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p><strong>TODO</strong> write section
<strong>TODO</strong> describe the general experimental design, and mention
that this design is also applicable when having multiple
samples from patients</p>
<p>In this vignette,…</p>
<p>If you never used <code>msqrob2</code>, we suggest to familiarise yourself
with the <a href="sec-basics.html#sec-basics">concepts chapter</a> first.</p>
</div>
<div id="load-packages-3" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Load packages<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('load-packages-3')" onmouseout="reset_tooltip('load-packages-3-tooltip')"><span class="tooltiptext" id="load-packages-3-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>First, we load the <code>msqrob2</code> package and additional packages for data
manipulation and visualisation.</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="sec-francisella.html#cb193-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb193-2"><a href="sec-francisella.html#cb193-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb193-3"><a href="sec-francisella.html#cb193-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb193-4"><a href="sec-francisella.html#cb193-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code></pre></div>
<p>We also configure the <a href="sec-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="sec-francisella.html#cb194-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb194-2"><a href="sec-francisella.html#cb194-2" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code></pre></div>
</div>
<div id="load-data" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> Load data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('load-data')" onmouseout="reset_tooltip('load-data-tooltip')"><span class="tooltiptext" id="load-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<div id="experimental-context-1" class="section level3" number="5.3.1">
<h3>
<span class="header-section-number">5.3.1</span> Experimental context<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('experimental-context-1')" onmouseout="reset_tooltip('experimental-context-1-tooltip')"><span class="tooltiptext" id="experimental-context-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>A study on the facultative pathogen Francisella tularensis was
conceived by <a href="https://www.mcponline.org/article/S1535-9476(20)33188-1/fulltext">Ramond et al.
(2015)</a>.
F. tularensis enters the cells of its host by phagocytosis. The
authors showed that F. tularensis is arginine deficient and imports
arginine from the host cell via an arginine transporter, ArgP, in
order to efficiently escape from the phagosome and reach the cytosolic
compartment, where it can actively multiply. In their study, they
compared the proteome of wild type F. tularensis (WT) to ArgP-gene
deleted F. tularensis (knock-out, D8). For this exercise, we use a
subset of the F. tularensis dataset where bacterial cultures were
grown in biological triplicate and each biorep was run in technical
triplicate on a nanoRSLC-Q Exactive PLUS instrument.</p>
</div>
<div id="getting-the-data" class="section level3" number="5.3.2">
<h3>
<span class="header-section-number">5.3.2</span> Getting the data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('getting-the-data')" onmouseout="reset_tooltip('getting-the-data-tooltip')"><span class="tooltiptext" id="getting-the-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The data were
searched with MaxQuant version 1.4.1.2. and are available on the PRIDE
repository:
<a href="https://www.ebi.ac.uk/pride/archive/projects/PXD001584">PXD001584</a>.</p>
<p><strong>TODO</strong> put on Zenodo and use BiocFileCache</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="sec-francisella.html#cb195-1" tabindex="-1"></a>pepFile <span class="ot">&lt;-</span> <span class="st">"./data/francisella/peptides.txt"</span></span></code></pre></div>
<p>After downloading the files, we can load the peptide table. Contrarily
a <a href="sec-advanced.html#sec-psm_table">PSM table</a>, the peptide table is provided in a
“wide format”, meaning that each row represents a single peptide and
that each quantification column (that starts with <code>"Intensity"</code>)
represents a single sample.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="sec-francisella.html#cb196-1" tabindex="-1"></a>peps <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(pepFile)</span>
<span id="cb196-2"><a href="sec-francisella.html#cb196-2" tabindex="-1"></a>quantcols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Intensity</span><span class="sc">\\</span><span class="st">."</span>, <span class="fu">names</span>(peps), <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table>
<colgroup>
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="4%">
<col width="10%">
<col width="0%">
<col width="0%">
</colgroup>
<thead><tr class="header">
<th align="left">Sequence</th>
<th align="left">Amino.acid.before</th>
<th align="left">First.amino.acid</th>
<th align="left">Second.amino.acid</th>
<th align="left">Second.last.amino.acid</th>
<th align="left">Last.amino.acid</th>
<th align="left">Amino.acid.after</th>
<th align="right">A.Count</th>
<th align="right">R.Count</th>
<th align="right">N.Count</th>
<th align="right">D.Count</th>
<th align="right">C.Count</th>
<th align="right">Q.Count</th>
<th align="right">E.Count</th>
<th align="right">G.Count</th>
<th align="right">H.Count</th>
<th align="right">I.Count</th>
<th align="right">L.Count</th>
<th align="right">K.Count</th>
<th align="right">M.Count</th>
<th align="right">F.Count</th>
<th align="right">P.Count</th>
<th align="right">S.Count</th>
<th align="right">T.Count</th>
<th align="right">W.Count</th>
<th align="right">Y.Count</th>
<th align="right">V.Count</th>
<th align="right">U.Count</th>
<th align="right">Length</th>
<th align="right">Missed.cleavages</th>
<th align="right">Mass</th>
<th align="left">Proteins</th>
<th align="left">Leading.razor.protein</th>
<th align="left">Unique..Groups.</th>
<th align="left">Unique..Proteins.</th>
<th align="left">Charges</th>
<th align="right">PEP</th>
<th align="left">Score</th>
<th align="right">Experiment.1WT_20_2h_n3_1</th>
<th align="right">Experiment.1WT_20_2h_n3_2</th>
<th align="right">Experiment.1WT_20_2h_n3_3</th>
<th align="right">Experiment.1WT_20_2h_n4_1</th>
<th align="right">Experiment.1WT_20_2h_n4_2</th>
<th align="right">Experiment.1WT_20_2h_n4_3</th>
<th align="right">Experiment.1WT_20_2h_n5_1</th>
<th align="right">Experiment.1WT_20_2h_n5_2</th>
<th align="right">Experiment.1WT_20_2h_n5_3</th>
<th align="right">Experiment.2WT_100_2h_n3_1</th>
<th align="right">Experiment.2WT_100_2h_n3_2</th>
<th align="right">Experiment.2WT_100_2h_n3_3</th>
<th align="right">Experiment.2WT_100_2h_n4_1</th>
<th align="right">Experiment.2WT_100_2h_n4_2</th>
<th align="right">Experiment.2WT_100_2h_n4_3</th>
<th align="right">Experiment.2WT_100_2h_n5_1</th>
<th align="right">Experiment.2WT_100_2h_n5_2</th>
<th align="right">Experiment.2WT_100_2h_n5_3</th>
<th align="right">Experiment.3D8_20_2h_n3_1</th>
<th align="right">Experiment.3D8_20_2h_n3_2</th>
<th align="right">Experiment.3D8_20_2h_n3_3</th>
<th align="right">Experiment.3D8_20_2h_n4_1</th>
<th align="right">Experiment.3D8_20_2h_n4_2</th>
<th align="right">Experiment.3D8_20_2h_n4_3</th>
<th align="right">Experiment.3D8_20_2h_n5_1</th>
<th align="right">Experiment.3D8_20_2h_n5_2</th>
<th align="right">Experiment.3D8_20_2h_n5_3</th>
<th align="right">Experiment.3WT_20_2h_1</th>
<th align="right">Experiment.3WT_20_2h_2</th>
<th align="right">Experiment.3WT_20_2h_3</th>
<th align="right">Experiment.4D8_100_2h_n3_1</th>
<th align="right">Experiment.4D8_100_2h_n3_2</th>
<th align="right">Experiment.4D8_100_2h_n3_3</th>
<th align="right">Experiment.4D8_100_2h_n4_1</th>
<th align="right">Experiment.4D8_100_2h_n4_2</th>
<th align="right">Experiment.4D8_100_2h_n4_3</th>
<th align="right">Experiment.4D8_100_2h_n5_1</th>
<th align="right">Experiment.4D8_100_2h_n5_2</th>
<th align="right">Experiment.4D8_100_2h_n5_3</th>
<th align="right">Experiment.4WT_100_2h_1</th>
<th align="right">Experiment.4WT_100_2h_2</th>
<th align="right">Experiment.4WT_100_2h_3</th>
<th align="right">Experiment.7D8_20_2h_1</th>
<th align="right">Experiment.7D8_20_2h_2</th>
<th align="right">Experiment.7D8_20_2h_3</th>
<th align="right">Experiment.8D8_100_2h_1</th>
<th align="right">Experiment.8D8_100_2h_2</th>
<th align="right">Experiment.8D8_100_2h_3</th>
<th align="right">Intensity</th>
<th align="right">Intensity.1WT_20_2h_n3_1</th>
<th align="right">Intensity.1WT_20_2h_n3_2</th>
<th align="right">Intensity.1WT_20_2h_n3_3</th>
<th align="right">Intensity.1WT_20_2h_n4_1</th>
<th align="right">Intensity.1WT_20_2h_n4_2</th>
<th align="right">Intensity.1WT_20_2h_n4_3</th>
<th align="right">Intensity.1WT_20_2h_n5_1</th>
<th align="right">Intensity.1WT_20_2h_n5_2</th>
<th align="right">Intensity.1WT_20_2h_n5_3</th>
<th align="right">Intensity.2WT_100_2h_n3_1</th>
<th align="right">Intensity.2WT_100_2h_n3_2</th>
<th align="right">Intensity.2WT_100_2h_n3_3</th>
<th align="right">Intensity.2WT_100_2h_n4_1</th>
<th align="right">Intensity.2WT_100_2h_n4_2</th>
<th align="right">Intensity.2WT_100_2h_n4_3</th>
<th align="right">Intensity.2WT_100_2h_n5_1</th>
<th align="right">Intensity.2WT_100_2h_n5_2</th>
<th align="right">Intensity.2WT_100_2h_n5_3</th>
<th align="right">Intensity.3D8_20_2h_n3_1</th>
<th align="right">Intensity.3D8_20_2h_n3_2</th>
<th align="right">Intensity.3D8_20_2h_n3_3</th>
<th align="right">Intensity.3D8_20_2h_n4_1</th>
<th align="right">Intensity.3D8_20_2h_n4_2</th>
<th align="right">Intensity.3D8_20_2h_n4_3</th>
<th align="right">Intensity.3D8_20_2h_n5_1</th>
<th align="right">Intensity.3D8_20_2h_n5_2</th>
<th align="right">Intensity.3D8_20_2h_n5_3</th>
<th align="right">Intensity.3WT_20_2h_1</th>
<th align="right">Intensity.3WT_20_2h_2</th>
<th align="right">Intensity.3WT_20_2h_3</th>
<th align="right">Intensity.4D8_100_2h_n3_1</th>
<th align="right">Intensity.4D8_100_2h_n3_2</th>
<th align="right">Intensity.4D8_100_2h_n3_3</th>
<th align="right">Intensity.4D8_100_2h_n4_1</th>
<th align="right">Intensity.4D8_100_2h_n4_2</th>
<th align="right">Intensity.4D8_100_2h_n4_3</th>
<th align="right">Intensity.4D8_100_2h_n5_1</th>
<th align="right">Intensity.4D8_100_2h_n5_2</th>
<th align="right">Intensity.4D8_100_2h_n5_3</th>
<th align="right">Intensity.4WT_100_2h_1</th>
<th align="right">Intensity.4WT_100_2h_2</th>
<th align="right">Intensity.4WT_100_2h_3</th>
<th align="right">Intensity.7D8_20_2h_1</th>
<th align="right">Intensity.7D8_20_2h_2</th>
<th align="right">Intensity.7D8_20_2h_3</th>
<th align="right">Intensity.8D8_100_2h_1</th>
<th align="right">Intensity.8D8_100_2h_2</th>
<th align="right">Intensity.8D8_100_2h_3</th>
<th align="left">Reverse</th>
<th align="left">Contaminant</th>
<th align="right">id</th>
<th align="left">Protein.group.IDs</th>
<th align="left">Mod..peptide.IDs</th>
<th align="left">Evidence.IDs</th>
<th align="left">MS.MS.IDs</th>
<th align="right">Best.MS.MS</th>
<th align="left">Oxidation..M..site.IDs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AAAEELDTR</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">T</td>
<td align="left">R</td>
<td align="left">K</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">974.4669</td>
<td align="left">gi|118497196</td>
<td align="left">gi|118497196</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">1,2</td>
<td align="right">0.0001253</td>
<td align="left">59.35</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1.3295e+09</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9290400</td>
<td align="right">92996000</td>
<td align="right">0</td>
<td align="right">98059000</td>
<td align="right">0</td>
<td align="right">80803000</td>
<td align="right">0</td>
<td align="right">77073000</td>
<td align="right">96987000</td>
<td align="right">88792000</td>
<td align="right">92370000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">64715000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">95433000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">7130200</td>
<td align="right">0</td>
<td align="right">86907000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">92052000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">80270000</td>
<td align="right">70885000</td>
<td align="right">59782000</td>
<td align="right">44146000</td>
<td align="right">46969000</td>
<td align="right">44875000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
<td align="left">419</td>
<td align="left">0</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td align="right">15</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAAGFVITASHNK</td>
<td align="left">R</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">N</td>
<td align="left">K</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="right">0</td>
<td align="right">1285.6779</td>
<td align="left">gi|118498194</td>
<td align="left">gi|118498194</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000530</td>
<td align="left">45.825</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.6719e+08</td>
<td align="right">0</td>
<td align="right">20679000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">17358000</td>
<td align="right">13841000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">14142000</td>
<td align="right">10098000</td>
<td align="right">0</td>
<td align="right">12278000</td>
<td align="right">10712000</td>
<td align="right">0</td>
<td align="right">9500700</td>
<td align="right">11111000</td>
<td align="right">8723500</td>
<td align="right">8125200</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">14059000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">8245100</td>
<td align="right">8318400</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
<td align="left">1150</td>
<td align="left">1</td>
<td align="left">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td align="left">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td align="right">26</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">AAANEYELALAYSIEEVAPDLHK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">H</td>
<td align="left">K</td>
<td align="left">Y</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">0</td>
<td align="right">2516.2435</td>
<td align="left">gi|118497331</td>
<td align="left">gi|118497331</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">3</td>
<td align="right">0.0000000</td>
<td align="left">84.327</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">5.5675e+08</td>
<td align="right">28853000</td>
<td align="right">28091000</td>
<td align="right">30436000</td>
<td align="right">19476000</td>
<td align="right">22050000</td>
<td align="right">20687000</td>
<td align="right">36588000</td>
<td align="right">28654000</td>
<td align="right">7009100</td>
<td align="right">13881000</td>
<td align="right">15549000</td>
<td align="right">30239000</td>
<td align="right">13921000</td>
<td align="right">27386000</td>
<td align="right">32287000</td>
<td align="right">12201000</td>
<td align="right">0</td>
<td align="right">14278000</td>
<td align="right">14174000</td>
<td align="right">8548000</td>
<td align="right">7859200</td>
<td align="right">14871000</td>
<td align="right">11690000</td>
<td align="right">4470500</td>
<td align="right">0</td>
<td align="right">4048100</td>
<td align="right">3422300</td>
<td align="right">15967000</td>
<td align="right">0</td>
<td align="right">12772000</td>
<td align="right">8623400</td>
<td align="right">8716100</td>
<td align="right">17835000</td>
<td align="right">12707000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">11324000</td>
<td align="right">0</td>
<td align="right">6540800</td>
<td align="right">4177900</td>
<td align="right">7417900</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">2</td>
<td align="left">513</td>
<td align="left">2</td>
<td align="left">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76</td>
<td align="left">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79</td>
<td align="right">39</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAANNPQLEAFK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">F</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1272.6463</td>
<td align="left">gi|118496879</td>
<td align="left">gi|118496879</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="left">108.56</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1.4830e+10</td>
<td align="right">339830000</td>
<td align="right">420390000</td>
<td align="right">393930000</td>
<td align="right">235060000</td>
<td align="right">381090000</td>
<td align="right">360270000</td>
<td align="right">255710000</td>
<td align="right">311370000</td>
<td align="right">271440000</td>
<td align="right">331550000</td>
<td align="right">349710000</td>
<td align="right">357230000</td>
<td align="right">327850000</td>
<td align="right">361880000</td>
<td align="right">379820000</td>
<td align="right">237000000</td>
<td align="right">204480000</td>
<td align="right">211470000</td>
<td align="right">225140000</td>
<td align="right">289880000</td>
<td align="right">282500000</td>
<td align="right">209490000</td>
<td align="right">223610000</td>
<td align="right">219700000</td>
<td align="right">169120000</td>
<td align="right">178150000</td>
<td align="right">170400000</td>
<td align="right">389360000</td>
<td align="right">428050000</td>
<td align="right">405040000</td>
<td align="right">232100000</td>
<td align="right">358170000</td>
<td align="right">376940000</td>
<td align="right">319950000</td>
<td align="right">291330000</td>
<td align="right">362520000</td>
<td align="right">183570000</td>
<td align="right">198970000</td>
<td align="right">190690000</td>
<td align="right">437080000</td>
<td align="right">411460000</td>
<td align="right">392540000</td>
<td align="right">481300000</td>
<td align="right">487260000</td>
<td align="right">453840000</td>
<td align="right">231800000</td>
<td align="right">265460000</td>
<td align="right">234340000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">3</td>
<td align="left">199</td>
<td align="left">3</td>
<td align="left">77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124</td>
<td align="left">80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172</td>
<td align="right">159</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">AAASAGLVDEK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">E</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="right">0</td>
<td align="right">1030.5295</td>
<td align="left">gi|118497152</td>
<td align="left">gi|118497152</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000951</td>
<td align="left">67.385</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">4.8209e+09</td>
<td align="right">240280000</td>
<td align="right">0</td>
<td align="right">241890000</td>
<td align="right">222250000</td>
<td align="right">189930000</td>
<td align="right">0</td>
<td align="right">160440000</td>
<td align="right">147230000</td>
<td align="right">143340000</td>
<td align="right">221650000</td>
<td align="right">0</td>
<td align="right">220950000</td>
<td align="right">221400000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">107540000</td>
<td align="right">0</td>
<td align="right">135910000</td>
<td align="right">185740000</td>
<td align="right">168620000</td>
<td align="right">160240000</td>
<td align="right">158000000</td>
<td align="right">0</td>
<td align="right">92542000</td>
<td align="right">102350000</td>
<td align="right">100090000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">172000000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">160480000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">97827000</td>
<td align="right">107600000</td>
<td align="right">105750000</td>
<td align="right">0</td>
<td align="right">129440000</td>
<td align="right">0</td>
<td align="right">163580000</td>
<td align="right">160210000</td>
<td align="right">140770000</td>
<td align="right">117920000</td>
<td align="right">126570000</td>
<td align="right">118410000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">4</td>
<td align="left">388</td>
<td align="left">4</td>
<td align="left">125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155</td>
<td align="left">173;174;175;176;177;178;179;180;181;182;183;184;185;186;187;188;189;190;191;192;193;194;195;196;197;198;199;200;201;202;203</td>
<td align="right">191</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAASLDLYSYPK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">P</td>
<td align="left">K</td>
<td align="left">V</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1297.6554</td>
<td align="left">gi|118497492</td>
<td align="left">gi|118497492</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0125630</td>
<td align="left">31.675</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.3715e+08</td>
<td align="right">0</td>
<td align="right">31069000</td>
<td align="right">31448000</td>
<td align="right">0</td>
<td align="right">27721000</td>
<td align="right">28657000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">18253000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">5</td>
<td align="left">624</td>
<td align="left">5</td>
<td align="left">156;157;158;159;160</td>
<td align="left">204;205;206;207;208</td>
<td align="right">205</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>We now extract the <a href="sec-basics.html#sec-annotation_table">sample annotations</a>. We
will build a table where each row in the annotation table contains
information for one sample (the table below shows the first 6 rows).
This information is extracted from the sample names.</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="sec-francisella.html#cb197-1" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">quantCols =</span> quantcols) <span class="sc">|&gt;</span> </span>
<span id="cb197-2"><a href="sec-francisella.html#cb197-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"_20_"</span>, quantCols) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"_n</span><span class="sc">\\</span><span class="st">d"</span>, quantCols)) <span class="sc">|&gt;</span> </span>
<span id="cb197-3"><a href="sec-francisella.html#cb197-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype  =</span> <span class="fu">substr</span>(quantCols, <span class="dv">12</span>, <span class="dv">13</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb197-4"><a href="sec-francisella.html#cb197-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">biorep  =</span> <span class="fu">paste0</span>(genotype, <span class="st">"_"</span>, <span class="fu">substr</span>(quantCols, <span class="dv">21</span>, <span class="dv">22</span>)))</span></code></pre></div>
<table>
<thead><tr class="header">
<th align="left">quantCols</th>
<th align="left">genotype</th>
<th align="left">biorep</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n3_1</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n3_2</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
</tr>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n3_3</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n4_1</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
</tr>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n4_2</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n4_3</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
</tr>
</tbody>
</table>
</div>
<div id="the-qfeatures-data-class" class="section level3" number="5.3.3">
<h3>
<span class="header-section-number">5.3.3</span> The <code>QFeatures</code> data class<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('the-qfeatures-data-class')" onmouseout="reset_tooltip('the-qfeatures-data-class-tooltip')"><span class="tooltiptext" id="the-qfeatures-data-class-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We combine the two tables into a <a href="sec-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="sec-francisella.html#cb198-1" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb198-2"><a href="sec-francisella.html#cb198-2" tabindex="-1"></a>  peps, <span class="at">colData =</span> coldata, <span class="at">fnames =</span> <span class="st">"Sequence"</span>, <span class="at">name =</span> <span class="st">"peptides"</span></span>
<span id="cb198-3"><a href="sec-francisella.html#cb198-3" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 1 set:
## 
##  [1] peptides: SummarizedExperiment with 10693 rows and 18 columns</code></pre>
<p>We now have a <code>QFeatures</code> object with 1 set, containing <code>r nrows(pe)[[1]]</code> rows (peptides) and 18 columns
(samples).</p>
</div>
</div>
<div id="data-preprocessing-2" class="section level2" number="5.4">
<h2>
<span class="header-section-number">5.4</span> Data preprocessing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-preprocessing-2')" onmouseout="reset_tooltip('data-preprocessing-2-tooltip')"><span class="tooltiptext" id="data-preprocessing-2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p><code>msqrob2</code> relies on the <code>QFeatures</code> data structure, meaning that we
can directly make use of <code>QFeatures</code>’ data preprocessing functionality
(see also the <code>QFeatures</code>
<a href="https://rformassspectrometry.github.io/QFeatures/articles/Processing.html">documentation</a>).</p>
<div id="encoding-missing-values-1" class="section level3" number="5.4.1">
<h3>
<span class="header-section-number">5.4.1</span> Encoding missing values<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('encoding-missing-values-1')" onmouseout="reset_tooltip('encoding-missing-values-1-tooltip')"><span class="tooltiptext" id="encoding-missing-values-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Peptides with zero intensities are missing peptides and should be
represent with a <code>NA</code> value rather than <code>0</code> (see <a href="sec-basics.html#sec-encode_missing">Encoding missing
values</a>).</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="sec-francisella.html#cb200-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">"peptides"</span>)</span></code></pre></div>
<p>We calculate how many non zero intensities we have per peptide and
this is often useful for filtering.</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="sec-francisella.html#cb201-1" tabindex="-1"></a>naResults <span class="ot">&lt;-</span> <span class="fu">nNA</span>(pe, <span class="st">"peptides"</span>)</span>
<span id="cb201-2"><a href="sec-francisella.html#cb201-2" tabindex="-1"></a><span class="fu">data.frame</span>(naResults<span class="sc">$</span>nNArows) <span class="sc">|&gt;</span> </span>
<span id="cb201-3"><a href="sec-francisella.html#cb201-3" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb201-4"><a href="sec-francisella.html#cb201-4" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> nNA) <span class="sc">+</span></span>
<span id="cb201-5"><a href="sec-francisella.html#cb201-5" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-223-1.png" width="672"></p>
<p>Because every biorep is assessed in technical triplicate, we will also
calculate the number of biorepeats in which each peptide is observed.</p>
<p><strong>TODO</strong> make a function for this?</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="sec-francisella.html#cb202-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">"peptides"</span>]])<span class="sc">$</span>nNonZeroBiorep <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb202-2"><a href="sec-francisella.html#cb202-2" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">"peptides"</span>]]), <span class="dv">1</span>, <span class="cf">function</span>(intensity)</span>
<span id="cb202-3"><a href="sec-francisella.html#cb202-3" tabindex="-1"></a>    pe<span class="sc">$</span>biorep[intensity <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="sc">|&gt;</span> </span>
<span id="cb202-4"><a href="sec-francisella.html#cb202-4" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">|&gt;</span> </span>
<span id="cb202-5"><a href="sec-francisella.html#cb202-5" tabindex="-1"></a>    <span class="fu">length</span>()</span>
<span id="cb202-6"><a href="sec-francisella.html#cb202-6" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="psm-filtering" class="section level3" number="5.4.2">
<h3>
<span class="header-section-number">5.4.2</span> PSM filtering<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('psm-filtering')" onmouseout="reset_tooltip('psm-filtering-tooltip')"><span class="tooltiptext" id="psm-filtering-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We filter features based on 3 criteria (see <a href="sec-basics.html#sec-filter">PSM filtering</a>).</p>
<ol style="list-style-type: decimal">
<li>Handling overlapping protein groups</li>
</ol>
<p>In our approach a peptide can map to multiple proteins, as long as
there is none of these proteins present in a smaller subgroup.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="sec-francisella.html#cb203-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,</span>
<span id="cb203-2"><a href="sec-francisella.html#cb203-2" tabindex="-1"></a>  <span class="sc">~</span> Proteins <span class="sc">%in%</span> <span class="fu">smallestUniqueGroups</span>(Proteins)</span>
<span id="cb203-3"><a href="sec-francisella.html#cb203-3" tabindex="-1"></a>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences (decoys) and contaminants</li>
</ol>
<p>We now remove the contaminants and peptides that map to decoy
sequences. These features bear no information of interest and will
reduce the statistical power upon multiple test adjustment.</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="sec-francisella.html#cb204-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> Contaminant <span class="sc">!=</span> <span class="st">"+"</span>)</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in a single biorepeat</li>
</ol>
<p>Note, that in experiments without technical repeats we filter on the
number of samples in which a peptide is picked up (this is typically
performed using <code>filterNA()</code>). Here, we will require that a peptide is
picked up in at least two biorepeats. We already computed the number
of biorepeats that were observed for each peptide (that is the number
of biorepeats that contain at least one observed value).</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="sec-francisella.html#cb205-1" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> nNonZeroBiorep <span class="sc">&gt;=</span> <span class="dv">2</span>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 1 set:
## 
##  [1] peptides: SummarizedExperiment with 8640 rows and 18 columns</code></pre>
<p>We keep 8640 peptides upon filtering.</p>
</div>
<div id="standard-preprocessing-workflow" class="section level3" number="5.4.3">
<h3>
<span class="header-section-number">5.4.3</span> Standard preprocessing workflow<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('standard-preprocessing-workflow')" onmouseout="reset_tooltip('standard-preprocessing-workflow-tooltip')"><span class="tooltiptext" id="standard-preprocessing-workflow-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can now prepare the data for modelling. The workflow ensures the
data complies to <code>msqrob2</code>’s requirements:</p>
<ol style="list-style-type: decimal">
<li>Intensities are <a href="sec-basics.html#sec-log2">log-transformed</a>.</li>
</ol>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="sec-francisella.html#cb207-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">"peptides"</span>, <span class="at">name =</span> <span class="st">"peptides_log"</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Samples are normalised by substracting the sample median (see
<a href="sec-basics.html#sec-norm">Normalisation</a>)</li>
</ol>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="sec-francisella.html#cb208-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">normalize</span>(</span>
<span id="cb208-2"><a href="sec-francisella.html#cb208-2" tabindex="-1"></a>  pe, <span class="at">i =</span> <span class="st">"peptides_log"</span>, <span class="at">name =</span> <span class="st">"peptides_norm"</span>, <span class="at">method =</span> <span class="st">"center.median"</span></span>
<span id="cb208-3"><a href="sec-francisella.html#cb208-3" tabindex="-1"></a>)</span></code></pre></div>
<p>Upon the normalisation the density curves should be nicely centred. To
confirm this, we will plot the intensity distributions for each
biorepeat (mouse). <code>longForm()</code> seamlessly combines the quantification
and annotation data into a table suitable for <code>ggplot2</code> visualisation.
We also subset the object with the data before and after normalisation.</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="sec-francisella.html#cb209-1" tabindex="-1"></a><span class="fu">longForm</span>(pe[, , <span class="fu">c</span>(<span class="st">"peptides_log"</span>, <span class="st">"peptides_norm"</span>)], <span class="at">colvar =</span> <span class="st">"biorep"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb209-2"><a href="sec-francisella.html#cb209-2" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb209-3"><a href="sec-francisella.html#cb209-3" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">group =</span> colname, <span class="at">color =</span> biorep) <span class="sc">+</span></span>
<span id="cb209-4"><a href="sec-francisella.html#cb209-4" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb209-5"><a href="sec-francisella.html#cb209-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> assay, <span class="at">scale =</span> <span class="st">"free"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-230-1.png" width="672"></p>
<ol start="3" style="list-style-type: decimal">
<li>
<a href="sec-basics.html#sec-summarisation">Summarisation</a> to protein level. We use the [robust summary] approach
to infer protein-level data from peptide-level data, accounting for
the fact that different peptides have ionisation efficiencies hence
leading to different intensity baselines.</li>
</ol>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="sec-francisella.html#cb210-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb210-2"><a href="sec-francisella.html#cb210-2" tabindex="-1"></a>  pe, <span class="at">i =</span> <span class="st">"peptides_norm"</span>, <span class="at">fcol =</span> <span class="st">"Proteins"</span>, </span>
<span id="cb210-3"><a href="sec-francisella.html#cb210-3" tabindex="-1"></a>  <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">name =</span> <span class="st">"proteins"</span></span>
<span id="cb210-4"><a href="sec-francisella.html#cb210-4" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="data-exploration-1" class="section level2" number="5.5">
<h2>
<span class="header-section-number">5.5</span> Data exploration<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-exploration-1')" onmouseout="reset_tooltip('data-exploration-1-tooltip')"><span class="tooltiptext" id="data-exploration-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>We will explore the main sources of variation in the
data using (see <a href="sec-basics.html#sec_data_exploration">Data exploration</a>).</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="sec-francisella.html#cb211-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb211-2"><a href="sec-francisella.html#cb211-2" tabindex="-1"></a><span class="fu">getWithColData</span>(pe, <span class="st">"proteins"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb211-3"><a href="sec-francisella.html#cb211-3" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">"SingleCellExperiment"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb211-4"><a href="sec-francisella.html#cb211-4" tabindex="-1"></a>  <span class="fu">runMDS</span>(<span class="at">exprs_values =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb211-5"><a href="sec-francisella.html#cb211-5" tabindex="-1"></a>  <span class="fu">plotMDS</span>(<span class="at">colour_by =</span> <span class="st">"genotype"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-232-1.png" width="672"></p>
<p>Note that the samples upon robust summarisation show a clear
separation according to the genotype in the first dimension of the MDS
plot.</p>
</div>
<div id="data-modelling-2" class="section level2" number="5.6">
<h2>
<span class="header-section-number">5.6</span> Data modelling<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-modelling-2')" onmouseout="reset_tooltip('data-modelling-2-tooltip')"><span class="tooltiptext" id="data-modelling-2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The preprocessed data can now be modelled to answer biologically
relevant questions. As described above, samples (bacterial cultures)
originate from either a wildtype (WT) or a ArgP knockout (D8). Each
genotype was cultured in biological triplicate. Each biological
triplicate was acquired in technical triplicate, leading to <span class="math inline">\(2 \times
3 \times 3 = 18\)</span> samples. In this context, we are interested in the
effects of genotype on the protein abundances.</p>
<p>The table below confirms we have a balanced design
for each condition and biological triplicate.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="sec-francisella.html#cb212-1" tabindex="-1"></a><span class="fu">table</span>(<span class="at">genotype =</span> pe<span class="sc">$</span>genotype, <span class="at">biorep =</span> pe<span class="sc">$</span>biorep)</span></code></pre></div>
<pre><code>##         biorep
## genotype D8_n3 D8_n4 D8_n5 WT_n3 WT_n4 WT_n5
##       D8     3     3     3     0     0     0
##       WT     0     0     0     3     3     3</code></pre>
<div id="sources-of-variation-1" class="section level3" number="5.6.1">
<h3>
<span class="header-section-number">5.6.1</span> Sources of variation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sources-of-variation-1')" onmouseout="reset_tooltip('sources-of-variation-1-tooltip')"><span class="tooltiptext" id="sources-of-variation-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We will model two <a href="sec-basics.html#sec-modelling">sources of variation</a>:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Genotype</strong>: we model the source of variation induced by the
experimental group of interest as a <strong>fixed effect</strong>. Fixed effects
are effect that are considered non-random, i.e. the treatment
effect is assumed to be the same and reproducible across repeated
experiments, but it is unknown and has to be estimated. We will
include <code>genotype</code> as a fixed effect that models the fact that a
change in genotype can induce changes in protein abundance.</p></li>
<li><p><strong>Biological replicate effect</strong>: the experiment involves biological
replication as the bacterial cultures are repeated. Replicate-specific
effects occurs due to uncontrollable factors, such as variation in
the number of bacterium seeded, position in the incubator,
transient contamination,… Two bacterial cultures will never
provide exactly the same sample material. These effects are typically
modelled as random effects which are considered as a random sample
from the population of all possible mice and are assumed to be
i.i.d normally distributed with mean 0 and constant variance,
<span class="math inline">\(u_{biorep} \sim
N(0,\sigma^{2,\text{biorep}})\)</span>. The use of random effects thus
models the correlation in the data, explicitly. We expect that
intensities from the same bacterial culture are more alike than
intensities between cultures.</p></li>
</ol>
<p>We model the protein level expression values using <code>msqrob2</code>.
<code>msqrob2</code> workflows rely on linear mixed models, which are models that
can estimate and predict fixed and random effects, respectively. The
fixed effect are estimated using robust regression to avoid that
outliers distort the statistical outcome.</p>
<p>Now we have identified the sources of variation in the experiment, we
can define a model.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="sec-francisella.html#cb214-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="er">~</span> genotype <span class="sc">+</span> <span class="do">## (1) fixed effect for genotype</span></span>
<span id="cb214-2"><a href="sec-francisella.html#cb214-2" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> biorep)  <span class="do">## (2) random effect for biological replicate (culture)</span></span></code></pre></div>
</div>
<div id="estimate-the-model" class="section level3" number="5.6.2">
<h3>
<span class="header-section-number">5.6.2</span> Estimate the model<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('estimate-the-model')" onmouseout="reset_tooltip('estimate-the-model-tooltip')"><span class="tooltiptext" id="estimate-the-model-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We estimate the model with <code>msqrob()</code> (see <a href="sec-advanced.html#sec-run_model">the modelling
section</a>). Recall that
variables defined in <code>model</code> are automatically retrieved from the
<code>colData</code> (i.e. <code>"genotype"</code>, and <code>"biorep"</code>). Note, that <code>msqrob2</code>
also features ridge regression for stabilising the parameter
estimation, but it is irrelevant in this context as the genotype
factor only has 2 levels (WT and D8), so ridge regression . We will
therefore leave the ridge regression disabled (default).</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="sec-francisella.html#cb215-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">formula =</span> model, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="statistical-inference-1" class="section level2" number="5.7">
<h2>
<span class="header-section-number">5.7</span> Statistical inference<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('statistical-inference-1')" onmouseout="reset_tooltip('statistical-inference-1-tooltip')"><span class="tooltiptext" id="statistical-inference-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Once the models are estimated, we can start answering biological
questions by performing <a href="sec-basics.html#sec-inference">Statistical inference</a>. We must convert the
biological question “does the bacterial genotype affect the protein
intensities?” into a statistical hypothesis. In other words, we must
convert this question in a combination of the model parameters, also
referred to as a contrast. To aid defining contrasts, we will
visualise the experimental design using the <code>ExploreModelMatrix</code>
package. Since we are not interested in technical effects, we will
only focus on the variable of interest, here <code>genotype</code>.</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="sec-francisella.html#cb216-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb216-2"><a href="sec-francisella.html#cb216-2" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(</span>
<span id="cb216-3"><a href="sec-francisella.html#cb216-3" tabindex="-1"></a>    <span class="at">sampleData =</span>  <span class="fu">colData</span>(pe),</span>
<span id="cb216-4"><a href="sec-francisella.html#cb216-4" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> genotype,</span>
<span id="cb216-5"><a href="sec-francisella.html#cb216-5" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb216-6"><a href="sec-francisella.html#cb216-6" tabindex="-1"></a>)</span>
<span id="cb216-7"><a href="sec-francisella.html#cb216-7" tabindex="-1"></a>vd<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-235-1.png" width="672"></p>
<p>This plot shows that the average protein intensity for the D8 group is
estimated by <code>(Intercept)</code>, and the the average protein intensity for
the WT group is estimated by <code>(Intercept) + genotypeWT</code>.</p>
<div id="hypothesis-testing-3" class="section level3" number="5.7.1">
<h3>
<span class="header-section-number">5.7.1</span> Hypothesis testing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('hypothesis-testing-3')" onmouseout="reset_tooltip('hypothesis-testing-3-tooltip')"><span class="tooltiptext" id="hypothesis-testing-3-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Assessing the difference between the WT and D8 groups boils down to
<code>(Intercept) + genotypeWT - (Intercept)</code>, which simplifies to the
<code>genotypeWT</code> parameter. The null hypothesis is that the differences
between the two groups is zero, hence the estimated parameter is zero.</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="sec-francisella.html#cb218-1" tabindex="-1"></a>hypothesis <span class="ot">&lt;-</span> <span class="st">"genotypeWT = 0"</span></span></code></pre></div>
<p>We next use <code>makeContrast()</code> to build the contrast matrix.</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="sec-francisella.html#cb219-1" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(hypothesis, <span class="at">parameterNames =</span> <span class="st">"genotypeWT"</span>))</span></code></pre></div>
<pre><code>##            genotypeWT
## genotypeWT          1</code></pre>
<p>In this case, the contrast matrix is trivial, but it becomes a matrix
but for <a href="sec-basics.html#sec-multiple_contrasts">more complex designs</a>. We can now
test our null hypothesis:</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="sec-francisella.html#cb221-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
<p>Let us retrieve the result table from the <code>rowData</code>. Note that the
hypothesis testing results are stored in <code>rowData</code> columns named after
the column names (here <code>genotypeWT</code>) of the contrast matrix <code>L</code>.</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="sec-francisella.html#cb222-1" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">"proteins"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb222-2"><a href="sec-francisella.html#cb222-2" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb222-3"><a href="sec-francisella.html#cb222-3" tabindex="-1"></a><span class="fu">head</span>(inference)</span></code></pre></div>
<pre><code>##                   logFC         se       df         t         pval      adjPval      Protein
## gi|118496616 0.11784231 0.05986775 16.37914 1.9683773 6.618849e-02 0.2289531725 gi|118496616
## gi|118496617 0.01583769 0.07893303 17.99716 0.2006472 8.432245e-01 0.9374721247 gi|118496617
## gi|118496621 0.66085954 0.26954611 12.99716 2.4517495 2.912069e-02 0.1283594396 gi|118496621
## gi|118496635 0.24384779 0.03838535 17.21457 6.3526263 6.804599e-06 0.0002832729 gi|118496635
## gi|118496636 0.17315742 0.06042136 16.03249 2.8658310 1.119025e-02 0.0706619984 gi|118496636
## gi|118496637 0.35403038 0.09239590 15.22182 3.8316675 1.595094e-03 0.0187831770 gi|118496637</code></pre>
<p>Notice that some rows contain missing values. This is because data
modelling resulted in a <code>fitError</code> for some proteins, probably because
not enough data was available for model fitting due to missing values
in the quantitative data (see <a href="sec-advanced.html#sec-fiterror">how to deal with
<code>fitError</code>s</a>).</p>
</div>
<div id="volcano-plot" class="section level3" number="5.7.2">
<h3>
<span class="header-section-number">5.7.2</span> Volcano-plot<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('volcano-plot')" onmouseout="reset_tooltip('volcano-plot-tooltip')"><span class="tooltiptext" id="volcano-plot-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Volcano plots are straightforward to generate from the inference table
above. We also use <code>ggrepel</code> to annotate the 20 most significant
proteins.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="sec-francisella.html#cb224-1" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb224-2"><a href="sec-francisella.html#cb224-2" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb224-3"><a href="sec-francisella.html#cb224-3" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb224-4"><a href="sec-francisella.html#cb224-4" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">slice_min</span>(inference, adjPval, <span class="at">n =</span> <span class="dv">20</span>),</span>
<span id="cb224-5"><a href="sec-francisella.html#cb224-5" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb224-6"><a href="sec-francisella.html#cb224-6" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb224-7"><a href="sec-francisella.html#cb224-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on differences between WT and D8"</span>,</span>
<span id="cb224-8"><a href="sec-francisella.html#cb224-8" tabindex="-1"></a>          <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">colnames</span>(L), <span class="st">"= 0"</span>))</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-240-1.png" width="672"></p>
<p>Note, that 149 proteins
are found to be differentially abundant.</p>
</div>
<div id="heatmap" class="section level3" number="5.7.3">
<h3>
<span class="header-section-number">5.7.3</span> Heatmap<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('heatmap')" onmouseout="reset_tooltip('heatmap-tooltip')"><span class="tooltiptext" id="heatmap-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can also build a heatmap for the significant proteins which are
obtained by filtering the inference table.</p>
<p><strong>TODO</strong> use ComplexHeatmap for adding annotations?</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="sec-francisella.html#cb225-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> inference <span class="sc">|&gt;</span> </span>
<span id="cb225-2"><a href="sec-francisella.html#cb225-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(adjPval), adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb225-3"><a href="sec-francisella.html#cb225-3" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb225-4"><a href="sec-francisella.html#cb225-4" tabindex="-1"></a><span class="fu">heatmap</span>(<span class="fu">assay</span>(pe[[<span class="st">"proteins"</span>]])[sigNames, ])</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-241-1.png" width="672"></p>
</div>
<div id="detail-plots-1" class="section level3" number="5.7.4">
<h3>
<span class="header-section-number">5.7.4</span> Detail plots<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('detail-plots-1')" onmouseout="reset_tooltip('detail-plots-1-tooltip')"><span class="tooltiptext" id="detail-plots-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Let us visualise the most significant protein. We perform this with a
little data manipulation pipeline:</p>
<ol style="list-style-type: decimal">
<li>Identify the target protein with largest logFC.</li>
<li>We use the <code>QFeatures</code> subsetting functionality to retrieve all
data related to the target protein, focusing on the <code>proteins</code>
set that contains the preprocessed data used for modelling.</li>
<li>We use <code>longForm()</code> to convert the object into a table suitable
for plotting.</li>
<li>We remove missing values for plotting.</li>
<li>Plot the data with <code>ggplot2</code>.</li>
</ol>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="sec-francisella.html#cb226-1" tabindex="-1"></a>targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)] <span class="co">#1</span></span>
<span id="cb226-2"><a href="sec-francisella.html#cb226-2" tabindex="-1"></a>pe[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span> <span class="co">#2</span></span>
<span id="cb226-3"><a href="sec-francisella.html#cb226-3" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="st">"genotype"</span>) <span class="sc">|&gt;</span>  <span class="co">#3</span></span>
<span id="cb226-4"><a href="sec-francisella.html#cb226-4" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb226-5"><a href="sec-francisella.html#cb226-5" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">|&gt;</span> <span class="co">#4</span></span>
<span id="cb226-6"><a href="sec-francisella.html#cb226-6" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="co">#5</span></span>
<span id="cb226-7"><a href="sec-francisella.html#cb226-7" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb226-8"><a href="sec-francisella.html#cb226-8" tabindex="-1"></a>      <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb226-9"><a href="sec-francisella.html#cb226-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> genotype)) <span class="sc">+</span></span>
<span id="cb226-10"><a href="sec-francisella.html#cb226-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2 intensity"</span>) <span class="sc">+</span></span>
<span id="cb226-11"><a href="sec-francisella.html#cb226-11" tabindex="-1"></a>  <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb226-12"><a href="sec-francisella.html#cb226-12" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-242-1.png" width="672"></p>
</div>
</div>
<div id="conclusion-2" class="section level2" number="5.8">
<h2>
<span class="header-section-number">5.8</span> Conclusion<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('conclusion-2')" onmouseout="reset_tooltip('conclusion-2-tooltip')"><span class="tooltiptext" id="conclusion-2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p><strong>TODO</strong> mention peptide workflows exist as well and link to first
chapter?</p>
<p><strong>TODO</strong> write conclusion</p>

</div>
</div>
<p style="text-align: center;">
<a href="sec-data_levels.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-mouse_diet.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2025-10-07
 using 
R version 4.5.1 (2025-06-13)
</p>
</div>
</div>



</body>
</html>
