<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="Chapter 2 Statistical analysis with msqrob2 | Statistical analysis for proteomics data" />
<meta property="og:type" content="book" />




<meta name="author" content="Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement" />

<meta name="date" content="2025-10-07" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="Chapter 2 Statistical analysis with msqrob2 | Statistical analysis for proteomics data">

<title>Chapter 2 Statistical analysis with msqrob2 | Statistical analysis for proteomics data</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }

code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Statistical analysis for proteomics data<p><p class="author">Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html" id="toc-preamble"><span class="toc-section-number">1</span> Preamble</a>
<a id="active-page" href="sec-basics.html" id="toc-sec-basics"><span class="toc-section-number">2</span> Statistical analysis with msqrob2</a><ul class="toc-sections">
<li class="toc"><a href="#background"> Background</a></li>
<li class="toc"><a href="#software"> Software</a></li>
<li class="toc"><a href="#data"> Data</a></li>
<li class="toc"><a href="#NA"> Data preprocessing</a></li>
<li class="toc"><a href="#NA"> Data exploration</a></li>
<li class="toc"><a href="#sec-modelling"> Data modelling</a></li>
<li class="toc"><a href="#sec-inference"> Statistical inference</a></li>
<li class="toc"><a href="#going-further"> Going further</a></li>
<li class="toc"><a href="#conclusion"> Conclusion</a></li>
</ul>
<a href="sec-advanced.html" id="toc-sec-advanced"><span class="toc-section-number">3</span> Advanced statistical analysis with msqrob2</a>
<a href="sec-data_levels.html#sec-data_levels" id="toc-sec-data_levels"><span class="toc-section-number">4</span> PSM, peptide or protein-level modelling?</a>
<a href="sec-francisella.html" id="toc-sec-francisella"><span class="toc-section-number">5</span> The francisella use case: a MaxQuant LFQ DDA dataset</a>
<a href="sec-mouse_diet.html#sec-mouse_diet" id="toc-sec-mouse_diet"><span class="toc-section-number">6</span> The mouse diet use case: a Skyline TMT DDA dataset</a>
<a href="sec-end.html" id="toc-sec-end"><span class="toc-section-number">7</span> Usefull links and session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec-basics" class="section level1" number="2">
<h1>
<span class="header-section-number">Chapter 2</span> Statistical analysis with msqrob2</h1>
<p>This chapter explains the main concepts for statistical analysis of
proteomics data using <code>msqrob2</code>. To illustrate these concepts, we will
use a spike-in dataset from the Clinical Proteomic Tumor Analysis
Consortium (CPTAC), published by <span class="citation">Paulovich et al. (<label for="tufte-mn-4" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-4" class="margin-toggle">2010<span class="marginnote">Paulovich, Amanda G., Dean Billheimer, Amy-Joan L. Ham, Lorenzo Vega-Montoto, Paul A. Rudnick, David L. Tabb, Pei Wang, et al. 2010. <span>“Interlaboratory Study Characterizing a Yeast Performance Standard for Benchmarking LC-MS Platform Performance*.”</span> <em>Molecular &amp; Cellular Proteomics</em> 9 (2): 242–54. https://doi.org/<a href="https://doi.org/10.1074/mcp.M900222-MCP200">https://doi.org/10.1074/mcp.M900222-MCP200</a>.</span>)</span> and reanalysed by
<span class="citation">Goeminne, Gevaert, and Clement (<label for="tufte-mn-5" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-5" class="margin-toggle">2016<span class="marginnote">Goeminne, Ludger J E, Kris Gevaert, and Lieven Clement. 2016. <span>“Peptide-Level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-Dependent Quantitative Label-Free Shotgun Proteomics.”</span> <em>Mol. Cell. Proteomics</em> 15 (2): 657–68.</span>)</span>. This is an excellent example dataset since it
contains ground truth information about which proteins are
differentially abundant, enabling us to objectively demonstrate the
performance of <code>msqrob2</code>. This dataset also implies a relatively
simple experimental design (which does not imply that the analysis is
easy), allowing us to answer simple research questions using simple
data analysis workflow. For a more advanced analysis with a complex
expiremental design, we refer to our <a href="sec-advanced.html#sec-advanced">advanced concepts
chapter</a></p>
<div id="background" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Background<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('background')" onmouseout="reset_tooltip('background-tooltip')"><span class="tooltiptext" id="background-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Mass spectrometry (MS)-based proteomics aims at characterising the
proteome content of biological samples. The most popular approach is
label-free quantification (LFQ), where every sample is analysed in a
separate MS run. This section provides an overview of the analytical
workflow and its main challenges regarding data modelling.</p>
<div id="sec-lfq_workflow" class="section level3" number="2.1.1">
<h3>
<span class="header-section-number">2.1.1</span> LFQ workflow<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-lfq_workflow')" onmouseout="reset_tooltip('sec-lfq_workflow-tooltip')"><span class="tooltiptext" id="sec-lfq_workflow-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>In a nutshell, the workflow starts with sample preparation where the
samples are collected, and the protein content is extracted and
digested into peptides. To reduce the sample complexity, the peptides
are then separated based on physicochemical properties (mostly
hydrophobicity) thanks to liquid chromatography (LC). Peptides are
then ionised by an electrospray as they elute from the chromatographic
column. The signal over time generated by the eluting ions is called
the total ion chromatogram. The ions are then sent for a first round
of MS to record their m/z distribution for the intact ions. This
provides an overview of the ions that elute from the column and allows
for further separation of the ions in the m/z space. The second round
of MS (MS2) records the fragmented ions for a selection of ions,
generally the most intense MS1 peaks<label for="tufte-sn-1" class="margin-toggle sidenote-number">1</label><input type="checkbox" id="tufte-sn-1" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">1</span> The ion selection for MS2
depends on the data recorded in MS1. Therefore, this approach is
referred to as data dependent acquisition (DDA).</span>. This process is
repeated for every sample so that every sample is acquired in one MS
run. This provides the ion’s mass fingerprint. For LFQ workflows, the
accumulated MS1 intensity over time, also known as the area under the
curve, around the target mass is used as a quantitification measure.
On the other hand, the ion mass fingerprint, called the MS2 spectrum,
enables computational identification of the corresponding peptide
using search engines (e.g. Andromeda has been used for this data set)
that will provide peptide-to-spectrum matches (PSM). The quantified
PSM are further processed by the software (MaxQuant) to obtain a
peptide table<label for="tufte-sn-2" class="margin-toggle sidenote-number">2</label><input type="checkbox" id="tufte-sn-2" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">2</span> MaxQuant also computes a protein table. However, we
found that starting from MaxQuant’s protein table leads to a decrease
in performance. We will illustrate in this tutorial how to build the
protein table.</span>, where every row corresponds to an identified peptide
and every column contains information about the peptide and its
quantification in one of the samples.</p>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-58"></span>
<p class="caption marginnote shownote">
Figure 2.1: Overview of an LFQ-based proteomics workflow.
</p>
<img src="figs/lfq_workflow.png" alt="Overview of an LFQ-based proteomics workflow." width="40%">
</div>
</div>
<div id="sec-lfq_challenges" class="section level3" number="2.1.2">
<h3>
<span class="header-section-number">2.1.2</span> Challenges<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-lfq_challenges')" onmouseout="reset_tooltip('sec-lfq_challenges-tooltip')"><span class="tooltiptext" id="sec-lfq_challenges-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Behind this workflow lies several challenges that will affect the data
modelling:</p>
<ul>
<li>MS-based proteomics doesn’t measure proteins directly, but their
constituting <strong>peptide ions</strong>. The protein-level information needs
to be reconstructed from the ion data. In this tutorial, we will
start from the peptide data, which has been constructed from the ion
data by MaxQuant.</li>
<li>All peptides cannot be ionised with the same efficiency. Poor
ionisation will lead to reduced signal as less ions will hit the
detector, hence leading to a huge variability in intensity among
different peptide species, even when they originate from the same
protein.</li>
<li>The identification step is not trivial and prone to
errors<label for="tufte-sn-3" class="margin-toggle sidenote-number">3</label><input type="checkbox" id="tufte-sn-3" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">3</span> Improving peptide identification is outside the scope of
this tutorial</span>. PSM misidentification leads to the assignment of a
quantitative values from another peptide with likely another
ionisation efficiency and relative abundance. Hence this misassigned
value will become an outlier.</li>
<li>Moreover, the ion selection for MS2 depends on its
intensity<label for="tufte-sn-4" class="margin-toggle sidenote-number">4</label><input type="checkbox" id="tufte-sn-4" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">4</span> Recall that only the top most intense ion peaks are send
for MS2</span>. Therefore, the chance to measure and, subsequently,
identify a peptide will depend on its abundance. Non identified
peptides will lead to data missingness, which is related to the
underlying quantification value. This phenomenon is known as
missingness not at random. Next to that, many reasons can lead to
ions not being selected or identified irrespective of their
quantification value leading to missingness that is not related to
its quantitative value. This is referred to as missingness
completely at random. The missingness issue is not negligible: only
41% of all proteins are quantified across all samples, and the
number drops to 6.6% when considering peptides.</li>
<li>The identification issues lead to unbalanced peptide missingness
across samples, and the patterns of missing values are potentially
different for every peptide, highlighting the need for an
automatised solution that is robust against missing values.</li>
<li>Technical variations during the experiment can lead to systematic
fluctuations across samples. The most obvious reason is when
different sample amounts are injected into the instruments, due to
small pipetting inconsistencies for instance. However, these
differences lead to unwanted variation that should be discarded when
answering biological questions.</li>
</ul>
</div>
<div id="sec-cptac_experiment" class="section level3" number="2.1.3">
<h3>
<span class="header-section-number">2.1.3</span> Experimental context<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-cptac_experiment')" onmouseout="reset_tooltip('sec-cptac_experiment-tooltip')"><span class="tooltiptext" id="sec-cptac_experiment-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The CPTAC <span class="citation">(<label for="tufte-mn-6" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-6" class="margin-toggle">Paulovich et al. 2010<span class="marginnote">Paulovich, Amanda G., Dean Billheimer, Amy-Joan L. Ham, Lorenzo Vega-Montoto, Paul A. Rudnick, David L. Tabb, Pei Wang, et al. 2010. <span>“Interlaboratory Study Characterizing a Yeast Performance Standard for Benchmarking LC-MS Platform Performance*.”</span> <em>Molecular &amp; Cellular Proteomics</em> 9 (2): 242–54. https://doi.org/<a href="https://doi.org/10.1074/mcp.M900222-MCP200">https://doi.org/10.1074/mcp.M900222-MCP200</a>.</span>)</span> study provides a proteomics data set
with ground truth. The samples were syntethically constructed,
starting from a trypsin-digested yeast (L Saccharomyces cerevisiae
strain BY4741) proteome background, hence yeast proteins are known to
be constant across samples. Next, the authors added a trypsin-digested
Sigma Universal Protein Standard mixture 1 (UPS1) standard, which
contains 48 different human proteins. They spiked in these UPS1
proteins at 5 different concentrations: 0.25 fmol/µl (treatment 6A),
0.74 fmol/µl (treatment 6B), 2.22 fmol/µl (treatment 6C), 6.67 fmol/µl
(treatment 6D), and 20 fmol/µl (treatment 6E). So the UPS1 proteins
are known to be differentially expressed, and we know exactly in what
amount they differ. The samples were constructed in triplicate and
each triplicate was sent to one of three labs. In total, the data set
contains 45 samples.</p>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-59"></span>
<p class="caption marginnote shownote">
Figure 2.2: Overview of the experimental design.
</p>
<img src="figs/cptac_experiment.png" alt="Overview of the experimental design." width="50%">
</div>
<p>We will start from the peptide data generated by MaxQuant and infer
protein-level differences between samples. To achieve this goal, we
will apply an <code>msqrob2</code> workflow, a data processing and modelling
workflow dedicated to the analysis of MS-based proteomics datasets. We
will demonstrate how the workflow can retrieve the spiked-in proteins
from the CPTAC data set, along with explaination of the key
statistical concepts behind proteomics data analysis. Before delving
into the analysis, let us prepare our computational environment.</p>
</div>
</div>
<div id="software" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Software<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('software')" onmouseout="reset_tooltip('software-tooltip')"><span class="tooltiptext" id="software-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<div id="load-packages" class="section level3" number="2.2.1">
<h3>
<span class="header-section-number">2.2.1</span> Load packages<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('load-packages')" onmouseout="reset_tooltip('load-packages-tooltip')"><span class="tooltiptext" id="load-packages-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We load the <code>msqrob2</code> package, along with additional packages for
data manipulation and visualisation.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="sec-basics.html#cb2-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb2-2"><a href="sec-basics.html#cb2-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb2-3"><a href="sec-basics.html#cb2-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb2-4"><a href="sec-basics.html#cb2-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"patchwork"</span>)</span></code></pre></div>
</div>
<div id="sec-parallel" class="section level3" number="2.2.2">
<h3>
<span class="header-section-number">2.2.2</span> Parallelisation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-parallel')" onmouseout="reset_tooltip('sec-parallel-tooltip')"><span class="tooltiptext" id="sec-parallel-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p><code>msqrob2</code> can parallelise computations during the model estimation
to improve speed. However, we will disable parallelisation to ensure
this vignette can be run regardless of hardware. Parallelisation is
controlled using the <code>BiocParallel</code> package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="sec-basics.html#cb3-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb3-2"><a href="sec-basics.html#cb3-2" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code></pre></div>
<p>If you want to use <code>msqrob2</code> with parallelisation enabled and using
4 cores, you can run the following:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="sec-basics.html#cb4-1" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">MulticoreParam</span>(<span class="at">workers =</span> <span class="dv">4</span>))</span></code></pre></div>
<p>Be mindful that, while parallelisation can improve speed, it will also
consume more RAM because part of the data will be copied multiple
times over your different workers. If you experience crashes because
you exceeded the amount of available RAM on your machine, you should
reduce the number of requested workers.</p>
</div>
</div>
<div id="data" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data')" onmouseout="reset_tooltip('data-tooltip')"><span class="tooltiptext" id="data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The data were reanalysed by <span class="citation">Goeminne, Gevaert, and Clement (<label for="tufte-mn-7" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-7" class="margin-toggle">2016<span class="marginnote">Goeminne, Ludger J E, Kris Gevaert, and Lieven Clement. 2016. <span>“Peptide-Level Robust Ridge Regression Improves Estimation, Sensitivity, and Specificity in Data-Dependent Quantitative Label-Free Shotgun Proteomics.”</span> <em>Mol. Cell. Proteomics</em> 15 (2): 657–68.</span>)</span> using MaxQuant and
deposited on <code>MsDataHub</code>, a Bioconductor data package with
quantification data files from MS-based experiments<label for="tufte-sn-5" class="margin-toggle sidenote-number">5</label><input type="checkbox" id="tufte-sn-5" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">5</span> Note: the data
will be downloaded the first time the code is called. Afterthat,
running the same code will automatically fetch the data from your
local machine</span>. We here retrieve MaxQuant’s <code>peptides.txt</code> for the
full CPTAC study 6.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="sec-basics.html#cb5-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"MsDataHub"</span>)</span>
<span id="cb5-2"><a href="sec-basics.html#cb5-2" tabindex="-1"></a>peptideFile <span class="ot">&lt;-</span> <span class="fu">cptac_peptides.txt</span>()</span></code></pre></div>
<div id="sec-peptide_table" class="section level4" number="2.3.0.1">
<h4>
<span class="header-section-number">2.3.0.1</span> Peptide table</h4>
<p>Each row in the peptide data table contains information about one
peptide (the table below shows the first 6 rows). The columns contains
various descriptors about the peptide, such as its sequence, its
charge, the amino acid composition, etc. Some of these columns (those
starting with <code>Intensity.</code>) contain the quantification values for each
sample. The table format where the quantitative values for each sample
are contained in a separate column is depicted as the “wide format”,
as opposed to the “long format” (eg, the <a href="sec-advanced.html#sec-psm_table">PSM table</a>).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="sec-basics.html#cb6-1" tabindex="-1"></a>peptides <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(peptideFile)</span>
<span id="cb6-2"><a href="sec-basics.html#cb6-2" tabindex="-1"></a>quantCols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Intensity[.]"</span>, <span class="fu">names</span>(peptides), <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="5%">
<col width="3%">
<col width="0%">
<col width="0%">
<col width="0%">
</colgroup>
<thead><tr class="header">
<th align="left">Sequence</th>
<th align="left">N.term.cleavage.window</th>
<th align="left">C.term.cleavage.window</th>
<th align="left">Amino.acid.before</th>
<th align="left">First.amino.acid</th>
<th align="left">Second.amino.acid</th>
<th align="left">Second.last.amino.acid</th>
<th align="left">Last.amino.acid</th>
<th align="left">Amino.acid.after</th>
<th align="right">A.Count</th>
<th align="right">R.Count</th>
<th align="right">N.Count</th>
<th align="right">D.Count</th>
<th align="right">C.Count</th>
<th align="right">Q.Count</th>
<th align="right">E.Count</th>
<th align="right">G.Count</th>
<th align="right">H.Count</th>
<th align="right">I.Count</th>
<th align="right">L.Count</th>
<th align="right">K.Count</th>
<th align="right">M.Count</th>
<th align="right">F.Count</th>
<th align="right">P.Count</th>
<th align="right">S.Count</th>
<th align="right">T.Count</th>
<th align="right">W.Count</th>
<th align="right">Y.Count</th>
<th align="right">V.Count</th>
<th align="right">U.Count</th>
<th align="right">Length</th>
<th align="right">Missed.cleavages</th>
<th align="right">Mass</th>
<th align="left">Proteins</th>
<th align="left">Leading.razor.protein</th>
<th align="right">Start.position</th>
<th align="right">End.position</th>
<th align="left">Unique..Groups.</th>
<th align="left">Unique..Proteins.</th>
<th align="left">Charges</th>
<th align="right">PEP</th>
<th align="right">Score</th>
<th align="left">Identification.type.6A_1</th>
<th align="left">Identification.type.6A_2</th>
<th align="left">Identification.type.6A_3</th>
<th align="left">Identification.type.6A_4</th>
<th align="left">Identification.type.6A_5</th>
<th align="left">Identification.type.6A_6</th>
<th align="left">Identification.type.6A_7</th>
<th align="left">Identification.type.6A_8</th>
<th align="left">Identification.type.6A_9</th>
<th align="left">Identification.type.6B_1</th>
<th align="left">Identification.type.6B_2</th>
<th align="left">Identification.type.6B_3</th>
<th align="left">Identification.type.6B_4</th>
<th align="left">Identification.type.6B_5</th>
<th align="left">Identification.type.6B_6</th>
<th align="left">Identification.type.6B_7</th>
<th align="left">Identification.type.6B_8</th>
<th align="left">Identification.type.6B_9</th>
<th align="left">Identification.type.6C_1</th>
<th align="left">Identification.type.6C_2</th>
<th align="left">Identification.type.6C_3</th>
<th align="left">Identification.type.6C_4</th>
<th align="left">Identification.type.6C_5</th>
<th align="left">Identification.type.6C_6</th>
<th align="left">Identification.type.6C_7</th>
<th align="left">Identification.type.6C_8</th>
<th align="left">Identification.type.6C_9</th>
<th align="left">Identification.type.6D_1</th>
<th align="left">Identification.type.6D_2</th>
<th align="left">Identification.type.6D_3</th>
<th align="left">Identification.type.6D_4</th>
<th align="left">Identification.type.6D_5</th>
<th align="left">Identification.type.6D_6</th>
<th align="left">Identification.type.6D_7</th>
<th align="left">Identification.type.6D_8</th>
<th align="left">Identification.type.6D_9</th>
<th align="left">Identification.type.6E_1</th>
<th align="left">Identification.type.6E_2</th>
<th align="left">Identification.type.6E_3</th>
<th align="left">Identification.type.6E_4</th>
<th align="left">Identification.type.6E_5</th>
<th align="left">Identification.type.6E_6</th>
<th align="left">Identification.type.6E_7</th>
<th align="left">Identification.type.6E_8</th>
<th align="left">Identification.type.6E_9</th>
<th align="right">Experiment.6A_1</th>
<th align="right">Experiment.6A_2</th>
<th align="right">Experiment.6A_3</th>
<th align="right">Experiment.6A_4</th>
<th align="right">Experiment.6A_5</th>
<th align="right">Experiment.6A_6</th>
<th align="right">Experiment.6A_7</th>
<th align="right">Experiment.6A_8</th>
<th align="right">Experiment.6A_9</th>
<th align="right">Experiment.6B_1</th>
<th align="right">Experiment.6B_2</th>
<th align="right">Experiment.6B_3</th>
<th align="right">Experiment.6B_4</th>
<th align="right">Experiment.6B_5</th>
<th align="right">Experiment.6B_6</th>
<th align="right">Experiment.6B_7</th>
<th align="right">Experiment.6B_8</th>
<th align="right">Experiment.6B_9</th>
<th align="right">Experiment.6C_1</th>
<th align="right">Experiment.6C_2</th>
<th align="right">Experiment.6C_3</th>
<th align="right">Experiment.6C_4</th>
<th align="right">Experiment.6C_5</th>
<th align="right">Experiment.6C_6</th>
<th align="right">Experiment.6C_7</th>
<th align="right">Experiment.6C_8</th>
<th align="right">Experiment.6C_9</th>
<th align="right">Experiment.6D_1</th>
<th align="right">Experiment.6D_2</th>
<th align="right">Experiment.6D_3</th>
<th align="right">Experiment.6D_4</th>
<th align="right">Experiment.6D_5</th>
<th align="right">Experiment.6D_6</th>
<th align="right">Experiment.6D_7</th>
<th align="right">Experiment.6D_8</th>
<th align="right">Experiment.6D_9</th>
<th align="right">Experiment.6E_1</th>
<th align="right">Experiment.6E_2</th>
<th align="right">Experiment.6E_3</th>
<th align="right">Experiment.6E_4</th>
<th align="right">Experiment.6E_5</th>
<th align="right">Experiment.6E_6</th>
<th align="right">Experiment.6E_7</th>
<th align="right">Experiment.6E_8</th>
<th align="right">Experiment.6E_9</th>
<th align="right">Intensity</th>
<th align="right">Intensity.6A_1</th>
<th align="right">Intensity.6A_2</th>
<th align="right">Intensity.6A_3</th>
<th align="right">Intensity.6A_4</th>
<th align="right">Intensity.6A_5</th>
<th align="right">Intensity.6A_6</th>
<th align="right">Intensity.6A_7</th>
<th align="right">Intensity.6A_8</th>
<th align="right">Intensity.6A_9</th>
<th align="right">Intensity.6B_1</th>
<th align="right">Intensity.6B_2</th>
<th align="right">Intensity.6B_3</th>
<th align="right">Intensity.6B_4</th>
<th align="right">Intensity.6B_5</th>
<th align="right">Intensity.6B_6</th>
<th align="right">Intensity.6B_7</th>
<th align="right">Intensity.6B_8</th>
<th align="right">Intensity.6B_9</th>
<th align="right">Intensity.6C_1</th>
<th align="right">Intensity.6C_2</th>
<th align="right">Intensity.6C_3</th>
<th align="right">Intensity.6C_4</th>
<th align="right">Intensity.6C_5</th>
<th align="right">Intensity.6C_6</th>
<th align="right">Intensity.6C_7</th>
<th align="right">Intensity.6C_8</th>
<th align="right">Intensity.6C_9</th>
<th align="right">Intensity.6D_1</th>
<th align="right">Intensity.6D_2</th>
<th align="right">Intensity.6D_3</th>
<th align="right">Intensity.6D_4</th>
<th align="right">Intensity.6D_5</th>
<th align="right">Intensity.6D_6</th>
<th align="right">Intensity.6D_7</th>
<th align="right">Intensity.6D_8</th>
<th align="right">Intensity.6D_9</th>
<th align="right">Intensity.6E_1</th>
<th align="right">Intensity.6E_2</th>
<th align="right">Intensity.6E_3</th>
<th align="right">Intensity.6E_4</th>
<th align="right">Intensity.6E_5</th>
<th align="right">Intensity.6E_6</th>
<th align="right">Intensity.6E_7</th>
<th align="right">Intensity.6E_8</th>
<th align="right">Intensity.6E_9</th>
<th align="left">Reverse</th>
<th align="left">Potential.contaminant</th>
<th align="right">id</th>
<th align="left">Protein.group.IDs</th>
<th align="left">Mod..peptide.IDs</th>
<th align="left">Evidence.IDs</th>
<th align="left">MS.MS.IDs</th>
<th align="right">Best.MS.MS</th>
<th align="left">Oxidation..M..site.IDs</th>
<th align="right">MS.MS.Count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AAAAGAGGAGDSGDAVTK</td>
<td align="left">EHQHDEQKAAAAGAGG</td>
<td align="left">DSGDAVTKIGSEDVKL</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">T</td>
<td align="left">K</td>
<td align="left">I</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">18</td>
<td align="right">0</td>
<td align="right">1445.6746</td>
<td align="left">sp|P38915|SPT8_YEAST</td>
<td align="left">sp|P38915|SPT8_YEAST</td>
<td align="right">97</td>
<td align="right">114</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000118</td>
<td align="right">82.942</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1190800</td>
<td align="right">0</td>
<td align="right">76034</td>
<td align="right">0</td>
<td align="right">56831</td>
<td align="right">83835</td>
<td align="right">92136</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">66760</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">34806</td>
<td align="right">48499</td>
<td align="right">0</td>
<td align="right">37436</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">54720</td>
<td align="right">56483</td>
<td align="right">45058</td>
<td align="right">0</td>
<td align="right">44097</td>
<td align="right">34553</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">37445</td>
<td align="right">77859</td>
<td align="right">74664</td>
<td align="right">75781</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">72670</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">69556</td>
<td align="right">51540</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
<td align="left">859</td>
<td align="left">0</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20;21;22;23</td>
<td align="left">0;1;2;3;4;5;6;7;8;9</td>
<td align="right">0</td>
<td align="left"></td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">AAAALAGGK</td>
<td align="left">QQLSKAAKAAAALAGG</td>
<td align="left">AAALAGGKKSKKKWSK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">728.4181</td>
<td align="left">sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST</td>
<td align="left">sp|Q3E792|RS25A_YEAST</td>
<td align="right">13</td>
<td align="right">21</td>
<td align="left">yes</td>
<td align="left">no</td>
<td align="left">2</td>
<td align="right">0.0000075</td>
<td align="right">134.810</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">280990000</td>
<td align="right">0</td>
<td align="right">19017000</td>
<td align="right">17021000</td>
<td align="right">16406000</td>
<td align="right">52547</td>
<td align="right">282260</td>
<td align="right">2441300</td>
<td align="right">1220000</td>
<td align="right">1337600</td>
<td align="right">12141000</td>
<td align="right">15732000</td>
<td align="right">15365000</td>
<td align="right">17052000</td>
<td align="right">21378</td>
<td align="right">0</td>
<td align="right">2850900</td>
<td align="right">935580</td>
<td align="right">1606200</td>
<td align="right">0</td>
<td align="right">13666000</td>
<td align="right">23575000</td>
<td align="right">13721000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2989800</td>
<td align="right">931380</td>
<td align="right">1444200</td>
<td align="right">0</td>
<td align="right">17021000</td>
<td align="right">17211000</td>
<td align="right">8967900</td>
<td align="right">153320</td>
<td align="right">0</td>
<td align="right">3023700</td>
<td align="right">1142500</td>
<td align="right">2408500</td>
<td align="right">0</td>
<td align="right">13562000</td>
<td align="right">16498000</td>
<td align="right">14772000</td>
<td align="right">16829</td>
<td align="right">0</td>
<td align="right">3285200</td>
<td align="right">902500</td>
<td align="right">2216600</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
<td align="left">230</td>
<td align="left">1</td>
<td align="left">24;25;26;27;28;29;30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73</td>
<td align="left">10;11;12;13;14;15;16;17;18;19;20;21;22;23;24;25;26;27;28;29</td>
<td align="right">21</td>
<td align="left"></td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">AAAALAGGKK</td>
<td align="left">QQLSKAAKAAAALAGG</td>
<td align="left">AALAGGKKSKKKWSKK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="left">S</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">856.5131</td>
<td align="left">sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST</td>
<td align="left">sp|Q3E792|RS25A_YEAST</td>
<td align="right">13</td>
<td align="right">22</td>
<td align="left">yes</td>
<td align="left">no</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="right">143.730</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">33360000</td>
<td align="right">0</td>
<td align="right">5074300</td>
<td align="right">0</td>
<td align="right">4330300</td>
<td align="right">0</td>
<td align="right">28816</td>
<td align="right">1029200</td>
<td align="right">668040</td>
<td align="right">638990</td>
<td align="right">0</td>
<td align="right">4617600</td>
<td align="right">834020</td>
<td align="right">89655</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">777030</td>
<td align="right">641270</td>
<td align="right">562840</td>
<td align="right">0</td>
<td align="right">3453200</td>
<td align="right">0</td>
<td align="right">1436100</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">784050</td>
<td align="right">653270</td>
<td align="right">597720</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2467500</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">921940</td>
<td align="right">728810</td>
<td align="right">729680</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">48994</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1030900</td>
<td align="right">489850</td>
<td align="right">725630</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">2</td>
<td align="left">230</td>
<td align="left">2</td>
<td align="left">74;75;76;77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98</td>
<td align="left">30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50</td>
<td align="right">31</td>
<td align="left"></td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">AAADALSDLEIK</td>
<td align="left">MPKETPSKAAADALSD</td>
<td align="left">ALSDLEIKDSKSNLNK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="left">K</td>
<td align="left">D</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1215.6347</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="right">9</td>
<td align="right">20</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="right">182.230</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">54622000</td>
<td align="right">1111200</td>
<td align="right">1578700</td>
<td align="right">1311400</td>
<td align="right">1277500</td>
<td align="right">2192900</td>
<td align="right">1889700</td>
<td align="right">515460</td>
<td align="right">670780</td>
<td align="right">712140</td>
<td align="right">822300</td>
<td align="right">1210400</td>
<td align="right">1410400</td>
<td align="right">824260</td>
<td align="right">1053200</td>
<td align="right">1214600</td>
<td align="right">426580</td>
<td align="right">620510</td>
<td align="right">737780</td>
<td align="right">1178300</td>
<td align="right">838050</td>
<td align="right">1403400</td>
<td align="right">762630</td>
<td align="right">1493000</td>
<td align="right">1616700</td>
<td align="right">463860</td>
<td align="right">887640</td>
<td align="right">764890</td>
<td align="right">2250300</td>
<td align="right">2586500</td>
<td align="right">2817700</td>
<td align="right">573900</td>
<td align="right">1944100</td>
<td align="right">1304300</td>
<td align="right">672760</td>
<td align="right">785430</td>
<td align="right">662070</td>
<td align="right">2550100</td>
<td align="right">1469300</td>
<td align="right">1764100</td>
<td align="right">946030</td>
<td align="right">1324300</td>
<td align="right">1132100</td>
<td align="right">787870</td>
<td align="right">1081800</td>
<td align="right">981190</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">3</td>
<td align="left">229</td>
<td align="left">3</td>
<td align="left">99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143</td>
<td align="left">51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79;80;81;82;83;84</td>
<td align="right">72</td>
<td align="left"></td>
<td align="right">29</td>
</tr>
<tr class="odd">
<td align="left">AAADALSDLEIKDSK</td>
<td align="left">MPKETPSKAAADALSD</td>
<td align="left">DLEIKDSKSNLNKELE</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">S</td>
<td align="left">K</td>
<td align="left">S</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">1545.7886</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="right">9</td>
<td align="right">23</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">3</td>
<td align="right">0.0001532</td>
<td align="right">73.927</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">18910000</td>
<td align="right">78739</td>
<td align="right">61204</td>
<td align="right">0</td>
<td align="right">667190</td>
<td align="right">1035200</td>
<td align="right">1087400</td>
<td align="right">331130</td>
<td align="right">420900</td>
<td align="right">365560</td>
<td align="right">0</td>
<td align="right">75201</td>
<td align="right">0</td>
<td align="right">473100</td>
<td align="right">475520</td>
<td align="right">672870</td>
<td align="right">329250</td>
<td align="right">380820</td>
<td align="right">414490</td>
<td align="right">90564</td>
<td align="right">87833</td>
<td align="right">56581</td>
<td align="right">412110</td>
<td align="right">573950</td>
<td align="right">736840</td>
<td align="right">278620</td>
<td align="right">416740</td>
<td align="right">413980</td>
<td align="right">486660</td>
<td align="right">413430</td>
<td align="right">451440</td>
<td align="right">279890</td>
<td align="right">1370800</td>
<td align="right">743060</td>
<td align="right">247740</td>
<td align="right">538740</td>
<td align="right">358390</td>
<td align="right">510500</td>
<td align="right">290860</td>
<td align="right">472840</td>
<td align="right">510350</td>
<td align="right">784520</td>
<td align="right">641070</td>
<td align="right">0</td>
<td align="right">413240</td>
<td align="right">460840</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">4</td>
<td align="left">229</td>
<td align="left">4</td>
<td align="left">144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172;173;174;175;176;177;178;179;180;181;182;183;184</td>
<td align="left">85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116</td>
<td align="right">94</td>
<td align="left"></td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">AAAEEFQR</td>
<td align="left">RERKEREKAAAEEFQR</td>
<td align="left">AAAEEFQRQQELLRQQ</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">Q</td>
<td align="left">R</td>
<td align="left">Q</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">920.4352</td>
<td align="left">sp|P53075|SHE10_YEAST</td>
<td align="left">sp|P53075|SHE10_YEAST</td>
<td align="right">538</td>
<td align="right">545</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0175880</td>
<td align="right">79.474</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1158600</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">240490</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">51558</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">149420</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">76970</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">173740</td>
<td align="right">0</td>
<td align="right">200230</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">153720</td>
<td align="right">0</td>
<td align="right">43453</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">69019</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">5</td>
<td align="left">1143</td>
<td align="left">5</td>
<td align="left">185;186;187;188;189;190;191;192;193;194</td>
<td align="left">117;118</td>
<td align="right">117</td>
<td align="left"></td>
<td align="right">2</td>
</tr>
</tbody>
</table>
</div>
<div id="sec-annotation_table" class="section level4" number="2.3.0.2">
<h4>
<span class="header-section-number">2.3.0.2</span> Sample annotation table</h4>
<p>Each row in the annotation table contains information about one
sample. The columns contain various descriptors about the sample, such
as the name of the sample or the MS run, the treatment (here the
spike-in condition), the lab that acquired the sample or any other
biological or technical information that may impact the data quality
or the quantification. Without an annotation table, no analysis
can be performed. The sample annotations are generated by the
researcher. In this example, the annotations are extracted from the
sample names, although reporting a detailed design of experiments in a
table is seen as better practice <span class="citation">(<label for="tufte-mn-8" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-8" class="margin-toggle">Gatto et al. 2023<span class="marginnote">Gatto, Laurent, Ruedi Aebersold, Juergen Cox, Vadim Demichev, Jason Derks, Edward Emmott, Alexander M Franks, et al. 2023. <span>“Initial Recommendations for Performing, Benchmarking and Reporting Single-Cell Proteomics Experiments.”</span> <em>Nat. Methods</em> 20 (3): 375–86.</span>)</span>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="sec-basics.html#cb7-1" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">quantCols =</span> quantCols)</span>
<span id="cb7-2"><a href="sec-basics.html#cb7-2" tabindex="-1"></a>coldata<span class="sc">$</span>lab <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">"lab"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>), <span class="at">each =</span> <span class="dv">3</span>), <span class="dv">5</span>)</span>
<span id="cb7-3"><a href="sec-basics.html#cb7-3" tabindex="-1"></a>coldata<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Intensity..(.)_.*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, quantCols)</span>
<span id="cb7-4"><a href="sec-basics.html#cb7-4" tabindex="-1"></a>concentrations <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">A =</span> <span class="fl">0.25</span>, <span class="at">B =</span> <span class="fl">0.74</span>, <span class="at">C =</span> <span class="fl">2.22</span>, <span class="at">D =</span> <span class="fl">6.67</span>, <span class="at">E =</span> <span class="dv">20</span>)</span>
<span id="cb7-5"><a href="sec-basics.html#cb7-5" tabindex="-1"></a>coldata<span class="sc">$</span>concentration <span class="ot">&lt;-</span> concentrations[coldata<span class="sc">$</span>condition]</span></code></pre></div>
<table>
<thead><tr class="header">
<th align="left">quantCols</th>
<th align="left">lab</th>
<th align="left">condition</th>
<th align="right">concentration</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Intensity.6A_1</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_2</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="odd">
<td align="left">Intensity.6A_3</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_4</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="odd">
<td align="left">Intensity.6A_5</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_6</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
</tbody>
</table>
</div>
<div id="sec-qfeatures" class="section level3" number="2.3.1">
<h3>
<span class="header-section-number">2.3.1</span> The QFeatures class<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-qfeatures')" onmouseout="reset_tooltip('sec-qfeatures-tooltip')"><span class="tooltiptext" id="sec-qfeatures-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p><code>msqrob2</code> is built around the <code>QFeatures</code> class. We refer to the <a href="https://rformassspectrometry.github.io/book/sec-quant.html">R
for mass spectrometry
book</a> for
a comprehensive description of the class. In a nutshell, the
<code>QFeatures</code> package provides infrastructure to manage and analyse
quantitative features from mass spectrometry experiments. It is based
on the <code>SummarizedExperiment</code> and <code>MultiAssayExperiment</code> classes. It
leverages the hierarchical structure of proteomics experiments: data
proteins are composed of peptides, themselves produced by spectra.
Each pieces of information in stored in an individual
<code>SummarizedExperiment</code> object, later referred to as a “set”.
Throughout the aggregation and processing of these data, the relations
between sets are tracked and recorded, thus allowing users to easily
navigate across spectra, peptide and protein quantitative data.</p>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-68"></span>
<p class="caption marginnote shownote">
Figure 2.3: Illustration of the <code>QFeatures</code> data class.
</p>
<img src="figs/QFeatures.png" alt="Illustration of the `QFeatures` data class." width="80%">
</div>
<p>The <code>readQFeatures()</code> enables a seamless conversion of tabular data
into a <code>QFeatures</code> object. We provide the peptide table and the sample
annotation table. The function will use the <code>quantCols</code> column in the
sample annotation table to understand which columns in <code>peptides</code>
contain the quantitative values, and automatically link the
corresponding sample annotation with the quantitative values. We also
tell the function to use the <code>Sequence</code> column as peptide identifier,
which will be used as rownames. See <code>?readQFeatures()</code> for more
details.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="sec-basics.html#cb8-1" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb8-2"><a href="sec-basics.html#cb8-2" tabindex="-1"></a>  peptides, coldata, <span class="at">name =</span> <span class="st">"peptides"</span>, <span class="at">fnames =</span> <span class="st">"Sequence"</span></span>
<span id="cb8-3"><a href="sec-basics.html#cb8-3" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 1 set:
## 
##  [1] peptides: SummarizedExperiment with 11466 rows and 45 columns</code></pre>
<p>We now have a <code>QFeatures</code> object with 1 set, called <code>peptides</code> (which
we specified using the <code>name</code> argument). In this toy example, we have
information for 11466 peptides across 45 samples, as
expected (recall the <a href="sec-basics.html#sec-cptac_experiment">experimental design</a>).</p>
<p>The sample annotations can be retrieved using <code>colData()</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="sec-basics.html#cb10-1" tabindex="-1"></a><span class="fu">colData</span>(spikein)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="sec-basics.html#cb11-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">colData</span>(spikein)))</span></code></pre></div>
<table>
<thead><tr class="header">
<th align="left"></th>
<th align="left">quantCols</th>
<th align="left">lab</th>
<th align="left">condition</th>
<th align="right">concentration</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Intensity.6A_1</td>
<td align="left">Intensity.6A_1</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_2</td>
<td align="left">Intensity.6A_2</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="odd">
<td align="left">Intensity.6A_3</td>
<td align="left">Intensity.6A_3</td>
<td align="left">lab1</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_4</td>
<td align="left">Intensity.6A_4</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="odd">
<td align="left">Intensity.6A_5</td>
<td align="left">Intensity.6A_5</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
<tr class="even">
<td align="left">Intensity.6A_6</td>
<td align="left">Intensity.6A_6</td>
<td align="left">lab2</td>
<td align="left">A</td>
<td align="right">0.25</td>
</tr>
</tbody>
</table>
<p>We can get a sample annotation directly using the <code>$</code> accessor.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="sec-basics.html#cb12-1" tabindex="-1"></a><span class="fu">head</span>(spikein<span class="sc">$</span>lab)</span></code></pre></div>
<pre><code>## [1] "lab1" "lab1" "lab1" "lab2" "lab2" "lab2"</code></pre>
<p>We can extract the <code>SummarizedExperiment</code> object for the <code>peptides</code>
set using double bracket subsetting<label for="tufte-sn-6" class="margin-toggle sidenote-number">6</label><input type="checkbox" id="tufte-sn-6" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">6</span> A <code>QFeatures</code> object can be seen
as a special list of <code>SummarizedExperiment</code> objects.</span></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="sec-basics.html#cb14-1" tabindex="-1"></a>spikein[[<span class="st">"peptides"</span>]]</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 11466 45 
## metadata(0):
## assays(1): ''
## rownames(11466): AAAAGAGGAGDSGDAVTK AAAALAGGK ... YYTVFDRDNNR YYTVFDRDNNRVGFAEAAR
## rowData names(143): Sequence N.term.cleavage.window ... Oxidation..M..site.IDs MS.MS.Count
## colnames(45): Intensity.6A_1 Intensity.6A_2 ... Intensity.6E_8 Intensity.6E_9
## colData names(0):</code></pre>
<p>But notice that the sample annotations were not extracted along with
the SummarizedExperiment (<code>colData names(0):</code>). This can be performed
using <code>getWithColData()</code>, which extracts the set of interest (like
<code>[[]]</code>) along with all the associated sample annotations.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="sec-basics.html#cb16-1" tabindex="-1"></a><span class="fu">getWithColData</span>(spikein, <span class="st">"peptides"</span>)</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 11466 45 
## metadata(0):
## assays(1): ''
## rownames(11466): AAAAGAGGAGDSGDAVTK AAAALAGGK ... YYTVFDRDNNR YYTVFDRDNNRVGFAEAAR
## rowData names(143): Sequence N.term.cleavage.window ... Oxidation..M..site.IDs MS.MS.Count
## colnames(45): Intensity.6A_1 Intensity.6A_2 ... Intensity.6E_8 Intensity.6E_9
## colData names(4): quantCols lab condition concentration</code></pre>
<p>The peptides annotations are available for in the corresponding
<code>rowData</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="sec-basics.html#cb18-1" tabindex="-1"></a><span class="fu">rowData</span>(spikein[[<span class="st">"peptides"</span>]])</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="sec-basics.html#cb19-1" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">rowData</span>(spikein[[<span class="st">"peptides"</span>]])))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="2%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="6%">
<col width="3%">
<col width="0%">
<col width="0%">
<col width="0%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Sequence</th>
<th align="left">N.term.cleavage.window</th>
<th align="left">C.term.cleavage.window</th>
<th align="left">Amino.acid.before</th>
<th align="left">First.amino.acid</th>
<th align="left">Second.amino.acid</th>
<th align="left">Second.last.amino.acid</th>
<th align="left">Last.amino.acid</th>
<th align="left">Amino.acid.after</th>
<th align="right">A.Count</th>
<th align="right">R.Count</th>
<th align="right">N.Count</th>
<th align="right">D.Count</th>
<th align="right">C.Count</th>
<th align="right">Q.Count</th>
<th align="right">E.Count</th>
<th align="right">G.Count</th>
<th align="right">H.Count</th>
<th align="right">I.Count</th>
<th align="right">L.Count</th>
<th align="right">K.Count</th>
<th align="right">M.Count</th>
<th align="right">F.Count</th>
<th align="right">P.Count</th>
<th align="right">S.Count</th>
<th align="right">T.Count</th>
<th align="right">W.Count</th>
<th align="right">Y.Count</th>
<th align="right">V.Count</th>
<th align="right">U.Count</th>
<th align="right">Length</th>
<th align="right">Missed.cleavages</th>
<th align="right">Mass</th>
<th align="left">Proteins</th>
<th align="left">Leading.razor.protein</th>
<th align="right">Start.position</th>
<th align="right">End.position</th>
<th align="left">Unique..Groups.</th>
<th align="left">Unique..Proteins.</th>
<th align="left">Charges</th>
<th align="right">PEP</th>
<th align="right">Score</th>
<th align="left">Identification.type.6A_1</th>
<th align="left">Identification.type.6A_2</th>
<th align="left">Identification.type.6A_3</th>
<th align="left">Identification.type.6A_4</th>
<th align="left">Identification.type.6A_5</th>
<th align="left">Identification.type.6A_6</th>
<th align="left">Identification.type.6A_7</th>
<th align="left">Identification.type.6A_8</th>
<th align="left">Identification.type.6A_9</th>
<th align="left">Identification.type.6B_1</th>
<th align="left">Identification.type.6B_2</th>
<th align="left">Identification.type.6B_3</th>
<th align="left">Identification.type.6B_4</th>
<th align="left">Identification.type.6B_5</th>
<th align="left">Identification.type.6B_6</th>
<th align="left">Identification.type.6B_7</th>
<th align="left">Identification.type.6B_8</th>
<th align="left">Identification.type.6B_9</th>
<th align="left">Identification.type.6C_1</th>
<th align="left">Identification.type.6C_2</th>
<th align="left">Identification.type.6C_3</th>
<th align="left">Identification.type.6C_4</th>
<th align="left">Identification.type.6C_5</th>
<th align="left">Identification.type.6C_6</th>
<th align="left">Identification.type.6C_7</th>
<th align="left">Identification.type.6C_8</th>
<th align="left">Identification.type.6C_9</th>
<th align="left">Identification.type.6D_1</th>
<th align="left">Identification.type.6D_2</th>
<th align="left">Identification.type.6D_3</th>
<th align="left">Identification.type.6D_4</th>
<th align="left">Identification.type.6D_5</th>
<th align="left">Identification.type.6D_6</th>
<th align="left">Identification.type.6D_7</th>
<th align="left">Identification.type.6D_8</th>
<th align="left">Identification.type.6D_9</th>
<th align="left">Identification.type.6E_1</th>
<th align="left">Identification.type.6E_2</th>
<th align="left">Identification.type.6E_3</th>
<th align="left">Identification.type.6E_4</th>
<th align="left">Identification.type.6E_5</th>
<th align="left">Identification.type.6E_6</th>
<th align="left">Identification.type.6E_7</th>
<th align="left">Identification.type.6E_8</th>
<th align="left">Identification.type.6E_9</th>
<th align="right">Experiment.6A_1</th>
<th align="right">Experiment.6A_2</th>
<th align="right">Experiment.6A_3</th>
<th align="right">Experiment.6A_4</th>
<th align="right">Experiment.6A_5</th>
<th align="right">Experiment.6A_6</th>
<th align="right">Experiment.6A_7</th>
<th align="right">Experiment.6A_8</th>
<th align="right">Experiment.6A_9</th>
<th align="right">Experiment.6B_1</th>
<th align="right">Experiment.6B_2</th>
<th align="right">Experiment.6B_3</th>
<th align="right">Experiment.6B_4</th>
<th align="right">Experiment.6B_5</th>
<th align="right">Experiment.6B_6</th>
<th align="right">Experiment.6B_7</th>
<th align="right">Experiment.6B_8</th>
<th align="right">Experiment.6B_9</th>
<th align="right">Experiment.6C_1</th>
<th align="right">Experiment.6C_2</th>
<th align="right">Experiment.6C_3</th>
<th align="right">Experiment.6C_4</th>
<th align="right">Experiment.6C_5</th>
<th align="right">Experiment.6C_6</th>
<th align="right">Experiment.6C_7</th>
<th align="right">Experiment.6C_8</th>
<th align="right">Experiment.6C_9</th>
<th align="right">Experiment.6D_1</th>
<th align="right">Experiment.6D_2</th>
<th align="right">Experiment.6D_3</th>
<th align="right">Experiment.6D_4</th>
<th align="right">Experiment.6D_5</th>
<th align="right">Experiment.6D_6</th>
<th align="right">Experiment.6D_7</th>
<th align="right">Experiment.6D_8</th>
<th align="right">Experiment.6D_9</th>
<th align="right">Experiment.6E_1</th>
<th align="right">Experiment.6E_2</th>
<th align="right">Experiment.6E_3</th>
<th align="right">Experiment.6E_4</th>
<th align="right">Experiment.6E_5</th>
<th align="right">Experiment.6E_6</th>
<th align="right">Experiment.6E_7</th>
<th align="right">Experiment.6E_8</th>
<th align="right">Experiment.6E_9</th>
<th align="right">Intensity</th>
<th align="left">Reverse</th>
<th align="left">Potential.contaminant</th>
<th align="right">id</th>
<th align="left">Protein.group.IDs</th>
<th align="left">Mod..peptide.IDs</th>
<th align="left">Evidence.IDs</th>
<th align="left">MS.MS.IDs</th>
<th align="right">Best.MS.MS</th>
<th align="left">Oxidation..M..site.IDs</th>
<th align="right">MS.MS.Count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AAAAGAGGAGDSGDAVTK</td>
<td align="left">AAAAGAGGAGDSGDAVTK</td>
<td align="left">EHQHDEQKAAAAGAGG</td>
<td align="left">DSGDAVTKIGSEDVKL</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">T</td>
<td align="left">K</td>
<td align="left">I</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">18</td>
<td align="right">0</td>
<td align="right">1445.6746</td>
<td align="left">sp|P38915|SPT8_YEAST</td>
<td align="left">sp|P38915|SPT8_YEAST</td>
<td align="right">97</td>
<td align="right">114</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000118</td>
<td align="right">82.942</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1190800</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
<td align="left">859</td>
<td align="left">0</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18;19;20;21;22;23</td>
<td align="left">0;1;2;3;4;5;6;7;8;9</td>
<td align="right">0</td>
<td align="left"></td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">AAAALAGGK</td>
<td align="left">AAAALAGGK</td>
<td align="left">QQLSKAAKAAAALAGG</td>
<td align="left">AAALAGGKKSKKKWSK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">728.4181</td>
<td align="left">sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST</td>
<td align="left">sp|Q3E792|RS25A_YEAST</td>
<td align="right">13</td>
<td align="right">21</td>
<td align="left">yes</td>
<td align="left">no</td>
<td align="left">2</td>
<td align="right">0.0000075</td>
<td align="right">134.810</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">280990000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
<td align="left">230</td>
<td align="left">1</td>
<td align="left">24;25;26;27;28;29;30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73</td>
<td align="left">10;11;12;13;14;15;16;17;18;19;20;21;22;23;24;25;26;27;28;29</td>
<td align="right">21</td>
<td align="left"></td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">AAAALAGGKK</td>
<td align="left">AAAALAGGKK</td>
<td align="left">QQLSKAAKAAAALAGG</td>
<td align="left">AALAGGKKSKKKWSKK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="left">S</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">856.5131</td>
<td align="left">sp|Q3E792|RS25A_YEAST;sp|P0C0T4|RS25B_YEAST</td>
<td align="left">sp|Q3E792|RS25A_YEAST</td>
<td align="right">13</td>
<td align="right">22</td>
<td align="left">yes</td>
<td align="left">no</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="right">143.730</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">33360000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">2</td>
<td align="left">230</td>
<td align="left">2</td>
<td align="left">74;75;76;77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98</td>
<td align="left">30;31;32;33;34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50</td>
<td align="right">31</td>
<td align="left"></td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">AAADALSDLEIK</td>
<td align="left">AAADALSDLEIK</td>
<td align="left">MPKETPSKAAADALSD</td>
<td align="left">ALSDLEIKDSKSNLNK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">I</td>
<td align="left">K</td>
<td align="left">D</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1215.6347</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="right">9</td>
<td align="right">20</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="right">182.230</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">54622000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">3</td>
<td align="left">229</td>
<td align="left">3</td>
<td align="left">99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143</td>
<td align="left">51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79;80;81;82;83;84</td>
<td align="right">72</td>
<td align="left"></td>
<td align="right">29</td>
</tr>
<tr class="odd">
<td align="left">AAADALSDLEIKDSK</td>
<td align="left">AAADALSDLEIKDSK</td>
<td align="left">MPKETPSKAAADALSD</td>
<td align="left">DLEIKDSKSNLNKELE</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">S</td>
<td align="left">K</td>
<td align="left">S</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">1545.7886</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="left">sp|P09938|RIR2_YEAST</td>
<td align="right">9</td>
<td align="right">23</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">3</td>
<td align="right">0.0001532</td>
<td align="right">73.927</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By MS/MS</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">18910000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">4</td>
<td align="left">229</td>
<td align="left">4</td>
<td align="left">144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172;173;174;175;176;177;178;179;180;181;182;183;184</td>
<td align="left">85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116</td>
<td align="right">94</td>
<td align="left"></td>
<td align="right">32</td>
</tr>
<tr class="even">
<td align="left">AAAEEFQR</td>
<td align="left">AAAEEFQR</td>
<td align="left">RERKEREKAAAEEFQR</td>
<td align="left">AAAEEFQRQQELLRQQ</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">Q</td>
<td align="left">R</td>
<td align="left">Q</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">920.4352</td>
<td align="left">sp|P53075|SHE10_YEAST</td>
<td align="left">sp|P53075|SHE10_YEAST</td>
<td align="right">538</td>
<td align="right">545</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0175880</td>
<td align="right">79.474</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By MS/MS</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="left">By matching</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1158600</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">5</td>
<td align="left">1143</td>
<td align="left">5</td>
<td align="left">185;186;187;188;189;190;191;192;193;194</td>
<td align="left">117;118</td>
<td align="right">117</td>
<td align="left"></td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<p>We can also retrieve the quantitative values for each set using
<code>assays()</code>.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="sec-basics.html#cb20-1" tabindex="-1"></a><span class="fu">assay</span>(spikein[[<span class="dv">1</span>]])[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##                    Intensity.6A_1 Intensity.6A_2 Intensity.6A_3 Intensity.6A_4 Intensity.6A_5
## AAAAGAGGAGDSGDAVTK              0          76034              0          56831          83835
## AAAALAGGK                       0       19017000       17021000       16406000          52547
## AAAALAGGKK                      0        5074300              0        4330300              0
## AAADALSDLEIK              1111200        1578700        1311400        1277500        2192900
## AAADALSDLEIKDSK             78739          61204              0         667190        1035200</code></pre>
</div>
</div>
<div id="sec-basic_preprocess" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Data preprocessing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-basic_preprocess')" onmouseout="reset_tooltip('sec-basic_preprocess-tooltip')"><span class="tooltiptext" id="sec-basic_preprocess-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Since we have a <code>QFeatures</code> object, we can directly make use of
<code>QFeatures</code>’ data preprocessing functionality<label for="tufte-sn-7" class="margin-toggle sidenote-number">7</label><input type="checkbox" id="tufte-sn-7" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">7</span> see also the <a href="https://rformassspectrometry.github.io/book/sec-quant.html">R for
mass spectrometry
book</a></span>. A
major advantage of <code>QFeatures</code> is it provides the power to build
highly modular workflows, where each step is carried out by a
dedicated function with a large choice of available methods and
parameters. This means that users can adapt the workflow to their
specific use case and their specific needs.</p>
<div id="sec-encode_missing" class="section level3" number="2.4.1">
<h3>
<span class="header-section-number">2.4.1</span> Encoding missing values<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-encode_missing')" onmouseout="reset_tooltip('sec-encode_missing-tooltip')"><span class="tooltiptext" id="sec-encode_missing-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The first preprocessing step is to correctly encode the missing
values. It is important that missing values are encoded using <code>NA</code>.
For instance, non-observed values should not be
encoded with a zero because true zeros (the proteomic feature
is absent from the sample) cannot be distinguished from technical
zeros (the feature was missed by the instrument or could not be
identified). We therefore replace any zero in the quantitative data
with an <code>NA</code>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="sec-basics.html#cb22-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(spikein, <span class="fu">names</span>(spikein))</span></code></pre></div>
<p>Note that <code>msqrob2</code> can handle missing data without having to rely on
hard-to-verify imputation assumptions. However, <code>msqrob2</code> does not
prevent users from using imputation, which can be performed with
<code>impute()</code> from the <code>QFeatures</code> package. Below, we show how one could
perform KNN imputation<label for="tufte-sn-8" class="margin-toggle sidenote-number">8</label><input type="checkbox" id="tufte-sn-8" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">8</span> We will however not evaluate the code in this
tutorial to exclude imputation from the current workflow.</span> (see
<code>?impute</code> for more options).</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="sec-basics.html#cb23-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">impute</span>(</span>
<span id="cb23-2"><a href="sec-basics.html#cb23-2" tabindex="-1"></a>  spikein, <span class="at">method =</span> <span class="st">"knn"</span>,</span>
<span id="cb23-3"><a href="sec-basics.html#cb23-3" tabindex="-1"></a>  <span class="at">i =</span> <span class="fu">names</span>(spikein), <span class="at">name =</span> <span class="fu">paste0</span>(<span class="fu">names</span>(spikein), <span class="st">"_imput"</span>)</span>
<span id="cb23-4"><a href="sec-basics.html#cb23-4" tabindex="-1"></a>)</span></code></pre></div>
<p><strong>TODO</strong> mention the hurdle model here?</p>
</div>
<div id="sec-log2" class="section level3" number="2.4.2">
<h3>
<span class="header-section-number">2.4.2</span> Log2 transformation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-log2')" onmouseout="reset_tooltip('sec-log2-tooltip')"><span class="tooltiptext" id="sec-log2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We need to log2 transform the intensities. We illustrate the rationale
using the spike-in data. For this, we perform a short data
manipulation pipeline:</p>
<ol style="list-style-type: decimal">
<li>We use <code>longForm()</code> to convert the <code>QFeatures</code> object into a long
table, where each row contains the quantitative information about
one observation, in which column, row and set it was found. Long
tables are particularly useful for manipulating data with the
<code>tidyverse</code> ecosystem, namely with <code>ggplot2</code> for visualisation.
<code>longForm()</code> also allows to include annotations, and we here
include <code>Mixture</code> and <code>TechRepMixture</code> for filtering and colouring.</li>
<li>
<code>longForm()</code> returns a <code>DataFrame</code> which we convert to a
<code>data.frame</code>.</li>
<li>We filter the data to keep only the data from AALEELVK (mapping to
a UPS1 protein) measured in lab1.</li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="sec-basics.html#cb24-1" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">longForm</span>(spikein, <span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein))) <span class="sc">|&gt;</span> <span class="do">## 1.</span></span>
<span id="cb24-2"><a href="sec-basics.html#cb24-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span> <span class="do">## 2.</span></span>
<span id="cb24-3"><a href="sec-basics.html#cb24-3" tabindex="-1"></a>  <span class="fu">filter</span>(rowname <span class="sc">==</span> <span class="st">"AALEELVK"</span> <span class="sc">&amp;</span> lab <span class="sc">==</span> <span class="st">"lab1"</span>) <span class="do">## 3.</span></span></code></pre></div>
<p>Next, we visualise the data using <code>ggplot2</code>. We will show the
intensities and log2-intensities for one of the peptides, AALEELVK, in
function of the UPS spike-in concentration. We first define a common
plot from which we generate two plots, one without and the other with
log2 transformation of the quantitative values.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="sec-basics.html#cb25-1" tabindex="-1"></a><span class="do">## Common plot</span></span>
<span id="cb25-2"><a href="sec-basics.html#cb25-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat) <span class="sc">+</span></span>
<span id="cb25-3"><a href="sec-basics.html#cb25-3" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> concentration) <span class="sc">+</span></span>
<span id="cb25-4"><a href="sec-basics.html#cb25-4" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb25-5"><a href="sec-basics.html#cb25-5" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log2"</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="sec-basics.html#cb25-6" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Spike-in concentration (fmol/µL)"</span>)</span>
<span id="cb25-7"><a href="sec-basics.html#cb25-7" tabindex="-1"></a><span class="do">## Add the y axis as linear or log2-transformed quant values</span></span>
<span id="cb25-8"><a href="sec-basics.html#cb25-8" tabindex="-1"></a>p <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb25-9"><a href="sec-basics.html#cb25-9" tabindex="-1"></a>  p <span class="sc">+</span> <span class="fu">aes</span>(<span class="at">y =</span> <span class="fu">log2</span>(value)) <span class="sc">+</span></span>
<span id="cb25-10"><a href="sec-basics.html#cb25-10" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">"UPS1 peptide AALEELVK"</span>) <span class="sc">+</span></span>
<span id="cb25-11"><a href="sec-basics.html#cb25-11" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">axis_titles =</span> <span class="st">"collect_x"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-81-1.png" width="672"></p>
<p>We can see that the variation around the mean for each dilution factor
increases as the mean increases. This is known as heteroskedasticity.
We will later see that <code>msqrob2</code> assumes that variance of the error
should be consistent across the different conditions. Upon,
log2-transformation we can see that the problem of unequal variation
is solved.</p>
<p>Another advantages of log2-transformation is that it provides a scale
that directly relates to biological interpretation. In biology, a
change induced by some condition often results in a fold change in
concentration. Interestingly, the log2 fold change (logFC), which is
the log of the ratio between two conditions, is identical to the
difference between the log of each condition.</p>
<p><span class="math display">\[log_2FC_{B-A} = log_2B - log_2A = log_2 \frac{B}{A}\]</span></p>
<p>This simplifies the modelling since the effects are now additive and
it simplifies the interpretration since a logFC of 1 means that the
abundance in B are <span class="math inline">\(2\times\)</span> higher in <span class="math inline">\(B\)</span> than in <span class="math inline">\(A\)</span>, a logFC of 2
means an increase of <span class="math inline">\(4\times\)</span>, for instance.</p>
<p>We perform log2-transformation with <code>logTransform()</code> from the
<code>QFeatures</code> package.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="sec-basics.html#cb26-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(</span>
<span id="cb26-2"><a href="sec-basics.html#cb26-2" tabindex="-1"></a>  spikein, <span class="st">"peptides"</span>, <span class="at">name =</span> <span class="st">"peptides_log"</span>, <span class="at">base =</span> <span class="dv">2</span></span>
<span id="cb26-3"><a href="sec-basics.html#cb26-3" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="sec-filter" class="section level3" number="2.4.3">
<h3>
<span class="header-section-number">2.4.3</span> Peptide filtering<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-filter')" onmouseout="reset_tooltip('sec-filter-tooltip')"><span class="tooltiptext" id="sec-filter-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Filtering removes low-quality and unreliable peptides that would
otherwise introduce noise and artefacts in the data. There are many
possible criteria for peptide filtering:</p>
<ul>
<li>Reverse sequences</li>
<li>Only identified by modification site (only modified peptides
detected)</li>
<li>Razor peptides: non-unique peptides assigned to the protein group
with the most other peptides</li>
<li>Contaminants</li>
<li>Peptides few identifications</li>
<li>Proteins that are only identified with one or a few peptides</li>
<li>FDR of identification</li>
<li>…</li>
</ul>
<p>The filtering will not induce bias as long as the criterion is
independent from the downstream data analysis.</p>
<p><strong>TODO</strong> the sentence above is taken from the PDA course, but I should
clarify this a bit further.</p>
<p>We use <code>filterFeatures()</code> to perform the filtering. It uses
information from the <code>rowData</code> and a formula to generate a filter for
each feature (row) in each set across the object. If the filter
returns <code>TRUE</code>, the corresponding row is retained, otherwise it is
removed. Defining a filter through a formula offers a flexible
approach, allowing for any customised filter. This dataset requires an
extensive PSM filtering which is an ideal use case to demonstrate the
customisation of a filtering workflow.</p>
<div id="remove-failed-protein-inference" class="section level4" number="2.4.3.1">
<h4>
<span class="header-section-number">2.4.3.1</span> Remove failed protein inference</h4>
<p>We remove peptides that could not be mapped to a protein. We also
remove peptides that cannot be uniquely mapped to a single proteins
because its sequence is shared across multiple proteins. This often
occurs for homologs or for proteins that share similar functional
domains. Shared peptides are often mapped to protein groups, which are
the set of proteins that share the given peptides. Protein groups are
encoded by appending the constituting protein identifiers, separated
by a <code>";"</code>. While these peptides may serve as interesting QC controls, the</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="sec-basics.html#cb27-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(</span>
<span id="cb27-2"><a href="sec-basics.html#cb27-2" tabindex="-1"></a>  spikein, <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb27-3"><a href="sec-basics.html#cb27-3" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
</div>
<div id="remove-reverse-sequences-and-contaminants" class="section level4" number="2.4.3.2">
<h4>
<span class="header-section-number">2.4.3.2</span> Remove reverse sequences and contaminants</h4>
<p>We now remove the peptides that map to decoy sequences or to
contaminants proteins. Decoy peptides, which consist of fake peptide
sequences generated by reversing a real amino-acid sequence, are used
during the database search to mitigating the number of random-hit PSM,
hence mitigating misidentification. Contaminant proteins are proteins
known to be artificially incorporated during the experiment, but that
are irrelevant to the biological system under study. Typical
contaminants are human or animal keratins (deposited by the
experimenter) or trypsin (used for protein digestion). It is important
to remove these peptides before performing statistical modelling,
otherwise these proteins will be accounted for during the multiple
test adjustment which will cause an unnecessary decrease of
statistical power.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="sec-basics.html#cb28-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(spikein, <span class="sc">~</span> Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> <span class="do">## Remove decoys</span></span>
<span id="cb28-2"><a href="sec-basics.html#cb28-2" tabindex="-1"></a>                       Potential.contaminant <span class="sc">!=</span> <span class="st">"+"</span>) <span class="do">## Remove contaminants</span></span></code></pre></div>
</div>
<div id="remove-highly-missing-peptides" class="section level4" number="2.4.3.3">
<h4>
<span class="header-section-number">2.4.3.3</span> Remove highly missing peptides</h4>
<p>We keep peptides that were observed at last 20 times out of the <span class="math inline">\(n
= 45\)</span> samples. We tolerate the following proportion of NAs:
<span class="math inline">\(\text{pNA} = \frac{(n - 20)}{n} = 0.556\)</span>, so we keep peptides that
are at most missing in about half of the samples. This is an arbitrary
value that may need to be adjusted depending on the experiment and the
data set.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="sec-basics.html#cb29-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb29-2"><a href="sec-basics.html#cb29-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(spikein[[<span class="st">"peptides_log"</span>]])</span>
<span id="cb29-3"><a href="sec-basics.html#cb29-3" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(spikein, <span class="at">i =</span> <span class="st">"peptides_log"</span>, <span class="at">pNA =</span> (n <span class="sc">-</span> nObs) <span class="sc">/</span> n))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 2 sets:
## 
##  [1] peptides: SummarizedExperiment with 10393 rows and 45 columns 
##  [2] peptides_log: SummarizedExperiment with 6494 rows and 45 columns</code></pre>
<p>We keep 6494 peptides upon filtering.</p>
</div>
</div>
<div id="sec-norm" class="section level3" number="2.4.4">
<h3>
<span class="header-section-number">2.4.4</span> Normalisation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-norm')" onmouseout="reset_tooltip('sec-norm-tooltip')"><span class="tooltiptext" id="sec-norm-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The most common objective of MS-based proteomics experiments is to
understand the biological changes in protein abundance between
experimental conditions. However, changes in measurements between
groups can be caused due to technical factors. For instance, there are
systematic fluctuations from run-to-run that shift the measured
intensity distribution. We can this explore as follows:</p>
<ol style="list-style-type: decimal">
<li>We extract the sets containing the log transformed data. This is
performed using <code>QFeatures</code>’ 3-way subsetting<label for="tufte-sn-9" class="margin-toggle sidenote-number">9</label><input type="checkbox" id="tufte-sn-9" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">9</span> We will explain this
with more details at the end of the <a href="sec-basics.html#sec-summarisation">summarisation step</a></span>.</li>
<li>We use <code>longForm()</code> to convert the <code>QFeatures</code> object into a long
table, including <code>lab</code> and <code>concentration</code> for filtering and
colouring.</li>
<li>To facilitate interpretation, we filter the data to keep only one
of the UPS1 spike-in concentrations.</li>
<li>We visualise the density of the quantitative values within each
sample. We colour each sample based on its corresponding lab.
Remember that each spike-in concentration has been acquired in
triplicate for each lab.</li>
</ol>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="sec-basics.html#cb31-1" tabindex="-1"></a>spikein[, , <span class="st">"peptides_log"</span>] <span class="sc">|&gt;</span> <span class="do">## 1.</span></span>
<span id="cb31-2"><a href="sec-basics.html#cb31-2" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">c</span>(<span class="st">"lab"</span>, <span class="st">"concentration"</span>)) <span class="sc">|&gt;</span> <span class="do">## 2.</span></span>
<span id="cb31-3"><a href="sec-basics.html#cb31-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb31-4"><a href="sec-basics.html#cb31-4" tabindex="-1"></a>  <span class="fu">filter</span>(concentration <span class="sc">==</span> <span class="fl">6.67</span>) <span class="sc">|&gt;</span> <span class="do">## 3.</span></span>
<span id="cb31-5"><a href="sec-basics.html#cb31-5" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="do">## 4.</span></span>
<span id="cb31-6"><a href="sec-basics.html#cb31-6" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> value,</span>
<span id="cb31-7"><a href="sec-basics.html#cb31-7" tabindex="-1"></a>      <span class="at">colour =</span> lab,</span>
<span id="cb31-8"><a href="sec-basics.html#cb31-8" tabindex="-1"></a>      <span class="at">group =</span> colname) <span class="sc">+</span></span>
<span id="cb31-9"><a href="sec-basics.html#cb31-9" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb31-10"><a href="sec-basics.html#cb31-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Intensity distribution for all samples with 6.667 fmol/µL UPS1"</span>,</span>
<span id="cb31-11"><a href="sec-basics.html#cb31-11" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Before normalisation"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-86-1.png" width="672"></p>
<p>Even in this clean synthetic data set (same background, only 48 UPS
proteins can be different), the marginal peptide intensity
distributions across samples are not well aligned. The intensity
distributions seem to impact by within and between lab variability.
Since we show the data for a single experimental condition, we know
that these difference stem from technical variability and not
biological variability. Ignoring this effect will increase the noise
and reduce the statistical power of the experiment, and may also, in
case of unbalanced designs, introduce confounding effects that will
bias the results.</p>
<p>Therefore, normalisation will transform the data to put all the
distributions on the same location, such as centering on the mean or
the median, so that the distributions better coincide and overlap.
Here, we will use <code>normalize()</code> to subtract the median intensity within
each run.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="sec-basics.html#cb32-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">normalize</span>(</span>
<span id="cb32-2"><a href="sec-basics.html#cb32-2" tabindex="-1"></a>  spikein, <span class="st">"peptides_log"</span>, <span class="at">name =</span> <span class="st">"peptides_norm"</span>,</span>
<span id="cb32-3"><a href="sec-basics.html#cb32-3" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"center.median"</span></span>
<span id="cb32-4"><a href="sec-basics.html#cb32-4" tabindex="-1"></a>)</span></code></pre></div>
<p>Formally, the function applies the follow operation on each sample <span class="math inline">\(i\)</span>
across all PSMs <span class="math inline">\(p\)</span>:</p>
<p><span class="math display">\[
y_{ip}^{\text{norm}} = y_{ip} - \hat{\mu}_i
\]</span></p>
<p>with <span class="math inline">\(\hat{\mu}_i\)</span> the median intensity over all observed peptides in
sample <span class="math inline">\(i\)</span>. Upon normalisation, we can see that the distribution
nicely overlap (using the same code as above)</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="sec-basics.html#cb33-1" tabindex="-1"></a>spikein[, , <span class="st">"peptides_norm"</span>] <span class="sc">|&gt;</span> <span class="do">## 1.</span></span>
<span id="cb33-2"><a href="sec-basics.html#cb33-2" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">c</span>(<span class="st">"lab"</span>, <span class="st">"concentration"</span>)) <span class="sc">|&gt;</span> <span class="do">## 2.</span></span>
<span id="cb33-3"><a href="sec-basics.html#cb33-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb33-4"><a href="sec-basics.html#cb33-4" tabindex="-1"></a>  <span class="fu">filter</span>(concentration <span class="sc">==</span> <span class="fl">6.67</span>) <span class="sc">|&gt;</span> <span class="do">## 3.</span></span>
<span id="cb33-5"><a href="sec-basics.html#cb33-5" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="do">## 4.</span></span>
<span id="cb33-6"><a href="sec-basics.html#cb33-6" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> value,</span>
<span id="cb33-7"><a href="sec-basics.html#cb33-7" tabindex="-1"></a>      <span class="at">colour =</span> lab,</span>
<span id="cb33-8"><a href="sec-basics.html#cb33-8" tabindex="-1"></a>      <span class="at">group =</span> colname) <span class="sc">+</span></span>
<span id="cb33-9"><a href="sec-basics.html#cb33-9" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb33-10"><a href="sec-basics.html#cb33-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Intensity distribution for all samples with 6.667 fmol/µL UPS1"</span>,</span>
<span id="cb33-11"><a href="sec-basics.html#cb33-11" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"After normalisation"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-88-1.png" width="672"></p>
<p>Beware there exist numerous types of normalisation methods (see
<code>?normalize</code>) and which method to use may be data set dependent. For
instance, some data set may show low overlap of distribution tails
upon normalisation indicating that a simple shift is not sufficient.
In micro-array literature, quantile normalisation is used to force the
median and all other quantiles to be equal across samples, but in
proteomics, quantile normalisation often introduces artifacts due to a
difference in missing peptides across samples (c.f. <a href="sec-basics.html#sec-lfq_challenges">Challenges</a>).</p>
<p>It is important to understand that most normalisation procedures
assume that the majority of the proteins do not change across
conditions and only a small proportion of the proteins are
differentially abundant. This assumption may not be valid in poorly
designed spike-in studies <span class="citation">(<label for="tufte-mn-9" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-9" class="margin-toggle">O’Brien et al. 2024<span class="marginnote">O’Brien, Jonathon J, Anil Raj, Aleksandr Gaun, Adam Waite, Wenzhou Li, David G Hendrickson, Niclas Olsson, and Fiona E McAllister. 2024. <span>“A Data Analysis Framework for Combining Multiple Batches Increases the Power of Isobaric Proteomics Experiments.”</span> <em>Nat. Methods</em> 21 (2): 290–300.</span>)</span> or for pull-down studies,
for example. Dedicated normalisation strategies are then required.</p>
</div>
<div id="sec-summarisation" class="section level3" number="2.4.5">
<h3>
<span class="header-section-number">2.4.5</span> Summarisation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-summarisation')" onmouseout="reset_tooltip('sec-summarisation-tooltip')"><span class="tooltiptext" id="sec-summarisation-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The objective of summarisation (also referred to as aggregation) is to
summarise the peptide-level intensities into a protein expression value.
We illustrate the motivation behind summarisation using the data for
one of the UPS proteins in Mixture 1 (separating the data for each
technical replicate). We also focus on the 0.125x and the 1x
spike-in conditions. We illustrate the different peptide ions on the
x axis and plot the log2 normalised intensities across samples on y
axis. All the points belonging to the same sample are linked through a
grey line.</p>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-89-1.png" width="576"></p>
<ol style="list-style-type: decimal">
<li>We can see that data for a protein can consist of many peptides,
hence the need for summarisation.</li>
<li>We observe that different peptides have different intensities
within the same sample (same line). This is because different
peptides have different properties and hence different ionisation
properties which will influence the detectability in the MS
(<a href="sec-basics.html#sec-lfq_challenges">Challenges</a>).</li>
<li>The peptide identification are inconsistent between groups of
interest. We can see the low-concentration group (condition A, red)
display more missing values than the high-concentration group
(condition B, blue). Moreover, which value is missing depends on
the peptide intensities. All the peptides found in the
low-concentration group are the most intense peptides in the high
concentration group.</li>
<li>We also find outliers. For instance, the third peptide
(AQLGPDESKQK) is detected in three samples, but one of the two
samples is strongly offset compared to the two others. These
outliers can be the result of misidentification or fluctuations
during MS acquisition.</li>
<li>These data also suggest pseudo-replication. The peptide intensities
in one same sample (dots connected by a line) are correlated, i.e.
they more alike than the peptide intensities between samples
(the lines do not perfectly align).</li>
</ol>
<p>The fact that different peptides have different intensities (2.), may
be inconsistently identified (3.) and/or quantified (4.), and show
sample correlations (5.) can lead to bias if we use simple
summarisation approaches such as summing or averaging the peptide
intensities for each sample. Instead, we will resort to more advanced
summarisation approaches to accommodate for these issues.</p>
<p>Here, we summarise the peptide-level data into protein intensities
through the robust summarisation approach <span class="citation">(<label for="tufte-mn-10" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-10" class="margin-toggle">Sticker et al. 2020<span class="marginnote">Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven Clement. 2020. <span>“Robust Summarization and Inference in Proteome-Wide Label-Free Quantification.”</span> <em>Mol. Cell. Proteomics</em> 19 (7): 1209–19.</span>)</span> which is a
model approach that estimates for each protein <span class="math inline">\(P\)</span> separately:</p>
<p><span class="math display">\[
y_{ip} = \beta_p^\text{pep} + \beta_i^\text{samp} + \epsilon_{ip}
\]</span></p>
<p>where <span class="math inline">\(y_{ip}\)</span> is the log-normalised peptide intensity for peptide <span class="math inline">\(p\)</span>
belonging to protein <span class="math inline">\(P\)</span> in sample <span class="math inline">\(i\)</span>. <span class="math inline">\(\beta_p^\text{pep}\)</span> is the
average effect of peptide <span class="math inline">\(p\)</span>, which account for the fact that
different peptide yield different baseline intensities (see issue 2.
above). <span class="math inline">\(\beta_i^\text{samp}\)</span> is the average effect of sample <span class="math inline">\(i\)</span>. In
other words, it provides the estimated intensity of protein <span class="math inline">\(P\)</span> in
sample <span class="math inline">\(i\)</span> which will be used as the summarised protein value.
<span class="math inline">\(\epsilon_{ip}\)</span> is the residual effect that cannot be explained by the
average sample and peptide effects. The method is called robust
summarisation because the estimation process will minimize <code>\epsilon</code>
using a robust estimator<label for="tufte-sn-10" class="margin-toggle sidenote-number">10</label><input type="checkbox" id="tufte-sn-10" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">10</span> During robust estimation, the least squared
error minimization, that is minimizing <span class="math inline">\(\sum\mathbf{\epsilon}^2\)</span>, is
iteratively reweighted using Huber weights. This is internally
performed using <code>MASS::rlm()</code>.</span> that will down-weigh extreme values,
effectively tackling issue 4.</p>
<p><code>aggregateFeatures()</code> streamlines summarisation. It requires the name
of a <code>rowData</code> column to group the peptides into proteins (or
protein groups), here <code>Proteins</code>. We provide the summarisation
approach through th <code>fun</code> argument. Other summarisation methods
are available from the <code>MsCoreUtils</code> package, see <code>?aggregateFeatures</code>
for a comprehensive list. The function will return a <code>QFeatures</code>
object with a new set that we called <code>proteins</code>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="sec-basics.html#cb34-1" tabindex="-1"></a>(spikein <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb34-2"><a href="sec-basics.html#cb34-2" tabindex="-1"></a>  spikein, <span class="at">i =</span> <span class="st">"peptides_norm"</span>, </span>
<span id="cb34-3"><a href="sec-basics.html#cb34-3" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">"proteins"</span>,</span>
<span id="cb34-4"><a href="sec-basics.html#cb34-4" tabindex="-1"></a>  <span class="at">fcol =</span> <span class="st">"Proteins"</span>, </span>
<span id="cb34-5"><a href="sec-basics.html#cb34-5" tabindex="-1"></a>  <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary,</span>
<span id="cb34-6"><a href="sec-basics.html#cb34-6" tabindex="-1"></a>  <span class="at">na.rm =</span> <span class="cn">TRUE</span></span>
<span id="cb34-7"><a href="sec-basics.html#cb34-7" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 4 sets:
## 
##  [1] peptides: SummarizedExperiment with 10393 rows and 45 columns 
##  [2] peptides_log: SummarizedExperiment with 6494 rows and 45 columns 
##  [3] peptides_norm: SummarizedExperiment with 6494 rows and 45 columns 
##  [4] proteins: SummarizedExperiment with 1285 rows and 45 columns</code></pre>
<p>Note that all the links between peptides and proteins are kept<label for="tufte-sn-11" class="margin-toggle sidenote-number">11</label><input type="checkbox" id="tufte-sn-11" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">11</span> In
fact, this is also true for the previous transformation where the
peptides are linked across sets.</span>. This come particularly handy when
we want to extract all the data from one protein, P12081ups for
instance. This can be performed using the 3-way indexing. Every
<code>QFeatures</code> object contains one or more sets, each characterised by
multiple rows (peptides or proteins) and multiple columns (samples).
Therefore, <code>QFeatures</code> can subset data based on one or more of these
indices, <code>data[set_k, feature_i, sample_j]</code>. The first entry will
subset particular features. This can be the name of a peptide (e.g.
<code>DAIAQELSLQR</code>) or the name of a protein (<code>P12081ups|SYHC_HUMAN_UPS</code>).
The second entry selects the samples columns of interest. The third
entry selects the sets of interest. If an entry is left blank, all the
corresponding features, samples or sets will be selected. Let’s
extract all the data (all samples and all sets) related to the protein
(<code>P12081ups|SYHC_HUMAN_UPS</code>).</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="sec-basics.html#cb36-1" tabindex="-1"></a>spikein[<span class="st">"P12081ups|SYHC_HUMAN_UPS"</span>, , ]</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 4 sets:
## 
##  [1] peptides: SummarizedExperiment with 19 rows and 45 columns 
##  [2] peptides_log: SummarizedExperiment with 19 rows and 45 columns 
##  [3] peptides_norm: SummarizedExperiment with 19 rows and 45 columns 
##  [4] proteins: SummarizedExperiment with 1 rows and 45 columns</code></pre>
<p>The function extracted one protein and its 33 related peptides. This
functionality is particularly useful for plotting and exploring the
changes induces by the data processing. For instance:</p>
<ol style="list-style-type: decimal">
<li>We use the <code>QFeatures</code> subsetting functionality to keep only the
data for protein P12081ups and the sets before and after
summarisation.</li>
<li>We use <code>longForm()</code> to convert the object for plotting.</li>
<li>We plot the intensities for each sample, facetting by set and
colouring by experimental condition, linking dots from the same
peptide or protein with a line.</li>
</ol>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="sec-basics.html#cb38-1" tabindex="-1"></a>spikein[<span class="st">"P12081ups|SYHC_HUMAN_UPS"</span>, , <span class="fu">c</span>(<span class="st">"peptides_norm"</span>, <span class="st">"proteins"</span>)] <span class="sc">|&gt;</span> <span class="do">## 1.</span></span>
<span id="cb38-2"><a href="sec-basics.html#cb38-2" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein))) <span class="sc">|&gt;</span> <span class="do">## 2.</span></span>
<span id="cb38-3"><a href="sec-basics.html#cb38-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb38-4"><a href="sec-basics.html#cb38-4" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="do">## 3.</span></span>
<span id="cb38-5"><a href="sec-basics.html#cb38-5" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> colname, </span>
<span id="cb38-6"><a href="sec-basics.html#cb38-6" tabindex="-1"></a>      <span class="at">y =</span> value,</span>
<span id="cb38-7"><a href="sec-basics.html#cb38-7" tabindex="-1"></a>      <span class="at">group =</span> rowname) <span class="sc">+</span></span>
<span id="cb38-8"><a href="sec-basics.html#cb38-8" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb38-9"><a href="sec-basics.html#cb38-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> condition)) <span class="sc">+</span></span>
<span id="cb38-10"><a href="sec-basics.html#cb38-10" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> assay) <span class="sc">+</span></span>
<span id="cb38-11"><a href="sec-basics.html#cb38-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb38-12"><a href="sec-basics.html#cb38-12" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Data for protein P12081ups before and after summarisation"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-92-1.png" width="672">
### Protein filtering</p>
<p>As a final step, we filter the protein data to keep at least 4 observed
proteins so that most proteins have at least 2 observations in each
group. So we tolerate a proportion of <span class="math inline">\(\frac{n - 4}{n}\)</span> NAs.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="sec-basics.html#cb39-1" tabindex="-1"></a>nObs <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb39-2"><a href="sec-basics.html#cb39-2" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">ncol</span>(spikein[[<span class="st">"proteins"</span>]])</span>
<span id="cb39-3"><a href="sec-basics.html#cb39-3" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">filterNA</span>(spikein, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">pNA =</span> (n <span class="sc">-</span> nObs) <span class="sc">/</span> n)</span></code></pre></div>
<p><strong>TODO</strong> update nObs as it makes no sense when considering all
comparisons and all labs</p>
<p>The data processing is complete.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="sec-basics.html#cb40-1" tabindex="-1"></a><span class="fu">plot</span>(spikein)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-94-1.png" width="672"></p>
</div>
</div>
<div id="sec_data_exploration" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Data exploration<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec_data_exploration')" onmouseout="reset_tooltip('sec_data_exploration-tooltip')"><span class="tooltiptext" id="sec_data_exploration-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Data exploration aims to highlight the main sources of variation in
the data prior to data modelling and can pinpoint to outlying or
off-behaving samples. A common approach for data exploration is to
perform dimension reduction, such as Multi Dimensional Scaling (MDS).
We will first extract the set to explore along the sample annotations
(used for plot colouring).</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="sec-basics.html#cb41-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(spikein, <span class="st">"proteins"</span>)</span></code></pre></div>
<p>We then use the <code>scater</code> package to compute and plot the PCA. For
technical reasons, it requires <code>SingleCellExperiment</code> class object,
but these can easily be generated from a <code>SummarizedExperiment</code>
object.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="sec-basics.html#cb42-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb42-2"><a href="sec-basics.html#cb42-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">runMDS</span>(<span class="fu">as</span>(se, <span class="st">"SingleCellExperiment"</span>), <span class="at">exprs_values =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>We can now explore the data structure while colouring for factors of
interest, here <code>condition</code>, and <code>lab</code>.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="sec-basics.html#cb43-1" tabindex="-1"></a><span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"condition"</span>) <span class="sc">+</span></span>
<span id="cb43-2"><a href="sec-basics.html#cb43-2" tabindex="-1"></a>  <span class="fu">plotMDS</span>(se, <span class="at">colour_by =</span> <span class="st">"lab"</span>) </span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-97-1.png" width="768"></p>
<p>This plot reveals interesting information. First, we see there the
data contains lab effects, as the first MDS axis separates the
different labs. Second, as expected, there are condition effects as
each condition is separated by the second MDS axis. Interestingly, the
conditions are sorted by the concentration<label for="tufte-sn-12" class="margin-toggle sidenote-number">12</label><input type="checkbox" id="tufte-sn-12" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">12</span> Recall that A to E
condition imply low to high concentration.</span>. Finally, we can also spot
an issue in the data generated by lab1. Conditions A-C in lab1 are
much further away than expected from the other groups. This has been
reported as ionisation issues in the original study
<span class="citation">(<label for="tufte-mn-11" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-11" class="margin-toggle">Paulovich et al. 2010<span class="marginnote">Paulovich, Amanda G., Dean Billheimer, Amy-Joan L. Ham, Lorenzo Vega-Montoto, Paul A. Rudnick, David L. Tabb, Pei Wang, et al. 2010. <span>“Interlaboratory Study Characterizing a Yeast Performance Standard for Benchmarking LC-MS Platform Performance*.”</span> <em>Molecular &amp; Cellular Proteomics</em> 9 (2): 242–54. https://doi.org/<a href="https://doi.org/10.1074/mcp.M900222-MCP200">https://doi.org/10.1074/mcp.M900222-MCP200</a>.</span>)</span> and demonstrates the power of data exploration for
detecting data defects.</p>
</div>
<div id="sec-modelling" class="section level2" number="2.6">
<h2>
<span class="header-section-number">2.6</span> Data modelling<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-modelling')" onmouseout="reset_tooltip('sec-modelling-tooltip')"><span class="tooltiptext" id="sec-modelling-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>We model the preprocessed data to answer biologically relevant
questions. As described above, each sample has been synthetically
created to contain a constant yest background where UPS1 proteins were
spiked in at 5 different concentration. Each sample preparation has
been replicated three times and each replicate has been sent to 3
different labs. So, the research quest is “can our model retrieve the
proteins that have been spiked in across the conditions?” Moreover,
since we know the expected protein abundance in each sample, we will
also be able to assess how accurate the model can estimate the protein
changes.</p>
<div id="sources-of-variation" class="section level3" number="2.6.1">
<h3>
<span class="header-section-number">2.6.1</span> Sources of variation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sources-of-variation')" onmouseout="reset_tooltip('sources-of-variation-tooltip')"><span class="tooltiptext" id="sources-of-variation-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>In this experimental design, we have two sources of variation: lab and
spike-in concentration<label for="tufte-sn-13" class="margin-toggle sidenote-number">13</label><input type="checkbox" id="tufte-sn-13" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">13</span> Note that we have the spike-in concentration
encoded in two ways. <code>concentration</code> is encoded as a numerical value
and provides the actual concentration amount in fmol/µL. <code>condition</code>
is encoded as a factor, meaning that every concentration is regarded
as an independent group. We will use the latter encoding as most
common research questions consist of group-wise comparisons. The
modelling of concentration as a numeric will be discussed in a
<a href="sec-basics.html#sec-numeric_model">dedicated section</a></span>. The table below confirms we
have a balanced design for each condition and lab.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="sec-basics.html#cb44-1" tabindex="-1"></a><span class="fu">table</span>(<span class="at">condition =</span> spikein<span class="sc">$</span>condition, <span class="at">lab =</span> spikein<span class="sc">$</span>lab)</span></code></pre></div>
<pre><code>##          lab
## condition lab1 lab2 lab3
##         A    3    3    3
##         B    3    3    3
##         C    3    3    3
##         D    3    3    3
##         E    3    3    3</code></pre>
<p>We will model the two sources of variation:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Spike-in condition effects</strong>: we model the source of variation
induced by the experimental treatment of interest as a fixed
effect, which we consider non-random, i.e. the treatment effect is
assumed to be the same in repeated experiments, but it is unknown
and has to be estimated.</li>
<li>
<strong>Lab effects</strong>: since each replicate has been analysed by three
different labs, we need to account for potential lab effects.
Because all spike-in conditions are present in each lab, the lab
effects can be estimated using fixed effects<label for="tufte-sn-14" class="margin-toggle sidenote-number">14</label><input type="checkbox" id="tufte-sn-14" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">14</span> If this is not the
case, you will need to model the lab effect as a random effect.
This is introduced in the <a href="sec-advanced.html#sec-advanced">advanced tutorial</a></span>.
Indeed, for these designs, lab acts as a blocking variable since
all spike-in condition effects can be estimated within each lab.</li>
</ol>
<p>When modelling a typical label-free experiment at the protein level,
the model boils down to the following linear model:</p>
<p><span class="math display">\[
y_r = \mathbf{x}^T_i \boldsymbol{\beta} + \epsilon_i
\]</span></p>
<p>with <span class="math inline">\(y_i\)</span> the <span class="math inline">\(\log_2\)</span>-normalised protein intensities in sample <span class="math inline">\(i\)</span>;
<span class="math inline">\(\mathbf{x}_i\)</span> a vector with the covariate pattern for the sample
encoding the intercept, spike-in condition, and lab<label for="tufte-sn-15" class="margin-toggle sidenote-number">15</label><input type="checkbox" id="tufte-sn-15" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">15</span> Categorical
variables are encoded using dummy coding.</span>; <span class="math inline">\(\boldsymbol{\beta}\)</span> the
vector of parameters that model the association between the covariates
and the outcome; and <span class="math inline">\(\epsilon_i\)</span> the residuals reflecting variation
that is not captured by the fixed effects. Note that <span class="math inline">\(\mathbf{x}_i\)</span>
allows for a flexible parameterisation of the treatment beyond a
single covariate, i.e. including a 1 for the intercept, continuous and
categorical variables as well as their interactions. We assume the
residuals to be independent and identically distributed (i.i.d)
according to a normal distribution with zero mean and constant
variance, i.e. <span class="math inline">\(\epsilon_{r} \sim N(0,\sigma_\epsilon^2)\)</span>, that can
differ from protein to protein.</p>
<p>In R, this model is encoded using the following simple formula<label for="tufte-sn-16" class="margin-toggle sidenote-number">16</label><input type="checkbox" id="tufte-sn-16" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">16</span> We
don’t need to specify the intercept as R automatically adds it during
model estimation. you can explicitly add it using <code>~ 1 + ...</code>, you can
also suppress it using <code>~ 0 +</code>. When an intercept is included, one of
the groups is defined as the reference group and its corresponding
model parameter is absorbed in the intercept. The residuals don’t need
to be specified as they are part of the model estimation process.</span>:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="sec-basics.html#cb46-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="er">~</span> condition <span class="sc">+</span> lab</span></code></pre></div>
</div>
<div id="sec-msqrob" class="section level3" number="2.6.2">
<h3>
<span class="header-section-number">2.6.2</span> Model estimation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-msqrob')" onmouseout="reset_tooltip('sec-msqrob-tooltip')"><span class="tooltiptext" id="sec-msqrob-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We estimate the model with <code>msqrob()</code>. The function takes the
<code>QFeatures</code> object, extracts the quantitative values from the
<code>"proteins"</code> set generated during summarisation, and fits a simple
linear model with <code>condition</code> and <code>lab</code> as covariates, which are
automatically retrieved from <code>colData(spikein)</code>.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="sec-basics.html#cb47-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb47-2"><a href="sec-basics.html#cb47-2" tabindex="-1"></a>  spikein,  <span class="at">i =</span> <span class="st">"proteins"</span>,</span>
<span id="cb47-3"><a href="sec-basics.html#cb47-3" tabindex="-1"></a>  <span class="at">formula =</span> model,</span>
<span id="cb47-4"><a href="sec-basics.html#cb47-4" tabindex="-1"></a>  <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span></span>
<span id="cb47-5"><a href="sec-basics.html#cb47-5" tabindex="-1"></a>)</span></code></pre></div>
<p><strong>TODO</strong> discuss that msqrob2 assumes MCAR when correcting for
peptide-specific missingness.</p>
<p><strong>TODO</strong> need to expand more robust ridge?</p>
<p>We enable M-estimation (<code>robust = TRUE</code>) for improved robustness
against outliers. We also enable ridge regression (<code>ridge = TRUE</code>).
Ridge regression stabilises the parameter estimation of fixed effects
in case where the number of parameters are close to the number of
available samples. Keep in mind that ridge regression is irrelevant
when modelling a single fixed effect with only 2 factors.</p>
<p>The fitting results are available in the <code>msqrobModels</code> column of the
<code>rowData</code>. More specifically, the modelling output is stored in the
<code>rowData</code> as a <code>statModel</code> object, one model per row (protein). We
will see in a later section how to perform statistical inference on
the estimated parameters.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="sec-basics.html#cb48-1" tabindex="-1"></a>models <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins"</span>]])[[<span class="st">"msqrobModels"</span>]]</span>
<span id="cb48-2"><a href="sec-basics.html#cb48-2" tabindex="-1"></a>models[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>## $CON__P00761
## Object of class "StatModel"
## The type is "lmer"
## There number of elements is 5 
## 
## $CON__P02769
## Object of class "StatModel"
## The type is "lmer"
## There number of elements is 5 
## 
## $CON__P04264
## Object of class "StatModel"
## The type is "lmer"
## There number of elements is 5</code></pre>
</div>
</div>
<div id="sec-inference" class="section level2" number="2.7">
<h2>
<span class="header-section-number">2.7</span> Statistical inference<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-inference')" onmouseout="reset_tooltip('sec-inference-tooltip')"><span class="tooltiptext" id="sec-inference-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>We can now convert the biological question “does the spike-in
condition affect the protein intensities?” into a statistical
hypothesis. In other words, we must convert this question in a
combination of the model parameters, also referred to as a contrast.
To aid defining contrasts, we will visualise the experimental design
using the <code>ExploreModelMatrix</code> package.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="sec-basics.html#cb50-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb50-2"><a href="sec-basics.html#cb50-2" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(</span>
<span id="cb50-3"><a href="sec-basics.html#cb50-3" tabindex="-1"></a>  <span class="at">sampleData =</span>  <span class="fu">colData</span>(spikein),</span>
<span id="cb50-4"><a href="sec-basics.html#cb50-4" tabindex="-1"></a>  <span class="at">designFormula =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> condition,</span>
<span id="cb50-5"><a href="sec-basics.html#cb50-5" tabindex="-1"></a>  <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb50-6"><a href="sec-basics.html#cb50-6" tabindex="-1"></a>)<span class="sc">$</span>plotlist[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-101-1.png" width="672"></p>
<p>Spike-in condition <code>A</code> is the reference class. So the mean log2
intensity for samples from condition A is <code>(Intercept)</code>. The mean
log2 expression for samples from condition B is ‘(Intercept) +
conditionB’. Hence, the average log2 fold change between condition B
and condition A is modelled using the parameter ‘conditionB’. However,
the plot above does not depict the estimated model as it does is not
include the blocking variable <code>lab</code>.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="sec-basics.html#cb51-1" tabindex="-1"></a><span class="fu">VisualizeDesign</span>(</span>
<span id="cb51-2"><a href="sec-basics.html#cb51-2" tabindex="-1"></a>  <span class="at">sampleData =</span>  <span class="fu">colData</span>(spikein),</span>
<span id="cb51-3"><a href="sec-basics.html#cb51-3" tabindex="-1"></a>  <span class="at">designFormula =</span> model,</span>
<span id="cb51-4"><a href="sec-basics.html#cb51-4" tabindex="-1"></a>  <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb51-5"><a href="sec-basics.html#cb51-5" tabindex="-1"></a>)<span class="sc">$</span>plotlist[[<span class="dv">1</span>]]</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-102-1.png" width="672"></p>
<p>While the plot seems more complicated, the interpretation of the
condition parameters does not change<label for="tufte-sn-17" class="margin-toggle sidenote-number">17</label><input type="checkbox" id="tufte-sn-17" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">17</span> Demonstration: the mean log2
intensity for samples from condition A <strong>in lab1</strong> is <code>(Intercept)</code>.
The mean log2 intensity for samples from condition A across labs is
<code>(Intercept) + (lablab2 + lablab3) / 3</code>. Similarly, The mean log2
intensity for samples from condition B across labs is <code>(Intercept) + conditionB + (lablab2 + lablab3) / 3</code>. Most parameters cancel out when
computing the difference between A and B, so that the average log2
fold change between condition B and condition A is sill
‘conditionB’.</span>.</p>
<p>With <code>getCoef()</code>, we can retrieve the estimated model parameters.
We start with extracting the model output, stored as <code>StatModel</code>
objects, from the <code>rowData</code>. Next, <code>getCoef()</code> retrieves the estimated
model parameters:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="sec-basics.html#cb52-1" tabindex="-1"></a>models <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins"</span>]])<span class="sc">$</span>msqrobModels</span>
<span id="cb52-2"><a href="sec-basics.html#cb52-2" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">getCoef</span>(models[[<span class="dv">1</span>]])</span>
<span id="cb52-3"><a href="sec-basics.html#cb52-3" tabindex="-1"></a><span class="fu">head</span>(params)</span></code></pre></div>
<pre><code>##     (Intercept) ridgeconditionB ridgeconditionC ridgeconditionD ridgeconditionE    ridgelablab2 
##     4.390702233     0.005113902    -0.087367286    -0.025562006    -0.056063425    -0.726253949</code></pre>
<p>Note the <code>"ridge"</code> tag in front of the parameter names that indicates
the parameters have been estimated using ridge penalisation. The model
estimated 7 parameters to model the fixed and random
effects. However, we are interested, for this data set, in the
parameters that model the effect of <code>condition</code>.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="sec-basics.html#cb54-1" tabindex="-1"></a>params[<span class="fu">grep</span>(<span class="st">"condition"</span>, <span class="fu">names</span>(params))]</span></code></pre></div>
<pre><code>## ridgeconditionB ridgeconditionC ridgeconditionD ridgeconditionE 
##     0.005113902    -0.087367286    -0.025562006    -0.056063425</code></pre>
<p>For this protein we can see that the effects of condition are very
close to zero, as expected since the protein is part of the yeast
constant background. However, we can explore the parameters for one of
the UPS proteins.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="sec-basics.html#cb56-1" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">getCoef</span>(models[[<span class="st">"O00762ups|UBE2C_HUMAN_UPS"</span>]])</span>
<span id="cb56-2"><a href="sec-basics.html#cb56-2" tabindex="-1"></a>params[<span class="fu">grep</span>(<span class="st">"condition"</span>, <span class="fu">names</span>(params))]</span></code></pre></div>
<pre><code>## ridgeconditionB ridgeconditionC ridgeconditionD ridgeconditionE 
##        1.655465        2.900635        4.705510        6.389053</code></pre>
<p>Since this is a benchmark study and we know the concentration of A
(0.25 fmol/µL) and B (0.74 fmol/µL), the obvious answer is
<span class="math inline">\(log_2(0.74) - log_2(0.25) = 1.566\)</span>. The average log2 difference in
intensity between condition B and condition A that has been estimated
by the model is 1.6554652, rather close. Now,
the question is how statistically significant is the estimated log2
fold change.</p>
<div id="hypothesis-testing" class="section level3" number="2.7.1">
<h3>
<span class="header-section-number">2.7.1</span> Hypothesis testing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('hypothesis-testing')" onmouseout="reset_tooltip('hypothesis-testing-tooltip')"><span class="tooltiptext" id="hypothesis-testing-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>As shown above, the average difference intensity between condition B
and A is <code>ridgeconditionB</code>, obtained after combining the relevant
parameters. This combination of parameters is also called a
contrast. Thus, we assess the contrast ‘ridgeconditionB = 0’ with our
statistical test.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="sec-basics.html#cb58-1" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="st">"ridgeconditionB = 0"</span></span></code></pre></div>
<p><code>makeContrast()</code> converts the hypothesis into a formal contrast matrix
with parameter names as rows and hypotheses in columns<label for="tufte-sn-18" class="margin-toggle sidenote-number">18</label><input type="checkbox" id="tufte-sn-18" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">18</span> We will
<a href="sec-basics.html#sec-multiple_contrasts">later</a> show how to asses multiple hypotheses
at once. In this case, the contrast matrix is trivial, but it becomes
testing multiple hypothesis will lead to a matrix with multiple
columns</span>.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="sec-basics.html#cb59-1" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb59-2"><a href="sec-basics.html#cb59-2" tabindex="-1"></a>  <span class="st">"ridgeconditionB = 0"</span>,</span>
<span id="cb59-3"><a href="sec-basics.html#cb59-3" tabindex="-1"></a>  <span class="st">"ridgeconditionB"</span></span>
<span id="cb59-4"><a href="sec-basics.html#cb59-4" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>##                 ridgeconditionB
## ridgeconditionB               1</code></pre>
<p>We can now test our null hypothesis using <code>hypothesisTest()</code> which
takes the <code>QFeatures</code> object with the fitted model and the contrast we
just built. <code>msqrob2</code> automatically applies the hypothesis
testing to all proteins in the data.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="sec-basics.html#cb61-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(spikein, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
<p>The results are stored in the set containing the model, here
<code>proteins</code>. We retrieve the results from the <code>rowData</code>.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="sec-basics.html#cb62-1" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins"</span>]])<span class="sc">$</span>ridgeconditionB</span>
<span id="cb62-2"><a href="sec-basics.html#cb62-2" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb62-3"><a href="sec-basics.html#cb62-3" tabindex="-1"></a>inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb62-4"><a href="sec-basics.html#cb62-4" tabindex="-1"></a><span class="fu">head</span>(inference)</span></code></pre></div>
<pre><code>##                                  logFC        se       df           t         pval     adjPval                   Protein isUps
## CON__P00761                0.005113902 0.2111368 42.77668  0.02422080 9.807891e-01 1.00000e+00               CON__P00761 FALSE
## CON__P02769                0.445819463 0.4142068 20.09988  1.07632085 2.945337e-01 1.00000e+00               CON__P02769 FALSE
## CON__P04264               -0.166414197 0.1632785 21.45718 -1.01920456 3.194607e-01 1.00000e+00               CON__P04264 FALSE
## CON__P13645               -0.003255728 0.1765379 30.16291 -0.01844209 9.854076e-01 1.00000e+00               CON__P13645 FALSE
## CON__P35527                         NA        NA       NA          NA           NA          NA               CON__P35527 FALSE
## O00762ups|UBE2C_HUMAN_UPS  1.655465232 0.1840909 39.10317  8.99265239 4.621329e-11 3.29106e-09 O00762ups|UBE2C_HUMAN_UPS  TRUE</code></pre>
<p>The fifth row is filled with missing values because data modelling
resulted in a <code>fitError</code> (the <a href="sec-advanced.html#sec-fiterror">advanced vignette</a>
describes how to deal with proteins that could not be fit).</p>
<p><strong>TODO</strong> discuss multiple test adjustments and FDR correction. Use the
material from the PDA course.</p>
</div>
<div id="volcano-plots" class="section level3" number="2.7.2">
<h3>
<span class="header-section-number">2.7.2</span> Volcano plots<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('volcano-plots')" onmouseout="reset_tooltip('volcano-plots-tooltip')"><span class="tooltiptext" id="volcano-plots-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>A volcano plot is a common visualisation that provides an overview of
the hypothesis testing results, plottig the p-value as a function of
the estimated log fold change. Volcano plots can be used to highlight
the most interesting proteins that have large fold changes and/or are
highly significant. We can use the table above directly to build a
volcano plot using <code>ggplot2</code> functionality. We also highlight which
proteins are UPS standards, known to be differentially abundant by
experimental design.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="sec-basics.html#cb64-1" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb64-2"><a href="sec-basics.html#cb64-2" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb64-3"><a href="sec-basics.html#cb64-3" tabindex="-1"></a>      <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(adjPval),</span>
<span id="cb64-4"><a href="sec-basics.html#cb64-4" tabindex="-1"></a>      <span class="at">color =</span> isUps) <span class="sc">+</span></span>
<span id="cb64-5"><a href="sec-basics.html#cb64-5" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb64-6"><a href="sec-basics.html#cb64-6" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">log2</span>(<span class="fl">0.74</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="fl">0.25</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb64-7"><a href="sec-basics.html#cb64-7" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb64-8"><a href="sec-basics.html#cb64-8" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb64-9"><a href="sec-basics.html#cb64-9" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb64-10"><a href="sec-basics.html#cb64-10" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb64-11"><a href="sec-basics.html#cb64-11" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Statistical inference results"</span>,</span>
<span id="cb64-12"><a href="sec-basics.html#cb64-12" tabindex="-1"></a>          <span class="fu">paste0</span>(<span class="st">"Hypothesis test: "</span>, <span class="fu">colnames</span>(L), <span class="st">" = 0"</span>))</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-110-1.png" width="672"></p>
<p>We retrieve the proteins that are significantly differentially
abundant based on the FDR-adjusted p-value (<code>adjPval</code>). Note that the
majority of the differentially abundant proteins are UPS1 proteins.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="sec-basics.html#cb65-1" tabindex="-1"></a><span class="fu">table</span>(<span class="at">is_significant =</span> inference<span class="sc">$</span>adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>, </span>
<span id="cb65-2"><a href="sec-basics.html#cb65-2" tabindex="-1"></a>      <span class="at">is_ups =</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein))</span></code></pre></div>
<pre><code>##               is_ups
## is_significant FALSE TRUE
##          FALSE  1146    3
##          TRUE      2   33</code></pre>
</div>
<div id="heatmaps" class="section level3" number="2.7.3">
<h3>
<span class="header-section-number">2.7.3</span> Heatmaps<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('heatmaps')" onmouseout="reset_tooltip('heatmaps-tooltip')"><span class="tooltiptext" id="heatmaps-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can explore the quantitative data of the significant proteins using
a heatmap. First, we select the names of the proteins that were
declared significant and extract their quantitative data.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="sec-basics.html#cb67-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> inference<span class="sc">$</span>Protein[<span class="sc">!</span><span class="fu">is.na</span>(inference<span class="sc">$</span>adjPval) <span class="sc">&amp;</span> inference<span class="sc">$</span>adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>]</span>
<span id="cb67-2"><a href="sec-basics.html#cb67-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(spikein, <span class="st">"proteins"</span>)</span>
<span id="cb67-3"><a href="sec-basics.html#cb67-3" tabindex="-1"></a>se <span class="ot">&lt;-</span> se[sigNames, se<span class="sc">$</span>condition <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>)]</span></code></pre></div>
<p>Next, we extract the quantitative data and scale by rows<label for="tufte-sn-19" class="margin-toggle sidenote-number">19</label><input type="checkbox" id="tufte-sn-19" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">19</span> The
<code>scales()</code> function aligns the means and the centered standard
deviations by column. Since we want to scale by row, we need to
transpose (<code>t())</code>) before scaling and transpose back after scaling.
Scaling the data will remove changes in feature intensities that are
caused by uninteresting changes in ionisation efficiency.</span> with
<code>assay()</code>. We will create a heatmap using the <code>ComplexHeatmap</code>
package, which enables heatmap annotations. We will annotate the
heatmap using our model variables <code>condition</code> and <code>lab</code>.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="sec-basics.html#cb68-1" tabindex="-1"></a>quants <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(<span class="fu">assay</span>(se))))</span>
<span id="cb68-2"><a href="sec-basics.html#cb68-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ComplexHeatmap"</span>)</span>
<span id="cb68-3"><a href="sec-basics.html#cb68-3" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">columnAnnotation</span>(</span>
<span id="cb68-4"><a href="sec-basics.html#cb68-4" tabindex="-1"></a>  <span class="at">condition =</span> se<span class="sc">$</span>condition,</span>
<span id="cb68-5"><a href="sec-basics.html#cb68-5" tabindex="-1"></a>  <span class="at">lab =</span> se<span class="sc">$</span>lab</span>
<span id="cb68-6"><a href="sec-basics.html#cb68-6" tabindex="-1"></a>)</span></code></pre></div>
<p>We now make the heatmap.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="sec-basics.html#cb69-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>) <span class="do">## annotation colours are randomly generated by default</span></span>
<span id="cb69-2"><a href="sec-basics.html#cb69-2" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb69-3"><a href="sec-basics.html#cb69-3" tabindex="-1"></a>  quants, <span class="at">name =</span> <span class="st">"log2 intensity"</span>,</span>
<span id="cb69-4"><a href="sec-basics.html#cb69-4" tabindex="-1"></a>  <span class="at">top_annotation =</span> annotations</span>
<span id="cb69-5"><a href="sec-basics.html#cb69-5" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-114-1.png" width="960"></p>
<p>We can see that the majority of the proteins are indeed UPS proteins,
and that 2 yeast proteins that are returned significant. The first
yeast protein is significant probably due to the outlying lab1<label for="tufte-sn-20" class="margin-toggle sidenote-number">20</label><input type="checkbox" id="tufte-sn-20" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">20</span> Note
that <code>msqrob2</code> estimation is robust against outlying observations, but
here the outlying experimental unit (lab1) contains three
observations.</span>; the significant result for the second yeast protein
probably stems from technical variations that could not be correctly
captured due to the high missingness (lab1 did not detected the
protein). Note, that the yeast protein indeed shows evidence for
differential abundance.</p>
</div>
<div id="fold-change-distributions" class="section level3" number="2.7.4">
<h3>
<span class="header-section-number">2.7.4</span> Fold change distributions<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('fold-change-distributions')" onmouseout="reset_tooltip('fold-change-distributions-tooltip')"><span class="tooltiptext" id="fold-change-distributions-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>As this is a spike-in study with known ground truth, we can also plot
the log2 fold change distributions against the expected values, in
this case 0 for the yeast proteins and 1.57 for the UPS standards.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="sec-basics.html#cb70-1" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb70-2"><a href="sec-basics.html#cb70-2" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">y =</span> logFC,</span>
<span id="cb70-3"><a href="sec-basics.html#cb70-3" tabindex="-1"></a>      <span class="at">x =</span> isUps,</span>
<span id="cb70-4"><a href="sec-basics.html#cb70-4" tabindex="-1"></a>      <span class="at">colour =</span> isUps) <span class="sc">+</span></span>
<span id="cb70-5"><a href="sec-basics.html#cb70-5" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb70-6"><a href="sec-basics.html#cb70-6" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">log2</span>(<span class="fl">0.74</span>) <span class="sc">-</span> <span class="fu">log2</span>(<span class="fl">0.25</span>)), </span>
<span id="cb70-7"><a href="sec-basics.html#cb70-7" tabindex="-1"></a>             <span class="at">colour =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>)) <span class="sc">+</span></span>
<span id="cb70-8"><a href="sec-basics.html#cb70-8" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb70-9"><a href="sec-basics.html#cb70-9" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb70-10"><a href="sec-basics.html#cb70-10" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Yeast background"</span>, <span class="st">"UPS proteins"</span>)</span>
<span id="cb70-11"><a href="sec-basics.html#cb70-11" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb70-12"><a href="sec-basics.html#cb70-12" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Distribution of the log2 fold changes"</span>,</span>
<span id="cb70-13"><a href="sec-basics.html#cb70-13" tabindex="-1"></a>          <span class="fu">paste0</span>(<span class="st">"Hypothesis test: "</span>, <span class="fu">colnames</span>(L), <span class="st">" = 0"</span>))</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-115-1.png" width="672"></p>
<p>Estimated log2 fold change for HeLa proteins are closely distributed
around 0, as expected. log2 fold changes for UPS standard proteins are
distributed around the fold changes expected from the experimental
design.</p>
</div>
<div id="sec-detail_plot" class="section level3" number="2.7.5">
<h3>
<span class="header-section-number">2.7.5</span> Detail plots<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-detail_plot')" onmouseout="reset_tooltip('sec-detail_plot-tooltip')"><span class="tooltiptext" id="sec-detail_plot-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can explore the data for a protein to validate the statistical
inference results. For example, let’s explore the peptide and the
summarised protein intensities for the protein with the most
significant difference.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="sec-basics.html#cb71-1" tabindex="-1"></a>(targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)])</span></code></pre></div>
<pre><code>## [1] "P07339ups|CATD_HUMAN_UPS"</code></pre>
<p>To obtain the required data, we perform a little data manipulation
pipeline:</p>
<p>We use the <code>QFeatures</code> subsetting functionality to retrieve all data
related to and focusing on the <code>peptides_log</code> and
<code>proteins</code> sets that contains the peptide ion data used for model
fitting. We then convert the data with <code>longForm()</code> for plotting.
Finally, we plot the log2 normalised intensities for each sample at
the protein and at the peptide level. Since multiple peptides are
recorded for the protein, we link peptides across samples using a grey
line. Samples are colored according to UPS spike-in condition.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="sec-basics.html#cb73-1" tabindex="-1"></a>spikein[targetProtein, , <span class="fu">c</span>(<span class="st">"peptides_log"</span>, <span class="st">"proteins"</span>)] <span class="sc">|&gt;</span> <span class="co">#1</span></span>
<span id="cb73-2"><a href="sec-basics.html#cb73-2" tabindex="-1"></a>     <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikein)), <span class="co">#2</span></span>
<span id="cb73-3"><a href="sec-basics.html#cb73-3" tabindex="-1"></a>              <span class="at">rowvars =</span> <span class="st">"Proteins"</span>) <span class="sc">|&gt;</span></span>
<span id="cb73-4"><a href="sec-basics.html#cb73-4" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb73-5"><a href="sec-basics.html#cb73-5" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb73-6"><a href="sec-basics.html#cb73-6" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb73-7"><a href="sec-basics.html#cb73-7" tabindex="-1"></a>        <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb73-8"><a href="sec-basics.html#cb73-8" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> rowname), <span class="at">linewidth =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb73-9"><a href="sec-basics.html#cb73-9" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> condition)) <span class="sc">+</span></span>
<span id="cb73-10"><a href="sec-basics.html#cb73-10" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> assay, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb73-11"><a href="sec-basics.html#cb73-11" tabindex="-1"></a>    <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb73-12"><a href="sec-basics.html#cb73-12" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb73-13"><a href="sec-basics.html#cb73-13" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-117-1.png" width="768"></p>
</div>
</div>
<div id="going-further" class="section level2" number="2.8">
<h2>
<span class="header-section-number">2.8</span> Going further<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('going-further')" onmouseout="reset_tooltip('going-further-tooltip')"><span class="tooltiptext" id="going-further-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<div id="sec-multiple_contrasts" class="section level3" number="2.8.1">
<h3>
<span class="header-section-number">2.8.1</span> Testing muliple contrasts at once<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-multiple_contrasts')" onmouseout="reset_tooltip('sec-multiple_contrasts-tooltip')"><span class="tooltiptext" id="sec-multiple_contrasts-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We showed how to perform a hypothesis test to answer the question
“which protein change in abundance between condition B and condition
A?”. However, since we have 5 conditions, we can imagine further
comparisons. <code>msqrob2</code> can assess multiple hypothesis at once. Like
above we must provide which hypotheses we want to compare.
<code>createPairwiseContrasts()</code> will generate all possible comparisons
between <code>condition</code> level. The function requires the model
specification that has been estimated and the sample annotation from
the data. We also specify that we estimated the model using ridge
regression, which influences parameter names.</p>
<p><strong>TODO</strong>: add automatic pairwise comparisons function to msqrob2</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="sec-basics.html#cb74-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"utils.R"</span>)</span>
<span id="cb74-2"><a href="sec-basics.html#cb74-2" tabindex="-1"></a>(allHypotheses <span class="ot">&lt;-</span> <span class="fu">createPairwiseContrasts</span>(</span>
<span id="cb74-3"><a href="sec-basics.html#cb74-3" tabindex="-1"></a>  model, <span class="fu">colData</span>(spikein), <span class="st">"condition"</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb74-4"><a href="sec-basics.html#cb74-4" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>##  [1] "ridgeconditionC - ridgeconditionB = 0" "ridgeconditionD - ridgeconditionB = 0" "ridgeconditionE - ridgeconditionB = 0"
##  [4] "ridgeconditionD - ridgeconditionC = 0" "ridgeconditionE - ridgeconditionC = 0" "ridgeconditionE - ridgeconditionD = 0"
##  [7] "ridgeconditionB = 0"                   "ridgeconditionC = 0"                   "ridgeconditionD = 0"                  
## [10] "ridgeconditionE = 0"</code></pre>
<p>We now run the same worklfow as above for a single hypothesis, except
that here <code>allHypotheses</code> is a vector of contrasts.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="sec-basics.html#cb76-1" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb76-2"><a href="sec-basics.html#cb76-2" tabindex="-1"></a>  allHypotheses,</span>
<span id="cb76-3"><a href="sec-basics.html#cb76-3" tabindex="-1"></a>  <span class="at">parameterNames =</span> <span class="fu">paste0</span>(<span class="st">"ridgecondition"</span>, <span class="fu">c</span>(<span class="st">"B"</span>, <span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"E"</span>))</span>
<span id="cb76-4"><a href="sec-basics.html#cb76-4" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>##                 ridgeconditionC - ridgeconditionB ridgeconditionD - ridgeconditionB ridgeconditionE - ridgeconditionB
## ridgeconditionB                                -1                                -1                                -1
## ridgeconditionC                                 1                                 0                                 0
## ridgeconditionD                                 0                                 1                                 0
## ridgeconditionE                                 0                                 0                                 1
##                 ridgeconditionD - ridgeconditionC ridgeconditionE - ridgeconditionC ridgeconditionE - ridgeconditionD
## ridgeconditionB                                 0                                 0                                 0
## ridgeconditionC                                -1                                -1                                 0
## ridgeconditionD                                 1                                 0                                -1
## ridgeconditionE                                 0                                 1                                 1
##                 ridgeconditionB ridgeconditionC ridgeconditionD ridgeconditionE
## ridgeconditionB               1               0               0               0
## ridgeconditionC               0               1               0               0
## ridgeconditionD               0               0               1               0
## ridgeconditionE               0               0               0               1</code></pre>
<p>The contrast contains mulitple hypotheses (multiple column) that
involve multiple parameters (multiple rows).</p>
<p>We use again <code>hypothesisTest()</code> for performing the hypothesis test. We
already generated results for the contrast <code>ridgeconditionB = 0</code> and
the function will throw an error by default, but we can overwrite the
results with the argument <code>overwrite = TRUE</code>.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="sec-basics.html#cb78-1" tabindex="-1"></a>spikein <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(spikein, <span class="at">i =</span> <span class="st">"proteins"</span>, L, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We retrieve the inference tables from the <code>rowData</code> to generate the
volcano plots.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="sec-basics.html#cb79-1" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikein[[<span class="st">"proteins"</span>]])[, <span class="fu">colnames</span>(L)]</span></code></pre></div>
<p>We here use a <code>lapply()</code> loop to generate additional information, and
then combine all the tables in a single table.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="sec-basics.html#cb80-1" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(inferences), <span class="cf">function</span>(i) {</span>
<span id="cb80-2"><a href="sec-basics.html#cb80-2" tabindex="-1"></a>  inference <span class="ot">&lt;-</span> inferences[[i]]</span>
<span id="cb80-3"><a href="sec-basics.html#cb80-3" tabindex="-1"></a>  inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb80-4"><a href="sec-basics.html#cb80-4" tabindex="-1"></a>  inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb80-5"><a href="sec-basics.html#cb80-5" tabindex="-1"></a>  inference<span class="sc">$</span>Comparison <span class="ot">&lt;-</span> i</span>
<span id="cb80-6"><a href="sec-basics.html#cb80-6" tabindex="-1"></a>  inference</span>
<span id="cb80-7"><a href="sec-basics.html#cb80-7" tabindex="-1"></a>})</span>
<span id="cb80-8"><a href="sec-basics.html#cb80-8" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, inferences) <span class="do">## combine in a single table</span></span></code></pre></div>
<p>We plot the volcano plots with each comparison in a separate facet.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="sec-basics.html#cb81-1" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb81-2"><a href="sec-basics.html#cb81-2" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb81-3"><a href="sec-basics.html#cb81-3" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb81-4"><a href="sec-basics.html#cb81-4" tabindex="-1"></a>        <span class="at">color =</span> isUps) <span class="sc">+</span></span>
<span id="cb81-5"><a href="sec-basics.html#cb81-5" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb81-6"><a href="sec-basics.html#cb81-6" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb81-7"><a href="sec-basics.html#cb81-7" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb81-8"><a href="sec-basics.html#cb81-8" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb81-9"><a href="sec-basics.html#cb81-9" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb81-10"><a href="sec-basics.html#cb81-10" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb81-11"><a href="sec-basics.html#cb81-11" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison) <span class="sc">+</span></span>
<span id="cb81-12"><a href="sec-basics.html#cb81-12" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>) </span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-123-1.png" width="864"></p>
<p><strong>TODO</strong> interpret the results</p>
</div>
<div id="sec-numeric_model" class="section level3" number="2.8.2">
<h3>
<span class="header-section-number">2.8.2</span> What about lab1?<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sec-numeric_model')" onmouseout="reset_tooltip('sec-numeric_model-tooltip')"><span class="tooltiptext" id="sec-numeric_model-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p><strong>TODO</strong>: discuss this</p>
<p>We saw during data exploration that lab1 displayed unexpected data
patterns compared to lab2 and lab3, and it was reported in the
original study that lab1 experienced ionisation issues. In a real-life
experiment, it would be advised to remove these low-quality
measurements<label for="tufte-sn-21" class="margin-toggle sidenote-number">21</label><input type="checkbox" id="tufte-sn-21" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">21</span> Watch out: the decision of removing low-quality samples
should be performed
<strong>before</strong> any statistical analysis and be independent of its outcome,
otherwise this could be regarded as data snooping. Relying solely on
the data exploration for sample exclusion is also not recommended,
since the same quantitative data is used for exploration and
statistical analysis. In this example, the data exploration allowed to
pinpoint issues with the data and led to searching for possible
explanations, which where confirmed based on data metrics that are
independent of quantification.</span> So, we will conduct the same analysis
as above, but removing any data coming from lab1.</p>
<p>Filtering samples is performed using <code>subsetByColData()</code> which takes a
<code>QFeatures</code> object and logical vector which is assumed to be generated
from the <code>colData</code> annotations, here we keep all samples that don’t
belong to <code>lab1</code>. The function will automatically retrieve the samples
that pass the <code>colData</code> filter and subset all sets for those samples.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="sec-basics.html#cb82-1" tabindex="-1"></a>(spikeinNoLab1 <span class="ot">&lt;-</span> <span class="fu">subsetByColData</span>(spikein, spikein<span class="sc">$</span>lab <span class="sc">!=</span> <span class="st">"lab1"</span>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 4 sets:
## 
##  [1] peptides: SummarizedExperiment with 10429 rows and 30 columns 
##  [2] peptides_log: SummarizedExperiment with 6501 rows and 30 columns 
##  [3] peptides_norm: SummarizedExperiment with 6501 rows and 30 columns 
##  [4] proteins: SummarizedExperiment with 1290 rows and 30 columns</code></pre>
<p>Whereas previously every set contained 45 samples, we here see that
only 30 samples are retained, all 15 samples belonging to lab1 were
discarded.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="sec-basics.html#cb84-1" tabindex="-1"></a><span class="fu">table</span>(<span class="at">condition =</span> spikeinNoLab1<span class="sc">$</span>condition,</span>
<span id="cb84-2"><a href="sec-basics.html#cb84-2" tabindex="-1"></a>      <span class="at">lab =</span> spikeinNoLab1<span class="sc">$</span>lab)</span></code></pre></div>
<pre><code>##          lab
## condition lab2 lab3
##         A    3    3
##         B    3    3
##         C    3    3
##         D    3    3
##         E    3    3</code></pre>
<p>We now run the same statistical workflow as described above.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="sec-basics.html#cb86-1" tabindex="-1"></a>spikeinNoLab1 <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb86-2"><a href="sec-basics.html#cb86-2" tabindex="-1"></a>  spikeinNoLab1,  <span class="at">i =</span> <span class="st">"proteins"</span>,</span>
<span id="cb86-3"><a href="sec-basics.html#cb86-3" tabindex="-1"></a>  <span class="at">formula =</span> model,</span>
<span id="cb86-4"><a href="sec-basics.html#cb86-4" tabindex="-1"></a>  <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span>, </span>
<span id="cb86-5"><a href="sec-basics.html#cb86-5" tabindex="-1"></a>  <span class="at">overwrite =</span> <span class="cn">TRUE</span></span>
<span id="cb86-6"><a href="sec-basics.html#cb86-6" tabindex="-1"></a>)</span>
<span id="cb86-7"><a href="sec-basics.html#cb86-7" tabindex="-1"></a>(allHypotheses <span class="ot">&lt;-</span> <span class="fu">createPairwiseContrasts</span>(</span>
<span id="cb86-8"><a href="sec-basics.html#cb86-8" tabindex="-1"></a>  model, <span class="fu">colData</span>(spikeinNoLab1), <span class="st">"condition"</span>, <span class="at">ridge =</span> <span class="cn">TRUE</span></span>
<span id="cb86-9"><a href="sec-basics.html#cb86-9" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>##  [1] "ridgeconditionC - ridgeconditionB = 0" "ridgeconditionD - ridgeconditionB = 0" "ridgeconditionE - ridgeconditionB = 0"
##  [4] "ridgeconditionD - ridgeconditionC = 0" "ridgeconditionE - ridgeconditionC = 0" "ridgeconditionE - ridgeconditionD = 0"
##  [7] "ridgeconditionB = 0"                   "ridgeconditionC = 0"                   "ridgeconditionD = 0"                  
## [10] "ridgeconditionE = 0"</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="sec-basics.html#cb88-1" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(</span>
<span id="cb88-2"><a href="sec-basics.html#cb88-2" tabindex="-1"></a>  allHypotheses,</span>
<span id="cb88-3"><a href="sec-basics.html#cb88-3" tabindex="-1"></a>  <span class="at">parameterNames =</span> <span class="fu">paste0</span>(<span class="st">"ridgecondition"</span>, <span class="fu">c</span>(<span class="st">"B"</span>, <span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"E"</span>))</span>
<span id="cb88-4"><a href="sec-basics.html#cb88-4" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>##                 ridgeconditionC - ridgeconditionB ridgeconditionD - ridgeconditionB ridgeconditionE - ridgeconditionB
## ridgeconditionB                                -1                                -1                                -1
## ridgeconditionC                                 1                                 0                                 0
## ridgeconditionD                                 0                                 1                                 0
## ridgeconditionE                                 0                                 0                                 1
##                 ridgeconditionD - ridgeconditionC ridgeconditionE - ridgeconditionC ridgeconditionE - ridgeconditionD
## ridgeconditionB                                 0                                 0                                 0
## ridgeconditionC                                -1                                -1                                 0
## ridgeconditionD                                 1                                 0                                -1
## ridgeconditionE                                 0                                 1                                 1
##                 ridgeconditionB ridgeconditionC ridgeconditionD ridgeconditionE
## ridgeconditionB               1               0               0               0
## ridgeconditionC               0               1               0               0
## ridgeconditionD               0               0               1               0
## ridgeconditionE               0               0               0               1</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="sec-basics.html#cb90-1" tabindex="-1"></a>spikeinNoLab1 <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb90-2"><a href="sec-basics.html#cb90-2" tabindex="-1"></a>  spikeinNoLab1, <span class="at">i =</span> <span class="st">"proteins"</span>, L, <span class="at">overwrite =</span> <span class="cn">TRUE</span></span>
<span id="cb90-3"><a href="sec-basics.html#cb90-3" tabindex="-1"></a>)</span>
<span id="cb90-4"><a href="sec-basics.html#cb90-4" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikeinNoLab1[[<span class="st">"proteins"</span>]])[, <span class="fu">colnames</span>(L)]</span></code></pre></div>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="sec-basics.html#cb91-1" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">colnames</span>(inferences), <span class="cf">function</span>(i) {</span>
<span id="cb91-2"><a href="sec-basics.html#cb91-2" tabindex="-1"></a>  inference <span class="ot">&lt;-</span> inferences[[i]]</span>
<span id="cb91-3"><a href="sec-basics.html#cb91-3" tabindex="-1"></a>  inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb91-4"><a href="sec-basics.html#cb91-4" tabindex="-1"></a>  inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb91-5"><a href="sec-basics.html#cb91-5" tabindex="-1"></a>  inference<span class="sc">$</span>Comparison <span class="ot">&lt;-</span> i</span>
<span id="cb91-6"><a href="sec-basics.html#cb91-6" tabindex="-1"></a>  inference</span>
<span id="cb91-7"><a href="sec-basics.html#cb91-7" tabindex="-1"></a>})</span>
<span id="cb91-8"><a href="sec-basics.html#cb91-8" tabindex="-1"></a>inferences <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, inferences) <span class="do">## combine in a single table</span></span>
<span id="cb91-9"><a href="sec-basics.html#cb91-9" tabindex="-1"></a><span class="fu">ggplot</span>(inferences) <span class="sc">+</span></span>
<span id="cb91-10"><a href="sec-basics.html#cb91-10" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb91-11"><a href="sec-basics.html#cb91-11" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval),</span>
<span id="cb91-12"><a href="sec-basics.html#cb91-12" tabindex="-1"></a>        <span class="at">color =</span> isUps) <span class="sc">+</span></span>
<span id="cb91-13"><a href="sec-basics.html#cb91-13" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb91-14"><a href="sec-basics.html#cb91-14" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb91-15"><a href="sec-basics.html#cb91-15" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(</span>
<span id="cb91-16"><a href="sec-basics.html#cb91-16" tabindex="-1"></a>      <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb91-17"><a href="sec-basics.html#cb91-17" tabindex="-1"></a>      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb91-18"><a href="sec-basics.html#cb91-18" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb91-19"><a href="sec-basics.html#cb91-19" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Comparison) <span class="sc">+</span></span>
<span id="cb91-20"><a href="sec-basics.html#cb91-20" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"Statistical inference for all pairwise comparisons"</span>,</span>
<span id="cb91-21"><a href="sec-basics.html#cb91-21" tabindex="-1"></a>            <span class="at">subtitle =</span> <span class="st">"Removing all data from lab1"</span>)</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-127-1.png" width="864"></p>
</div>
<div id="model-the-spike-in-concentration-as-a-numeric" class="section level3" number="2.8.3">
<h3>
<span class="header-section-number">2.8.3</span> Model the spike-in concentration as a numeric<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('model-the-spike-in-concentration-as-a-numeric')" onmouseout="reset_tooltip('model-the-spike-in-concentration-as-a-numeric-tooltip')"><span class="tooltiptext" id="model-the-spike-in-concentration-as-a-numeric-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p><strong>TODO</strong>: discuss this</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="sec-basics.html#cb92-1" tabindex="-1"></a>spikeinNoLab1<span class="sc">$</span>concentration_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(spikeinNoLab1<span class="sc">$</span>concentration)</span>
<span id="cb92-2"><a href="sec-basics.html#cb92-2" tabindex="-1"></a>spikeinNoLab1 <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(</span>
<span id="cb92-3"><a href="sec-basics.html#cb92-3" tabindex="-1"></a>  spikeinNoLab1, <span class="at">i =</span> <span class="st">"proteins"</span>,</span>
<span id="cb92-4"><a href="sec-basics.html#cb92-4" tabindex="-1"></a>  <span class="at">formula =</span> <span class="sc">~</span> concentration_log2 <span class="sc">+</span> lab,</span>
<span id="cb92-5"><a href="sec-basics.html#cb92-5" tabindex="-1"></a>  <span class="at">ridge =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span>, </span>
<span id="cb92-6"><a href="sec-basics.html#cb92-6" tabindex="-1"></a>  <span class="at">modelColumnName =</span> <span class="st">"concentration_model"</span>,</span>
<span id="cb92-7"><a href="sec-basics.html#cb92-7" tabindex="-1"></a>  <span class="at">overwrite =</span> T</span>
<span id="cb92-8"><a href="sec-basics.html#cb92-8" tabindex="-1"></a>)</span>
<span id="cb92-9"><a href="sec-basics.html#cb92-9" tabindex="-1"></a>L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(<span class="st">"ridgeconcentration_log2 = 0"</span>, <span class="at">parameterNames =</span> <span class="st">"ridgeconcentration_log2"</span>)</span>
<span id="cb92-10"><a href="sec-basics.html#cb92-10" tabindex="-1"></a>spikeinNoLab1 <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(</span>
<span id="cb92-11"><a href="sec-basics.html#cb92-11" tabindex="-1"></a>  spikeinNoLab1, <span class="at">i =</span> <span class="st">"proteins"</span>, L, <span class="at">modelColumn =</span> <span class="st">"concentration_model"</span>,</span>
<span id="cb92-12"><a href="sec-basics.html#cb92-12" tabindex="-1"></a>  <span class="at">overwrite =</span> T</span>
<span id="cb92-13"><a href="sec-basics.html#cb92-13" tabindex="-1"></a>)</span>
<span id="cb92-14"><a href="sec-basics.html#cb92-14" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(spikeinNoLab1[[<span class="st">"proteins"</span>]])[, <span class="fu">colnames</span>(L)]</span>
<span id="cb92-15"><a href="sec-basics.html#cb92-15" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb92-16"><a href="sec-basics.html#cb92-16" tabindex="-1"></a>inference<span class="sc">$</span>isUps <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="st">"ups"</span>, inference<span class="sc">$</span>Protein)</span>
<span id="cb92-17"><a href="sec-basics.html#cb92-17" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb92-18"><a href="sec-basics.html#cb92-18" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> logFC,</span>
<span id="cb92-19"><a href="sec-basics.html#cb92-19" tabindex="-1"></a>      <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(adjPval),</span>
<span id="cb92-20"><a href="sec-basics.html#cb92-20" tabindex="-1"></a>      <span class="at">color =</span> isUps) <span class="sc">+</span></span>
<span id="cb92-21"><a href="sec-basics.html#cb92-21" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb92-22"><a href="sec-basics.html#cb92-22" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb92-23"><a href="sec-basics.html#cb92-23" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(</span>
<span id="cb92-24"><a href="sec-basics.html#cb92-24" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey20"</span>, <span class="st">"firebrick"</span>), <span class="at">name =</span> <span class="st">""</span>,</span>
<span id="cb92-25"><a href="sec-basics.html#cb92-25" tabindex="-1"></a>    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"HeLA background"</span>, <span class="st">"UPS standard"</span>)</span>
<span id="cb92-26"><a href="sec-basics.html#cb92-26" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb92-27"><a href="sec-basics.html#cb92-27" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Statistical inference results"</span>,</span>
<span id="cb92-28"><a href="sec-basics.html#cb92-28" tabindex="-1"></a>          <span class="fu">paste0</span>(<span class="st">"Hypothesis test: "</span>, <span class="fu">colnames</span>(L), <span class="st">" = 0"</span>))</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-128-1.png" width="672"></p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="sec-basics.html#cb93-1" tabindex="-1"></a>targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)]</span>
<span id="cb93-2"><a href="sec-basics.html#cb93-2" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">getCoef</span>(<span class="fu">rowData</span>(spikeinNoLab1)[[<span class="st">"proteins"</span>]][[<span class="st">"concentration_model"</span>]][[targetProtein]])</span>
<span id="cb93-3"><a href="sec-basics.html#cb93-3" tabindex="-1"></a>spikeinNoLab1[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span></span>
<span id="cb93-4"><a href="sec-basics.html#cb93-4" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikeinNoLab1)),</span>
<span id="cb93-5"><a href="sec-basics.html#cb93-5" tabindex="-1"></a>           <span class="at">rowvars =</span> <span class="st">"Proteins"</span>) <span class="sc">|&gt;</span></span>
<span id="cb93-6"><a href="sec-basics.html#cb93-6" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb93-7"><a href="sec-basics.html#cb93-7" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb93-8"><a href="sec-basics.html#cb93-8" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log2</span>(concentration),</span>
<span id="cb93-9"><a href="sec-basics.html#cb93-9" tabindex="-1"></a>      <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb93-10"><a href="sec-basics.html#cb93-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> lab)) <span class="sc">+</span></span>
<span id="cb93-11"><a href="sec-basics.html#cb93-11" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="fu">c</span>(<span class="dv">1</span>, params[[<span class="st">"ridgeconcentration_log2"</span>]]), </span>
<span id="cb93-12"><a href="sec-basics.html#cb93-12" tabindex="-1"></a>              <span class="at">intercept =</span> params[[<span class="st">"(Intercept)"</span>]],</span>
<span id="cb93-13"><a href="sec-basics.html#cb93-13" tabindex="-1"></a>              <span class="at">linetype =</span> <span class="fu">c</span>(<span class="st">"solid"</span>, <span class="st">"dashed"</span>)) <span class="sc">+</span></span>
<span id="cb93-14"><a href="sec-basics.html#cb93-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb93-15"><a href="sec-basics.html#cb93-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-129-1.png" width="672"></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="sec-basics.html#cb94-1" tabindex="-1"></a>targetProtein <span class="ot">&lt;-</span> inference<span class="sc">$</span>Protein[<span class="fu">which.min</span>(inference<span class="sc">$</span>logFC <span class="sc">*</span> <span class="fu">as.numeric</span>(inference<span class="sc">$</span>adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>))]</span>
<span id="cb94-2"><a href="sec-basics.html#cb94-2" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">getCoef</span>(<span class="fu">rowData</span>(spikeinNoLab1)[[<span class="st">"proteins"</span>]][[<span class="st">"concentration_model"</span>]][[targetProtein]])</span>
<span id="cb94-3"><a href="sec-basics.html#cb94-3" tabindex="-1"></a>spikeinNoLab1[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span></span>
<span id="cb94-4"><a href="sec-basics.html#cb94-4" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="fu">colnames</span>(<span class="fu">colData</span>(spikeinNoLab1)),</span>
<span id="cb94-5"><a href="sec-basics.html#cb94-5" tabindex="-1"></a>           <span class="at">rowvars =</span> <span class="st">"Proteins"</span>) <span class="sc">|&gt;</span></span>
<span id="cb94-6"><a href="sec-basics.html#cb94-6" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb94-7"><a href="sec-basics.html#cb94-7" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb94-8"><a href="sec-basics.html#cb94-8" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log2</span>(concentration),</span>
<span id="cb94-9"><a href="sec-basics.html#cb94-9" tabindex="-1"></a>      <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb94-10"><a href="sec-basics.html#cb94-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> lab)) <span class="sc">+</span></span>
<span id="cb94-11"><a href="sec-basics.html#cb94-11" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="fu">c</span>(<span class="dv">0</span>, params[[<span class="st">"ridgeconcentration_log2"</span>]]), </span>
<span id="cb94-12"><a href="sec-basics.html#cb94-12" tabindex="-1"></a>              <span class="at">intercept =</span> params[[<span class="st">"(Intercept)"</span>]],</span>
<span id="cb94-13"><a href="sec-basics.html#cb94-13" tabindex="-1"></a>              <span class="at">linetype =</span> <span class="fu">c</span>(<span class="st">"solid"</span>, <span class="st">"dashed"</span>)) <span class="sc">+</span></span>
<span id="cb94-14"><a href="sec-basics.html#cb94-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb94-15"><a href="sec-basics.html#cb94-15" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="msqrob2_book_files/figure-html/unnamed-chunk-130-1.png" width="672"></p>
</div>
</div>
<div id="conclusion" class="section level2" number="2.9">
<h2>
<span class="header-section-number">2.9</span> Conclusion<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('conclusion')" onmouseout="reset_tooltip('conclusion-tooltip')"><span class="tooltiptext" id="conclusion-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>In this vignette, we have demonstrated a typical data analysis
workflow for LFQ data using <code>msqrob2</code>. Because the packages relies on
the <code>QFeatures</code> data class, we could demonstrate the implementation of
a complete pre-processing workflow starting from MaxQuant’s peptide
table: log2-transformation, PSM filtering, missing value management,
normalisation, and summarisation. The data exploration revealed
expected patterns of variation among samples, as this is a ground
truth data set, but also unexpected patterns which were reported by
the original study.</p>
<p>Once pre-processed, we use the <code>msqrob2</code> package to model the (known)
effect of spike-in condition while accounting for effects of lab. We
showed how to run statistical inference on the modelling results to
retrieve the significance of differentially abundant proteins. We
explored statistical results through volcano plots and boxplots of the
log2 fold changes and visually validated the results for one protein
by plotting the input data. We have shown the inference pipeline can
be streamlined when having multiple hypotheses to test.</p>
<p><strong>TODO</strong>: extend conclusion with the going further topics.
<strong>TODO</strong>: introduce more complex design, current limitations which
will be tackled in the advancded vignette</p>

</div>
</div>
<p style="text-align: center;">
<a href="index.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-advanced.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2025-10-07
 using 
R version 4.5.1 (2025-06-13)
</p>
</div>
</div>



</body>
</html>
