<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="6 The francisella use case: a MaxQuant LFQ DDA dataset with technical replication | Statistical analysis of mass spectrometry-based proteomics data" />
<meta property="og:type" content="book" />




<meta name="author" content="Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement" />

<meta name="date" content="2025-12-10" />

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="6 The francisella use case: a MaxQuant LFQ DDA dataset with technical replication | Statistical analysis of mass spectrometry-based proteomics data">

<title>6 The francisella use case: a MaxQuant LFQ DDA dataset with technical replication | Statistical analysis of mass spectrometry-based proteomics data</title>

<link href="libs/tufte-css-2015.12.29/tufte.css" rel="stylesheet" />
<link href="libs/tufte-css-2015.12.29/envisioned.css" rel="stylesheet" />
<link href="libs/msmb-css-0/msmb.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);

e.style.display = ((e.style.display!='none') ? 'none' : 'block');

if(f.classList.contains('fa-plus-square')) {
    f.classList.add('fa-minus-square')
    f.classList.remove('fa-plus-square')
} else {
    f.classList.add('fa-plus-square')
    f.classList.remove('fa-minus-square')
}

}
</script>
<script>
function copy_link(id) {
  var dummy = document.createElement('input'),
  text = window.location.href.split(/[?#]/)[0] + '#' + id;
  document.body.appendChild(dummy);
  dummy.value = text;
  dummy.select();
  document.execCommand('copy');
  document.body.removeChild(dummy);
  
  var tooltip = document.getElementById(id + '-tooltip');
  tooltip.innerHTML = 'Copied!';
}

function reset_tooltip(id) {
  var tooltip = document.getElementById(id);
  tooltip.innerHTML = 'Copy link';
}
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }

code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />

</head>

<body>



<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul class="navbar">
<li class="msmb"><p class="title">Statistical analysis of mass spectrometry-based proteomics data<p><p class="author">Christophe Vanderaa, Stijn Vandenbulcke, Lieven Clement</p>
<li class="dropdown" style="float:right">
<a href="javascript:void(0)" class="dropbtn">&#x25BE; Chapters</a>
<div class="dropdown-content">
<a href="index.html" id="toc-preamble"><span class="toc-section-number">1</span> Preamble</a>
<a href="sec-basics.html" id="toc-sec-basics"><span class="toc-section-number">2</span> Statistical analysis with msqrob2</a>
<a href="sec-advanced.html" id="toc-sec-advanced"><span class="toc-section-number">3</span> Advanced statistical analysis with msqrob2</a>
<a href="sec-benchmarking.html" id="toc-sec-benchmarking"><span class="toc-section-number">4</span> Benchmarking workflows</a>
<a href="sec-workflow-optimisation.html" id="toc-sec-workflow-optimisation"><span class="toc-section-number">5</span> Optimisation of a data analysis workflow</a>
<a id="active-page" href="sec-francisella.html" id="toc-sec-francisella"><span class="toc-section-number">6</span> The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</a><ul class="toc-sections">
<li class="toc"><a href="#introduction"> Introduction</a></li>
<li class="toc"><a href="#load-packages-4"> Load packages</a></li>
<li class="toc"><a href="#load-data"> Load data</a></li>
<li class="toc"><a href="#data-preprocessing-3"> Data preprocessing</a></li>
<li class="toc"><a href="#data-exploration-1"> Data exploration</a></li>
<li class="toc"><a href="#data-modelling-2"> Data modelling</a></li>
<li class="toc"><a href="#statistical-inference-1"> Statistical inference</a></li>
<li class="toc"><a href="#conclusion-3"> Conclusion</a></li>
</ul>
<a href="sec-heart.html" id="toc-sec-heart"><span class="toc-section-number">7</span> Heart use case: a MaxQuant LFQ DDA dataset with a more complex design</a>
<a href="sec-mouse-diet.html" id="toc-sec-mouse-diet"><span class="toc-section-number">8</span> The mouse diet use case: a Skyline TMT DDA dataset</a>
<a href="sec-end.html" id="toc-sec-end"><span class="toc-section-number">9</span> Useful links and session information</a>
</div>
</li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="sec-francisella" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> The francisella use case: a MaxQuant LFQ DDA dataset with technical replication</h1>
<div id="introduction" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> Introduction<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('introduction')" onmouseout="reset_tooltip('introduction-tooltip')"><span class="tooltiptext" id="introduction-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>In this chapter we show how to analyse LFQ data with technical
replication. In the data analysis we have to acknowledge
pseudo-replication, i.e. data for multiple technical replicates from
the same biological repeat are correlated and we cannot treat these as
independent repeats.</p>
<p>We will build upon the mixed model framework in msqrob2 to address
this correlation by introducing a random effect for each biological
repeat.</p>
<p>If you never used <code>msqrob2</code>, we suggest to familiarise yourself
with the <a href="sec-basics.html#sec-basics">concepts chapter</a> first.</p>
</div>
<div id="load-packages-4" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> Load packages<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('load-packages-4')" onmouseout="reset_tooltip('load-packages-4-tooltip')"><span class="tooltiptext" id="load-packages-4-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>First, we load the <code>msqrob2</code> package and additional packages for data
manipulation and visualisation.</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="sec-francisella.html#cb251-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"msqrob2"</span>)</span>
<span id="cb251-2"><a href="sec-francisella.html#cb251-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb251-3"><a href="sec-francisella.html#cb251-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ggrepel"</span>)</span>
<span id="cb251-4"><a href="sec-francisella.html#cb251-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span></code></pre></div>
<p>We also configure the <a href="sec-basics.html#sec-parallel">parallelisation</a> framework.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="sec-francisella.html#cb252-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocParallel"</span>)</span>
<span id="cb252-2"><a href="sec-francisella.html#cb252-2" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SerialParam</span>())</span></code></pre></div>
</div>
<div id="load-data" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> Load data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('load-data')" onmouseout="reset_tooltip('load-data-tooltip')"><span class="tooltiptext" id="load-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<div id="experimental-context-1" class="section level3" number="6.3.1">
<h3>
<span class="header-section-number">6.3.1</span> Experimental context<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('experimental-context-1')" onmouseout="reset_tooltip('experimental-context-1-tooltip')"><span class="tooltiptext" id="experimental-context-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>A study on the facultative pathogen Francisella tularensis was
conceived by <span class="citation">(<label for="tufte-mn-23" class="margin-toggle">⊕</label><input type="checkbox" id="tufte-mn-23" class="margin-toggle">Ramond et al. 2015<span class="marginnote">Ramond, Elodie, Gael Gesbert, Ida Chiara Guerrera, Cerina Chhuon, Marion Dupuis, Mélanie Rigard, Thomas Henry, Monique Barel, and Alain Charbit. 2015. <span>“Importance of Host Cell Arginine Uptake in Francisella Phagosomal Escape and Ribosomal Protein Amounts.”</span> <em>Mol. Cell. Proteomics</em> 14 (4): 870–81.</span>)</span>. F. tularensis enters the cells of its
host by phagocytosis. The authors showed that F. tularensis is
arginine deficient and imports arginine from the host cell via an
arginine transporter, ArgP, in order to efficiently escape from the
phagosome and reach the cytosolic compartment, where it can actively
multiply. In their study, they compared the proteome of wild type F.
tularensis (WT) to ArgP-gene deleted F. tularensis (knock-out, D8).
For this exercise, we use a subset of the F. tularensis dataset where
bacterial cultures were grown in biological triplicate and each biorep
was run in technical triplicate on a nanoRSLC-Q Exactive PLUS
instrument.</p>
</div>
<div id="getting-the-data" class="section level3" number="6.3.2">
<h3>
<span class="header-section-number">6.3.2</span> Getting the data<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('getting-the-data')" onmouseout="reset_tooltip('getting-the-data-tooltip')"><span class="tooltiptext" id="getting-the-data-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>The data were searched with MaxQuant version 1.4.1.2. and are
available on the PRIDE repository:
<a href="https://www.ebi.ac.uk/pride/archive/projects/PXD001584">PXD001584</a>.
We download the peptides file<label for="tufte-sn-37" class="margin-toggle sidenote-number">37</label><input type="checkbox" id="tufte-sn-37" class="margin-toggle"><span class="sidenote"><span class="sidenote-number">37</span> The file is
<a href="sec-advanced.html#sec-caching">locally cached</a></span>.</p>
<p><strong>TODO</strong> put on Zenodo and use BiocFileCache</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="sec-francisella.html#cb253-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"BiocFileCache"</span>)</span>
<span id="cb253-2"><a href="sec-francisella.html#cb253-2" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> <span class="fu">BiocFileCache</span>()</span>
<span id="cb253-3"><a href="sec-francisella.html#cb253-3" tabindex="-1"></a>pepFile <span class="ot">&lt;-</span> <span class="fu">bfcrpath</span>(bfc, <span class="st">"https://raw.githubusercontent.com/statOmics/MSqRobSumPaper/master/Francisella/data/maxquant/peptides.txt"</span>)</span></code></pre></div>
<p>After downloading the files, we can load the <a href="%7B#sec-peptide_table%7D">peptide
table</a>. Contrarily a <a href="sec-advanced.html#sec-psm_table">PSM
table</a>, the peptide table is provided in a “wide
format”, meaning that each row represents a single peptide and that
each quantification column (that starts with <code>"Intensity"</code>) represents
a single sample.</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="sec-francisella.html#cb254-1" tabindex="-1"></a>peps <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(pepFile)</span>
<span id="cb254-2"><a href="sec-francisella.html#cb254-2" tabindex="-1"></a>quantcols <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"Intensity</span><span class="sc">\\</span><span class="st">."</span>, <span class="fu">names</span>(peps), <span class="at">value =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="3%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="8%">
<col width="17%">
<col width="0%">
<col width="1%">
</colgroup>
<thead><tr class="header">
<th align="left">Sequence</th>
<th align="left">Amino.acid.before</th>
<th align="left">First.amino.acid</th>
<th align="left">Second.amino.acid</th>
<th align="left">Second.last.amino.acid</th>
<th align="left">Last.amino.acid</th>
<th align="left">Amino.acid.after</th>
<th align="right">A.Count</th>
<th align="right">R.Count</th>
<th align="right">N.Count</th>
<th align="right">D.Count</th>
<th align="right">C.Count</th>
<th align="right">Q.Count</th>
<th align="right">E.Count</th>
<th align="right">G.Count</th>
<th align="right">H.Count</th>
<th align="right">I.Count</th>
<th align="right">L.Count</th>
<th align="right">K.Count</th>
<th align="right">M.Count</th>
<th align="right">F.Count</th>
<th align="right">P.Count</th>
<th align="right">S.Count</th>
<th align="right">T.Count</th>
<th align="right">W.Count</th>
<th align="right">Y.Count</th>
<th align="right">V.Count</th>
<th align="right">U.Count</th>
<th align="right">Length</th>
<th align="right">Missed.cleavages</th>
<th align="right">Mass</th>
<th align="left">Proteins</th>
<th align="left">GI.number</th>
<th align="left">Leading.razor.protein</th>
<th align="left">Protein.names</th>
<th align="left">Unique..Groups.</th>
<th align="left">Unique..Proteins.</th>
<th align="left">Charges</th>
<th align="right">PEP</th>
<th align="left">Score</th>
<th align="right">Experiment.1WT_20_2h_n3_1</th>
<th align="right">Experiment.1WT_20_2h_n3_2</th>
<th align="right">Experiment.1WT_20_2h_n3_3</th>
<th align="right">Experiment.1WT_20_2h_n4_1</th>
<th align="right">Experiment.1WT_20_2h_n4_2</th>
<th align="right">Experiment.1WT_20_2h_n4_3</th>
<th align="right">Experiment.1WT_20_2h_n5_1</th>
<th align="right">Experiment.1WT_20_2h_n5_2</th>
<th align="right">Experiment.1WT_20_2h_n5_3</th>
<th align="right">Experiment.3D8_20_2h_n3_1</th>
<th align="right">Experiment.3D8_20_2h_n3_2</th>
<th align="right">Experiment.3D8_20_2h_n3_3</th>
<th align="right">Experiment.3D8_20_2h_n4_1</th>
<th align="right">Experiment.3D8_20_2h_n4_2</th>
<th align="right">Experiment.3D8_20_2h_n4_3</th>
<th align="right">Experiment.3D8_20_2h_n5_1</th>
<th align="right">Experiment.3D8_20_2h_n5_2</th>
<th align="right">Experiment.3D8_20_2h_n5_3</th>
<th align="right">Intensity</th>
<th align="right">Intensity.1WT_20_2h_n3_1</th>
<th align="right">Intensity.1WT_20_2h_n3_2</th>
<th align="right">Intensity.1WT_20_2h_n3_3</th>
<th align="right">Intensity.1WT_20_2h_n4_1</th>
<th align="right">Intensity.1WT_20_2h_n4_2</th>
<th align="right">Intensity.1WT_20_2h_n4_3</th>
<th align="right">Intensity.1WT_20_2h_n5_1</th>
<th align="right">Intensity.1WT_20_2h_n5_2</th>
<th align="right">Intensity.1WT_20_2h_n5_3</th>
<th align="right">Intensity.3D8_20_2h_n3_1</th>
<th align="right">Intensity.3D8_20_2h_n3_2</th>
<th align="right">Intensity.3D8_20_2h_n3_3</th>
<th align="right">Intensity.3D8_20_2h_n4_1</th>
<th align="right">Intensity.3D8_20_2h_n4_2</th>
<th align="right">Intensity.3D8_20_2h_n4_3</th>
<th align="right">Intensity.3D8_20_2h_n5_1</th>
<th align="right">Intensity.3D8_20_2h_n5_2</th>
<th align="right">Intensity.3D8_20_2h_n5_3</th>
<th align="left">Reverse</th>
<th align="left">Contaminant</th>
<th align="right">id</th>
<th align="left">Protein.group.IDs</th>
<th align="left">Mod..peptide.IDs</th>
<th align="left">Evidence.IDs</th>
<th align="left">MS.MS.IDs</th>
<th align="right">Best.MS.MS</th>
<th align="left">Oxidation..M..site.IDs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AAAEELDTR</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">T</td>
<td align="left">R</td>
<td align="left">K</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">974.4669</td>
<td align="left">WP_003038655</td>
<td align="left">gi|118497196</td>
<td align="left">gi|118497196</td>
<td align="left">hypothetical protein [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">1,2</td>
<td align="right">0.0001253</td>
<td align="left">59.35</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.3295e+09</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9290400</td>
<td align="right">92996000</td>
<td align="right">0</td>
<td align="right">98059000</td>
<td align="right">0</td>
<td align="right">80803000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">95433000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">0</td>
<td align="left">419</td>
<td align="left">0</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td align="left">0;1;2;3;4;5;6;7;8;9;10;11;12;13;14;15;16;17;18</td>
<td align="right">15</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAAGFVITASHNK</td>
<td align="left">R</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">N</td>
<td align="left">K</td>
<td align="left">F</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="right">0</td>
<td align="right">1285.6779</td>
<td align="left">WP_003041237</td>
<td align="left">gi|118498194</td>
<td align="left">gi|118498194</td>
<td align="left">phosphoglucosamine mutase [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000530</td>
<td align="left">45.825</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.6719e+08</td>
<td align="right">0</td>
<td align="right">20679000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">17358000</td>
<td align="right">13841000</td>
<td align="right">12278000</td>
<td align="right">10712000</td>
<td align="right">0</td>
<td align="right">9500700</td>
<td align="right">11111000</td>
<td align="right">8723500</td>
<td align="right">8125200</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">1</td>
<td align="left">1150</td>
<td align="left">1</td>
<td align="left">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td align="left">19;20;21;22;23;24;25;26;27;28;29;30;31;32;33</td>
<td align="right">26</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">AAANEYELALAYSIEEVAPDLHK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">H</td>
<td align="left">K</td>
<td align="left">Y</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">23</td>
<td align="right">0</td>
<td align="right">2516.2435</td>
<td align="left">WP_003038915</td>
<td align="left">gi|118497331</td>
<td align="left">gi|118497331</td>
<td align="left">glycine–tRNA ligase subunit beta [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">3</td>
<td align="right">0.0000000</td>
<td align="left">84.327</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">5.5675e+08</td>
<td align="right">28853000</td>
<td align="right">28091000</td>
<td align="right">30436000</td>
<td align="right">19476000</td>
<td align="right">22050000</td>
<td align="right">20687000</td>
<td align="right">36588000</td>
<td align="right">28654000</td>
<td align="right">7009100</td>
<td align="right">14174000</td>
<td align="right">8548000</td>
<td align="right">7859200</td>
<td align="right">14871000</td>
<td align="right">11690000</td>
<td align="right">4470500</td>
<td align="right">0</td>
<td align="right">4048100</td>
<td align="right">3422300</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">2</td>
<td align="left">513</td>
<td align="left">2</td>
<td align="left">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76</td>
<td align="left">34;35;36;37;38;39;40;41;42;43;44;45;46;47;48;49;50;51;52;53;54;55;56;57;58;59;60;61;62;63;64;65;66;67;68;69;70;71;72;73;74;75;76;77;78;79</td>
<td align="right">39</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAANNPQLEAFK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">F</td>
<td align="left">K</td>
<td align="left">K</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1272.6463</td>
<td align="left">WP_003038264</td>
<td align="left">gi|118496879</td>
<td align="left">gi|118496879</td>
<td align="left">molecular chaperone HtpG [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000000</td>
<td align="left">108.56</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1.4830e+10</td>
<td align="right">339830000</td>
<td align="right">420390000</td>
<td align="right">393930000</td>
<td align="right">235060000</td>
<td align="right">381090000</td>
<td align="right">360270000</td>
<td align="right">255710000</td>
<td align="right">311370000</td>
<td align="right">271440000</td>
<td align="right">225140000</td>
<td align="right">289880000</td>
<td align="right">282500000</td>
<td align="right">209490000</td>
<td align="right">223610000</td>
<td align="right">219700000</td>
<td align="right">169120000</td>
<td align="right">178150000</td>
<td align="right">170400000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">3</td>
<td align="left">199</td>
<td align="left">3</td>
<td align="left">77;78;79;80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124</td>
<td align="left">80;81;82;83;84;85;86;87;88;89;90;91;92;93;94;95;96;97;98;99;100;101;102;103;104;105;106;107;108;109;110;111;112;113;114;115;116;117;118;119;120;121;122;123;124;125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155;156;157;158;159;160;161;162;163;164;165;166;167;168;169;170;171;172</td>
<td align="right">159</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">AAASAGLVDEK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">E</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="right">0</td>
<td align="right">1030.5295</td>
<td align="left">WP_003035781</td>
<td align="left">gi|118497152</td>
<td align="left">gi|118497152</td>
<td align="left">delta-aminolevulinic acid dehydratase [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0000951</td>
<td align="left">67.385</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">4.8209e+09</td>
<td align="right">240280000</td>
<td align="right">0</td>
<td align="right">241890000</td>
<td align="right">222250000</td>
<td align="right">189930000</td>
<td align="right">0</td>
<td align="right">160440000</td>
<td align="right">147230000</td>
<td align="right">143340000</td>
<td align="right">135910000</td>
<td align="right">185740000</td>
<td align="right">168620000</td>
<td align="right">160240000</td>
<td align="right">158000000</td>
<td align="right">0</td>
<td align="right">92542000</td>
<td align="right">102350000</td>
<td align="right">100090000</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">4</td>
<td align="left">388</td>
<td align="left">4</td>
<td align="left">125;126;127;128;129;130;131;132;133;134;135;136;137;138;139;140;141;142;143;144;145;146;147;148;149;150;151;152;153;154;155</td>
<td align="left">173;174;175;176;177;178;179;180;181;182;183;184;185;186;187;188;189;190;191;192;193;194;195;196;197;198;199;200;201;202;203</td>
<td align="right">191</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">AAASLDLYSYPK</td>
<td align="left">K</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">P</td>
<td align="left">K</td>
<td align="left">V</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="right">0</td>
<td align="right">1297.6554</td>
<td align="left">WP_003039212</td>
<td align="left">gi|118497492</td>
<td align="left">gi|118497492</td>
<td align="left">phosphorylase [Francisella tularensis]</td>
<td align="left">yes</td>
<td align="left">yes</td>
<td align="left">2</td>
<td align="right">0.0125630</td>
<td align="left">31.675</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">1.3715e+08</td>
<td align="right">0</td>
<td align="right">31069000</td>
<td align="right">31448000</td>
<td align="right">0</td>
<td align="right">27721000</td>
<td align="right">28657000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">18253000</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left"></td>
<td align="left"></td>
<td align="right">5</td>
<td align="left">624</td>
<td align="left">5</td>
<td align="left">156;157;158;159;160</td>
<td align="left">204;205;206;207;208</td>
<td align="right">205</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<p>We now extract the <a href="sec-basics.html#sec-annotation_table">sample annotations</a>. We
will build a table where each row in the annotation table contains
information for one sample (the table below shows the first 6 rows).
This information is extracted from the sample names.</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="sec-francisella.html#cb255-1" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">quantCols =</span> quantcols) <span class="sc">|&gt;</span> </span>
<span id="cb255-2"><a href="sec-francisella.html#cb255-2" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"_20_"</span>, quantCols) <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"_n</span><span class="sc">\\</span><span class="st">d"</span>, quantCols)) <span class="sc">|&gt;</span> </span>
<span id="cb255-3"><a href="sec-francisella.html#cb255-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype  =</span> <span class="fu">substr</span>(quantCols, <span class="dv">12</span>, <span class="dv">13</span>), </span>
<span id="cb255-4"><a href="sec-francisella.html#cb255-4" tabindex="-1"></a>         <span class="at">biorep  =</span> <span class="fu">paste0</span>(genotype, <span class="st">"_"</span>, <span class="fu">substr</span>(quantCols, <span class="dv">21</span>, <span class="dv">22</span>)),</span>
<span id="cb255-5"><a href="sec-francisella.html#cb255-5" tabindex="-1"></a>         <span class="at">run =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(quantcols))</span></code></pre></div>
<table>
<thead><tr class="header">
<th align="left">quantCols</th>
<th align="left">genotype</th>
<th align="left">biorep</th>
<th align="right">run</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n3_1</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n3_2</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n3_3</td>
<td align="left">WT</td>
<td align="left">WT_n3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n4_1</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">Intensity.1WT_20_2h_n4_2</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">Intensity.1WT_20_2h_n4_3</td>
<td align="left">WT</td>
<td align="left">WT_n4</td>
<td align="right">6</td>
</tr>
</tbody>
</table>
</div>
<div id="the-qfeatures-data-class" class="section level3" number="6.3.3">
<h3>
<span class="header-section-number">6.3.3</span> The <code>QFeatures</code> data class<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('the-qfeatures-data-class')" onmouseout="reset_tooltip('the-qfeatures-data-class-tooltip')"><span class="tooltiptext" id="the-qfeatures-data-class-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We combine the two tables into a <a href="sec-basics.html#sec-qfeatures"><code>QFeatures</code> object</a>.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="sec-francisella.html#cb256-1" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">readQFeatures</span>(</span>
<span id="cb256-2"><a href="sec-francisella.html#cb256-2" tabindex="-1"></a>  peps, <span class="at">colData =</span> coldata, <span class="at">fnames =</span> <span class="st">"Sequence"</span>, <span class="at">name =</span> <span class="st">"peptides"</span></span>
<span id="cb256-3"><a href="sec-francisella.html#cb256-3" tabindex="-1"></a>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 1 set:
## 
##  [1] peptides: SummarizedExperiment with 10693 rows and 18 columns</code></pre>
<p>We now have a <code>QFeatures</code> object with 1 set, containing <code>r nrows(pe)[[1]]</code> rows (peptides) and 18 columns
(samples).</p>
</div>
</div>
<div id="data-preprocessing-3" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> Data preprocessing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-preprocessing-3')" onmouseout="reset_tooltip('data-preprocessing-3-tooltip')"><span class="tooltiptext" id="data-preprocessing-3-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p><code>msqrob2</code> relies on the <code>QFeatures</code> data structure, meaning that we
can directly make use of <code>QFeatures</code>’ data preprocessing functionality
(see also the <code>QFeatures</code>
<a href="https://rformassspectrometry.github.io/QFeatures/articles/Processing.html">documentation</a>).</p>
<div id="encoding-missing-values-1" class="section level3" number="6.4.1">
<h3>
<span class="header-section-number">6.4.1</span> Encoding missing values<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('encoding-missing-values-1')" onmouseout="reset_tooltip('encoding-missing-values-1-tooltip')"><span class="tooltiptext" id="encoding-missing-values-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Peptides with zero intensities should be
<a href="sec-basics.html#sec-encode_missing">encoded</a> using <code>NA</code>.</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="sec-francisella.html#cb258-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">zeroIsNA</span>(pe, <span class="st">"peptides"</span>)</span></code></pre></div>
<p>We calculate how many non zero intensities we have per peptide and
this is often useful for filtering.</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="sec-francisella.html#cb259-1" tabindex="-1"></a>naResults <span class="ot">&lt;-</span> <span class="fu">nNA</span>(pe, <span class="st">"peptides"</span>)</span>
<span id="cb259-2"><a href="sec-francisella.html#cb259-2" tabindex="-1"></a><span class="fu">data.frame</span>(naResults<span class="sc">$</span>nNArows) <span class="sc">|&gt;</span> </span>
<span id="cb259-3"><a href="sec-francisella.html#cb259-3" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb259-4"><a href="sec-francisella.html#cb259-4" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> nNA) <span class="sc">+</span></span>
<span id="cb259-5"><a href="sec-francisella.html#cb259-5" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span></code></pre></div>
<p><img src="figure/francisella_nNA_hist-1.png"></p>
</div>
<div id="peptide-filtering" class="section level3" number="6.4.2">
<h3>
<span class="header-section-number">6.4.2</span> Peptide filtering<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('peptide-filtering')" onmouseout="reset_tooltip('peptide-filtering-tooltip')"><span class="tooltiptext" id="peptide-filtering-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We filter peptides based on 3 criteria (see <a href="sec-advanced.html#psm-filtering">PSM filtering</a>).</p>
<ol style="list-style-type: decimal">
<li>Remove failed protein inference</li>
</ol>
<p>We remove peptides that could not be uniquely mapped to a protein.</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="sec-francisella.html#cb260-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe,</span>
<span id="cb260-2"><a href="sec-francisella.html#cb260-2" tabindex="-1"></a>  <span class="sc">~</span> Proteins <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="do">## Remove failed protein inference</span></span>
<span id="cb260-3"><a href="sec-francisella.html#cb260-3" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">";"</span>, Proteins)) <span class="do">## Remove protein groups</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Remove reverse sequences and contaminants</li>
</ol>
<p>We now remove the contaminants and peptides that map to decoy
sequences. These features bear no information of interest and will
reduce the statistical power upon multiple test adjustment.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="sec-francisella.html#cb261-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> Reverse <span class="sc">!=</span> <span class="st">"+"</span> <span class="sc">&amp;</span> Contaminant <span class="sc">!=</span> <span class="st">"+"</span>)</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Drop peptides that were only identified in a single biorepeat</li>
</ol>
<p>Note, that in experiments without technical repeats we filter on the
number of samples in which a peptide is picked up (this is typically
performed using <code>filterNA()</code>). Here, we will require that a peptide is
picked up in at least two biorepeats. We compute the number
of biorepeats that were observed for each peptide (that is the number
of biorepeats that contain at least one observed value).</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="sec-francisella.html#cb262-1" tabindex="-1"></a><span class="fu">rowData</span>(pe[[<span class="st">"peptides"</span>]])<span class="sc">$</span>nNonZeroBiorep <span class="ot">&lt;-</span> <span class="fu">apply</span>(</span>
<span id="cb262-2"><a href="sec-francisella.html#cb262-2" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">"peptides"</span>]]), <span class="dv">1</span>, <span class="cf">function</span>(intensity)</span>
<span id="cb262-3"><a href="sec-francisella.html#cb262-3" tabindex="-1"></a>    pe<span class="sc">$</span>biorep[<span class="sc">!</span><span class="fu">is.na</span>(intensity)] <span class="sc">|&gt;</span> </span>
<span id="cb262-4"><a href="sec-francisella.html#cb262-4" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">|&gt;</span> </span>
<span id="cb262-5"><a href="sec-francisella.html#cb262-5" tabindex="-1"></a>    <span class="fu">length</span>()</span>
<span id="cb262-6"><a href="sec-francisella.html#cb262-6" tabindex="-1"></a>)</span></code></pre></div>
<p>We keep peptides that are observed in at least two biorepeats.</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="sec-francisella.html#cb263-1" tabindex="-1"></a>(pe <span class="ot">&lt;-</span> <span class="fu">filterFeatures</span>(pe, <span class="sc">~</span> nNonZeroBiorep <span class="sc">&gt;=</span> <span class="dv">2</span>))</span></code></pre></div>
<pre><code>## An instance of class QFeatures (type: bulk) with 1 set:
## 
##  [1] peptides: SummarizedExperiment with 7542 rows and 18 columns</code></pre>
<p>We keep 7542 peptides upon filtering.</p>
</div>
<div id="standard-preprocessing-workflow" class="section level3" number="6.4.3">
<h3>
<span class="header-section-number">6.4.3</span> Standard preprocessing workflow<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('standard-preprocessing-workflow')" onmouseout="reset_tooltip('standard-preprocessing-workflow-tooltip')"><span class="tooltiptext" id="standard-preprocessing-workflow-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can now prepare the data for modelling. The workflow ensures the
data complies to <code>msqrob2</code>’s requirements:</p>
<ol style="list-style-type: decimal">
<li>Intensities are <a href="sec-basics.html#sec-log2">log-transformed</a>.</li>
</ol>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="sec-francisella.html#cb265-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">logTransform</span>(pe, <span class="at">base =</span> <span class="dv">2</span>, <span class="at">i =</span> <span class="st">"peptides"</span>, <span class="at">name =</span> <span class="st">"peptides_log"</span>)</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Normalisation with the Median of Ratios method.</li>
</ol>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="sec-francisella.html#cb266-1" tabindex="-1"></a>pseudoRef <span class="ot">&lt;-</span> <span class="fu">assay</span>(pe[[<span class="st">"peptides_log"</span>]]) <span class="sc">|&gt;</span> </span>
<span id="cb266-2"><a href="sec-francisella.html#cb266-2" tabindex="-1"></a>  <span class="fu">rowMeans</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#1. Calculate the row means </span></span>
<span id="cb266-3"><a href="sec-francisella.html#cb266-3" tabindex="-1"></a></span>
<span id="cb266-4"><a href="sec-francisella.html#cb266-4" tabindex="-1"></a>nfLog <span class="ot">&lt;-</span> <span class="fu">sweep</span>(</span>
<span id="cb266-5"><a href="sec-francisella.html#cb266-5" tabindex="-1"></a>  <span class="fu">assay</span>(pe[[<span class="st">"peptides_log"</span>]]), </span>
<span id="cb266-6"><a href="sec-francisella.html#cb266-6" tabindex="-1"></a>  <span class="at">MARGIN =</span> <span class="dv">1</span>, </span>
<span id="cb266-7"><a href="sec-francisella.html#cb266-7" tabindex="-1"></a>  pseudoRef) <span class="sc">|&gt;</span> <span class="co">#2. Subtract the row means row-by-row (MARGIN = 1)</span></span>
<span id="cb266-8"><a href="sec-francisella.html#cb266-8" tabindex="-1"></a>  <span class="fu">colMedians</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)  <span class="co">#3. Calculate the column median </span></span>
<span id="cb266-9"><a href="sec-francisella.html#cb266-9" tabindex="-1"></a></span>
<span id="cb266-10"><a href="sec-francisella.html#cb266-10" tabindex="-1"></a>pe <span class="ot">&lt;-</span> </span>
<span id="cb266-11"><a href="sec-francisella.html#cb266-11" tabindex="-1"></a>  <span class="fu">sweep</span>(pe, </span>
<span id="cb266-12"><a href="sec-francisella.html#cb266-12" tabindex="-1"></a>        <span class="at">MARGIN =</span> <span class="dv">2</span>, </span>
<span id="cb266-13"><a href="sec-francisella.html#cb266-13" tabindex="-1"></a>        <span class="at">STATS =</span> nfLog , </span>
<span id="cb266-14"><a href="sec-francisella.html#cb266-14" tabindex="-1"></a>        <span class="at">i =</span> <span class="st">"peptides_log"</span>, </span>
<span id="cb266-15"><a href="sec-francisella.html#cb266-15" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">"peptides_norm"</span>) <span class="co">#4. Subtract log2 norm factor column-by-column (MARGIN = 2)</span></span></code></pre></div>
<!--
2. Samples are normalised by substracting the sample median (see [Normalisation])
-->
<p>Upon the normalisation the density curves should be nicely centred. To
confirm this, we will plot the intensity distributions for each
biorepeat (francisella culture). <code>longForm()</code> seamlessly combines the quantification
and annotation data into a table suitable for <code>ggplot2</code> visualisation.
We also subset the object with the data before and after normalisation.</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="sec-francisella.html#cb267-1" tabindex="-1"></a><span class="fu">longForm</span>(pe[, , <span class="fu">c</span>(<span class="st">"peptides_log"</span>, <span class="st">"peptides_norm"</span>)], <span class="at">colvar =</span> <span class="st">"biorep"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb267-2"><a href="sec-francisella.html#cb267-2" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb267-3"><a href="sec-francisella.html#cb267-3" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">group =</span> colname, <span class="at">color =</span> biorep) <span class="sc">+</span></span>
<span id="cb267-4"><a href="sec-francisella.html#cb267-4" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb267-5"><a href="sec-francisella.html#cb267-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> assay, <span class="at">scale =</span> <span class="st">"free"</span>)</span></code></pre></div>
<p><img src="figure/francisella_after_norm-1.png"></p>
<ol start="3" style="list-style-type: decimal">
<li>
<a href="sec-basics.html#sec-summarisation">Summarisation</a> to protein level.</li>
</ol>
<p>We use the robust summary approach to infer protein-level data from
peptide-level data, accounting for the fact that different peptides
have ionisation efficiencies hence leading to different intensity
baselines.</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="sec-francisella.html#cb268-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">aggregateFeatures</span>(</span>
<span id="cb268-2"><a href="sec-francisella.html#cb268-2" tabindex="-1"></a>  pe, <span class="at">i =</span> <span class="st">"peptides_norm"</span>, <span class="at">fcol =</span> <span class="st">"Proteins"</span>, </span>
<span id="cb268-3"><a href="sec-francisella.html#cb268-3" tabindex="-1"></a>  <span class="at">fun =</span> MsCoreUtils<span class="sc">::</span>robustSummary, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">name =</span> <span class="st">"proteins"</span></span>
<span id="cb268-4"><a href="sec-francisella.html#cb268-4" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="data-exploration-1" class="section level2" number="6.5">
<h2>
<span class="header-section-number">6.5</span> Data exploration<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-exploration-1')" onmouseout="reset_tooltip('data-exploration-1-tooltip')"><span class="tooltiptext" id="data-exploration-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>We will explore the main sources of variation in the data using
<a href="sec-basics.html#sec_data_exploration">MDS</a>.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="sec-francisella.html#cb269-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"scater"</span>)</span>
<span id="cb269-2"><a href="sec-francisella.html#cb269-2" tabindex="-1"></a><span class="fu">getWithColData</span>(pe, <span class="st">"proteins"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb269-3"><a href="sec-francisella.html#cb269-3" tabindex="-1"></a>  <span class="fu">as</span>(<span class="st">"SingleCellExperiment"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb269-4"><a href="sec-francisella.html#cb269-4" tabindex="-1"></a>  <span class="fu">runMDS</span>(<span class="at">exprs_values =</span> <span class="dv">1</span>) <span class="sc">|&gt;</span> </span>
<span id="cb269-5"><a href="sec-francisella.html#cb269-5" tabindex="-1"></a>  <span class="fu">plotMDS</span>(<span class="at">colour_by =</span> <span class="st">"genotype"</span>)</span></code></pre></div>
<p><img src="figure/francisella_mds-1.png"></p>
<p>Note that the samples upon robust summarisation show a clear
separation according to the genotype in the first dimension of the MDS
plot.</p>
</div>
<div id="data-modelling-2" class="section level2" number="6.6">
<h2>
<span class="header-section-number">6.6</span> Data modelling<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('data-modelling-2')" onmouseout="reset_tooltip('data-modelling-2-tooltip')"><span class="tooltiptext" id="data-modelling-2-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>The preprocessed data can now be modelled to answer biologically
relevant questions. As described above, samples (bacterial cultures)
originate from either a wildtype (WT) or a ArgP knockout (D8). Each
genotype was cultured in biological triplicate. Each biological
triplicate was acquired in technical triplicate, leading to <span class="math inline">\(2 \times
3 \times 3 = 18\)</span> samples. In this context, we are interested in the
effects of genotype on the protein abundances.</p>
<p>The table below confirms we have a balanced design
for each condition and biological triplicate.</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="sec-francisella.html#cb270-1" tabindex="-1"></a><span class="fu">table</span>(<span class="at">genotype =</span> pe<span class="sc">$</span>genotype, <span class="at">biorep =</span> pe<span class="sc">$</span>biorep)</span></code></pre></div>
<pre><code>##         biorep
## genotype D8_n3 D8_n4 D8_n5 WT_n3 WT_n4 WT_n5
##       D8     3     3     3     0     0     0
##       WT     0     0     0     3     3     3</code></pre>
<div id="sources-of-variation-1" class="section level3" number="6.6.1">
<h3>
<span class="header-section-number">6.6.1</span> Sources of variation<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('sources-of-variation-1')" onmouseout="reset_tooltip('sources-of-variation-1-tooltip')"><span class="tooltiptext" id="sources-of-variation-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We will model two <a href="sec-basics.html#sec-modelling">sources of variation</a>:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Genotype</strong>: we model the source of variation induced by the
experimental group of interest as a <strong>fixed effect</strong>. Fixed effects
are effect that are considered non-random, i.e. the treatment
effect is assumed to be the same and reproducible across repeated
experiments, but it is unknown and has to be estimated. We will
include <code>genotype</code> as a fixed effect that models the fact that a
change in genotype can induce changes in protein abundance.</p></li>
<li><p><strong>Biological replicate effect</strong>: the experiment involves biological
replication as the bacterial cultures are repeated. Replicate-specific
effects occurs due to uncontrollable factors, such as variation in
the number of bacterium seeded, position in the incubator,
transient contamination,… Two bacterial cultures will never
provide exactly the same sample material. These effects are typically
modelled as random effects which are considered as a random sample
from the population of all possible mice and are assumed to be
i.i.d normally distributed with mean 0 and constant variance,
<span class="math inline">\(u_{biorep} \sim
N(0,\sigma^{2,\text{b}})\)</span>. The use of random effects thus
models the correlation in the data, explicitly. We expect that
intensities from the same bacterial culture are more alike than
intensities between cultures.</p></li>
</ol>
<p>Hence, the variance-covariance matrix of the 18 protein abundance
values <span class="math inline">\(\mathbf{Y}=(y_{wt\_n3,1}, y_{wt\_n3,2}, y_{wt\_n3,3},
y_{wt\_n4,1} \ldots y_{d8\_n5,1}, y_{d8\_n5,2}, y_{d8\_n5,3})^T\)</span> is
assumed to have a block diagonal structure, with as variance
<span class="math inline">\(\sigma_b^2 + \sigma_\epsilon^2\)</span> and the covariance between protein
abundance values of technical replicates from the same biorepeat
equals <span class="math inline">\(\sigma_b^2\)</span> (with _^2 the variance of the
residuals). More details on mixed models can be found in the <a href="sec-advanced.html#sec-random_effect">advanced
chapter</a>.</p>
<p>We model the protein level expression values using <code>msqrob2</code>.
<code>msqrob2</code> workflows rely on linear mixed models, which are models that
can estimate and predict fixed and random effects, respectively. The
fixed effect are estimated using robust regression to avoid that
outliers distort the statistical outcome.</p>
<p>Note, that we cannot use ridge regression here to further stabilize
the parameter estimation of the fixed effects. Ridge regression
typically performs better the more slope parameters there are in the
mean model (more complex designs). Here, the fixed effect consists of
the genotype: knockout (D8) vs wild type (WT). By default the first
group (D8) will become the reference group and its effect will be
absorbed in the intercept of the model. Hence, only a single slope
term is needed to model the average difference between D8 and WT (See
Section <a href="sec-basics.html#sec-inference">Statistical inference</a> below).</p>
<p>Now we have identified the sources of variation in the experiment, we
can define a model.</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="sec-francisella.html#cb272-1" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="er">~</span> genotype <span class="sc">+</span> <span class="do">## (1) fixed effect for genotype</span></span>
<span id="cb272-2"><a href="sec-francisella.html#cb272-2" tabindex="-1"></a>  (<span class="dv">1</span> <span class="sc">|</span> biorep)  <span class="do">## (2) random effect for biological replicate (culture)</span></span></code></pre></div>
</div>
<div id="estimate-the-model" class="section level3" number="6.6.2">
<h3>
<span class="header-section-number">6.6.2</span> Estimate the model<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('estimate-the-model')" onmouseout="reset_tooltip('estimate-the-model-tooltip')"><span class="tooltiptext" id="estimate-the-model-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We estimate the model with <code>msqrob()</code> (see <a href="sec-advanced.html#sec-run_model">the modelling
section</a>). Recall that
variables defined in <code>model</code> are automatically retrieved from the
<code>colData</code> (i.e. <code>"genotype"</code>, and <code>"biorep"</code>). Note, that <code>msqrob2</code>
also features ridge regression for stabilising the parameter
estimation, but it is irrelevant in this context as the genotype
factor only has 2 levels (WT and D8), so ridge regression . We will
therefore leave the ridge regression disabled (default).</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="sec-francisella.html#cb273-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">msqrob</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">formula =</span> model, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="statistical-inference-1" class="section level2" number="6.7">
<h2>
<span class="header-section-number">6.7</span> Statistical inference<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('statistical-inference-1')" onmouseout="reset_tooltip('statistical-inference-1-tooltip')"><span class="tooltiptext" id="statistical-inference-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>Once the models are estimated, we can start answering biological
questions by performing <a href="sec-basics.html#sec-inference">Statistical inference</a>. We must convert the
biological question “does the bacterial genotype affect the protein
intensities?” into a statistical hypothesis. In other words, we must
convert this question in a combination of the model parameters, also
referred to as a contrast. To aid defining contrasts, we will
visualise the experimental design using the <code>ExploreModelMatrix</code>
package. Note that with ExploreModelMatrix we can only visualise
fixed effects part of the model. This is fine as the mean protein
abundances can only systematically differ from each other according to
the <code>genotype</code> (fixed effect).</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="sec-francisella.html#cb274-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ExploreModelMatrix"</span>)</span>
<span id="cb274-2"><a href="sec-francisella.html#cb274-2" tabindex="-1"></a>vd <span class="ot">&lt;-</span> <span class="fu">VisualizeDesign</span>(</span>
<span id="cb274-3"><a href="sec-francisella.html#cb274-3" tabindex="-1"></a>    <span class="at">sampleData =</span>  <span class="fu">colData</span>(pe),</span>
<span id="cb274-4"><a href="sec-francisella.html#cb274-4" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> genotype,</span>
<span id="cb274-5"><a href="sec-francisella.html#cb274-5" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb274-6"><a href="sec-francisella.html#cb274-6" tabindex="-1"></a>)</span>
<span id="cb274-7"><a href="sec-francisella.html#cb274-7" tabindex="-1"></a>vd<span class="sc">$</span>plotlist</span></code></pre></div>
<pre><code>## [[1]]</code></pre>
<p><img src="figure/francisella_VisualizeDesign-1.png"></p>
<p>This plot shows that the average <span class="math inline">\(\log_2\)</span> protein intensity for the D8 group is
modelled by <code>(Intercept)</code>, and the the average <span class="math inline">\(\log_2\)</span> protein intensity for
the WT group is modelled by <code>(Intercept) + genotypeWT</code>.</p>
<div id="hypothesis-testing-1" class="section level3" number="6.7.1">
<h3>
<span class="header-section-number">6.7.1</span> Hypothesis testing<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('hypothesis-testing-1')" onmouseout="reset_tooltip('hypothesis-testing-1-tooltip')"><span class="tooltiptext" id="hypothesis-testing-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Hence we can translate the research hypothesis that there is an effect
of the genotype on the protein abundance to the average log2 fold
change (<span class="math inline">\(\log_2 FC\)</span>) between the WT and D8 groups, which boils down to
<code>(Intercept) + genotypeWT - (Intercept)</code>, and equals the <code>genotypeWT</code>
parameter. The null hypothesis of the hypothesis test for this
contrast is that the average <span class="math inline">\(\log2 FC\)</span> between D8 knock-out and WT is
zero, or in other words that the <code>genotypeWT</code> parameter is zero.</p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="sec-francisella.html#cb276-1" tabindex="-1"></a>hypothesis <span class="ot">&lt;-</span> <span class="st">"genotypeWT = 0"</span></span></code></pre></div>
<p>We next use <code>makeContrast()</code> to build the contrast matrix.</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="sec-francisella.html#cb277-1" tabindex="-1"></a>(L <span class="ot">&lt;-</span> <span class="fu">makeContrast</span>(hypothesis, <span class="at">parameterNames =</span> <span class="st">"genotypeWT"</span>))</span></code></pre></div>
<pre><code>##            genotypeWT
## genotypeWT          1</code></pre>
<p>In this case, the contrast matrix is trivial, but it becomes a matrix
but for <a href="sec-basics.html#sec-multiple_contrasts">more complex designs</a>. We can now
test our null hypothesis:</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="sec-francisella.html#cb279-1" tabindex="-1"></a>pe <span class="ot">&lt;-</span> <span class="fu">hypothesisTest</span>(pe, <span class="at">i =</span> <span class="st">"proteins"</span>, <span class="at">contrast =</span> L)</span></code></pre></div>
<p>Let us retrieve the result table from the <code>rowData</code>. Note that the
hypothesis testing results are stored in <code>rowData</code> columns named after
the column names (here <code>genotypeWT</code>) of the contrast matrix <code>L</code>.</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="sec-francisella.html#cb280-1" tabindex="-1"></a>inference <span class="ot">&lt;-</span> <span class="fu">rowData</span>(pe[[<span class="st">"proteins"</span>]])[[<span class="fu">colnames</span>(L)]]</span>
<span id="cb280-2"><a href="sec-francisella.html#cb280-2" tabindex="-1"></a>inference<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)</span>
<span id="cb280-3"><a href="sec-francisella.html#cb280-3" tabindex="-1"></a><span class="fu">head</span>(inference)</span></code></pre></div>
<pre><code>##                    logFC         se        df          t         pval
## WP_003013731  0.08607313 0.13622311 14.515249  0.6318541 0.5373074487
## WP_003013860 -0.24361570 0.96333877  4.863216 -0.2528868 0.8106957489
## WP_003013909 -0.29604787 0.06878765 15.007538 -4.3037941 0.0006263236
## WP_003014068  0.11520522 0.15938527 14.142659  0.7228097 0.4815800326
## WP_003014122  0.20050168 0.11484365 17.854759  1.7458665 0.0980174216
## WP_003014123  0.11735634 0.14937755 14.599255  0.7856357 0.4446451519
##                  adjPval      Protein
## WP_003013731 0.740335646 WP_003013731
## WP_003013860 0.909709125 WP_003013860
## WP_003013909 0.006914364 WP_003013909
## WP_003014068 0.695546291 WP_003014068
## WP_003014122 0.253571752 WP_003014122
## WP_003014123 0.666917832 WP_003014123</code></pre>
<p>Notice that some rows contain missing values. This is because data
modelling resulted in a <code>fitError</code> for some proteins, probably because
not enough data was available for model fitting due to missing values
in the quantitative data (see <a href="sec-advanced.html#sec-fiterror">how to deal with
<code>fitError</code>s</a>).</p>
</div>
<div id="volcano-plot" class="section level3" number="6.7.2">
<h3>
<span class="header-section-number">6.7.2</span> Volcano-plot<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('volcano-plot')" onmouseout="reset_tooltip('volcano-plot-tooltip')"><span class="tooltiptext" id="volcano-plot-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Volcano plots are straightforward to generate from the inference table
above. We also use <code>ggrepel</code> to annotate the 20 most significant
proteins.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="sec-francisella.html#cb282-1" tabindex="-1"></a><span class="fu">ggplot</span>(inference) <span class="sc">+</span></span>
<span id="cb282-2"><a href="sec-francisella.html#cb282-2" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">color =</span> adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb282-3"><a href="sec-francisella.html#cb282-3" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb282-4"><a href="sec-francisella.html#cb282-4" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> <span class="fu">slice_min</span>(inference, adjPval, <span class="at">n =</span> <span class="dv">20</span>),</span>
<span id="cb282-5"><a href="sec-francisella.html#cb282-5" tabindex="-1"></a>                  <span class="fu">aes</span>(<span class="at">label =</span> Protein)) <span class="sc">+</span></span>
<span id="cb282-6"><a href="sec-francisella.html#cb282-6" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">alpha</span>(<span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"red"</span>), <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb282-7"><a href="sec-francisella.html#cb282-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Statistical inference on differences between WT and D8"</span>,</span>
<span id="cb282-8"><a href="sec-francisella.html#cb282-8" tabindex="-1"></a>          <span class="fu">paste</span>(<span class="st">"Hypothesis test:"</span>, <span class="fu">colnames</span>(L), <span class="st">"= 0"</span>))</span></code></pre></div>
<p><img src="figure/francisella_volcano-1.png"></p>
<p>Note, that 195 proteins
are found to be differentially abundant.</p>
</div>
<div id="heatmap" class="section level3" number="6.7.3">
<h3>
<span class="header-section-number">6.7.3</span> Heatmap<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('heatmap')" onmouseout="reset_tooltip('heatmap-tooltip')"><span class="tooltiptext" id="heatmap-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>We can also build a <a href="sec-basics.html#sec-heatmaps">heatmap</a> for the significant
proteins which are obtained by filtering the inference table. We first
retrieve the data with proteins that are differentially abundant
between the WT and the D8 genotype.</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="sec-francisella.html#cb283-1" tabindex="-1"></a>sigNames <span class="ot">&lt;-</span> inference<span class="sc">$</span>Protein[<span class="sc">!</span><span class="fu">is.na</span>(inference<span class="sc">$</span>adjPval) <span class="sc">&amp;</span> inference<span class="sc">$</span>adjPval <span class="sc">&lt;</span> <span class="fl">0.05</span>]</span>
<span id="cb283-2"><a href="sec-francisella.html#cb283-2" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getWithColData</span>(pe, <span class="st">"proteins"</span>)[sigNames, ]</span></code></pre></div>
<p>We then plot the protein-wise standardised data as an annotated
heatmap.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="sec-francisella.html#cb284-1" tabindex="-1"></a>quants <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(<span class="fu">assay</span>(se))))</span>
<span id="cb284-2"><a href="sec-francisella.html#cb284-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"ComplexHeatmap"</span>)</span>
<span id="cb284-3"><a href="sec-francisella.html#cb284-3" tabindex="-1"></a>annotations <span class="ot">&lt;-</span> <span class="fu">columnAnnotation</span>(</span>
<span id="cb284-4"><a href="sec-francisella.html#cb284-4" tabindex="-1"></a>  <span class="at">condition =</span> se<span class="sc">$</span>genotype</span>
<span id="cb284-5"><a href="sec-francisella.html#cb284-5" tabindex="-1"></a>)</span>
<span id="cb284-6"><a href="sec-francisella.html#cb284-6" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>) <span class="do">## annotation colours are randomly generated by default</span></span>
<span id="cb284-7"><a href="sec-francisella.html#cb284-7" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb284-8"><a href="sec-francisella.html#cb284-8" tabindex="-1"></a> quants, <span class="at">name =</span> <span class="st">"log2 intensity"</span>,</span>
<span id="cb284-9"><a href="sec-francisella.html#cb284-9" tabindex="-1"></a> <span class="at">top_annotation =</span> annotations</span>
<span id="cb284-10"><a href="sec-francisella.html#cb284-10" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/francisella_heatmap-1.png"></p>
</div>
<div id="detail-plots-1" class="section level3" number="6.7.4">
<h3>
<span class="header-section-number">6.7.4</span> Detail plots<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('detail-plots-1')" onmouseout="reset_tooltip('detail-plots-1-tooltip')"><span class="tooltiptext" id="detail-plots-1-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h3>
<p>Let us visualise the most significant protein. We perform this with a
little data manipulation pipeline:</p>
<ol style="list-style-type: decimal">
<li>Identify the target protein with largest logFC.</li>
<li>We use the <code>QFeatures</code> subsetting functionality to retrieve all
data related to the target protein, focusing on the <code>proteins</code>
set that contains the preprocessed data used for modelling.</li>
<li>We use <code>longForm()</code> to convert the object into a table suitable
for plotting.</li>
<li>We remove missing values for plotting.</li>
<li>Plot the data with <code>ggplot2</code>.</li>
</ol>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="sec-francisella.html#cb285-1" tabindex="-1"></a>targetProtein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(inference)[<span class="fu">which.min</span>(inference<span class="sc">$</span>adjPval)] <span class="co">#1</span></span>
<span id="cb285-2"><a href="sec-francisella.html#cb285-2" tabindex="-1"></a>pe[targetProtein, , <span class="st">"proteins"</span>] <span class="sc">|&gt;</span> <span class="co">#2</span></span>
<span id="cb285-3"><a href="sec-francisella.html#cb285-3" tabindex="-1"></a>  <span class="fu">longForm</span>(<span class="at">colvars =</span> <span class="st">"genotype"</span>) <span class="sc">|&gt;</span>  <span class="co">#3</span></span>
<span id="cb285-4"><a href="sec-francisella.html#cb285-4" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb285-5"><a href="sec-francisella.html#cb285-5" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">|&gt;</span> <span class="co">#4</span></span>
<span id="cb285-6"><a href="sec-francisella.html#cb285-6" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span> <span class="co">#5</span></span>
<span id="cb285-7"><a href="sec-francisella.html#cb285-7" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> colname,</span>
<span id="cb285-8"><a href="sec-francisella.html#cb285-8" tabindex="-1"></a>      <span class="at">y =</span> value) <span class="sc">+</span></span>
<span id="cb285-9"><a href="sec-francisella.html#cb285-9" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> genotype)) <span class="sc">+</span></span>
<span id="cb285-10"><a href="sec-francisella.html#cb285-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2 intensity"</span>) <span class="sc">+</span></span>
<span id="cb285-11"><a href="sec-francisella.html#cb285-11" tabindex="-1"></a>  <span class="fu">ggtitle</span>(targetProtein) <span class="sc">+</span></span>
<span id="cb285-12"><a href="sec-francisella.html#cb285-12" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="figure/francisella_detail_plot-1.png"></p>
</div>
</div>
<div id="conclusion-3" class="section level2" number="6.8">
<h2>
<span class="header-section-number">6.8</span> Conclusion<div class="tooltip"><button class="internal-link-btn" onclick="copy_link('conclusion-3')" onmouseout="reset_tooltip('conclusion-3-tooltip')"><span class="tooltiptext" id="conclusion-3-tooltip">Copy link</span><i class="fa fa-link"></i></button></div>
</h2>
<p>In this chapter, we illustrated the analysis of a label-free
proteomics data set with technical replication. We followed the
workflow described in the previous chapters with minimal changes, with
a few exceptions.</p>
<p>First, we removed peptides that were missing across biological
replicates instead of across all samples. This filtering strategy uses
the experimental design to better define interesting features. Indeed,
peptides that are only found in a single francisella culture batch
bears no interesting biological information. The <code>QFeatures</code> framework
is amenable for this custom filtering.</p>
<p>Second, we could not perform ridge regression because the fixed
effects contained only two levels. However, it is possible to disable
ridge regression within <code>mssqrob()</code> thanks to the argument <code>ridge = FALSE</code>.</p>
<p>Third, this data set has a more complex experimental design since there
each biological replicate has been acquired in technical triplicate.
Hence, we had to model</p>
<ul>
<li>the variation according to the treatment, effect of interest and modelled using a fixed effect</li>
<li>variation between biological repeats, random effect as the francisella cultures will change each time we repeat the experiment, and</li>
<li>technical variability, run to run variability and other sources of technical variability which are lumped in the residual variance.</li>
</ul>
<p>Note, that we could have performed the differential abundance analysis
at the protein-level using <a href="sec-benchmarking.html#sec-ion_level_modelling">ion- or peptide-level
models</a>, i.e. by using the <code>msqrobAggregate</code>
on the peptide_log assay with a formula <code>peptideModel &lt;- ~ genotype + (1|biorep) + (1|run) + (1|Sequence)</code>.</p>

</div>
</div>
<p style="text-align: center;">
<a href="sec-workflow-optimisation.html"><button class="btn btn-default">Previous</button></a>
<a href="sec-heart.html"><button class="btn btn-default">Next</button></a>
</p>
<p class="build-date">Page built: 
2025-12-10
 using 
R version 4.5.2 (2025-10-31)
</p>
</div>
</div>



</body>
</html>
